{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2b7c219a-7ec9-45e9-a567-67bf056c2e46",
   "metadata": {},
   "source": [
    "## BRENDA pH activity data\n",
    "- BRENDA has \"pH optimum\" and \"pH range\" fields\n",
    "- We need to match up the data from these two sheets\n",
    "- This notebook will parse the \"pH optimum\" data and match it with the ph range data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f2998a61-76fd-419d-a01a-2da7e0ae55d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import re\n",
    "import regex\n",
    "from tqdm.auto import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec71a1a6-a2a6-4ea7-bad8-868fd8a7f2cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "inputs_dir = Path(\"/projects/robustmicrob/jlaw/inputs/brenda/202301_tables\")\n",
    "data_dir = \"/projects/robustmicrob/jlaw/projects/prot_stability_engineering/inputs/brenda\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a4b999be-1118-4976-b359-53d9c252f612",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "51028\n",
      "50084\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>name</th>\n",
       "      <th>ph_min</th>\n",
       "      <th>ph_max</th>\n",
       "      <th>comments</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>reduction of aldehydes</td>\n",
       "      <td>Thermococcus kodakarensis</td>\n",
       "      <td>Q5JI09</td>\n",
       "      <td>735486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>3.8</td>\n",
       "      <td>-</td>\n",
       "      <td>-4.3</td>\n",
       "      <td>Neurospora crassa</td>\n",
       "      <td>-</td>\n",
       "      <td>659773</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ec_num                   name ph_min ph_max                comments  \\\n",
       "2  1.1.1.1  alcohol dehydrogenase      3      -  reduction of aldehydes   \n",
       "3  1.1.1.1  alcohol dehydrogenase    3.8      -                    -4.3   \n",
       "\n",
       "                    organism uniprot_id     ref  \n",
       "2  Thermococcus kodakarensis     Q5JI09  735486  \n",
       "3          Neurospora crassa          -  659773  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ph_range_file = inputs_dir / \"202301_ph_opt.tsv.gz\"\n",
    "header = [\"ec_num\", \"name\", \"ph_min\", \"ph_max\", \"comments\", \"organism\", \"uniprot_id\", \"ref\", \"empty\"]\n",
    "data = pd.read_csv(ph_range_file, sep='\\t', names=header)\n",
    "data = data.drop(columns=['empty'])\n",
    "print(len(data))\n",
    "data = data[data.ph_min != \"additional information\"]\n",
    "# data = data[data.ph_min != \"additional information\"]\n",
    "print(len(data))\n",
    "data.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "460e0966-0550-4cdc-b391-bbe0015eea37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Homo sapiens                    3979\n",
       "Rattus norvegicus               2213\n",
       "Escherichia coli                1885\n",
       "Bos taurus                      1029\n",
       "Saccharomyces cerevisiae        1013\n",
       "                                ... \n",
       "Palaemon serratus                  1\n",
       "Parophrys vetulus                  1\n",
       "Pleuronichthys verticalis          1\n",
       "Pseudopleuronectes yokohamae       1\n",
       "Acyrthosiphon pisum                1\n",
       "Name: organism, Length: 5906, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.organism.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ac95c3e-e080-4de6-92bd-69e4cc169a1e",
   "metadata": {},
   "source": [
    "#### Find a uniprot ID for rows where it's missing\n",
    "See if we can use the mapping from the uniprot ID's ec numbers to get a protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "21c70f40-6e60-447b-82dc-7f63259f2317",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "274744\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Reviewed</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "      <th>EC number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A009IHW8</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>ABTIR_ACIB9</td>\n",
       "      <td>2' cyclic ADP-D-ribose synthase AbTIR (2'cADPR...</td>\n",
       "      <td>J512_3302</td>\n",
       "      <td>Acinetobacter baumannii (strain 1295743)</td>\n",
       "      <td>269</td>\n",
       "      <td>3.2.2.-; 3.2.2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A023I7E1</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>ENG1_RHIMI</td>\n",
       "      <td>Glucan endo-1,3-beta-D-glucosidase 1 (Endo-1,3...</td>\n",
       "      <td>ENG1 LAM81A</td>\n",
       "      <td>Rhizomucor miehei</td>\n",
       "      <td>796</td>\n",
       "      <td>3.2.1.39</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Entry  Reviewed   Entry Name  \\\n",
       "0  A0A009IHW8  reviewed  ABTIR_ACIB9   \n",
       "1  A0A023I7E1  reviewed   ENG1_RHIMI   \n",
       "\n",
       "                                       Protein names   Gene Names  \\\n",
       "0  2' cyclic ADP-D-ribose synthase AbTIR (2'cADPR...    J512_3302   \n",
       "1  Glucan endo-1,3-beta-D-glucosidase 1 (Endo-1,3...  ENG1 LAM81A   \n",
       "\n",
       "                                   Organism  Length         EC number  \n",
       "0  Acinetobacter baumannii (strain 1295743)     269  3.2.2.-; 3.2.2.6  \n",
       "1                         Rhizomucor miehei     796          3.2.1.39  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_ec_file = f\"{data_dir}/uniprot/uniprotkb_ec_AND_reviewed_true_2023_07_06.tsv.gz\"\n",
    "uniprot_ec_df = pd.read_table(uniprot_ec_file)\n",
    "print(len(uniprot_ec_df))\n",
    "uniprot_ec_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "92b22ecc-c146-46f5-b20c-d15bf1ddf8c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36326"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_no_uniprot = data[data.uniprot_id == \"-\"]\n",
    "len(data_no_uniprot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "daec7ada-73a1-45f1-a747-b9566b8bf088",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "73b9b570cf2d43e59f0259d8bcbb3b60",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/4539 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "33505\n"
     ]
    }
   ],
   "source": [
    "data_matched_uniprot = pd.DataFrame()\n",
    "for ec_num, df in tqdm(data_no_uniprot.groupby('ec_num')):\n",
    "    uniprot_ec_rows = uniprot_ec_df[uniprot_ec_df['EC number'].apply(lambda x: ec_num in x.split('; '))]\n",
    "    if len(uniprot_ec_rows) == 0:\n",
    "        continue\n",
    "\n",
    "    org_count = df.organism.value_counts().to_dict()\n",
    "    \n",
    "    def match_uniprot_ec_organism(org, \n",
    "                                  num_brenda_org_cutoff=2,\n",
    "                                  num_matching_rows_cutoff=2):\n",
    "        # only try and automatically match for brenda entries that have fewer than 3 entries for the same enzyme\n",
    "        if org_count[org] > num_brenda_org_cutoff:\n",
    "            return np.nan\n",
    "\n",
    "        matching_rows = uniprot_ec_rows[uniprot_ec_rows.Organism.apply(lambda x: org in x)]\n",
    "        if len(matching_rows) > num_matching_rows_cutoff:\n",
    "            return np.nan\n",
    "\n",
    "        return \",\".join(matching_rows.Entry.values)\n",
    "\n",
    "    df['uniprot_id'] = df.organism.apply(match_uniprot_ec_organism)\n",
    "    data_matched_uniprot = pd.concat([data_matched_uniprot, df])\n",
    "\n",
    "print(len(data_matched_uniprot))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "964e8cf8-c2c9-4783-89e3-72f5c8af5814",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_matched_uniprot.uniprot_id = data_matched_uniprot.uniprot_id.replace(\"\",np.nan)\n",
    "data_matched_uniprot = data_matched_uniprot.dropna(subset='uniprot_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7b11d976-4757-4634-a62c-72781755e1dc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5469 4268\n"
     ]
    }
   ],
   "source": [
    "print(len(data_matched_uniprot), data_matched_uniprot.uniprot_id.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "dfe55027-293d-4923-bdfa-29fecdc1d10e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13756 10465\n",
      "19225 14208\n",
      "19225 14208\n"
     ]
    }
   ],
   "source": [
    "data_uniprot = data[data.uniprot_id != '-'].dropna(subset='uniprot_id')\n",
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())\n",
    "data_uniprot = pd.concat([data_uniprot, data_matched_uniprot])\n",
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())\n",
    "data_uniprot = data_uniprot.drop_duplicates()\n",
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a3a879a6-ec11-4e93-a0bb-6b787455e1b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Q51723           17\n",
       "P08659           14\n",
       "P00784           13\n",
       "Q9AJE3           11\n",
       "O23791           11\n",
       "                 ..\n",
       "Q5AVY9            1\n",
       "Q9KIF2            1\n",
       "B3VS85            1\n",
       "C7A621            1\n",
       "P97046,Q9CHL8     1\n",
       "Name: uniprot_id, Length: 14208, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_uniprot.uniprot_id.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bcedfa5d-c50b-469a-b2a0-59ec35be656f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24674 15793\n"
     ]
    }
   ],
   "source": [
    "# split rows with multiple uniprot IDs\n",
    "data_uniprot[\"uniprot_id\"] = data_uniprot.uniprot_id.apply(lambda x: x.replace(' AND ',',').replace(' ','').split(','))\n",
    "data_uniprot = data_uniprot.explode(column='uniprot_id', ignore_index=True)\n",
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "6b4da9fd-ac6f-4c85-ba58-84d0f0b53e17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "P27169    34\n",
       "Q975F9    33\n",
       "Q51723    20\n",
       "P08659    19\n",
       "Q9NR71    19\n",
       "          ..\n",
       "Q2QC90     1\n",
       "Q6HA27     1\n",
       "P48147     1\n",
       "W8GKU3     1\n",
       "Q9CHL8     1\n",
       "Name: uniprot_id, Length: 15793, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_uniprot.uniprot_id.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5986a380-4a9b-4afc-ab32-4f5aad03f7c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>name</th>\n",
       "      <th>ph_min</th>\n",
       "      <th>ph_max</th>\n",
       "      <th>comments</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9203</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>7.4</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35213, 665028, 678376, 681730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9204</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>7.4</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35213, 665028, 678376, 681730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9208</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9209</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9210</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9213</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9214</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9215</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9216</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 35215, 663985, 664184, 665960, 677411, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9221</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>677892, 679634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9222</th>\n",
       "      <td>3.1.1.2</td>\n",
       "      <td>arylesterase</td>\n",
       "      <td>8.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>677892, 679634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9272</th>\n",
       "      <td>3.1.1.25</td>\n",
       "      <td>1,4-lactonase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>665960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9669</th>\n",
       "      <td>3.1.1.81</td>\n",
       "      <td>quorum-quenching N-acyl-homoserine lactonase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>749628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10451</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>7.4</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35213, 678376, 679759, 681730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10452</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>7.4</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35213, 678376, 679759, 681730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10454</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>7.5</td>\n",
       "      <td>8</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>749610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10457</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>664153, 666426, 707214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10461</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10462</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10463</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10464</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10465</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10466</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10467</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35212, 663614, 664184, 665987, 677411, 677893,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10478</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>665960, 677892, 679634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10480</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>665960, 677892, 679634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10481</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>8.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>665960, 677892, 679634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10496</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>10.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35208, 35215, 664184, 677884, 678464, 681055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10497</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>10.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35208, 35215, 664184, 677884, 678464, 681055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10498</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>10.5</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>35208, 35215, 664184, 677884, 678464, 681055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10499</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>10.5</td>\n",
       "      <td>-</td>\n",
       "      <td>paraoxonase activity of recombinant PON1-hFc o...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>752102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10500</th>\n",
       "      <td>3.1.8.1</td>\n",
       "      <td>aryldialkylphosphatase</td>\n",
       "      <td>10.5</td>\n",
       "      <td>-</td>\n",
       "      <td>the maximum rate of methylphosphonic acid hydr...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>749684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10503</th>\n",
       "      <td>3.1.8.2</td>\n",
       "      <td>diisopropyl-fluorophosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>749628, 752102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10504</th>\n",
       "      <td>3.1.8.2</td>\n",
       "      <td>diisopropyl-fluorophosphatase</td>\n",
       "      <td>8</td>\n",
       "      <td>-</td>\n",
       "      <td>assay at</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>P27169</td>\n",
       "      <td>749628, 752102</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ec_num                                          name ph_min ph_max  \\\n",
       "9203    3.1.1.2                                  arylesterase    7.4      -   \n",
       "9204    3.1.1.2                                  arylesterase    7.4      -   \n",
       "9208    3.1.1.2                                  arylesterase      8      -   \n",
       "9209    3.1.1.2                                  arylesterase      8      -   \n",
       "9210    3.1.1.2                                  arylesterase      8      -   \n",
       "9213    3.1.1.2                                  arylesterase      8      -   \n",
       "9214    3.1.1.2                                  arylesterase      8      -   \n",
       "9215    3.1.1.2                                  arylesterase      8      -   \n",
       "9216    3.1.1.2                                  arylesterase      8      -   \n",
       "9221    3.1.1.2                                  arylesterase    8.5      -   \n",
       "9222    3.1.1.2                                  arylesterase    8.5      -   \n",
       "9272   3.1.1.25                                 1,4-lactonase      8      -   \n",
       "9669   3.1.1.81  quorum-quenching N-acyl-homoserine lactonase      8      -   \n",
       "10451   3.1.8.1                        aryldialkylphosphatase    7.4      -   \n",
       "10452   3.1.8.1                        aryldialkylphosphatase    7.4      -   \n",
       "10454   3.1.8.1                        aryldialkylphosphatase    7.5      8   \n",
       "10457   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10461   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10462   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10463   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10464   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10465   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10466   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10467   3.1.8.1                        aryldialkylphosphatase      8      -   \n",
       "10478   3.1.8.1                        aryldialkylphosphatase    8.5      -   \n",
       "10480   3.1.8.1                        aryldialkylphosphatase    8.5      -   \n",
       "10481   3.1.8.1                        aryldialkylphosphatase    8.5      -   \n",
       "10496   3.1.8.1                        aryldialkylphosphatase   10.5      -   \n",
       "10497   3.1.8.1                        aryldialkylphosphatase   10.5      -   \n",
       "10498   3.1.8.1                        aryldialkylphosphatase   10.5      -   \n",
       "10499   3.1.8.1                        aryldialkylphosphatase   10.5      -   \n",
       "10500   3.1.8.1                        aryldialkylphosphatase   10.5      -   \n",
       "10503   3.1.8.2                 diisopropyl-fluorophosphatase      8      -   \n",
       "10504   3.1.8.2                 diisopropyl-fluorophosphatase      8      -   \n",
       "\n",
       "                                                comments      organism  \\\n",
       "9203                                            assay at  Homo sapiens   \n",
       "9204                                            assay at  Homo sapiens   \n",
       "9208                                            assay at  Homo sapiens   \n",
       "9209                                            assay at  Homo sapiens   \n",
       "9210                                            assay at  Homo sapiens   \n",
       "9213                                            assay at  Homo sapiens   \n",
       "9214                                            assay at  Homo sapiens   \n",
       "9215                                            assay at  Homo sapiens   \n",
       "9216                                            assay at  Homo sapiens   \n",
       "9221                                            assay at  Homo sapiens   \n",
       "9222                                            assay at  Homo sapiens   \n",
       "9272                                            assay at  Homo sapiens   \n",
       "9669                                            assay at  Homo sapiens   \n",
       "10451                                           assay at  Homo sapiens   \n",
       "10452                                           assay at  Homo sapiens   \n",
       "10454                                           assay at  Homo sapiens   \n",
       "10457                                                  -  Homo sapiens   \n",
       "10461                                           assay at  Homo sapiens   \n",
       "10462                                           assay at  Homo sapiens   \n",
       "10463                                           assay at  Homo sapiens   \n",
       "10464                                           assay at  Homo sapiens   \n",
       "10465                                           assay at  Homo sapiens   \n",
       "10466                                           assay at  Homo sapiens   \n",
       "10467                                           assay at  Homo sapiens   \n",
       "10478                                           assay at  Homo sapiens   \n",
       "10480                                           assay at  Homo sapiens   \n",
       "10481                                           assay at  Homo sapiens   \n",
       "10496                                           assay at  Homo sapiens   \n",
       "10497                                           assay at  Homo sapiens   \n",
       "10498                                           assay at  Homo sapiens   \n",
       "10499  paraoxonase activity of recombinant PON1-hFc o...  Homo sapiens   \n",
       "10500  the maximum rate of methylphosphonic acid hydr...  Homo sapiens   \n",
       "10503                                           assay at  Homo sapiens   \n",
       "10504                                           assay at  Homo sapiens   \n",
       "\n",
       "      uniprot_id                                                ref  \n",
       "9203      P27169                      35213, 665028, 678376, 681730  \n",
       "9204      P27169                      35213, 665028, 678376, 681730  \n",
       "9208      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9209      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9210      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9213      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9214      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9215      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9216      P27169  35212, 35215, 663985, 664184, 665960, 677411, ...  \n",
       "9221      P27169                                     677892, 679634  \n",
       "9222      P27169                                     677892, 679634  \n",
       "9272      P27169                                             665960  \n",
       "9669      P27169                                             749628  \n",
       "10451     P27169                      35213, 678376, 679759, 681730  \n",
       "10452     P27169                      35213, 678376, 679759, 681730  \n",
       "10454     P27169                                             749610  \n",
       "10457     P27169                             664153, 666426, 707214  \n",
       "10461     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10462     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10463     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10464     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10465     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10466     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10467     P27169  35212, 663614, 664184, 665987, 677411, 677893,...  \n",
       "10478     P27169                             665960, 677892, 679634  \n",
       "10480     P27169                             665960, 677892, 679634  \n",
       "10481     P27169                             665960, 677892, 679634  \n",
       "10496     P27169       35208, 35215, 664184, 677884, 678464, 681055  \n",
       "10497     P27169       35208, 35215, 664184, 677884, 678464, 681055  \n",
       "10498     P27169       35208, 35215, 664184, 677884, 678464, 681055  \n",
       "10499     P27169                                             752102  \n",
       "10500     P27169                                             749684  \n",
       "10503     P27169                                     749628, 752102  \n",
       "10504     P27169                                     749628, 752102  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_uniprot[data_uniprot.uniprot_id == \"P27169\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "69bb3755-8595-4eba-a647-46d3a06d8681",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_file = Path(data_dir, \"ph_opt/20230907_ph_opt_processed.csv\")\n",
    "data_uniprot.to_csv(out_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8c57d19-3a28-4027-89ef-82e0b3bbdf0f",
   "metadata": {},
   "source": [
    "Remove the \"assay at\" rows as well as the mutant rows (for now)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "08a446f4-3807-4b56-b07b-75f2620be3d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24674 15793\n",
      "14049 9819\n"
     ]
    }
   ],
   "source": [
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())\n",
    "data_uniprot = data_uniprot[data_uniprot.comments.apply(lambda x: \"assay at\" not in x)]\n",
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "37b93073-18f7-4b60-85ff-052f21155aec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14049 9819\n",
      "13781 9763\n"
     ]
    }
   ],
   "source": [
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())\n",
    "data_uniprot = data_uniprot[data_uniprot.comments.apply(lambda x: \"mutant\" not in x)]\n",
    "print(len(data_uniprot), data_uniprot.uniprot_id.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "53f2ce18-25ba-4c7d-8f23-95350a50c400",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_file = Path(data_dir, \"ph_opt/20230907_ph_opt_no_assay_at_no_mutant.csv\")\n",
    "data_uniprot.drop(columns='name').to_csv(out_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8578b42-f4e5-4ffd-8d76-5e3aacc71d17",
   "metadata": {},
   "source": [
    "### Match the data_opt rows with the ph_range rows\n",
    "- We should be able to match based on the references"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "889ac17f-618b-4665-87a1-3687ff932127",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6241\n",
      "3151\n"
     ]
    }
   ],
   "source": [
    "out_file = Path(data_dir, \"ph_range/20230907_ph_range_processed_act.csv\")\n",
    "data_act_ph_range = pd.read_csv(out_file)\n",
    "print(len(data_act_ph_range))\n",
    "data_act_ph_range\n",
    "\n",
    "out_file = Path(data_dir, \"ph_range/20230907_ph_range_processed.csv\")\n",
    "data_ph_range = pd.read_csv(out_file, index_col=0)\n",
    "data_ph_range = data_ph_range[data_ph_range.index.isin(data_act_ph_range.exp_index.unique())]\n",
    "print(len(data_ph_range))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "405bde2a-c1fe-45c5-970f-f10735587d5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exp_index</th>\n",
       "      <th>ph</th>\n",
       "      <th>activity</th>\n",
       "      <th>note</th>\n",
       "      <th>ph_range_comment</th>\n",
       "      <th>uniprot_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7.5</td>\n",
       "      <td>0.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4.9</td>\n",
       "      <td>0.25</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>5.8</td>\n",
       "      <td>0.40</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4.9</td>\n",
       "      <td>0.25</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Q4J9F2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6234</th>\n",
       "      <td>240</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>over 60% of maximal activity within this range...</td>\n",
       "      <td>A0A141BGH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6235</th>\n",
       "      <td>240</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>over 60% of maximal activity within this range...</td>\n",
       "      <td>A0A141BGH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6236</th>\n",
       "      <td>232</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>over 60% of maximal activity within this range...</td>\n",
       "      <td>A0A141BGH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6237</th>\n",
       "      <td>232</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>over 60% of maximal activity within this range...</td>\n",
       "      <td>A0A141BGH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6238</th>\n",
       "      <td>232</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>over 60% of maximal activity within this range...</td>\n",
       "      <td>A0A141BGH5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6239 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      exp_index    ph  activity                note  \\\n",
       "0             1   4.0      0.80                 NaN   \n",
       "1             1   7.5      0.60                 NaN   \n",
       "2             3   4.9      0.25  reduction reaction   \n",
       "3             3   5.8      0.40  reduction reaction   \n",
       "4             4   4.9      0.25  reduction reaction   \n",
       "...         ...   ...       ...                 ...   \n",
       "6234        240  10.0      0.60                 NaN   \n",
       "6235        240  11.0      1.00                 NaN   \n",
       "6236        232   5.0      1.00                 NaN   \n",
       "6237        232   4.0      0.60                 NaN   \n",
       "6238        232   8.0      0.60                 NaN   \n",
       "\n",
       "                                       ph_range_comment  uniprot_id  \n",
       "0     pH 4.0: about 80% of maximal activity, pH 7.5:...      B2KJ46  \n",
       "1     pH 4.0: about 80% of maximal activity, pH 7.5:...      B2KJ46  \n",
       "2     pH 4.9: about 25% of maximal activity, pH 5.8:...      Q4J702  \n",
       "3     pH 4.9: about 25% of maximal activity, pH 5.8:...      Q4J702  \n",
       "4     pH 4.9: about 25% of maximal activity, pH 5.8:...      Q4J9F2  \n",
       "...                                                 ...         ...  \n",
       "6234  over 60% of maximal activity within this range...  A0A141BGH5  \n",
       "6235  over 60% of maximal activity within this range...  A0A141BGH5  \n",
       "6236  over 60% of maximal activity within this range...  A0A141BGH5  \n",
       "6237  over 60% of maximal activity within this range...  A0A141BGH5  \n",
       "6238  over 60% of maximal activity within this range...  A0A141BGH5  \n",
       "\n",
       "[6239 rows x 6 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_act_ph_range"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "c45f5afa-e533-4e2f-b1cc-35492a6d93d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>name</th>\n",
       "      <th>ph_min</th>\n",
       "      <th>ph_max</th>\n",
       "      <th>comments</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>4</td>\n",
       "      <td>7.5</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>Candida parapsilosis</td>\n",
       "      <td>B2KJ46</td>\n",
       "      <td>684556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>4.9</td>\n",
       "      <td>5.8</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Sulfolobus acidocaldarius</td>\n",
       "      <td>Q4J702</td>\n",
       "      <td>719428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>4.9</td>\n",
       "      <td>5.8</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Sulfolobus acidocaldarius</td>\n",
       "      <td>Q4J9F2</td>\n",
       "      <td>719428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>5.5</td>\n",
       "      <td>7.5</td>\n",
       "      <td>pH 5.5: about 35% of maximal activity, pH 7.5:...</td>\n",
       "      <td>Sulfurisphaera tokodaii</td>\n",
       "      <td>F9VMI9</td>\n",
       "      <td>718627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>6</td>\n",
       "      <td>-</td>\n",
       "      <td>50% of maximum activity for reduction of aldeh...</td>\n",
       "      <td>Thermococcus kodakarensis</td>\n",
       "      <td>Q5JI09</td>\n",
       "      <td>735486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6382</th>\n",
       "      <td>6.5.1.7</td>\n",
       "      <td>DNA ligase (ATP, ADP or GTP)</td>\n",
       "      <td>6.5</td>\n",
       "      <td>9</td>\n",
       "      <td>pH 6.5: about 90% of maximal activity, 9.0: ab...</td>\n",
       "      <td>Hyperthermus butylicus</td>\n",
       "      <td>A2BJX6</td>\n",
       "      <td>727438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6391</th>\n",
       "      <td>7.1.2.2</td>\n",
       "      <td>H+-transporting two-sector ATPase</td>\n",
       "      <td>10</td>\n",
       "      <td>-</td>\n",
       "      <td>ATP synthesis in washed membranes of Haloferax...</td>\n",
       "      <td>Haloferax volcanii</td>\n",
       "      <td>Q48332</td>\n",
       "      <td>727822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6392</th>\n",
       "      <td>7.2.2.10</td>\n",
       "      <td>P-type Ca2+ transporter</td>\n",
       "      <td>5.5</td>\n",
       "      <td>8</td>\n",
       "      <td>about 35% of maximal activity at pH 5.5 and at...</td>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>P13586</td>\n",
       "      <td>210495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6393</th>\n",
       "      <td>7.2.2.10</td>\n",
       "      <td>P-type Ca2+ transporter</td>\n",
       "      <td>5.5</td>\n",
       "      <td>8</td>\n",
       "      <td>about 35% of maximal activity at pH 5.5 and at...</td>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>P38929</td>\n",
       "      <td>210495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6395</th>\n",
       "      <td>7.2.2.19</td>\n",
       "      <td>H+/K+-exchanging ATPase</td>\n",
       "      <td>6.8</td>\n",
       "      <td>7.8</td>\n",
       "      <td>pH 6.8: about 60% of maximal activity, pH 7.8:...</td>\n",
       "      <td>Sus scrofa</td>\n",
       "      <td>P19156</td>\n",
       "      <td>246853</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3151 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ec_num                               name ph_min ph_max  \\\n",
       "1      1.1.1.1              alcohol dehydrogenase      4    7.5   \n",
       "3      1.1.1.1              alcohol dehydrogenase    4.9    5.8   \n",
       "4      1.1.1.1              alcohol dehydrogenase    4.9    5.8   \n",
       "5      1.1.1.1              alcohol dehydrogenase    5.5    7.5   \n",
       "7      1.1.1.1              alcohol dehydrogenase      6      -   \n",
       "...        ...                                ...    ...    ...   \n",
       "6382   6.5.1.7       DNA ligase (ATP, ADP or GTP)    6.5      9   \n",
       "6391   7.1.2.2  H+-transporting two-sector ATPase     10      -   \n",
       "6392  7.2.2.10            P-type Ca2+ transporter    5.5      8   \n",
       "6393  7.2.2.10            P-type Ca2+ transporter    5.5      8   \n",
       "6395  7.2.2.19            H+/K+-exchanging ATPase    6.8    7.8   \n",
       "\n",
       "                                               comments  \\\n",
       "1     pH 4.0: about 80% of maximal activity, pH 7.5:...   \n",
       "3     pH 4.9: about 25% of maximal activity, pH 5.8:...   \n",
       "4     pH 4.9: about 25% of maximal activity, pH 5.8:...   \n",
       "5     pH 5.5: about 35% of maximal activity, pH 7.5:...   \n",
       "7     50% of maximum activity for reduction of aldeh...   \n",
       "...                                                 ...   \n",
       "6382  pH 6.5: about 90% of maximal activity, 9.0: ab...   \n",
       "6391  ATP synthesis in washed membranes of Haloferax...   \n",
       "6392  about 35% of maximal activity at pH 5.5 and at...   \n",
       "6393  about 35% of maximal activity at pH 5.5 and at...   \n",
       "6395  pH 6.8: about 60% of maximal activity, pH 7.8:...   \n",
       "\n",
       "                       organism uniprot_id     ref  \n",
       "1          Candida parapsilosis     B2KJ46  684556  \n",
       "3     Sulfolobus acidocaldarius     Q4J702  719428  \n",
       "4     Sulfolobus acidocaldarius     Q4J9F2  719428  \n",
       "5       Sulfurisphaera tokodaii     F9VMI9  718627  \n",
       "7     Thermococcus kodakarensis     Q5JI09  735486  \n",
       "...                         ...        ...     ...  \n",
       "6382     Hyperthermus butylicus     A2BJX6  727438  \n",
       "6391         Haloferax volcanii     Q48332  727822  \n",
       "6392   Saccharomyces cerevisiae     P13586  210495  \n",
       "6393   Saccharomyces cerevisiae     P38929  210495  \n",
       "6395                 Sus scrofa     P19156  246853  \n",
       "\n",
       "[3151 rows x 8 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_ph_range"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "ddda87e0-2a6b-46f0-8aea-df2c6fd0cc65",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3181\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>name</th>\n",
       "      <th>ph_min</th>\n",
       "      <th>ph_max</th>\n",
       "      <th>comments</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>ref</th>\n",
       "      <th>ph_range_ref_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>reduction of aldehydes</td>\n",
       "      <td>Thermococcus kodakarensis</td>\n",
       "      <td>Q5JI09</td>\n",
       "      <td>735486</td>\n",
       "      <td>735486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>4.5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Candida parapsilosis</td>\n",
       "      <td>B2KJ46</td>\n",
       "      <td>684556</td>\n",
       "      <td>684556</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ec_num                   name ph_min ph_max                comments  \\\n",
       "0  1.1.1.1  alcohol dehydrogenase      3      -  reduction of aldehydes   \n",
       "1  1.1.1.1  alcohol dehydrogenase    4.5      -                       -   \n",
       "\n",
       "                    organism uniprot_id     ref ph_range_ref_match  \n",
       "0  Thermococcus kodakarensis     Q5JI09  735486             735486  \n",
       "1       Candida parapsilosis     B2KJ46  684556             684556  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_opt_match = data_uniprot[data_uniprot.uniprot_id.isin(data_ph_range.uniprot_id.unique())]\n",
    "print(len(data_opt_match))\n",
    "data_opt_match.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5c293d45-198c-4e48-80df-06041d355832",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2276\n"
     ]
    }
   ],
   "source": [
    "ph_range_refs = set([r.strip() for refs in data_ph_range.ref.values for r in refs.split(',')])\n",
    "print(len(ph_range_refs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "2a5fcfa1-1003-47bd-9170-3e258d7e0de1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2751\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>name</th>\n",
       "      <th>ph_min</th>\n",
       "      <th>ph_max</th>\n",
       "      <th>comments</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>ref</th>\n",
       "      <th>ph_range_ref_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>reduction of aldehydes</td>\n",
       "      <td>Thermococcus kodakarensis</td>\n",
       "      <td>Q5JI09</td>\n",
       "      <td>735486</td>\n",
       "      <td>735486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>4.5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Candida parapsilosis</td>\n",
       "      <td>B2KJ46</td>\n",
       "      <td>684556</td>\n",
       "      <td>684556</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ec_num                   name ph_min ph_max                comments  \\\n",
       "0  1.1.1.1  alcohol dehydrogenase      3      -  reduction of aldehydes   \n",
       "1  1.1.1.1  alcohol dehydrogenase    4.5      -                       -   \n",
       "\n",
       "                    organism uniprot_id     ref ph_range_ref_match  \n",
       "0  Thermococcus kodakarensis     Q5JI09  735486             735486  \n",
       "1       Candida parapsilosis     B2KJ46  684556             684556  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the reference to matchup the optimum rather than the uniprot ID\n",
    "def match_refs(refs):\n",
    "    if not isinstance(refs, str):\n",
    "        return np.nan\n",
    "    for r in refs.replace(' ','').split(','):\n",
    "        if r in ph_range_refs:\n",
    "            return r\n",
    "    return np.nan\n",
    "\n",
    "data_uniprot['ph_range_ref_match'] = data_uniprot.ref.apply(match_refs)\n",
    "data_opt_match = data_uniprot.dropna(subset='ph_range_ref_match')\n",
    "print(len(data_opt_match))\n",
    "data_opt_match.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "f577dc7c-0f2c-4b16-adbd-7cbe5fdb8390",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>name</th>\n",
       "      <th>ph_min</th>\n",
       "      <th>ph_max</th>\n",
       "      <th>comments</th>\n",
       "      <th>organism</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>ref</th>\n",
       "      <th>ph_range_ref_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>reduction of aldehydes</td>\n",
       "      <td>Thermococcus kodakarensis</td>\n",
       "      <td>Q5JI09</td>\n",
       "      <td>735486</td>\n",
       "      <td>735486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>alcohol dehydrogenase</td>\n",
       "      <td>4.5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Candida parapsilosis</td>\n",
       "      <td>B2KJ46</td>\n",
       "      <td>684556</td>\n",
       "      <td>684556</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ec_num                   name ph_min ph_max                comments  \\\n",
       "0  1.1.1.1  alcohol dehydrogenase      3      -  reduction of aldehydes   \n",
       "1  1.1.1.1  alcohol dehydrogenase    4.5      -                       -   \n",
       "\n",
       "                    organism uniprot_id     ref ph_range_ref_match  \n",
       "0  Thermococcus kodakarensis     Q5JI09  735486             735486  \n",
       "1       Candida parapsilosis     B2KJ46  684556             684556  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_opt_match.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "41ee444a-4a96-4268-b62c-2d8cd15c268e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exp_index</th>\n",
       "      <th>ph</th>\n",
       "      <th>activity</th>\n",
       "      <th>note</th>\n",
       "      <th>ph_range_comment</th>\n",
       "      <th>uniprot_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7.5</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   exp_index   ph  activity note  \\\n",
       "0          1  4.0       0.8  NaN   \n",
       "1          1  7.5       0.6  NaN   \n",
       "\n",
       "                                    ph_range_comment uniprot_id  \n",
       "0  pH 4.0: about 80% of maximal activity, pH 7.5:...     B2KJ46  \n",
       "1  pH 4.0: about 80% of maximal activity, pH 7.5:...     B2KJ46  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_act_ph_range.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "d36c62b0-d04c-45a6-bcab-13f8ffdf52fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exp_index</th>\n",
       "      <th>ph</th>\n",
       "      <th>activity</th>\n",
       "      <th>note</th>\n",
       "      <th>ph_range_comment</th>\n",
       "      <th>uniprot_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3676</th>\n",
       "      <td>6265</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.21</td>\n",
       "      <td>3.1% and in the presence of 1 mM and 0.5 mM Mn...</td>\n",
       "      <td>3.1% and 21% of maximal activity at pH 6.0 and...</td>\n",
       "      <td>Q49W02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3677</th>\n",
       "      <td>6265</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.21</td>\n",
       "      <td>3.1% and in the presence of 1 mM and 0.5 mM Mn...</td>\n",
       "      <td>3.1% and 21% of maximal activity at pH 6.0 and...</td>\n",
       "      <td>Q49W02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      exp_index   ph  activity  \\\n",
       "3676       6265  6.0      0.21   \n",
       "3677       6265  7.0      0.21   \n",
       "\n",
       "                                                   note  \\\n",
       "3676  3.1% and in the presence of 1 mM and 0.5 mM Mn...   \n",
       "3677  3.1% and in the presence of 1 mM and 0.5 mM Mn...   \n",
       "\n",
       "                                       ph_range_comment uniprot_id  \n",
       "3676  3.1% and 21% of maximal activity at pH 6.0 and...     Q49W02  \n",
       "3677  3.1% and 21% of maximal activity at pH 6.0 and...     Q49W02  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_act_ph_range[data_act_ph_range.exp_index == 6265]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "e565878b-180f-458d-9788-e30fb4866797",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3.1% and 21% of maximal activity at pH 6.0 and pH 7.0 in the presence of 1 mM and 0.5 mM Mn2+, respectively'"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_ph_range.loc[6265].comments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "01a19eb3-f252-4d41-9405-fdbc121da123",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[3 4.9 0.25 'reduction reaction']\n",
      " [3 5.8 0.4 'reduction reaction']] [8.2] oxidation of alcohols\n",
      "[[3 4.9 0.25 'reduction reaction']\n",
      " [3 5.8 0.4 'reduction reaction']] [8.2] oxidation reaction\n",
      "[[4 4.9 0.25 'reduction reaction']\n",
      " [4 5.8 0.4 'reduction reaction']] [8.2] oxidation of alcohols\n",
      "[[4 4.9 0.25 'reduction reaction']\n",
      " [4 5.8 0.4 'reduction reaction']] [8.2] oxidation reaction\n",
      "[[5 5.5 0.35 'reduction of benzaldehyde']\n",
      " [5 7.5 0.55 'reduction of benzaldehyde']] [10.5] oxidation of 1-hexanol\n",
      "[[10 8.5 0.85 'oxidation reaction']\n",
      " [10 6.7 0.4 'oxidation reaction']] [5.1] reduction of ketones\n",
      "[[10 8.5 0.85 'oxidation reaction']\n",
      " [10 6.7 0.4 'oxidation reaction']] [5.1] reduction reaction\n",
      "[[11 6.7 0.4 'oxidation reaction']\n",
      " [11 8.5 0.85 'oxidation reaction']] [5.1] reduction of ketones\n",
      "[[11 6.7 0.4 'oxidation reaction']\n",
      " [11 8.5 0.85 'oxidation reaction']] [5.1] reduction reaction\n",
      "[[13 7.0 0.7 'reduction of 2-pentanone']\n",
      " [13 9.6 0.5 'reduction of 2-pentanone']] [10.5] oxidation of 2-pentanol\n",
      "[[15 7.5 0.35 'oxidation of 1-hexanol']\n",
      " [15 10.8 0.5 'oxidation of 1-hexanol']] [7.0] reduction of benzaldehyde\n",
      "[[16 11.5 0.3 'oxidation of 2-pentanol']\n",
      " [16 9.5 0.4 'oxidation of 2-pentanol']] [8.0] reduction of 2-pentanone\n",
      "[[22 7.0 0.8\n",
      "  'reduction of D-xylulose with NADH, within this range, recombinant His-tagged enzyme']\n",
      " [22 3.0 0.8\n",
      "  'reduction of D-xylulose with NADH, within this range, recombinant His-tagged enzyme']] [8.5] oxidation of D-arabitol with NAD+, recombinant His-tagged enzyme\n",
      "[[23 11.0 0.6\n",
      "  'oxidation of D-arabitol with NAD+, within this range, recombinant His-tagged enzyme']\n",
      " [23 6.0 0.6\n",
      "  'oxidation of D-arabitol with NAD+, within this range, recombinant His-tagged enzyme']] [5.5] reduction of D-xylulose with NADH, recombinant His-tagged enzyme\n",
      "[[43 6.0 0.4 nan]\n",
      " [43 8.0 0.4 nan]] [6.0, 7.0] -\n",
      "[[89 6.7 0.8 nan]\n",
      " [89 9.0 0.8 nan]] [10.0, 11.0] alcohol oxidation\n",
      "[[89 6.7 0.8 nan]\n",
      " [89 9.0 0.8 nan]] [10.0, 11.0] alcohol oxidation\n",
      "[[97 7.0 0.8 'for reduction of aldehyde']\n",
      " [97 8.5 0.8 'for reduction of aldehyde']] [10.5] oxidation of alcohol\n",
      "[[122 6.0 0.95 'reduction reaction']\n",
      " [122 4.5 0.1 'reduction reaction']\n",
      " [122 7.5 0.1 'reduction reaction']] [9.5] oxidation reaction\n",
      "[[123 7.5 0.5 'reduction reaction']\n",
      " [123 9.0 0.0 'reduction reaction']\n",
      " [123 5.0 0.25 'reduction reaction']] [9.5] oxidation reaction, recombinant enzyme\n",
      "[[124 9.5 0.95 'oxidation reaction']\n",
      " [124 10.5 0.1 'oxidation reaction']] [5.5, 6.0] recombinant enzyme, reduction reaction\n",
      "[[124 9.5 0.95 'oxidation reaction']\n",
      " [124 10.5 0.1 'oxidation reaction']] [5.5, 6.0] recombinant enzyme, reduction reaction\n",
      "[[124 9.5 0.95 'oxidation reaction']\n",
      " [124 10.5 0.1 'oxidation reaction']] [9.5] oxidation reaction, recombinant enzyme\n",
      "[[128 5.5 0.4 'reduction of D-ribulose']\n",
      " [128 8.5 0.1 'reduction of D-ribulose']] [8.5] oxidation of D-arabitol\n",
      "[[131 4.0 0.4 'for reduction of pyruvate within this range']\n",
      " [131 8.0 0.4 'for reduction of pyruvate within this range']] [8.5, 9.5] oxidation of lactate\n",
      "[[131 4.0 0.4 'for reduction of pyruvate within this range']\n",
      " [131 8.0 0.4 'for reduction of pyruvate within this range']] [8.5, 9.5] oxidation of lactate\n",
      "[[132 4.0 0.0 nan]\n",
      " [132 10.0 0.0 nan]] [11.5] recombinant enzyme, lactate oxidation\n",
      "[[134 8.0 0.2 'oxidation of lactate']\n",
      " [134 8.5 0.9 'oxidation of lactate']\n",
      " [134 9.0 0.95 'oxidation of lactate']\n",
      " [134 9.5 0.95 'oxidation of lactate']\n",
      " [134 7.5 0.0 'oxidation of lactate']] [5.0, 5.5] reduction of pyruvate\n",
      "[[134 8.0 0.2 'oxidation of lactate']\n",
      " [134 8.5 0.9 'oxidation of lactate']\n",
      " [134 9.0 0.95 'oxidation of lactate']\n",
      " [134 9.5 0.95 'oxidation of lactate']\n",
      " [134 7.5 0.0 'oxidation of lactate']] [5.0, 5.5] reduction of pyruvate\n",
      "[[134 8.0 0.2 'oxidation of lactate']\n",
      " [134 8.5 0.9 'oxidation of lactate']\n",
      " [134 9.0 0.95 'oxidation of lactate']\n",
      " [134 9.5 0.95 'oxidation of lactate']\n",
      " [134 7.5 0.0 'oxidation of lactate']] [8.5, 9.5] oxidation of lactate\n",
      "[[145 9.0 0.95 'L-HSE oxidation']\n",
      " [145 7.0 0.011 'L-HSE oxidation']] [9.0] L-HSE oxidation in the presence of 0.4 M NaCl\n",
      "[[153 8.0 0.8 'free cells']\n",
      " [153 7.0 0.8 'free cells']] [8.0] recombinant enzyme\n",
      "[[155 9.0 0.95 nan]\n",
      " [155 10.0 0.95 nan]\n",
      " [155 6.5 0.05 nan]] [9.0, 10.0] activity decreases rapidly with decreasing pH. At pH 6.5 the enzyme is almost inactive\n",
      "[[164 11.0 0.55 nan]\n",
      " [164 9.0 0.45 nan]] [5.5, 6.5] conversion of acetoacetyl-CoA to beta-hydroxybutyryl-CoA decreases above pH 6.5\n",
      "[[164 11.0 0.55 nan]\n",
      " [164 9.0 0.45 nan]] [5.5, 6.5] conversion of acetoacetyl-CoA to beta-hydroxybutyryl-CoA decreases above pH 6.5\n",
      "[[175 6.5 0.85 'reduction of acetoin']\n",
      " [175 7.0 0.85 'reduction of acetoin']] [6.5] for acetoin reduction\n",
      "[[175 6.5 0.85 'reduction of acetoin']\n",
      " [175 7.0 0.85 'reduction of acetoin']] [10.0] for (2R,3R)-butane-2,3-diol oxidation\n",
      "[[181 7.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [181 8.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [181 7.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [181 8.3 0.65 'pH: pH: recombinant NADP-ME3']] [7.3] recombinant NADP-ME2\n",
      "[[182 7.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [182 8.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [182 7.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [182 8.3 0.65 'pH: pH: recombinant NADP-ME3']] [7.3] recombinant NADP-ME2\n",
      "[[191 7.5 1.0 nan]\n",
      " [191 5.8 0.33 nan]] [7.5] -\n",
      "[[194 6.5 0.6 'isoenzyme IDP1']\n",
      " [194 9.0 0.7 'isoenzyme IDP1']] [6.5, 9.0] more than 90% of optimal Vmax, isoenzyme IDP2\n",
      "[[194 6.5 0.6 'isoenzyme IDP1']\n",
      " [194 9.0 0.7 'isoenzyme IDP1']] [6.5, 9.0] more than 90% of optimal Vmax, isoenzyme IDP2\n",
      "[[195 6.5 0.6 'isoenzyme IDP1']\n",
      " [195 9.0 0.7 'isoenzyme IDP1']] [6.5, 9.0] more than 90% of optimal Vmax, isoenzyme IDP2\n",
      "[[195 6.5 0.6 'isoenzyme IDP1']\n",
      " [195 9.0 0.7 'isoenzyme IDP1']] [6.5, 9.0] more than 90% of optimal Vmax, isoenzyme IDP2\n",
      "[[196 7.0 0.6 'isoenzyme IDP2']\n",
      " [196 8.0 0.4 'isoenzyme IDP2']] [8.0] isoenzyme IDP1\n",
      "[[196 7.0 0.6 'isoenzyme IDP2']\n",
      " [196 8.0 0.4 'isoenzyme IDP2']] [8.5] isoenzyme IDP3\n",
      "[[197 7.0 0.6 'isoenzyme IDP2']\n",
      " [197 8.0 0.4 'isoenzyme IDP2']] [8.0] isoenzyme IDP1\n",
      "[[197 7.0 0.6 'isoenzyme IDP2']\n",
      " [197 8.0 0.4 'isoenzyme IDP2']] [8.5] isoenzyme IDP3\n",
      "[[200 7.5 0.9 nan]\n",
      " [200 8.5 0.9 nan]] [8.5] -\n",
      "[[244 7.5 0.75 nan]\n",
      " [244 9.5 0.75 nan]] [5.2] enzyme possess a dual pH optimum. A significant decrease in the Ki for substrate inhibition at pH values corresponding to the valley between these optima is responsible for this phenomenon\n",
      "[[255 6.0 0.6 nan]\n",
      " [255 8.0 0.6 nan]] [9.0] -\n",
      "[[291 6.0 0.95 nan]\n",
      " [291 7.0 0.95 nan]] [6.0, 7.0] the enzyme activity retains more than 95% of its maximal activity within the pH range of 6-7 at 25°C\n",
      "[[291 6.0 0.95 nan]\n",
      " [291 7.0 0.95 nan]] [6.0, 7.0] the enzyme activity retains more than 95% of its maximal activity within the pH range of 6-7 at 25°C\n",
      "[[305 6.0 0.95 nan]\n",
      " [305 7.5 0.95 nan]\n",
      " [305 5.0 0.0 nan]] [6.0, 7.5] recombinant His-tagged enzyme\n",
      "[[321 3.2 0.5 'reaction with cellobiose and cytochrome c']\n",
      " [321 5.3 0.45 'reaction with cellobiose and cytochrome c']] [4.5, 7.5] two activity peaks, one at pH 3.5 and another at pH 4.5-4.75, the activity at pH 3.5 is 90% of the activity at pH 4.5-7.5, reaction with cellobiose and cytochrome c\n",
      "[[346 7.0 0.55\n",
      "  \"substrate: 2,2'-azino-bis(3-ethylbenzothiazoline-6-sulfonic acid)\"]\n",
      " [346 4.0 0.35\n",
      "  \"substrate: 2,2'-azino-bis(3-ethylbenzothiazoline-6-sulfonic acid)\"]] [7.0] substrate: 2,6-dimethoxyphenol\n",
      "[[353 6.0 0.75 'substrate: 2,6-dimethoxyphenol']\n",
      " [353 9.0 0.5 'substrate: 2,6-dimethoxyphenol']] [5.5] substrate: 2,2'-azino-bis(3-ethylbenzothiazoline-6-sulfonic acid)\n",
      "[[381 9.0 0.95 nan]\n",
      " [381 5.4 0.0 nan]] [9.0] -\n",
      "[[388 4.0 0.534 nan]\n",
      " [388 3.0 0.184 nan]] [5.0, 10.0] broad optimum\n",
      "[[388 4.0 0.534 nan]\n",
      " [388 3.0 0.184 nan]] [5.0, 10.0] broad optimum\n",
      "[[392 4.5 0.9 'above']\n",
      " [392 10.5 0.9 'above']] [10.5] -\n",
      "[[450 8.0 0.3 nan]\n",
      " [450 9.0 0.15 nan]] [6.0] -\n",
      "[[472 6.0 0.5 'within this range']\n",
      " [472 9.0 0.5 'within this range']] [6.0, 7.5] broad optimum\n",
      "[[479 10.0 0.1 'almost no activity at pH 4.5, i.e. a second optima']\n",
      " [479 8.0 0.7 'almost no activity at pH 4.5, i.e. a second optima']] [6.0, 6.5] -\n",
      "[[479 10.0 0.1 'almost no activity at pH 4.5, i.e. a second optima']\n",
      " [479 8.0 0.7 'almost no activity at pH 4.5, i.e. a second optima']] [6.0, 6.5] -\n",
      "[[582 6.0 0.5 'within this range']\n",
      " [582 9.0 0.5 'within this range']] [6.0, 7.5] broad optimum\n",
      "[[628 7.6 0.95 nan]\n",
      " [628 7.8 0.95 nan]] [7.0] -\n",
      "[[628 7.6 0.95 nan]\n",
      " [628 7.8 0.95 nan]] [7.0] broad optimum\n",
      "[[669 6.6 0.1 nan]\n",
      " [669 7.0 0.0 nan]] [6.0] activity assay, feruloyl-CoA as substrate\n",
      "[[762 7.0 0.7 nan]\n",
      " [762 8.5 0.7 nan]] [7.0] -\n",
      "[[810 6.5 0.35 'reduction of oxidized coenzyme F420 with NADPH']\n",
      " [810 4.5 0.45 'reduction of oxidized coenzyme F420 with NADPH']] [8.0] reduction of NADP+ with reduced coenzyme F420\n",
      "[[811 7.0 0.6 'reduction of NADP+ with reduced coenzyme F420']\n",
      " [811 9.0 0.5 'reduction of NADP+ with reduced coenzyme F420']] [5.5] reduction of oxidized coenzyme F420 with NADPH\n",
      "[[812 7.5 0.5 'reduction of NADP+ with reduced coenzyme F420']\n",
      " [812 10.0 0.9 'reduction of NADP+ with reduced coenzyme F420']] [5.5] pH optimum for reduction of oxidized coenzyme F420 with NADPH\n",
      "[[815 5.0 0.4 nan]\n",
      " [815 11.0 0.8 nan]] [4.0, 6.0] substrates: DELTA1-pyrroline-2-carboxylate + NADH + H+\n",
      "[[815 5.0 0.4 nan]\n",
      " [815 11.0 0.8 nan]] [11.0] substrates: L-proline + NAD+\n",
      "[[816 12.0 0.85 nan]\n",
      " [816 10.0 0.75 nan]] [4.0, 6.0] substrates: DELTA1-pyrroline-2-carboxylate + NADH + H+\n",
      "[[816 12.0 0.85 nan]\n",
      " [816 10.0 0.75 nan]] [4.0, 6.0] substrates: DELTA1-pyrroline-2-carboxylate + NADH + H+\n",
      "[[826 7.0 0.6 'substrate: spermine']\n",
      " [826 9.0 0.45 'substrate: spermine']] [6.5] substrate: thermospermine\n",
      "[[838 5.0 0.5 nan]\n",
      " [838 8.0 0.5 nan]] [8.0] -\n",
      "[[839 5.0 0.5 nan]\n",
      " [839 8.0 0.5 nan]] [8.0] -\n",
      "[[841 6.5 0.55 nan]\n",
      " [841 9.0 0.55 nan]] [6.5] -\n",
      "[[842 6.5 0.55 nan]\n",
      " [842 9.0 0.55 nan]] [6.5] -\n",
      "[[861 8.0 0.95 nan]\n",
      " [861 9.0 0.95 nan]] [7.0] quinone reductase activity\n",
      "[[873 6.0 0.95 'inferred']\n",
      " [873 7.0 0.95 'inferred']] [6.0] -\n",
      "[[903 4.5 0.7 nan]\n",
      " [903 5.5 0.35 nan]] [4.5, 5.0] -\n",
      "[[904 5.5 0.35 nan]\n",
      " [904 4.5 0.7 nan]] [4.5, 5.0] -\n",
      "[[960 7.5 0.95 nan]\n",
      " [960 9.0 0.95 nan]] [7.5] -\n",
      "[[979 5.0 0.8 'native enzyme,within this range']\n",
      " [979 6.0 0.8 'native enzyme,within this range']] [5.0] native enzyme\n",
      "[[979 5.0 0.8 'native enzyme,within this range']\n",
      " [979 6.0 0.8 'native enzyme,within this range']] [6.0] recombinant enzyme\n",
      "[[981 6.3 0.8 'recombinant enzyme,within this range']\n",
      " [981 5.4 0.8 'recombinant enzyme,within this range']] [5.0] native enzyme\n",
      "[[1023 1.0 0.0 nan]\n",
      " [1023 3.0 0.0 nan]] [7.5, 8.5] -\n",
      "[[1023 1.0 0.0 nan]\n",
      " [1023 3.0 0.0 nan]] [7.5, 8.5] -\n",
      "[[1024 9.5 0.05 'inferred']\n",
      " [1024 10.0 0.05 'inferred']] [7.5, 8.5] -\n",
      "[[1024 9.5 0.05 'inferred']\n",
      " [1024 10.0 0.05 'inferred']] [7.5, 8.5] -\n",
      "[[1032 6.5 0.9 nan]\n",
      " [1032 7.5 0.9 nan]\n",
      " [1032 8.0 0.5 nan]\n",
      " [1032 9.0 0.1 nan]] [6.5, 7.5] recombinant enzyme, broad optimum\n",
      "[[1053 10.0 1.0 'L-gamma-glutamyl-4-nitroanilide with diglycine acceptor']\n",
      " [1053 9.0 0.96 nan]\n",
      " [1053 8.0 0.926 nan]\n",
      " [1053 6.0 0.25 nan]] [10.0] -\n",
      "[[1069 8.5 0.8 'within this range']\n",
      " [1069 6.0 0.8 'within this range']] [8.0, 8.5] maximum activity\n",
      "[[1086 5.5 0.4 nan]\n",
      " [1086 7.5 0.4 nan]] [5.5] -\n",
      "[[1089 5.5 0.4 nan]\n",
      " [1089 7.5 0.4 nan]] [5.0] -\n",
      "[[1158 5.0 0.5 nan]\n",
      " [1158 7.0 0.5 nan]] [7.0, 7.5] -\n",
      "[[1158 5.0 0.5 nan]\n",
      " [1158 7.0 0.5 nan]] [7.0, 7.5] -\n",
      "[[1159 5.0 0.5 nan]\n",
      " [1159 7.0 0.5 nan]] [7.0, 7.5] -\n",
      "[[1159 5.0 0.5 nan]\n",
      " [1159 7.0 0.5 nan]] [7.0, 7.5] -\n",
      "[[1160 8.0 0.5 nan]\n",
      " [1160 6.5 0.5 nan]] [6.0, 6.5] -\n",
      "[[1160 8.0 0.5 nan]\n",
      " [1160 6.5 0.5 nan]] [6.0, 6.5] -\n",
      "[[1161 6.5 0.5 nan]\n",
      " [1161 8.0 0.5 nan]] [6.0, 6.5] -\n",
      "[[1161 6.5 0.5 nan]\n",
      " [1161 8.0 0.5 nan]] [6.0, 6.5] -\n",
      "[[1229 6.0 0.5 'glycosynthesis']\n",
      " [1229 7.0 0.5 'glycosynthesis']] [6.0] glycosynthesis\n",
      "[[1229 6.0 0.5 'glycosynthesis']\n",
      " [1229 7.0 0.5 'glycosynthesis']] [7.0, 7.5] phosphorolysis\n",
      "[[1229 6.0 0.5 'glycosynthesis']\n",
      " [1229 7.0 0.5 'glycosynthesis']] [7.0, 7.5] phosphorolysis\n",
      "[[1251 5.0 0.95 nan]\n",
      " [1251 6.0 0.95 nan]] [6.0] recombinant enzyme\n",
      "[[1298 8.5 1.0 nan]\n",
      " [1298 7.5 0.6 nan]] [8.5] -\n",
      "[[1365 8.0 0.5 'phosphoserine synthesis']\n",
      " [1365 9.0 0.6 'phosphoserine synthesis']] [9.0] phosphoserine synthesis\n",
      "[[1394 5.0 0.5 nan]\n",
      " [1394 7.0 0.65 nan]] [7.0] -\n",
      "[[1394 5.0 0.5 nan]\n",
      " [1394 7.0 0.65 nan]] [7.0] -\n",
      "[[1394 5.0 0.5 nan]\n",
      " [1394 7.0 0.65 nan]] [7.0] glucokinase activity\n",
      "[[1394 5.0 0.5 nan]\n",
      " [1394 7.0 0.65 nan]] [7.0] glucokinase activity\n",
      "[[1395 5.0 0.5 nan]\n",
      " [1395 7.0 0.65 nan]] [7.0] -\n",
      "[[1395 5.0 0.5 nan]\n",
      " [1395 7.0 0.65 nan]] [7.0] -\n",
      "[[1395 5.0 0.5 nan]\n",
      " [1395 7.0 0.65 nan]] [7.0] glucokinase activity\n",
      "[[1395 5.0 0.5 nan]\n",
      " [1395 7.0 0.65 nan]] [7.0] glucokinase activity\n",
      "[[1451 4.0 0.9 nan]\n",
      " [1451 10.0 0.9 nan]] [4.0, 10.0] -\n",
      "[[1451 4.0 0.9 nan]\n",
      " [1451 10.0 0.9 nan]] [4.0, 10.0] -\n",
      "[[1550 5.5 0.7 'maximal activity in 50 mM potassium phosphate buffer']\n",
      " [1550 8.0 0.2 'maximal activity in 50 mM potassium phosphate buffer']] [8.0] maximal activity in 50 mM Tris-HCl buffer\n",
      "[[1551 6.5 0.65 'potassium phosphate buffer']\n",
      " [1551 9.2 0.2 'potassium phosphate buffer']] [6.5] potassium phosphate buffer\n",
      "[[1552 6.5 0.65 'potassium phosphate buffer']\n",
      " [1552 9.2 0.2 'potassium phosphate buffer']] [6.5] potassium phosphate buffer\n",
      "[[1559 7.0 0.7 'in 50 mM Tris-HCl buffer']\n",
      " [1559 9.5 0.7 'in 50 mM Tris-HCl buffer']] [6.0] maximal activity in 50 mM potassium phosphate buffer\n",
      "[[1649 1.0 0.5 'inferred']\n",
      " [1649 3.0 0.5 'inferred']\n",
      " [1649 4.0 0.25 'inferred']\n",
      " [1649 7.0 0.01 nan]] [8.0] two pH optima, pH 2.0 and pH 8.0, are recorded\n",
      "[[1651 4.0 0.8 'recombinant enzyme, within this range']\n",
      " [1651 5.0 0.8 'recombinant enzyme, within this range']] [5.0] recombinant enzyme\n",
      "[[1677 5.0 0.55 nan]\n",
      " [1677 7.0 0.37 nan]] [9.0] -\n",
      "[[1717 3.0 0.96 nan]\n",
      " [1717 5.0 0.96 nan]] [3.0] -\n",
      "[[1731 6.0 0.6 'within this range']\n",
      " [1731 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1732 6.0 0.6 'within this range']\n",
      " [1732 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1733 6.0 0.6 'within this range']\n",
      " [1733 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1734 6.0 0.6 'within this range']\n",
      " [1734 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1735 7.0 0.6 'within this range']\n",
      " [1735 6.0 0.6 'within this range']] [7.0] -\n",
      "[[1736 6.0 0.6 'within this range']\n",
      " [1736 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1744 7.0 0.6 'within this range']\n",
      " [1744 8.0 0.6 'within this range']] [7.0] -\n",
      "[[1745 8.0 0.6 'within this range']\n",
      " [1745 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1746 7.0 0.6 'within this range']\n",
      " [1746 8.0 0.6 'within this range']] [7.0] -\n",
      "[[1747 7.0 0.6 'within this range']\n",
      " [1747 8.0 0.6 'within this range']] [7.0] -\n",
      "[[1748 7.0 0.6 'within this range']\n",
      " [1748 8.0 0.6 'within this range']] [7.0] -\n",
      "[[1749 8.0 0.6 'within this range']\n",
      " [1749 7.0 0.6 'within this range']] [7.0] -\n",
      "[[1768 4.1 0.0 nan]\n",
      " [1768 8.0 1.0 nan]\n",
      " [1768 4.5 0.1 nan]] [8.0] -\n",
      "[[1790 5.5 0.25 nan]\n",
      " [1790 6.5 0.25 nan]] [6.5] -\n",
      "[[1791 5.5 0.25 nan]\n",
      " [1791 6.5 0.25 nan]] [6.5] -\n",
      "[[1794 6.0 0.86 nan]\n",
      " [1794 9.0 0.86 nan]] [7.5, 9.0] -\n",
      "[[1880 6.0 0.5 nan]\n",
      " [1880 8.0 0.5 nan]] [6.0] -\n",
      "[[1922 7.1 0.85\n",
      "  'hydrolysis of 1-phosphatidyl-1D-myo-inositol 4,5-bisphosphate']\n",
      " [1922 7.6 0.65\n",
      "  'hydrolysis of 1-phosphatidyl-1D-myo-inositol 4,5-bisphosphate']] [8.0] hydrolysis of phosphatidylinositol\n",
      "[[1949 10.5 1.0 nan]\n",
      " [1949 7.5 0.0 nan]] [10.5] paraoxonase activity of recombinant PON1-hFc or PON1\n",
      "[[2031 4.0 0.85 'alpha-1,4-glycosidic linkage hydrolysis of maltotriose']\n",
      " [2031 6.0 0.6 'alpha-1,4-glycosidic linkage hydrolysis of maltotriose']] [6.0] alpha-1,6-glycosidic linkage hydrolysis\n",
      "[[2055 4.5 0.9 nan]\n",
      " [2055 6.0 0.9 nan]] [4.5] -\n",
      "[[2102 6.0 0.8 'within this range']\n",
      " [2102 4.0 0.8 'within this range']] [6.0] -\n",
      "[[2103 6.0 0.8 'within this range']\n",
      " [2103 4.0 0.8 'within this range']] [6.0] -\n",
      "[[2104 4.0 0.8 'within this range']\n",
      " [2104 6.0 0.8 'within this range']] [6.0] -\n",
      "[[2132 3.3 0.7 \"substrate: N,N',N''-triacetyl chitotrioside\"]\n",
      " [2132 4.7 0.5 \"substrate: N,N',N''-triacetyl chitotrioside\"]] [4.7] substrate: N,N',N''-triacetylchitotrioside\n",
      "[[2159 6.0 0.6 'alpha-1,4-glycosidic linkage hydrolysis of maltotriose']\n",
      " [2159 4.0 0.85 'alpha-1,4-glycosidic linkage hydrolysis of maltotriose']] [6.0] alpha-1,6-glycosidic linkage hydrolysis\n",
      "[[2163 1.0 0.45 nan]\n",
      " [2163 4.0 0.55 nan]] [3.5, 4.0] -\n",
      "[[2164 1.0 0.45 nan]\n",
      " [2164 4.0 0.55 nan]] [3.5, 4.0] -\n",
      "[[2165 1.0 0.45 nan]\n",
      " [2165 4.0 0.55 nan]] [3.5, 4.0] -\n",
      "[[2170 2.5 0.4 nan]\n",
      " [2170 5.5 0.4 nan]] [2.0] -\n",
      "[[2170 2.5 0.4 nan]\n",
      " [2170 5.5 0.4 nan]] [2.5, 3.0] -\n",
      "[[2171 2.5 0.4 nan]\n",
      " [2171 5.5 0.4 nan]] [2.0] -\n",
      "[[2171 2.5 0.4 nan]\n",
      " [2171 5.5 0.4 nan]] [2.5, 3.0] -\n",
      "[[2172 5.5 0.4 nan]\n",
      " [2172 2.5 0.4 nan]] [2.0] -\n",
      "[[2172 5.5 0.4 nan]\n",
      " [2172 2.5 0.4 nan]] [2.5, 3.0] -\n",
      "[[2175 3.5 0.8 'recombinant His-tagged enzyme, in this range']\n",
      " [2175 5.0 0.8 'recombinant His-tagged enzyme, in this range']] [3.5] recombinant His-tagged enzyme\n",
      "[[2189 5.0 0.8 nan]\n",
      " [2189 3.5 0.8 nan]] [3.5] -\n",
      "[[2194 5.0 0.7 'beta-glucosidase TpBGL3']\n",
      " [2194 3.0 0.65 'beta-glucosidase TpBGL3']] [6.0] beta-glucosidase TpBGL1\n",
      "[[2195 3.0 0.65 'beta-glucosidase TpBGL3']\n",
      " [2195 5.0 0.7 'beta-glucosidase TpBGL3']] [6.0] beta-glucosidase TpBGL1\n",
      "[[2206 4.0 0.45 'beta-glucosidase TpBGL1']\n",
      " [2206 8.0 0.6 'beta-glucosidase TpBGL1']] [4.0] beta-glucosidase TpBGL3\n",
      "[[2207 4.0 0.45 'beta-glucosidase TpBGL1']\n",
      " [2207 8.0 0.6 'beta-glucosidase TpBGL1']] [4.0] beta-glucosidase TpBGL3\n",
      "[[2219 6.5 0.6 'substrate: 4-nitrophenyl beta-D-glucopyranoside']\n",
      " [2219 5.0 0.3 'substrate: 4-nitrophenyl beta-D-glucopyranoside']] [5.0, 5.5] with 4-nitrophenyl beta-D-glucopyranoside as substrate\n",
      "[[2220 5.0 0.3 'substrate: 4-nitrophenyl beta-D-glucopyranoside']\n",
      " [2220 6.5 0.6 'substrate: 4-nitrophenyl beta-D-glucopyranoside']] [5.0, 5.5] with 4-nitrophenyl beta-D-glucopyranoside as substrate\n",
      "[[2221 5.0 0.3 'substrate: 4-nitrophenyl beta-D-glucopyranoside']\n",
      " [2221 6.5 0.6 'substrate: 4-nitrophenyl beta-D-glucopyranoside']] [5.0, 5.5] with 4-nitrophenyl beta-D-glucopyranoside as substrate\n",
      "[[2222 5.0 0.9 'substrate: 4-nitrophenyl beta-D-glucopyranoside']\n",
      " [2222 6.5 0.7 'substrate: 4-nitrophenyl beta-D-glucopyranoside']] [5.0, 5.5] with 4-nitrophenyl beta-D-glucopyranoside as substrate\n",
      "[[2223 5.0 0.9 'substrate: 4-nitrophenyl beta-D-glucopyranoside']\n",
      " [2223 6.5 0.7 'substrate: 4-nitrophenyl beta-D-glucopyranoside']] [5.0, 5.5] with 4-nitrophenyl beta-D-glucopyranoside as substrate\n",
      "[[2224 6.5 0.7 'substrate: 4-nitrophenyl beta-D-glucopyranoside']\n",
      " [2224 5.0 0.9 'substrate: 4-nitrophenyl beta-D-glucopyranoside']] [5.0, 5.5] with 4-nitrophenyl beta-D-glucopyranoside as substrate\n",
      "[[2232 7.0 0.8 nan]\n",
      " [2232 8.5 0.8 nan]] [7.0] -\n",
      "[[2250 6.5 0.9 'in the pH range of 6.5-8.5']\n",
      " [2250 8.5 0.9 'in the pH range of 6.5-8.5']] [6.5] -\n",
      "[[2255 2.0 0.9 'recombinant His-tagged isozyme betafruct2, % at pH 4.5']\n",
      " [2255 4.5 0.2 nan]] [4.0, 5.0] recombinant His-tagged isozyme betafruct3\n",
      "[[2256 4.5 0.2 'recombinant His-tagged isozyme betafruct']\n",
      " [2256 2.0 0.9 'recombinant His-tagged isozyme betafruct']] [4.0, 5.0] recombinant His-tagged isozyme betafruct3\n",
      "[[2257 2.0 0.9 'recombinant His-tagged isozyme betafruct2, % at pH 4.5']\n",
      " [2257 4.5 0.2 nan]] [4.0, 5.0] recombinant His-tagged isozyme betafruct3\n",
      "[[2261 3.5 0.5 'recombinant His-tagged isozyme betafruct3,']\n",
      " [2261 6.0 0.5 'recombinant His-tagged isozyme betafruct3,']] [3.0] recombinant His-tagged isozyme betafruct2\n",
      "[[2262 3.5 0.5 'recombinant His-tagged isozyme betafruct3,']\n",
      " [2262 6.0 0.5 'recombinant His-tagged isozyme betafruct3,']] [3.0] recombinant His-tagged isozyme betafruct2\n",
      "[[2263 3.5 0.5 'recombinant His-tagged isozyme betafruct3,']\n",
      " [2263 6.0 0.5 'recombinant His-tagged isozyme betafruct3,']] [3.0] recombinant His-tagged isozyme betafruct2\n",
      "[[2270 6.0 0.5 nan]\n",
      " [2270 9.0 0.7 nan]] [6.0, 7.5] -\n",
      "[[2318 7.0 0.8 'within']\n",
      " [2318 6.0 0.8 'within']] [7.0] substrate 4-nitrophenyl alpha-L-arabinofuranoside\n",
      "[[2322 7.0 0.6 nan]\n",
      " [2322 9.0 0.6 nan]] [7.0] -\n",
      "[[2356 4.5 0.45\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']\n",
      " [2356 3.5 0.7\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']] [3.0] chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA\n",
      "[[2356 4.5 0.45\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']\n",
      " [2356 3.5 0.7\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']] [3.0] hybrid protein (CelA-SSO1949-CelA) based on beta-endoglucanase from Sulfolobus solfataricus P2 and beta-endoglucanase from Thermotoga maritima\n",
      "[[2357 4.5 0.45\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']\n",
      " [2357 3.5 0.7\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']] [3.0] chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA\n",
      "[[2357 4.5 0.45\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']\n",
      " [2357 3.5 0.7\n",
      "  'chimeric fusion protein CelA-SSO1949-CelA, that consists of 70 amino acids CelA followed by 163 amino acids SSO1949 and 29 amino acids CelA']] [4.5] native enzyme\n",
      "[[2392 4.0 0.5 nan]\n",
      " [2392 7.9 0.5 nan]] [7.9] -\n",
      "[[2393 4.0 0.5 nan]\n",
      " [2393 7.9 0.5 nan]] [7.9] -\n",
      "[[2402 5.0 0.5 nan]\n",
      " [2402 8.7 0.5 nan]] [5.0, 6.9] -\n",
      "[[2403 8.7 0.5 nan]\n",
      " [2403 5.0 0.5 nan]] [5.0, 6.9] -\n",
      "[[2444 4.0 0.8 nan]\n",
      " [2444 6.0 0.8 nan]] [8.0] -\n",
      "[[2510 5.5 0.7 'within']\n",
      " [2510 1.0 0.7 'within']] [11.5] -\n",
      "[[2589 7.0 0.7 'recombinant enzyme, within this range']\n",
      " [2589 9.0 0.7 'recombinant enzyme, within this range']] [6.5, 8.5] recombinant enzyme\n",
      "[[2597 5.0 0.8 'isoform 1-FEH I shows']\n",
      " [2597 6.0 0.8 'isoform 1-FEH I shows']] [6.0] -\n",
      "[[2598 6.0 0.8 'isoform 1-FEH I shows']\n",
      " [2598 5.0 0.8 'isoform 1-FEH I shows']] [6.0] -\n",
      "[[2610 4.0 0.8 'in comparison to the activity at ph 4.0']\n",
      " [2610 7.0 0.8 'in comparison to the activity at ph 4.0']] [4.0] -\n",
      "[[2618 5.0 0.5 nan]\n",
      " [2618 7.0 0.5 nan]] [5.0] -\n",
      "[[2618 5.0 0.5 nan]\n",
      " [2618 7.0 0.5 nan]] [5.0] pH optimum using Avicel or cellotetraose as substrates\n",
      "[[2620 5.0 0.3\n",
      "  'optimum at pH 5, drop of activity between pH 6 and pH 7 to']\n",
      " [2620 7.0 0.3\n",
      "  'optimum at pH 5, drop of activity between pH 6 and pH 7 to']] [5.0] -\n",
      "[[2625 6.0 0.8 'retains']\n",
      " [2625 9.0 0.8 'retains']] [6.0] optimum pH of the enzyme for Galbeta1-3GalNAcalpha1-pNP hydrolysis is at pH 6.0 and 9.0, whereas optimum pH for other substrates hydrolyzed is around pH 6.0\n",
      "[[2645 5.5 0.8 nan]\n",
      " [2645 6.5 0.6 nan]] [5.5] -\n",
      "[[2646 6.0 0.6 'alpha-1,4-glycosidic linkage hydrolysis of maltotriose']\n",
      " [2646 4.0 0.85 'alpha-1,4-glycosidic linkage hydrolysis of maltotriose']] [6.0] alpha-1,6-glycosidic linkage hydrolysis\n",
      "[[2663 8.0 0.45 'native enzyme,, inactive at pH 5.0 and 9.0']\n",
      " [2663 6.0 0.45 'native enzyme,, inactive at pH 5.0 and 9.0']] [8.0] recombinant enzyme\n",
      "[[2670 7.5 0.4 'recombinant enzyme,']\n",
      " [2670 9.0 0.75 'recombinant enzyme,']] [7.0] native enzyme\n",
      "[[2767 7.5 0.55 nan]\n",
      " [2767 8.0 0.75 nan]] [7.5] -\n",
      "[[2801 7.0 0.9 'within this range, profile overview']\n",
      " [2801 9.0 0.9 'within this range, profile overview']] [9.0] -\n",
      "[[2819 4.0 0.5\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B2']\n",
      " [2819 6.0 0.7\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B2']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2820 4.0 0.5\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B2']\n",
      " [2820 6.0 0.7\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B2']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2821 4.0 0.5\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B2']\n",
      " [2821 6.0 0.7\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B2']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2822 6.0 0.8\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B3']\n",
      " [2822 4.0 0.55\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B3']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2823 4.0 0.55\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B3']\n",
      " [2823 6.0 0.8\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B3']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2824 4.0 0.55\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B3']\n",
      " [2824 6.0 0.8\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B3']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2825 4.0 0.6\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4']\n",
      " [2825 6.0 0.5\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2826 4.0 0.6\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4']\n",
      " [2826 6.0 0.5\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2827 6.0 0.5\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4']\n",
      " [2827 4.0 0.6\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2828 4.0 0.7\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B1']\n",
      " [2828 6.0 0.8\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B1']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2829 4.0 0.7\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B1']\n",
      " [2829 6.0 0.8\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B1']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2830 4.0 0.7\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B1']\n",
      " [2830 6.0 0.8\n",
      "  'substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B1']] [4.5, 6.0] substrate: benzyloxycarbonyl-Phe-Arg-7-amido-4-methylcoumarin, cathepsin B4\n",
      "[[2928 5.0 0.0 'about 20% activity at pH 2.0, 100% activity at pH 3.0,']\n",
      " [2928 4.0 0.8 'about 20% activity at pH 2.0, 100% activity at pH 3.0,']] [3.0] -\n",
      "[[3001 8.0 0.5 '112 kDa CDase from kidney,in the range']\n",
      " [3001 9.0 0.5 '112 kDa CDase from kidney,in the range']] [6.0, 7.0] 112 kDa CDase from kidney\n",
      "[[3001 8.0 0.5 '112 kDa CDase from kidney,in the range']\n",
      " [3001 9.0 0.5 '112 kDa CDase from kidney,in the range']] [6.0, 7.0] 112 kDa CDase from kidney\n",
      "[[3019 5.0 0.7 nan]\n",
      " [3019 9.0 0.7 nan]] [9.0] -\n",
      "[[3092 6.0 0.8 nan]\n",
      " [3092 8.0 0.8 nan]] [8.0] -\n",
      "[[3105 8.3 0.65 'pH: pH: recombinant NADP-ME3']\n",
      " [3105 7.3 0.65 'pH: pH: recombinant NADP-ME3']] [7.0] soluble and enzyme ecapsulated in ethyleneamine-mediated biosilica\n",
      "[[3151 7.5 0.6 nan]\n",
      " [3151 9.0 0.9 nan]] [8.5, 9.0] -\n",
      "[[3152 7.5 0.6 nan]\n",
      " [3152 9.0 0.9 nan]] [8.5, 9.0] -\n",
      "[[3208 8.0 0.3 nan]\n",
      " [3208 5.5 1.0 nan]] [5.5] recombinant enzyme\n",
      "[[3209 8.8 0.3 'soluble wild-type enzyme']\n",
      " [3209 6.2 0.95 'soluble wild-type enzyme']\n",
      " [3209 8.0 0.95 'soluble wild-type enzyme']] [6.2, 8.0] broad optimum, soluble wild-type enzyme\n",
      "[[3271 9.5 0.75 'retro-aldol addition']\n",
      " [3271 6.0 0.65 'retro-aldol addition']] [6.0] aldol addition\n",
      "[[3348 4.0 0.6 nan]\n",
      " [3348 6.0 0.6 nan]] [7.0] -\n",
      "[[3478 7.0 0.8\n",
      "  'substrate: apo-[phycoerythrocyanin beta-subunit] mutant C155']\n",
      " [3478 7.0 0.8\n",
      "  'substrate: apo-[phycoerythrocyanin beta-subunit] mutant C155']] [7.0, 9.0] substrate: apo-[C-phycocyanin beta-subunit]\n",
      "[[3478 7.0 0.8\n",
      "  'substrate: apo-[phycoerythrocyanin beta-subunit] mutant C155']\n",
      " [3478 7.0 0.8\n",
      "  'substrate: apo-[phycoerythrocyanin beta-subunit] mutant C155']] [7.0, 9.0] substrate: apo-[C-phycocyanin beta-subunit]\n",
      "[[3571 5.0 0.6 nan]\n",
      " [3571 7.5 0.6 nan]] [7.5] -\n",
      "[[3596 8.0 0.9 nan]\n",
      " [3596 9.0 0.9 nan]] [7.5] -\n",
      "[[3611 4.0 0.5 nan]\n",
      " [3611 6.0 0.5 nan]] [6.0] -\n",
      "[[3613 7.0 0.22 '2% and, respectively, at 1 mM Mn2+']\n",
      " [3613 6.0 0.22 '2% and, respectively, at 1 mM Mn2+']] [8.0] -\n",
      "[[3666 8.0 0.95 nan]\n",
      " [3666 9.0 0.95 nan]\n",
      " [3666 7.0 0.8 nan]] [8.0, 9.0] -\n",
      "[[3667 6.0 0.6 'at pH 6.0-7.0, the recombinant enzyme shows 15–']\n",
      " [3667 7.0 0.6 'at pH 6.0-7.0, the recombinant enzyme shows 15–']] [6.0] forward reaction\n",
      "[[3667 6.0 0.6 'at pH 6.0-7.0, the recombinant enzyme shows 15–']\n",
      " [3667 7.0 0.6 'at pH 6.0-7.0, the recombinant enzyme shows 15–']] [7.0] reverse reaction\n",
      "[[3667 6.0 0.6 'at pH 6.0-7.0, the recombinant enzyme shows 15–']\n",
      " [3667 7.0 0.6 'at pH 6.0-7.0, the recombinant enzyme shows 15–']] [8.0] -\n",
      "[[3669 7.0 0.2 'at pH 9,  at pH 9']\n",
      " [3669 8.0 0.65 'at pH 9,  at pH 9']] [9.0] or above. At a pH of above 9 the coupled spectrophotometric assay becomes limited due to instability of the substrates\n",
      "[[3673 7.0 0.2 'at pH 9,  at pH 9']\n",
      " [3673 8.0 0.65 'at pH 9,  at pH 9']] [9.0] or above. At a pH of above 9 the coupled spectrophotometric assay becomes limited due to instability of the substrates\n",
      "[[3744 11.5 0.85 'ethanol oxidation enzyme form ADH-3']\n",
      " [3744 9.5 0.4 'ethanol oxidation enzyme form ADH-3']] [6.5] reduction of octanal\n",
      "[[3745 11.0 0.75 'pyrazole-insensitive enzyme']\n",
      " [3745 10.0 0.5 'pyrazole-insensitive enzyme']] [11.2] oxidation of ethanol, pyrazole-sensitive enzyme form\n",
      "[[3754 6.0 0.5 'above and below, sugar reduction']\n",
      " [3754 7.2 0.5 'above and below, sugar reduction']] [6.0, 7.2] D-fructose reduction\n",
      "[[3754 6.0 0.5 'above and below, sugar reduction']\n",
      " [3754 7.2 0.5 'above and below, sugar reduction']] [6.0, 7.2] D-fructose reduction\n",
      "[[3754 6.0 0.5 'above and below, sugar reduction']\n",
      " [3754 7.2 0.5 'above and below, sugar reduction']] [11.0] sorbitol oxidation\n",
      "[[3755 9.2 0.5 'above and below, sorbitol oxidation']\n",
      " [3755 11.6 0.5 'above and below, sorbitol oxidation']] [6.0, 7.2] D-fructose reduction\n",
      "[[3755 9.2 0.5 'above and below, sorbitol oxidation']\n",
      " [3755 11.6 0.5 'above and below, sorbitol oxidation']] [6.0, 7.2] D-fructose reduction\n",
      "[[3795 6.5 0.5 nan]\n",
      " [3795 8.0 0.5 nan]\n",
      " [3795 4.5 0.0 nan]\n",
      " [3795 7.5 1.0 nan]] [9.8] -\n",
      "[[3801 5.8 0.5 ', formation of (7S)-salutaridinol']\n",
      " [3801 7.8 0.5 ', formation of (7S)-salutaridinol']] [9.0, 9.5] formation of salutaridine\n",
      "[[3801 5.8 0.5 ', formation of (7S)-salutaridinol']\n",
      " [3801 7.8 0.5 ', formation of (7S)-salutaridinol']] [9.0, 9.5] formation of salutaridine\n",
      "[[3802 5.8 0.5 ', formationof salutaridine']\n",
      " [3802 7.8 0.5 ', formationof salutaridine']] [9.0, 9.5] formation of salutaridine\n",
      "[[3802 5.8 0.5 ', formationof salutaridine']\n",
      " [3802 7.8 0.5 ', formationof salutaridine']] [9.0, 9.5] formation of salutaridine\n",
      "[[3804 8.5 0.6 nan]\n",
      " [3804 8.0 0.4 nan]\n",
      " [3804 10.0 0.95 nan]\n",
      " [3804 11.0 0.95 nan]] [10.0, 11.0] oxidation of D-arabinitol\n",
      "[[3805 4.2 0.4 nan]\n",
      " [3805 6.5 0.3 nan]] [7.5] oxidation of scytalone\n",
      "[[3806 7.0 0.95 nan]\n",
      " [3806 8.0 0.8 nan]] [5.5] reduction of 1,3,6,8-tetrahydroxynaphthalene\n",
      "[[3811 8.0 0.6 'reduction of pyruvate']\n",
      " [3811 5.3 0.5 'reduction of pyruvate']] [9.8] oxidation of lactate\n",
      "[[3812 8.5 0.6 'oxidation of lactate']\n",
      " [3812 10.7 0.7 'oxidation of lactate']] [7.0] reduction of pyruvate\n",
      "[[3822 7.5 0.5 nan]\n",
      " [3822 4.5 0.55 nan]] [8.9] xylitol oxidation\n",
      "[[3849 5.5 0.3 'isoform ICD-1,isoform ICD-2,']\n",
      " [3849 5.5 0.1 'isoform ICD-1,isoform ICD-2,']] [7.5] both isoforms ICD-1, ICD-2\n",
      "[[3850 5.5 0.1 'isoform ICD-1,isoform ICD-2,']\n",
      " [3850 5.5 0.3 'isoform ICD-1,isoform ICD-2,']] [7.5] both isoforms ICD-1, ICD-2\n",
      "[[3855 9.5 0.9 'isoform ICD-1,isoform ICD-2,']\n",
      " [3855 9.5 0.4 'isoform ICD-1,isoform ICD-2,']] [7.5] both isoforms ICD-1, ICD-2\n",
      "[[3856 9.5 0.9 'isoform ICD-1,isoform ICD-2,']\n",
      " [3856 9.5 0.4 'isoform ICD-1,isoform ICD-2,']] [7.5] both isoforms ICD-1, ICD-2\n",
      "[[3874 5.5 0.7 'reduction of dihydroxyacetone']\n",
      " [3874 9.0 0.45 'reduction of dihydroxyacetone']] [5.0, 6.0] reduction of dihydroxyacetone\n",
      "[[3874 5.5 0.7 'reduction of dihydroxyacetone']\n",
      " [3874 9.0 0.45 'reduction of dihydroxyacetone']] [11.0, 12.0] oxidation of glycerol\n",
      "[[3874 5.5 0.7 'reduction of dihydroxyacetone']\n",
      " [3874 9.0 0.45 'reduction of dihydroxyacetone']] [11.0, 12.0] oxidation of glycerol\n",
      "[[3875 9.0 0.45 'oxidation of glycerol']\n",
      " [3875 11.0 0.95 'oxidation of glycerol']\n",
      " [3875 12.0 0.95 'oxidation of glycerol']] [5.0, 6.0] reduction of dihydroxyacetone\n",
      "[[3875 9.0 0.45 'oxidation of glycerol']\n",
      " [3875 11.0 0.95 'oxidation of glycerol']\n",
      " [3875 12.0 0.95 'oxidation of glycerol']] [5.0, 6.0] reduction of dihydroxyacetone\n",
      "[[3875 9.0 0.45 'oxidation of glycerol']\n",
      " [3875 11.0 0.95 'oxidation of glycerol']\n",
      " [3875 12.0 0.95 'oxidation of glycerol']] [11.0, 12.0] oxidation of glycerol\n",
      "[[4033 8.0 0.7 'NADH-dehydrogenase']\n",
      " [4033 7.0 0.7 'NADH-dehydrogenase']] [5.0] NADH-dehydrogenase\n",
      "[[4034 8.0 0.7 'NADH-dehydrogenase']\n",
      " [4034 7.0 0.7 'NADH-dehydrogenase']] [5.0] NADH-dehydrogenase\n",
      "[[4126 7.0 0.9 'methylmalonate semialdehyde oxidation, above and below']\n",
      " [4126 8.5 0.9 'methylmalonate semialdehyde oxidation, above and below']] [9.0] propanal + NAD+\n",
      "[[4128 8.0 0.5 'propanal oxidation, above and below']\n",
      " [4128 9.5 0.5 'propanal oxidation, above and below']] [8.0] 2-methyl-3-oxopropanoate + CoA + NAD+\n",
      "[[4270 5.5 0.5 ', tauropine formation, enzyme form TaDH-1']\n",
      " [4270 7.9 0.5 ', tauropine formation, enzyme form TaDH-1']] [8.5] tauropine oxidation\n",
      "[[4271 6.8 0.5 'tauropine oxidation, enzyme form TaDH-1']\n",
      " [4271 8.5 1.0 'tauropine oxidation, enzyme form TaDH-1']] [6.6, 7.3] tauropine formation, enzyme form TaDH-1\n",
      "[[4271 6.8 0.5 'tauropine oxidation, enzyme form TaDH-1']\n",
      " [4271 8.5 1.0 'tauropine oxidation, enzyme form TaDH-1']] [8.5] tauropine oxidation\n",
      "[[4409 6.0 0.6 'cytosol: about 70% at 8.5, mitochondria: and 8.5']\n",
      " [4409 6.0 0.4 'cytosol: about 70% at 8.5, mitochondria: and 8.5']] [7.3] -\n",
      "[[4410 6.0 0.6 'cytosol: about 70% at 8.5, mitochondria: and 8.5']\n",
      " [4410 6.0 0.4 'cytosol: about 70% at 8.5, mitochondria: and 8.5']] [7.3] -\n",
      "[[4567 7.6 0.95 'pH 7.3-7.no activity below pH 6.7']\n",
      " [4567 7.3 0.95 'pH 7.3-7.no activity below pH 6.7']\n",
      " [4567 6.6 0.0 'pH 7.3-7.no activity below pH 6.7']] [7.3, 7.6] -\n",
      "[[4568 8.2 0.5 'and sucrose cleavage']\n",
      " [4568 5.0 0.9 'and sucrose cleavage']] [9.5] sucrose synthesis\n",
      "[[4571 10.0 0.9 'and sucrose synthesis']\n",
      " [4571 7.5 0.5 'and sucrose synthesis']] [6.0] sucrose cleavage\n",
      "[[4586 6.5 0.6 'more than 90% between pH 8.0, and 11.0']\n",
      " [4586 7.0 0.8 'more than 90% between pH 8.0, and 11.0']\n",
      " [4586 7.5 1.0 'more than 90% between pH 8.0, and 11.0']] [7.5] -\n",
      "[[4587 6.5 0.6 'more than 90% between pH 8.0, and 11.0']\n",
      " [4587 7.0 0.8 'more than 90% between pH 8.0, and 11.0']\n",
      " [4587 7.5 1.0 'more than 90% between pH 8.0, and 11.0']] [7.5] -\n",
      "[[4645 7.3 0.55 nan]\n",
      " [4645 6.4 0.5 nan]] [6.2] -\n",
      "[[4671 7.2 0.5 'maximum,  maximum']\n",
      " [4671 8.0 0.7 'maximum,  maximum']] [7.5, 8.0] -\n",
      "[[4782 6.0 0.8 nan]\n",
      " [4782 9.5 0.8 nan]] [10.0] -\n",
      "[[4824 5.0 0.6 'at least']\n",
      " [4824 7.0 0.6 'at least']] [8.0, 8.5] -\n",
      "[[4824 5.0 0.6 'at least']\n",
      " [4824 7.0 0.6 'at least']] [8.0, 8.5] -\n",
      "[[4839 5.0 0.5 ', pyridoxine as substrate']\n",
      " [4839 6.0 0.5 ', pyridoxine as substrate']] [6.0] substrate 2-methyl-4-amino-5-hydroxymethylpyrimidine\n",
      "[[4864 7.5 0.5 'maximum, pH 8.4 and  maximum']\n",
      " [4864 9.4 0.85 'maximum, pH 8.4 and  maximum']] [5.5] substrate DNA\n",
      "[[4865 7.5 0.5 'maximum, pH 8.4 and  maximum']\n",
      " [4865 9.4 0.85 'maximum, pH 8.4 and  maximum']] [5.5] substrate DNA\n",
      "[[4905 5.0 0.3 'ATP synthesis']\n",
      " [4905 8.2 0.3 'ATP synthesis']] [8.3] synthesis of N-phospho-L-arginine\n",
      "[[4906 5.5 0.9 'no activity at pH 5.0, synthesis of ATP']\n",
      " [4906 8.0 0.7 'no activity at pH 5.0, synthesis of ATP']] [9.0] synthesis of N-phospho-L-arginine\n",
      "[[4908 8.0 0.6 'synthesis of N-phospho-L-Arg']\n",
      " [4908 10.0 0.7 'synthesis of N-phospho-L-Arg']] [6.1] synthesis of ATP\n",
      "[[4924 5.2 0.5 nan]\n",
      " [4924 7.5 0.5 nan]] [7.5] -\n",
      "[[4925 5.7 0.5 ', dGMP-phosphorylation, calf thymus']\n",
      " [4925 10.9 0.5 ', dGMP-phosphorylation, calf thymus']] [7.0, 10.9] GMP-phosphorylation\n",
      "[[4928 5.0 0.45 nan]\n",
      " [4928 6.0 0.95 nan]\n",
      " [4928 9.0 0.95 nan]] [6.0, 9.0] -\n",
      "[[5017 6.0 0.75 nan]\n",
      " [5017 7.0 1.0 nan]] [7.0] with substrate 4-nitrophenyl 2-O-(methyl 4-O-methyl-alpha-D-glucopyranosyluronate)-beta-D-xylopyranoside\n",
      "[[5022 6.5 0.5 nan]\n",
      " [5022 5.9 0.0 nan]\n",
      " [5022 7.5 0.95 nan]\n",
      " [5022 9.0 0.95 nan]] [7.5, 9.0] -\n",
      "[[5040 5.8 0.5 nan]\n",
      " [5040 8.0 0.6 nan]] [5.5, 8.0] -\n",
      "[[5040 5.8 0.5 nan]\n",
      " [5040 8.0 0.6 nan]] [5.5, 8.0] -\n",
      "[[5041 5.8 0.5 nan]\n",
      " [5041 8.0 0.6 nan]] [5.5, 8.0] -\n",
      "[[5041 5.8 0.5 nan]\n",
      " [5041 8.0 0.6 nan]] [5.5, 8.0] -\n",
      "[[5101 6.8 0.6 'enzyme form H-2']\n",
      " [5101 8.4 0.35 'enzyme form H-2']] [8.4] enzyme form H-1\n",
      "[[5102 6.8 0.6 'enzyme form H-2']\n",
      " [5102 8.4 0.35 'enzyme form H-2']] [8.4] enzyme form H-1\n",
      "[[5103 7.5 0.45 ', enzyme form H-1']\n",
      " [5103 9.0 0.45 ', enzyme form H-1']] [7.5] enzyme form H-2\n",
      "[[5104 7.5 0.45 ', enzyme form H-1']\n",
      " [5104 9.0 0.45 ', enzyme form H-1']] [7.5] enzyme form H-2\n",
      "[[5176 5.0 0.2 'enzyme from animals without infection']\n",
      " [5176 7.0 0.55 'enzyme from animals without infection']] [7.0] enzyme from animals infected with Schistosoma mansoni\n",
      "[[5185 7.5 0.1 'below']\n",
      " [5185 6.0 0.1 'below']] [8.5] -\n",
      "[[5229 5.5 0.5 ', in the presence of soybean phosphatidylinositol']\n",
      " [5229 7.5 0.5 ', in the presence of soybean phosphatidylinositol']] [5.5, 6.5] in the presence of egg yolk lyso-phosphatidylcholine\n",
      "[[5925 8.8 0.95 'hydrolysis of 2-hydroxy-dATP']\n",
      " [5925 7.2 0.6 'hydrolysis of 2-hydroxy-dATP']] [7.7, 8.8] hydrolysis of 2-hydroxy-dATP\n",
      "[[5926 7.2 0.15 'hydrolysis of 8-oxo-dGTP']\n",
      " [5926 7.7 0.95 'hydrolysis of 8-oxo-dGTP']\n",
      " [5926 8.8 0.5 'hydrolysis of 8-oxo-dGTP']] [7.7, 8.8] hydrolysis of 2-hydroxy-dATP\n",
      "[[6069 6.8 0.5 'wild-type enzyme, above and below']\n",
      " [6069 8.0 0.5 'wild-type enzyme, above and below']] [8.0] enzyme expressed in Escherichia coli\n",
      "[[6194 3.3 0.95 nan]\n",
      " [6194 4.4 0.95 nan]\n",
      " [6194 5.2 0.75 nan]\n",
      " [6194 5.7 0.0 nan]] [3.0, 4.4] -\n",
      "[[6221 6.5 0.1 nan]\n",
      " [6221 10.0 0.95 nan]\n",
      " [6221 8.0 0.95 nan]] [8.0, 10.0] -\n",
      "[[6233 10.0 0.3 nan]\n",
      " [6233 6.0 0.35 nan]] [8.0, 10.0] -\n",
      "[[6241 6.3 0.5 nan]\n",
      " [6241 7.0 0.5 nan]] [6.3] -\n",
      "[[6264 6.0 0.24\n",
      "  '24% and in the presence of 1 mM and 0.5 mM Mn2+, respectively']\n",
      " [6264 7.0 0.36\n",
      "  '24% and in the presence of 1 mM and 0.5 mM Mn2+, respectively']] [8.0] -\n",
      "[[6265 6.0 0.03\n",
      "  '3.1% and in the presence of 1 mM and 0.5 mM Mn2+, respectively']\n",
      " [6265 7.0 0.21\n",
      "  '3.1% and in the presence of 1 mM and 0.5 mM Mn2+, respectively']] [8.0] -\n",
      "[[6335 7.0 0.1 '8.4-8.9: maximal activity']\n",
      " [6335 8.0 0.75 '8.4-8.9: maximal activity']] [8.4, 8.9] -\n",
      "[[6335 7.0 0.1 '8.4-8.9: maximal activity']\n",
      " [6335 8.0 0.75 '8.4-8.9: maximal activity']] [8.4, 8.9] -\n",
      "[[6350 7.4 0.9 nan]\n",
      " [6350 8.1 0.9 nan]] [4.7] -\n",
      "[[6361 5.5 0.6 nan]\n",
      " [6361 7.0 0.6 nan]] [7.5, 7.8] reaction of ATP + citrulline + L-argininosuccinate\n",
      "[[6361 5.5 0.6 nan]\n",
      " [6361 7.0 0.6 nan]] [7.5, 7.8] reaction of ATP + citrulline + L-argininosuccinate\n",
      "[[6362 6.5 0.55 'reaction of ATP + citrulline + L-argininosuccinate']\n",
      " [6362 8.5 0.7 'reaction of ATP + citrulline + L-argininosuccinate']] [6.0, 6.5] reaction of AMP + diphosphate + L-argininosuccinate\n",
      "[[6362 6.5 0.55 'reaction of ATP + citrulline + L-argininosuccinate']\n",
      " [6362 8.5 0.7 'reaction of ATP + citrulline + L-argininosuccinate']] [6.0, 6.5] reaction of AMP + diphosphate + L-argininosuccinate\n",
      "[[6364 7.8 0.4 'with NH4+ as amide donor']\n",
      " [6364 9.3 0.5 'with NH4+ as amide donor']] [6.2, 8.4] broad peak of activity with a maximum around 7.6, with Gln as the amide donor\n",
      "[[6378 7.2 0.95 nan]\n",
      " [6378 7.7 0.95 nan]\n",
      " [6378 8.4 0.5 nan]] [7.2, 7.7] in Tris-HCl buffer\n",
      "len(matched_data) = 3242, len(opt_not_in_range) = 336, matched_more_than_2_count = 434\n"
     ]
    }
   ],
   "source": [
    "# Match the ph optimum to the correct ph range\n",
    "opt_not_in_range = []\n",
    "matched_data = []\n",
    "matched_more_than_2_count = 0\n",
    "for exp_idx, df in data_act_ph_range.groupby(\"exp_index\"):\n",
    "    note = df.note.unique()[0]\n",
    "    row = data_ph_range.loc[exp_idx]\n",
    "    df2 = data_opt_match[(data_opt_match.ref == row.ref) & \n",
    "                         (data_opt_match.uniprot_id == row.uniprot_id) & \n",
    "                         (data_opt_match.ec_num == row.ec_num)]\n",
    "    if len(df2) == 0:\n",
    "        continue\n",
    "    if len(df2) > 1:\n",
    "        matched_more_than_2_count += 1\n",
    "    #     # skip for now\n",
    "    #     continue\n",
    "    # ph_opts = [ph for phs in df2.opt.values for ph in phs.split('-')]\n",
    "    # ph_opts = list(df2.ph_min.values)\n",
    "    # ph_opts += [ph for ph in df2.ph_max.values if ph != '-']\n",
    "    for ph_min, ph_max, note in df2[['ph_min', 'ph_max', 'comments']].values:\n",
    "        ph_opts = [float(ph_min)]\n",
    "        if ph_max != '-':\n",
    "            ph_opts += [float(ph_max)]\n",
    "        for ph_opt in ph_opts:\n",
    "            if len(df) == 1 or (\n",
    "                df.ph.min() < ph_opt and df.ph.max() > ph_opt):\n",
    "                # ph_opt is in range. Add to data\n",
    "                matched_data += [[exp_idx, ph_opt, 1.0, note, row.uniprot_id]]\n",
    "            else:\n",
    "                print(df[['exp_index', 'ph', 'activity', 'note']].values, ph_opts, note)\n",
    "                opt_not_in_range.append((exp_idx, ph_opts))\n",
    "                continue\n",
    "print(f\"{len(matched_data) = }, {len(opt_not_in_range) = }, {matched_more_than_2_count = }\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "f97f7658-f3ba-47ee-9f4f-2b69c366f420",
   "metadata": {},
   "outputs": [],
   "source": [
    "# some manual match fixes\n",
    "exp_idx_matches_to_fix = [6335, 6265, 6264]\n",
    "for exp_idx in exp_idx_matches_to_fix:\n",
    "    row = data_ph_range.loc[exp_idx]\n",
    "    df2 = data_opt_match[(data_opt_match.ref == row.ref) & \n",
    "                         (data_opt_match.uniprot_id == row.uniprot_id) & \n",
    "                         (data_opt_match.ec_num == row.ec_num)]\n",
    "    for ph_min, ph_max, note in df2[['ph_min', 'ph_max', 'comments']].values:\n",
    "        ph_opts = [float(ph_min)]\n",
    "        if ph_max != '-':\n",
    "            ph_opts += [float(ph_max)]\n",
    "        for ph_opt in ph_opts:\n",
    "            matched_data += [[exp_idx, ph_opt, 1.0, note, row.uniprot_id]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "43134360-1daa-4c2f-b760-ef0c0b77d646",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3246\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exp_index</th>\n",
       "      <th>ph</th>\n",
       "      <th>activity</th>\n",
       "      <th>note</th>\n",
       "      <th>uniprot_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>reduction of ketones</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>reduction of ketones</td>\n",
       "      <td>Q4J9F2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>Q4J9F2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3241</th>\n",
       "      <td>6393</td>\n",
       "      <td>6.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>P38929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3242</th>\n",
       "      <td>6335</td>\n",
       "      <td>8.4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Q53005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3243</th>\n",
       "      <td>6335</td>\n",
       "      <td>8.9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Q53005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3244</th>\n",
       "      <td>6265</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Q49W02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3245</th>\n",
       "      <td>6264</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>P39773</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3246 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      exp_index   ph  activity                  note uniprot_id\n",
       "0             1  4.5       1.0                     -     B2KJ46\n",
       "1             3  5.1       1.0  reduction of ketones     Q4J702\n",
       "2             3  5.1       1.0    reduction reaction     Q4J702\n",
       "3             4  5.1       1.0  reduction of ketones     Q4J9F2\n",
       "4             4  5.1       1.0    reduction reaction     Q4J9F2\n",
       "...         ...  ...       ...                   ...        ...\n",
       "3241       6393  6.7       1.0                     -     P38929\n",
       "3242       6335  8.4       1.0                     -     Q53005\n",
       "3243       6335  8.9       1.0                     -     Q53005\n",
       "3244       6265  8.0       1.0                     -     Q49W02\n",
       "3245       6264  8.0       1.0                     -     P39773\n",
       "\n",
       "[3246 rows x 5 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matched_ph_opts = pd.DataFrame(matched_data, columns=['exp_index', 'ph', 'activity', 'note', 'uniprot_id'])\n",
    "print(len(matched_ph_opts))\n",
    "matched_ph_opts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "f8293b7e-733d-4931-9a04-06bed4d51b5a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6241 9487\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exp_index</th>\n",
       "      <th>ph</th>\n",
       "      <th>activity</th>\n",
       "      <th>note</th>\n",
       "      <th>ph_range_comment</th>\n",
       "      <th>uniprot_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7.5</td>\n",
       "      <td>0.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pH 4.0: about 80% of maximal activity, pH 7.5:...</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6241</th>\n",
       "      <td>1</td>\n",
       "      <td>4.5</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B2KJ46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6243</th>\n",
       "      <td>3</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6242</th>\n",
       "      <td>3</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>reduction of ketones</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>5.8</td>\n",
       "      <td>0.40</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4.9</td>\n",
       "      <td>0.25</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Q4J702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4</td>\n",
       "      <td>5.8</td>\n",
       "      <td>0.40</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>pH 4.9: about 25% of maximal activity, pH 5.8:...</td>\n",
       "      <td>Q4J9F2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6245</th>\n",
       "      <td>4</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>reduction reaction</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q4J9F2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6244</th>\n",
       "      <td>4</td>\n",
       "      <td>5.1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>reduction of ketones</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q4J9F2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      exp_index   ph  activity                  note  \\\n",
       "0             1  4.0      0.80                   NaN   \n",
       "1             1  7.5      0.60                   NaN   \n",
       "6241          1  4.5      1.00                     -   \n",
       "6243          3  5.1      1.00    reduction reaction   \n",
       "6242          3  5.1      1.00  reduction of ketones   \n",
       "3             3  5.8      0.40    reduction reaction   \n",
       "2             3  4.9      0.25    reduction reaction   \n",
       "5             4  5.8      0.40    reduction reaction   \n",
       "6245          4  5.1      1.00    reduction reaction   \n",
       "6244          4  5.1      1.00  reduction of ketones   \n",
       "\n",
       "                                       ph_range_comment uniprot_id  \n",
       "0     pH 4.0: about 80% of maximal activity, pH 7.5:...     B2KJ46  \n",
       "1     pH 4.0: about 80% of maximal activity, pH 7.5:...     B2KJ46  \n",
       "6241                                                NaN     B2KJ46  \n",
       "6243                                                NaN     Q4J702  \n",
       "6242                                                NaN     Q4J702  \n",
       "3     pH 4.9: about 25% of maximal activity, pH 5.8:...     Q4J702  \n",
       "2     pH 4.9: about 25% of maximal activity, pH 5.8:...     Q4J702  \n",
       "5     pH 4.9: about 25% of maximal activity, pH 5.8:...     Q4J9F2  \n",
       "6245                                                NaN     Q4J9F2  \n",
       "6244                                                NaN     Q4J9F2  "
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_act2 = pd.concat([data_act_ph_range, matched_ph_opts], ignore_index=True).reset_index(drop=True)\n",
    "data_act2.sort_values('exp_index', inplace=True)\n",
    "print(len(data_act_ph_range), len(data_act2))\n",
    "# print(data_act2.uniprot_id.nunique(), len(data_act2))\n",
    "data_act2.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "4955afa0-7118-4276-a6ed-17a5c5f12b13",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty DataFrame\n",
      "Columns: [exp_index, ph, activity, note, ph_range_comment, uniprot_id]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [exp_index, ph, activity, note, ph_range_comment, uniprot_id]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "print(data_act2[data_act2.ph > 13])\n",
    "print(data_act2[data_act2.ph < 1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "d6707a23-a372-4601-9f48-e6eb78ee6d59",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_act2 = data_act2.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "afcd474e-cdab-4620-8605-0c820dcf2f3b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/projects/robustmicrob/jlaw/projects/prot_stability_engineering/inputs/brenda/ph_range/20230907_ph_range_processed_act_wopt.csv\n"
     ]
    }
   ],
   "source": [
    "# write to file\n",
    "out_file = Path(data_dir, \"ph_range/20230907_ph_range_processed_act_wopt.csv\")\n",
    "print(out_file)\n",
    "data_act2.to_csv(out_file, index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
