{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "610ccd59-0df6-4ef1-b0e9-4c895dc4bf01",
   "metadata": {},
   "source": [
    "## Topt Dataset\n",
    "- The current Topt dataset is from 2018\n",
    "- Re-download and parse the brenda data to see if there's more data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a2771156-dab8-4dd4-9137-3aeae074b567",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "13e02791-667a-4d94-8981-21b619835ed3",
   "metadata": {},
   "outputs": [],
   "source": [
    "brenda_file = \"/projects/robustmicrob/jlaw/inputs/brenda/brenda_2022_2.json\"\n",
    "brenda_dict = json.load(open(brenda_file))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9aeadaad-0581-4511-8dd6-217d306d0792",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['7.6.2.15',\n",
       " '7.6.2.16',\n",
       " '7.6.2.2',\n",
       " '7.6.2.3',\n",
       " '7.6.2.4',\n",
       " '7.6.2.5',\n",
       " '7.6.2.6',\n",
       " '7.6.2.7',\n",
       " '7.6.2.8',\n",
       " '7.6.2.9']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(brenda_dict['data'].keys())[-10:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "251b383c-d567-4938-895e-56f63475b16e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['id', 'name', 'systematic_name', 'generic_reaction'])"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['spontaneous'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4b931c55-fa33-44a5-a6d4-7f8c24097eca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['id', 'name', 'systematic_name', 'synonyms', 'localization', 'cofactor', 'activating_compound', 'inhibitor', 'metals_ions', 'subunits', 'engineering', 'cloned', 'general_information', 'crystallization', 'purification', 'renaturation', 'general_stability', 'generic_reaction', 'natural_reaction', 'reaction', 'turnover_number', 'km_value', 'ph_optimum', 'specific_activity', 'temperature_optimum', 'molecular_weight', 'organisms', 'references', 'proteins'])"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['7.6.2.8'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f0d402f8-bd52-48d4-8a66-b08cfc4b5d0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': '29',\n",
       " 'title': 'Mechanistic basis of vitamin B12 and cobinamide salvaging by the Vibrio species. Biochim. Biophys',\n",
       " 'authors': ['Agarwal, S.',\n",
       "  'Dey, S.',\n",
       "  'Ghosh, B.',\n",
       "  'Biswas, M.',\n",
       "  'Dasgupta, J.'],\n",
       " 'journal': 'Acta',\n",
       " 'year': 2019,\n",
       " 'pages': '140-151',\n",
       " 'vol': '1867',\n",
       " 'pmid': 30463026}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['7.6.2.8']['references']['29']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b644996a-7db7-4d96-9b89-1776ee7c9b95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'comment': '#12# BtuF binding assay at <29>',\n",
       "  'proteins': ['12'],\n",
       "  'organisms': ['12'],\n",
       "  'references': ['29'],\n",
       "  'num_value': 7.0},\n",
       " {'comment': '#5,10,11# assay at <15,31,37,38>',\n",
       "  'proteins': ['11', '5', '10'],\n",
       "  'organisms': ['11', '5', '10'],\n",
       "  'references': ['15', '31', '37', '38'],\n",
       "  'num_value': 7.5},\n",
       " {'comment': '#8# assay at <34>',\n",
       "  'proteins': ['8'],\n",
       "  'organisms': ['8'],\n",
       "  'references': ['34'],\n",
       "  'num_value': 8.0},\n",
       " {'comment': '#9# assay at <28>',\n",
       "  'proteins': ['9'],\n",
       "  'organisms': ['9'],\n",
       "  'references': ['28'],\n",
       "  'num_value': 7.6}]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['7.6.2.8']['ph_optimum']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5dbac1b3-97c0-4446-bb0a-e920d66de0a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'organisms': ['15'],\n",
       "  'references': ['42'],\n",
       "  'min_value': 4.0,\n",
       "  'max_value': 11.0},\n",
       " {'comment': '#27# in the biosynthetic direction <38>',\n",
       "  'proteins': ['27'],\n",
       "  'organisms': ['27'],\n",
       "  'references': ['38'],\n",
       "  'min_value': 5.0,\n",
       "  'max_value': 9.0},\n",
       " {'comment': '#27# in the degradative direction <38>',\n",
       "  'proteins': ['27'],\n",
       "  'organisms': ['27'],\n",
       "  'references': ['38'],\n",
       "  'min_value': 5.0,\n",
       "  'max_value': 7.8},\n",
       " {'comment': '#31# hydrolysis reaction <32>',\n",
       "  'proteins': ['31'],\n",
       "  'organisms': ['31'],\n",
       "  'references': ['32'],\n",
       "  'min_value': 7.8,\n",
       "  'max_value': 8.5},\n",
       " {'comment': '#31# synthesis reaction <32>',\n",
       "  'proteins': ['31'],\n",
       "  'organisms': ['31'],\n",
       "  'references': ['32'],\n",
       "  'min_value': 5.5,\n",
       "  'max_value': 6.9}]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['3.5.2.3']['ph_range']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e9ff0d57-e647-413f-b510-49f4911e3956",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'comment': '#10# assay at <37>',\n",
       "  'proteins': ['10'],\n",
       "  'organisms': ['10'],\n",
       "  'references': ['37'],\n",
       "  'num_value': 23.0},\n",
       " {'comment': '#11# assay at <31>',\n",
       "  'proteins': ['11'],\n",
       "  'organisms': ['11'],\n",
       "  'references': ['31'],\n",
       "  'num_value': 30.0},\n",
       " {'comment': '#12# BtuF binding assay at <29>',\n",
       "  'proteins': ['12'],\n",
       "  'organisms': ['12'],\n",
       "  'references': ['29'],\n",
       "  'num_value': 20.0},\n",
       " {'comment': '#5,10# assay at <15,38>; #8# assay at room temperature <34>',\n",
       "  'proteins': ['5', '8', '10'],\n",
       "  'organisms': ['5', '8', '10'],\n",
       "  'references': ['15', '34', '38'],\n",
       "  'num_value': 22.0},\n",
       " {'comment': '#9# assay at <28>',\n",
       "  'proteins': ['9'],\n",
       "  'organisms': ['9'],\n",
       "  'references': ['28'],\n",
       "  'num_value': 37.0}]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['7.6.2.8']['temperature_optimum']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "261068ab-7b00-4fbb-b2c3-14f9935d23a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'5': [{'accessions': ['P06611'],\n",
       "   'source': 'uniprot',\n",
       "   'comment': '#5# alpha-subunit of Fdh3 <12>'}],\n",
       " '6': [{'accessions': ['P37028'], 'source': 'uniprot'}],\n",
       " '7': [{'accessions': ['P06609'], 'source': 'uniprot'}],\n",
       " '8': [{'accessions': ['Q3SFD8'], 'source': 'uniprot'}],\n",
       " '9': [{'accessions': ['O14678'], 'source': 'uniprot'}],\n",
       " '10': [{'accessions': ['P06609', 'P06611', 'P37028'], 'source': 'uniprot'}],\n",
       " '11': [{'accessions': ['Q1GBI8'], 'source': 'uniprot'}],\n",
       " '12': [{'accessions': ['A0A0H3AMA6', 'A5F1V0', 'A5F5P5'],\n",
       "   'source': 'uniprot'}]}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['7.6.2.8']['proteins']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "88eda47f-b9b0-4882-975b-35dab156f0d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "example = brenda_dict['data']['4.1.1.101']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "6b1a7c70-924a-438f-be1e-88deae1f47c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "example = brenda_dict['data']['1.1.1.1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "23ac3a0e-8479-4c7c-a24d-54bbb13bdcf5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': '99',\n",
       " 'title': 'Purification and characterization of an oxygen-labile, NAD-dependent alcohol dehydrogenase from Desulfovibrio gigas. J',\n",
       " 'authors': ['Hensgens, C.M.H.',\n",
       "  'Vonck, J.',\n",
       "  'van Beeumen, J.',\n",
       "  'Bruggen, E.F.J.',\n",
       "  'Hansen, T.A.'],\n",
       " 'journal': 'Bacteriol.',\n",
       " 'year': 1993,\n",
       " 'pages': '2859-2863',\n",
       " 'vol': '175',\n",
       " 'pmid': 8491707}"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "example['references']['99']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ab1285c6-9c9b-4a60-85a9-16a286df84c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'proteins': ['10'],\n",
       "  'organisms': ['10'],\n",
       "  'references': ['1'],\n",
       "  'num_value': 45.0},\n",
       " {'comment': '#5# assay at <4>',\n",
       "  'organisms': ['5'],\n",
       "  'references': ['4'],\n",
       "  'num_value': 30.0},\n",
       " {'organisms': ['5'], 'references': ['10'], 'num_value': 35.0},\n",
       " {'organisms': ['6', '5'], 'references': ['13', '14'], 'num_value': 37.0}]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "example['temperature_optimum']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "012f86b7-ab31-4312-8ff9-d4bd01c148d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "62\n"
     ]
    }
   ],
   "source": [
    "temp_opt_data = defaultdict(list)\n",
    "for ec_number, ec_data in brenda_dict['data'].items():\n",
    "    if '4.1.1.' not in ec_number:\n",
    "        continue\n",
    "    if 'temperature_optimum' not in ec_data:\n",
    "        continue\n",
    "    for entry in ec_data['temperature_optimum']:\n",
    "        if 'num_value' in entry:\n",
    "            val = entry['num_value']\n",
    "        else:\n",
    "            val = entry['max_value']\n",
    "        temp_opt_data[ec_number].append(val)\n",
    "print(len(temp_opt_data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "59d7d371-25db-4998-be68-bfe72a99ab76",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4.1.1.48      5\n",
       "4.1.1.50      5\n",
       "4.1.1.61      5\n",
       "4.1.1.112     5\n",
       "4.1.1.32      6\n",
       "4.1.1.7       6\n",
       "4.1.1.72      6\n",
       "4.1.1.12      6\n",
       "4.1.1.23      7\n",
       "4.1.1.22      7\n",
       "4.1.1.17      7\n",
       "4.1.1.11      8\n",
       "4.1.1.25      9\n",
       "4.1.1.18     10\n",
       "4.1.1.2      10\n",
       "4.1.1.19     15\n",
       "4.1.1.39     15\n",
       "4.1.1.31     15\n",
       "4.1.1.15     16\n",
       "4.1.1.1      20\n",
       "dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "S = pd.Series({ec: len(data) for ec, data in temp_opt_data.items()})\n",
    "S.sort_values().tail(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "40e788f1-4bda-4685-8e14-3636d8b15c91",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'8': [{'accessions': ['F6LR19'], 'source': 'uniprot'}],\n",
       " '9': [{'accessions': ['Q48662'], 'source': 'uniprot'}],\n",
       " '10': [{'accessions': ['Q5S3L3'], 'source': 'uniprot'}],\n",
       " '11': [{'accessions': ['Q48796'], 'source': 'uniprot'}],\n",
       " '12': [{'accessions': ['W5XU69'], 'source': 'uniprot'}],\n",
       " '13': [{'accessions': ['Q03XG6'], 'source': 'uniprot'}]}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "example['proteins']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "7fc1595a-514b-48ab-a171-b26be030b93c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'value': 'Thermoanaerobacter brockii'}\n",
      "{'value': 'Sulfurisphaera tokodaii'}\n"
     ]
    }
   ],
   "source": [
    "print(example['organisms']['73'])\n",
    "print(example['organisms']['125'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "2d9053dc-a867-404a-b431-34c318109a76",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'proteins': ['103'],\n",
       "  'organisms': ['103'],\n",
       "  'references': ['173'],\n",
       "  'num_value': 83.0},\n",
       " {'comment': '#109# assay at <234>',\n",
       "  'proteins': ['163', '136', '109'],\n",
       "  'organisms': ['163', '136', '109'],\n",
       "  'references': ['234', '236', '285'],\n",
       "  'num_value': 80.0},\n",
       " {'organisms': ['11'],\n",
       "  'references': ['3'],\n",
       "  'min_value': 45.0,\n",
       "  'max_value': 50.0},\n",
       " {'comment': '#11# assay at <120>',\n",
       "  'proteins': ['178'],\n",
       "  'organisms': ['11', '178'],\n",
       "  'references': ['120', '307'],\n",
       "  'min_value': 25.0,\n",
       "  'max_value': 30.0},\n",
       " {'comment': '#29# assay at <129>; #11# Cu-ADH <122>; #11# enzyme covalently immobilized to magnetic Fe3O4 nanoparticles via glutaraldehyde <182>',\n",
       "  'organisms': ['11', '53', '57', '29'],\n",
       "  'references': ['99', '122', '129', '182', '278'],\n",
       "  'num_value': 40.0},\n",
       " {'comment': '#11# immobilized enzyme <196>; #11# Zn-ADH and Co-ADH <122>; #46# reductive reaction, recombinant enzymes expressed from Saccharomyces cerevisiae and Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Saccharomyces cerevisiae <231>; #11# reduction of glycolaldehyde, furfural, butyraldehyde, and propionaldehyde <284>',\n",
       "  'proteins': ['101'],\n",
       "  'organisms': ['11', '46', '101'],\n",
       "  'references': ['122', '168', '196', '231', '284'],\n",
       "  'num_value': 35.0},\n",
       " {'comment': '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>',\n",
       "  'proteins': ['170', '120', '171', '174', '175'],\n",
       "  'organisms': ['131',\n",
       "   '170',\n",
       "   '120',\n",
       "   '46',\n",
       "   '105',\n",
       "   '91',\n",
       "   '171',\n",
       "   '174',\n",
       "   '11',\n",
       "   '165',\n",
       "   '175',\n",
       "   '72'],\n",
       "  'references': ['84',\n",
       "   '144',\n",
       "   '182',\n",
       "   '185',\n",
       "   '205',\n",
       "   '224',\n",
       "   '284',\n",
       "   '287',\n",
       "   '296',\n",
       "   '303'],\n",
       "  'num_value': 30.0},\n",
       " {'proteins': ['115', '126', '127'],\n",
       "  'organisms': ['115', '126', '127'],\n",
       "  'references': ['213', '218'],\n",
       "  'num_value': 75.0},\n",
       " {'comment': '#120# assay at <299>',\n",
       "  'proteins': ['120'],\n",
       "  'organisms': ['120'],\n",
       "  'references': ['299'],\n",
       "  'num_value': 20.0},\n",
       " {'comment': '#127# the reaction rate increases up to 78°C and then decreases rapidly due to thermal inactivation <217>',\n",
       "  'proteins': ['127'],\n",
       "  'organisms': ['127'],\n",
       "  'references': ['217'],\n",
       "  'num_value': 78.0},\n",
       " {'comment': '#135# catalytic activity increased up to 95°C <238>',\n",
       "  'proteins': ['135'],\n",
       "  'organisms': ['135'],\n",
       "  'references': ['238'],\n",
       "  'num_value': 95.0},\n",
       " {'comment': '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>',\n",
       "  'proteins': ['132', '121', '96'],\n",
       "  'organisms': ['121', '14', '88', '132', '54', '92', '96', '33'],\n",
       "  'references': ['82', '118', '126', '144', '181', '205', '225', '229'],\n",
       "  'num_value': 70.0},\n",
       " {'proteins': ['146'],\n",
       "  'organisms': ['146'],\n",
       "  'references': ['138'],\n",
       "  'num_value': 100.0},\n",
       " {'proteins': ['152'],\n",
       "  'organisms': ['152'],\n",
       "  'references': ['242'],\n",
       "  'num_value': 90.0},\n",
       " {'organisms': ['16', '31'], 'references': ['133', '134'], 'num_value': 33.0},\n",
       " {'comment': '#25# assay at <288,289,292>',\n",
       "  'proteins': ['142', '140'],\n",
       "  'organisms': ['17', '142', '25', '140'],\n",
       "  'references': ['251', '288', '289', '292', '303'],\n",
       "  'num_value': 45.0},\n",
       " {'comment': '#26,130# assay at room temperature <128,221>',\n",
       "  'proteins': ['130'],\n",
       "  'organisms': ['130', '26'],\n",
       "  'references': ['128', '221'],\n",
       "  'num_value': 22.0},\n",
       " {'comment': '#153# oxidation of crotyl aclohol <243>',\n",
       "  'proteins': ['141', '153'],\n",
       "  'organisms': ['28', '141', '153'],\n",
       "  'references': ['188', '243', '251'],\n",
       "  'num_value': 55.0},\n",
       " {'organisms': ['40'],\n",
       "  'references': ['85'],\n",
       "  'min_value': 50.0,\n",
       "  'max_value': 55.0},\n",
       " {'comment': '#43,166,167# assay at <117,290,293>',\n",
       "  'proteins': ['167'],\n",
       "  'organisms': ['43', '167', '166'],\n",
       "  'references': ['117', '290', '293'],\n",
       "  'num_value': 37.0},\n",
       " {'organisms': ['46'],\n",
       "  'references': ['114'],\n",
       "  'min_value': 30.0,\n",
       "  'max_value': 50.0},\n",
       " {'comment': '#46# reductive reaction, recombinant enzyme expressed from Arxula adeninivorans <231>; #46# recombinant enzyme expressed in Arxula adeninivorans <231>',\n",
       "  'organisms': ['46'],\n",
       "  'references': ['231'],\n",
       "  'num_value': 39.0},\n",
       " {'comment': '#159# enzyme expressed in Pyrococcus furiosus <280>; #5# temperature optimum for oxidation reaction in glycine NaOH buffer using ethanol as the substrate <304>',\n",
       "  'proteins': ['118', '159'],\n",
       "  'organisms': ['5', '118', '159'],\n",
       "  'references': ['280', '304', '308'],\n",
       "  'num_value': 60.0},\n",
       " {'organisms': ['50'],\n",
       "  'references': ['222'],\n",
       "  'min_value': 50.0,\n",
       "  'max_value': 60.0},\n",
       " {'comment': '#6,9,11,121# assay at <107,115,119,121,124,228,294,297>; #63# assay at, forward and reverse reaction <113>; #11# free enzyme, at 25°C <196>',\n",
       "  'proteins': ['121'],\n",
       "  'organisms': ['121', '6', '63', '11', '9', '27', '42'],\n",
       "  'references': ['106',\n",
       "   '107',\n",
       "   '113',\n",
       "   '115',\n",
       "   '119',\n",
       "   '121',\n",
       "   '124',\n",
       "   '131',\n",
       "   '196',\n",
       "   '228',\n",
       "   '294',\n",
       "   '297'],\n",
       "  'num_value': 25.0},\n",
       " {'organisms': ['7'], 'references': ['169'], 'num_value': 73.0},\n",
       " {'comment': '#73# approximately 85°C <205>',\n",
       "  'proteins': ['125'],\n",
       "  'organisms': ['73', '125'],\n",
       "  'references': ['205', '216'],\n",
       "  'num_value': 85.0},\n",
       " {'comment': '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>',\n",
       "  'proteins': ['126', '153', '127'],\n",
       "  'organisms': ['73', '126', '153', '127'],\n",
       "  'references': ['121', '217', '218', '243'],\n",
       "  'num_value': 65.0},\n",
       " {'comment': '#9,43# assay at <116>',\n",
       "  'organisms': ['9', '43'],\n",
       "  'references': ['116'],\n",
       "  'min_value': 30.0,\n",
       "  'max_value': 37.0},\n",
       " {'comment': '#125# assay at <216>; #104# reaction with ethanol + NAD+ <171>',\n",
       "  'proteins': ['125', '139'],\n",
       "  'organisms': ['125', '139', '97', '104'],\n",
       "  'references': ['137', '171', '216', '251'],\n",
       "  'num_value': 50.0}]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "example['temperature_optimum']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1e17dbf0-b481-44c3-99ea-dbd79f195226",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(list,\n",
       "            {'5': ['P06611'],\n",
       "             '6': ['P37028'],\n",
       "             '7': ['P06609'],\n",
       "             '8': ['Q3SFD8'],\n",
       "             '9': ['O14678'],\n",
       "             '10': ['P06609', 'P06611', 'P37028'],\n",
       "             '11': ['Q1GBI8'],\n",
       "             '12': ['A0A0H3AMA6', 'A5F1V0', 'A5F5P5']})"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_proteins(ec_dict):\n",
    "    uniprot_ids = defaultdict(list)\n",
    "    for prot_id, sources in ec_dict['proteins'].items():\n",
    "        for source in sources:\n",
    "            if source['source'] == 'uniprot':\n",
    "                uniprot_ids[prot_id] += source['accessions']\n",
    "    if len(uniprot_ids) == 0:\n",
    "        print(\"Warning: no uniprot ids\")\n",
    "    return uniprot_ids\n",
    "get_proteins(example)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "75979757-de14-4853-9f6a-a48a239766aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1': 'Salmonella enterica subsp. enterica serovar Typhimurium',\n",
       " '2': 'Haemophilus influenzae',\n",
       " '3': 'Escherichia coli',\n",
       " '4': 'Mycobacterium tuberculosis',\n",
       " '5': 'Escherichia coli',\n",
       " '6': 'Escherichia coli',\n",
       " '7': 'Escherichia coli',\n",
       " '8': 'Thiobacillus denitrificans',\n",
       " '9': 'Homo sapiens',\n",
       " '10': 'Escherichia coli',\n",
       " '11': 'Lactobacillus delbrueckii subsp. bulgaricus',\n",
       " '12': 'Vibrio cholerae serotype O1'}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_organisms(ec_dict):\n",
    "    organisms = {}\n",
    "    for o_id, organism in ec_dict['organisms'].items():\n",
    "        organisms[o_id] = organism['value']\n",
    "    return organisms\n",
    "get_organisms(example)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b88d2466-f620-4cb3-99a0-9a27cd6cb637",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1': '3528129',\n",
       " '2': '2089218',\n",
       " '3': '2671656',\n",
       " '4': '3528128',\n",
       " '5': '2987192',\n",
       " '6': '2656634',\n",
       " '7': '16331991',\n",
       " '8': '17073449',\n",
       " '9': '17302441',\n",
       " '10': '18358587',\n",
       " '11': '17208973',\n",
       " '12': '17951296',\n",
       " '13': '17673622',\n",
       " '14': '19254551',\n",
       " '15': '19101549',\n",
       " '16': '20173761',\n",
       " '17': '19847921',\n",
       " '18': '20596946',\n",
       " '19': '22569249',\n",
       " '20': '21345797',\n",
       " '21': '21953468',\n",
       " '23': '26247943',\n",
       " '24': '24362024',\n",
       " '25': '25402482',\n",
       " '26': '23000901',\n",
       " '27': '23407640',\n",
       " '28': '29397936',\n",
       " '29': '30463026',\n",
       " '30': '27276259',\n",
       " '31': '29809140',\n",
       " '32': '29915233',\n",
       " '33': '29162829',\n",
       " '34': '30072686',\n",
       " '35': '27870912',\n",
       " '36': '27456980',\n",
       " '37': '29084999',\n",
       " '38': '28128319'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# TODO not all entries have pmid\n",
    "def get_references(ec_dict):\n",
    "    references = {}\n",
    "    for ref_id, data in ec_dict['references'].items():\n",
    "        if 'pmid' not in data:\n",
    "            continue\n",
    "        # print(data)\n",
    "        references[ref_id] = str(data['pmid'])\n",
    "    return references\n",
    "get_references(example)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "ef61b837-fa74-4553-b617-cc5b17eead1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(list,\n",
       "            {'Q6L0S1': [(83.0, 'Picrophilus torridus', [17989975], None)],\n",
       "             'A3MVR8': [(80.0,\n",
       "               'Pyrobaculum calidifontis',\n",
       "               [20377222, 22526808, 29022135],\n",
       "               '#109# assay at <234>')],\n",
       "             'D4GSN2': [(80.0,\n",
       "               'Haloferax volcanii',\n",
       "               [20377222, 22526808, 29022135],\n",
       "               '#109# assay at <234>')],\n",
       "             'P39462': [(80.0,\n",
       "               'Saccharolobus solfataricus',\n",
       "               [20377222, 22526808, 29022135],\n",
       "               '#109# assay at <234>')],\n",
       "             'P28625': [('25.0-30.0',\n",
       "               'Saccharomyces cerevisiae',\n",
       "               [12702265, 32967812],\n",
       "               '#11# assay at <120>')],\n",
       "             'B2KJ46': [(35.0,\n",
       "               'Candida parapsilosis',\n",
       "               [11893392, 17435004, 19578994, 22584819, 28265724],\n",
       "               '#11# immobilized enzyme <196>; #11# Zn-ADH and Co-ADH <122>; #46# reductive reaction, recombinant enzymes expressed from Saccharomyces cerevisiae and Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Saccharomyces cerevisiae <231>; #11# reduction of glycolaldehyde, furfural, butyraldehyde, and propionaldehyde <284>')],\n",
       "             'G2I689': [(30.0,\n",
       "               'Komagataeibacter medellinensis',\n",
       "               [17072683, 18456317, 17368401, 21670533, 28265724, 26216639],\n",
       "               '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>')],\n",
       "             'P00330': [(30.0,\n",
       "               'Saccharomyces cerevisiae',\n",
       "               [17072683, 18456317, 17368401, 21670533, 28265724, 26216639],\n",
       "               '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>'),\n",
       "              (20.0, 'Saccharomyces cerevisiae', [], '#120# assay at <299>')],\n",
       "             'A0A380WRS7': [(30.0,\n",
       "               'Aminobacter aminovorans',\n",
       "               [17072683, 18456317, 17368401, 21670533, 28265724, 26216639],\n",
       "               '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>')],\n",
       "             'A0A0U5FNC1': [(30.0,\n",
       "               'Acetobacter senegalensis',\n",
       "               [17072683, 18456317, 17368401, 21670533, 28265724, 26216639],\n",
       "               '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>')],\n",
       "             'A0A3T0KHC6': [(30.0,\n",
       "               'Komagataeibacter xylinus',\n",
       "               [17072683, 18456317, 17368401, 21670533, 28265724, 26216639],\n",
       "               '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>')],\n",
       "             'Q9HIM3': [(75.0,\n",
       "               'Thermoplasma acidophilum',\n",
       "               [19058034, 20049620],\n",
       "               None)],\n",
       "             'Q4J702': [(75.0,\n",
       "               'Sulfolobus acidocaldarius',\n",
       "               [19058034, 20049620],\n",
       "               None),\n",
       "              (65.0,\n",
       "               'Sulfolobus acidocaldarius',\n",
       "               [11809419, 22805786, 20049620, 24509923],\n",
       "               '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>')],\n",
       "             'Q4J9F2': [(75.0,\n",
       "               'Sulfolobus acidocaldarius',\n",
       "               [19058034, 20049620],\n",
       "               None),\n",
       "              (78.0,\n",
       "               'Sulfolobus acidocaldarius',\n",
       "               [22805786],\n",
       "               '#127# the reaction rate increases up to 78°C and then decreases rapidly due to thermal inactivation <217>'),\n",
       "              (65.0,\n",
       "               'Sulfolobus acidocaldarius',\n",
       "               [11809419, 22805786, 20049620, 24509923],\n",
       "               '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>')],\n",
       "             'Q9Y9P9': [(95.0,\n",
       "               'Aeropyrum pernix',\n",
       "               [16233615],\n",
       "               '#135# catalytic activity increased up to 95°C <238>')],\n",
       "             'Q8U259': [(70.0,\n",
       "               'Pyrococcus furiosus',\n",
       "               [12031468, 17072683, 18189118, 23160740, 16463078],\n",
       "               '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>')],\n",
       "             'P00327': [(70.0,\n",
       "               'Equus caballus',\n",
       "               [12031468, 17072683, 18189118, 23160740, 16463078],\n",
       "               '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>'),\n",
       "              (25.0,\n",
       "               'Equus caballus',\n",
       "               [12172614,\n",
       "                15369820,\n",
       "                12834271,\n",
       "                11964133,\n",
       "                15096205,\n",
       "                11809419,\n",
       "                11997393,\n",
       "                18629898,\n",
       "                19578994,\n",
       "                21784063,\n",
       "                21177091],\n",
       "               '#6,9,11,121# assay at <107,115,119,121,124,228,294,297>; #63# assay at, forward and reverse reaction <113>; #11# free enzyme, at 25°C <196>')],\n",
       "             'Q8L3C9': [(70.0,\n",
       "               'Flavobacterium frigidimaris',\n",
       "               [12031468, 17072683, 18189118, 23160740, 16463078],\n",
       "               '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>')],\n",
       "             'Q8TZM9': [(100.0, 'Pyrococcus furiosus', [16391048], None)],\n",
       "             'Q5JI09': [(90.0, 'Thermococcus kodakarensis', [23354700], None)],\n",
       "             'V9SFA1': [(45.0,\n",
       "               'Kluyveromyces marxianus',\n",
       "               [24885162, 20632378],\n",
       "               '#25# assay at <288,289,292>')],\n",
       "             'V9SDP6': [(45.0,\n",
       "               'Kluyveromyces marxianus',\n",
       "               [24885162, 20632378],\n",
       "               '#25# assay at <288,289,292>')],\n",
       "             'H9ZGN0': [(22.0,\n",
       "               'Ogataea angusta',\n",
       "               [12058720, 22249723],\n",
       "               '#26,130# assay at room temperature <128,221>')],\n",
       "             'V9SCJ1': [(55.0,\n",
       "               'Kluyveromyces marxianus',\n",
       "               [24509923, 24885162],\n",
       "               '#153# oxidation of crotyl aclohol <243>')],\n",
       "             'W6CX26': [(55.0,\n",
       "               'Yokenella sp.',\n",
       "               [24509923, 24885162],\n",
       "               '#153# oxidation of crotyl aclohol <243>'),\n",
       "              (65.0,\n",
       "               'Yokenella sp.',\n",
       "               [11809419, 22805786, 20049620, 24509923],\n",
       "               '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>')],\n",
       "             'E7FHL9': [(37.0,\n",
       "               'Pyrococcus furiosus',\n",
       "               [12184789],\n",
       "               '#43,166,167# assay at <117,290,293>')],\n",
       "             'A4IP64': [(60.0,\n",
       "               'Geobacillus thermodenitrificans',\n",
       "               [28194879, 28320283, 19383697],\n",
       "               '#159# enzyme expressed in Pyrococcus furiosus <280>; #5# temperature optimum for oxidation reaction in glycine NaOH buffer using ethanol as the substrate <304>')],\n",
       "             'C7IV28': [(60.0,\n",
       "               'Thermoanaerobacter ethanolicus',\n",
       "               [28194879, 28320283, 19383697],\n",
       "               '#159# enzyme expressed in Pyrococcus furiosus <280>; #5# temperature optimum for oxidation reaction in glycine NaOH buffer using ethanol as the substrate <304>')],\n",
       "             'F9VMI9': [(85.0,\n",
       "               'Sulfurisphaera tokodaii',\n",
       "               [19139244],\n",
       "               '#73# approximately 85°C <205>'),\n",
       "              (50.0,\n",
       "               'Sulfurisphaera tokodaii',\n",
       "               [16000771, 17646983, 19139244, 24885162],\n",
       "               '#125# assay at <216>; #104# reaction with ethanol + NAD+ <171>')],\n",
       "             'G3FFC9': [(50.0,\n",
       "               'Kluyveromyces marxianus',\n",
       "               [16000771, 17646983, 19139244, 24885162],\n",
       "               '#125# assay at <216>; #104# reaction with ethanol + NAD+ <171>')]})"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_topt(ec_dict, proteins_dict, organism_dict, ref_dict):\n",
    "    uniprot_topt = defaultdict(list)\n",
    "    # extra_data = {}\n",
    "    for data in ec_dict['temperature_optimum']:\n",
    "        if 'proteins' not in data:\n",
    "            continue\n",
    "        for i, prot_id in enumerate(data['proteins']):\n",
    "            org = organism_dict[prot_id]\n",
    "            refs = [ref_dict[r] for r in data['references'] if r in ref_dict]\n",
    "            for u_id in proteins_dict[prot_id]:\n",
    "                if 'num_value' in data:\n",
    "                    val = data['num_value']\n",
    "                elif 'min_value' in data:\n",
    "                    val = f\"{data['min_value']}-{data['max_value']}\"\n",
    "                # if u_id in uniprot_topt and uniprot_topt[u_id][0] != val:\n",
    "                #     print(f\"WARNING: {u_id} present twice. \"\n",
    "                #           f\"Orig topt: {uniprot_topt[u_id][0]}; New topt: {val}\")\n",
    "                uniprot_topt[u_id].append((val, org, refs, data.get('comment')))\n",
    "    return uniprot_topt\n",
    "get_topt(example, get_proteins(example), get_organisms(example), get_references(example))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "becadae3-c043-489b-afe6-fd3bac17c01a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7754"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(brenda_dict['data'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ec80a7a4-bc0e-42de-b1f4-9481fe4f2187",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9485"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(uniprot_topts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a622d4ac-7ac4-4773-b722-084345f1dbb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3754"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(ec_num for ec_nums in uniprot_ec_num.values() for ec_num in ec_nums))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "01436b27-6b08-4aa5-902e-672986264f0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "u_id: A4IP64 already in uniprot_topts\n",
      "u_id: A4ISB9 already in uniprot_topts\n",
      "u_id: P15121 already in uniprot_topts\n",
      "u_id: P16116 already in uniprot_topts\n",
      "u_id: H8XA83 already in uniprot_topts\n",
      "u_id: A0A097ZMY7 already in uniprot_topts\n",
      "u_id: J7M9D0 already in uniprot_topts\n",
      "u_id: P28625 already in uniprot_topts\n",
      "u_id: Q9X5C9 already in uniprot_topts\n",
      "u_id: A4QB65 already in uniprot_topts\n",
      "u_id: A0A0E4AY21 already in uniprot_topts\n",
      "u_id: A0A0E4AX59 already in uniprot_topts\n",
      "u_id: Q308C1 already in uniprot_topts\n",
      "u_id: A0A0M4HL56 already in uniprot_topts\n",
      "u_id: Q7SD67 already in uniprot_topts\n",
      "u_id: A0A166G7A0 already in uniprot_topts\n",
      "u_id: A0A172EJV6 already in uniprot_topts\n",
      "u_id: B2NI94 already in uniprot_topts\n",
      "u_id: P42330 already in uniprot_topts\n",
      "u_id: P23457 already in uniprot_topts\n",
      "u_id: Q04828 already in uniprot_topts\n",
      "u_id: B9WJQ5 already in uniprot_topts\n",
      "u_id: O08349 already in uniprot_topts\n",
      "u_id: A0A223LRZ7 already in uniprot_topts\n",
      "u_id: Q5UY95 already in uniprot_topts\n",
      "u_id: D4GP33 already in uniprot_topts\n",
      "u_id: O93715 already in uniprot_topts\n",
      "u_id: Q9I1X3 already in uniprot_topts\n",
      "u_id: P80702 already in uniprot_topts\n",
      "u_id: Q04828 already in uniprot_topts\n",
      "u_id: P23457 already in uniprot_topts\n",
      "u_id: P42330 already in uniprot_topts\n",
      "u_id: C9JRZ8 already in uniprot_topts\n",
      "u_id: P42330 already in uniprot_topts\n",
      "u_id: O54939 already in uniprot_topts\n",
      "u_id: C6A190 already in uniprot_topts\n",
      "u_id: U6CL97 already in uniprot_topts\n",
      "u_id: Q72IW9 already in uniprot_topts\n",
      "u_id: A0A1B4XTS0 already in uniprot_topts\n",
      "u_id: M5AJW4 already in uniprot_topts\n",
      "u_id: W6CX26 already in uniprot_topts\n",
      "u_id: W6CX26 already in uniprot_topts\n",
      "u_id: P16027 already in uniprot_topts\n",
      "u_id: P14775 already in uniprot_topts\n",
      "u_id: Q588Z1 already in uniprot_topts\n",
      "u_id: Q53U15 already in uniprot_topts\n",
      "u_id: E4QP00 already in uniprot_topts\n",
      "u_id: O94219 already in uniprot_topts\n",
      "u_id: E4QP00 already in uniprot_topts\n",
      "u_id: Q6NQ66 already in uniprot_topts\n",
      "u_id: O81030 already in uniprot_topts\n",
      "u_id: Q9LYD8 already in uniprot_topts\n",
      "u_id: Q5LLG6 already in uniprot_topts\n",
      "u_id: E7FI44 already in uniprot_topts\n",
      "u_id: Q8U2E5 already in uniprot_topts\n",
      "u_id: Q8U2E4 already in uniprot_topts\n",
      "u_id: E7FHU4 already in uniprot_topts\n",
      "u_id: P93184 already in uniprot_topts\n",
      "u_id: Q8VY26 already in uniprot_topts\n",
      "u_id: Q7X0P3 already in uniprot_topts\n",
      "u_id: O75164 already in uniprot_topts\n",
      "u_id: H9N289 already in uniprot_topts\n",
      "u_id: A0A1I9Q5Z0 already in uniprot_topts\n",
      "u_id: B1NF19 already in uniprot_topts\n",
      "u_id: B1NF19 already in uniprot_topts\n",
      "u_id: Q6V0J1 already in uniprot_topts\n",
      "u_id: A0A076U8J8 already in uniprot_topts\n",
      "u_id: P32021 already in uniprot_topts\n",
      "u_id: Q06069 already in uniprot_topts\n",
      "u_id: O60341 already in uniprot_topts\n",
      "u_id: P36649 already in uniprot_topts\n",
      "u_id: P36649 already in uniprot_topts\n",
      "u_id: P22985 already in uniprot_topts\n",
      "u_id: O54050 already in uniprot_topts\n",
      "u_id: Q988H5 already in uniprot_topts\n",
      "u_id: P39460 already in uniprot_topts\n",
      "u_id: Q976X5 already in uniprot_topts\n",
      "u_id: Q01651 already in uniprot_topts\n",
      "u_id: Q84DC3 already in uniprot_topts\n",
      "u_id: A0A0J9X1M8 already in uniprot_topts\n",
      "u_id: Q9RBF6 already in uniprot_topts\n",
      "u_id: A0A1P8VFW6 already in uniprot_topts\n",
      "u_id: Q82TA7 already in uniprot_topts\n",
      "u_id: Q17209 already in uniprot_topts\n",
      "u_id: P9WGR1 already in uniprot_topts\n",
      "u_id: G3FHJ1 already in uniprot_topts\n",
      "u_id: A3KEZ1 already in uniprot_topts\n",
      "u_id: P32370 already in uniprot_topts\n",
      "u_id: M4YFG7 already in uniprot_topts\n",
      "u_id: A0A0M3VGH5 already in uniprot_topts\n",
      "u_id: Q9SU79 already in uniprot_topts\n",
      "u_id: A0A0N7IZL9 already in uniprot_topts\n",
      "u_id: P09546 already in uniprot_topts\n",
      "u_id: Q5JGP4 already in uniprot_topts\n",
      "u_id: P41407 already in uniprot_topts\n",
      "u_id: Q9X4K2 already in uniprot_topts\n",
      "u_id: A7GXH2 already in uniprot_topts\n",
      "u_id: Q5JGP4 already in uniprot_topts\n",
      "u_id: Q5JGF8 already in uniprot_topts\n",
      "u_id: P57681 already in uniprot_topts\n",
      "u_id: P14930 already in uniprot_topts\n",
      "u_id: A8IJ34 already in uniprot_topts\n",
      "u_id: Q8H0D3 already in uniprot_topts\n",
      "u_id: Q5JD38 already in uniprot_topts\n",
      "u_id: A0QYR0 already in uniprot_topts\n",
      "u_id: O30640 already in uniprot_topts\n",
      "u_id: Q9XGP7 already in uniprot_topts\n",
      "u_id: Q29U70 already in uniprot_topts\n",
      "u_id: Q9SPV4 already in uniprot_topts\n",
      "u_id: Q9V2G1 already in uniprot_topts\n",
      "u_id: Q74NE4 already in uniprot_topts\n",
      "u_id: A8IJ34 already in uniprot_topts\n",
      "u_id: A0A125T1T5 already in uniprot_topts\n",
      "u_id: A4H4P8 already in uniprot_topts\n",
      "u_id: Q27793 already in uniprot_topts\n",
      "u_id: S5M3K7 already in uniprot_topts\n",
      "u_id: Q27783 already in uniprot_topts\n",
      "u_id: Q6ZD89 already in uniprot_topts\n",
      "u_id: A0A125T1T5 already in uniprot_topts\n",
      "u_id: A0A1B1W6Z7 already in uniprot_topts\n",
      "u_id: Q837U7 already in uniprot_topts\n",
      "u_id: Q0GA14 already in uniprot_topts\n",
      "u_id: C0SVZ6 already in uniprot_topts\n",
      "u_id: C6L7V9 already in uniprot_topts\n",
      "u_id: C6L7V8 already in uniprot_topts\n",
      "u_id: L7NCQ3 already in uniprot_topts\n",
      "u_id: Q3ZPN4 already in uniprot_topts\n",
      "u_id: Q9ZGI5 already in uniprot_topts\n",
      "u_id: Q9ZGI4 already in uniprot_topts\n",
      "u_id: Q9ZGI3 already in uniprot_topts\n",
      "u_id: Q9ZGI2 already in uniprot_topts\n",
      "u_id: Q9ZGI1 already in uniprot_topts\n",
      "u_id: I6YG32 already in uniprot_topts\n",
      "u_id: I6YG32 already in uniprot_topts\n",
      "u_id: P41227 already in uniprot_topts\n",
      "u_id: Q9BXJ9 already in uniprot_topts\n",
      "u_id: P20582 already in uniprot_topts\n",
      "u_id: Q975F9 already in uniprot_topts\n",
      "u_id: Q9HA82 already in uniprot_topts\n",
      "u_id: Q1A3B0 already in uniprot_topts\n",
      "u_id: D2Z028 already in uniprot_topts\n",
      "u_id: P41227 already in uniprot_topts\n",
      "u_id: Q3TFD2 already in uniprot_topts\n",
      "u_id: Q8NF37 already in uniprot_topts\n",
      "u_id: Q3UN02 already in uniprot_topts\n",
      "u_id: Q9AU11 already in uniprot_topts\n",
      "u_id: A0A1Z2RWY4 already in uniprot_topts\n",
      "u_id: A0A3G1ICG7 already in uniprot_topts\n",
      "u_id: Q6ZPD8 already in uniprot_topts\n",
      "u_id: Q2RK58 already in uniprot_topts\n",
      "u_id: Q2RJ94 already in uniprot_topts\n",
      "u_id: A0A346FEE6 already in uniprot_topts\n",
      "u_id: C3PTS7 already in uniprot_topts\n",
      "u_id: A0A346FEE5 already in uniprot_topts\n",
      "u_id: P63146 already in uniprot_topts\n",
      "u_id: P61086 already in uniprot_topts\n",
      "u_id: Q96B02 already in uniprot_topts\n",
      "u_id: Q9LCB4 already in uniprot_topts\n",
      "u_id: Q8N6G5 already in uniprot_topts\n",
      "u_id: Q86X52 already in uniprot_topts\n",
      "u_id: D0QYM5 already in uniprot_topts\n",
      "u_id: P43636 already in uniprot_topts\n",
      "u_id: E6UBR9 already in uniprot_topts\n",
      "u_id: B7IED6 already in uniprot_topts\n",
      "u_id: Q9SWH5 already in uniprot_topts\n",
      "u_id: D3Y5N8 already in uniprot_topts\n",
      "u_id: P0C037 already in uniprot_topts\n",
      "u_id: A8R0Y0 already in uniprot_topts\n",
      "u_id: P14324 already in uniprot_topts\n",
      "u_id: P29848 already in uniprot_topts\n",
      "u_id: Q7XYS8 already in uniprot_topts\n",
      "u_id: A2R1N3 already in uniprot_topts\n",
      "u_id: Q9YC66 already in uniprot_topts\n",
      "u_id: B1XNF8 already in uniprot_topts\n",
      "u_id: P73133 already in uniprot_topts\n",
      "u_id: P24298 already in uniprot_topts\n",
      "u_id: Q9SR86 already in uniprot_topts\n",
      "u_id: Q9Y617 already in uniprot_topts\n",
      "u_id: P46395 already in uniprot_topts\n",
      "u_id: Q71RI9 already in uniprot_topts\n",
      "u_id: Q9WVM8 already in uniprot_topts\n",
      "u_id: Q9SIE1 already in uniprot_topts\n",
      "u_id: Q9SIE1 already in uniprot_topts\n",
      "u_id: Q02635 already in uniprot_topts\n",
      "u_id: Q9S7N2 already in uniprot_topts\n",
      "u_id: D2Z027 already in uniprot_topts\n",
      "u_id: Q5JD05 already in uniprot_topts\n",
      "u_id: Q12WB9 already in uniprot_topts\n",
      "u_id: Q58999 already in uniprot_topts\n",
      "u_id: D7E8P3 already in uniprot_topts\n",
      "u_id: Q8PZL9 already in uniprot_topts\n",
      "u_id: Q9SBA5 already in uniprot_topts\n",
      "u_id: Q5JHF1 already in uniprot_topts\n",
      "u_id: Q0K080 already in uniprot_topts\n",
      "u_id: O00329 already in uniprot_topts\n",
      "u_id: P48736 already in uniprot_topts\n",
      "u_id: Q9WZ17 already in uniprot_topts\n",
      "u_id: O43314 already in uniprot_topts\n",
      "u_id: M9XB82 already in uniprot_topts\n",
      "u_id: Q08224 already in uniprot_topts\n",
      "u_id: P55882 already in uniprot_topts\n",
      "u_id: Q5SKG3 already in uniprot_topts\n",
      "u_id: Q9BZQ4 already in uniprot_topts\n",
      "u_id: A8BIC5 already in uniprot_topts\n",
      "u_id: Q59263 already in uniprot_topts\n",
      "u_id: P9WMN3 already in uniprot_topts\n",
      "u_id: P0ACC7 already in uniprot_topts\n",
      "u_id: A5U161 already in uniprot_topts\n",
      "u_id: Q975F9 already in uniprot_topts\n",
      "u_id: Q975F9 already in uniprot_topts\n",
      "u_id: Q975F9 already in uniprot_topts\n",
      "u_id: Q9C5I1 already in uniprot_topts\n",
      "u_id: Q975F9 already in uniprot_topts\n",
      "u_id: P03523 already in uniprot_topts\n",
      "u_id: Q975F9 already in uniprot_topts\n",
      "u_id: O67379 already in uniprot_topts\n",
      "u_id: O29976 already in uniprot_topts\n",
      "u_id: Q5JDA9 already in uniprot_topts\n",
      "u_id: Q8U1Z6 already in uniprot_topts\n",
      "u_id: P0AC78 already in uniprot_topts\n",
      "u_id: O95396 already in uniprot_topts\n",
      "u_id: Q3S3F7 already in uniprot_topts\n",
      "u_id: Q2XV97 already in uniprot_topts\n",
      "u_id: F0QXN6 already in uniprot_topts\n",
      "u_id: Q15166 already in uniprot_topts\n",
      "u_id: Q68FP2 already in uniprot_topts\n",
      "u_id: F0QXN6 already in uniprot_topts\n",
      "u_id: Q15165 already in uniprot_topts\n",
      "u_id: P27169 already in uniprot_topts\n",
      "u_id: Q4J6Z8 already in uniprot_topts\n",
      "u_id: F6LQK7 already in uniprot_topts\n",
      "u_id: Q8TZJ1 already in uniprot_topts\n",
      "u_id: O28511 already in uniprot_topts\n",
      "u_id: D4GP31 already in uniprot_topts\n",
      "u_id: F2X2G0 already in uniprot_topts\n",
      "u_id: A0A1B1WA13 already in uniprot_topts\n",
      "u_id: A0A1D0BQV8 already in uniprot_topts\n",
      "u_id: A0A1B1WA14 already in uniprot_topts\n",
      "u_id: Q97VU2 already in uniprot_topts\n",
      "u_id: F0QXN6 already in uniprot_topts\n",
      "u_id: P27169 already in uniprot_topts\n",
      "u_id: O00748 already in uniprot_topts\n",
      "u_id: P26811 already in uniprot_topts\n",
      "u_id: P0AGG2 already in uniprot_topts\n",
      "u_id: Q9R0X4 already in uniprot_topts\n",
      "u_id: Q8VHK0 already in uniprot_topts\n",
      "u_id: O14734 already in uniprot_topts\n",
      "u_id: P41903 already in uniprot_topts\n",
      "u_id: O00154 already in uniprot_topts\n",
      "u_id: P96273 already in uniprot_topts\n",
      "u_id: P03366 already in uniprot_topts\n",
      "u_id: G4RK44 already in uniprot_topts\n",
      "u_id: B1XLK5 already in uniprot_topts\n",
      "u_id: Q8X270 already in uniprot_topts\n",
      "u_id: Q8VWZ6 already in uniprot_topts\n",
      "u_id: Q0GYS2 already in uniprot_topts\n",
      "u_id: F6IQ86 already in uniprot_topts\n",
      "u_id: V5M269 already in uniprot_topts\n",
      "u_id: F6IQ86 already in uniprot_topts\n",
      "u_id: A2TBB4 already in uniprot_topts\n",
      "u_id: Q6RK08 already in uniprot_topts\n",
      "u_id: Q7CIZ7 already in uniprot_topts\n",
      "u_id: A0A125R6G8 already in uniprot_topts\n",
      "u_id: C4PKL2 already in uniprot_topts\n",
      "u_id: H9TUK6 already in uniprot_topts\n",
      "u_id: Q6YNE9 already in uniprot_topts\n",
      "u_id: D9I0I9 already in uniprot_topts\n",
      "u_id: A5A5J1 already in uniprot_topts\n",
      "u_id: B4X9S4 already in uniprot_topts\n",
      "u_id: Q0GYS2 already in uniprot_topts\n",
      "u_id: A0A0E3VIU5 already in uniprot_topts\n",
      "u_id: O00085 already in uniprot_topts\n",
      "u_id: B7GTV0 already in uniprot_topts\n",
      "u_id: B3VPB3 already in uniprot_topts\n",
      "u_id: B3VPB2 already in uniprot_topts\n",
      "u_id: C0BTR1 already in uniprot_topts\n",
      "u_id: H9TUK5 already in uniprot_topts\n",
      "u_id: Q8KTX7 already in uniprot_topts\n",
      "u_id: B6RGT1 already in uniprot_topts\n",
      "u_id: Q8NEB5 already in uniprot_topts\n",
      "u_id: Q5VZY2 already in uniprot_topts\n",
      "u_id: Q8TZI9 already in uniprot_topts\n",
      "u_id: A0QVM9 already in uniprot_topts\n",
      "u_id: Q97VT7 already in uniprot_topts\n",
      "u_id: F0QXN6 already in uniprot_topts\n",
      "u_id: Q68FP2 already in uniprot_topts\n",
      "u_id: P27169 already in uniprot_topts\n",
      "u_id: Q15166 already in uniprot_topts\n",
      "u_id: Q4J6Z8 already in uniprot_topts\n",
      "u_id: B5BLW5 already in uniprot_topts\n",
      "u_id: P27170 already in uniprot_topts\n",
      "u_id: P27169 already in uniprot_topts\n",
      "u_id: P0A434 already in uniprot_topts\n",
      "u_id: A0A0A0V7L5 already in uniprot_topts\n",
      "u_id: A3LY17 already in uniprot_topts\n",
      "u_id: Q38911 already in uniprot_topts\n",
      "u_id: P93046 already in uniprot_topts\n",
      "u_id: Q3MUH7 already in uniprot_topts\n",
      "u_id: Q9Y223 already in uniprot_topts\n",
      "u_id: A0A142G555 already in uniprot_topts\n",
      "u_id: H5U8W4 already in uniprot_topts\n",
      "u_id: L0EKQ7 already in uniprot_topts\n",
      "u_id: I3ZTN9 already in uniprot_topts\n",
      "u_id: Q9UWR7 already in uniprot_topts\n",
      "u_id: J9SJ08 already in uniprot_topts\n",
      "u_id: Q2V621 already in uniprot_topts\n",
      "u_id: A5IL43 already in uniprot_topts\n",
      "u_id: A0A401ETL2 already in uniprot_topts\n",
      "u_id: Q99YP5 already in uniprot_topts\n",
      "u_id: Q6XK24 already in uniprot_topts\n",
      "u_id: D3VYD8 already in uniprot_topts\n",
      "u_id: P94804 already in uniprot_topts\n",
      "u_id: Q9AJS0 already in uniprot_topts\n",
      "u_id: I3WU34 already in uniprot_topts\n",
      "u_id: Q4J9M2 already in uniprot_topts\n",
      "u_id: P14410 already in uniprot_topts\n",
      "u_id: Q97UH8 already in uniprot_topts\n",
      "u_id: K7T9D9 already in uniprot_topts\n",
      "u_id: F7YTI3 already in uniprot_topts\n",
      "u_id: A0A0K2SQZ9 already in uniprot_topts\n",
      "u_id: A0A0K2SQX3 already in uniprot_topts\n",
      "u_id: A0A0K2SQX4 already in uniprot_topts\n",
      "u_id: E7DXB6 already in uniprot_topts\n",
      "u_id: A0A0A0R2Z1 already in uniprot_topts\n",
      "u_id: A0A401ETL2 already in uniprot_topts\n",
      "u_id: C3W4Y2 already in uniprot_topts\n",
      "u_id: A0A0B4RUW2 already in uniprot_topts\n",
      "u_id: Q000P7 already in uniprot_topts\n",
      "u_id: A8N2G5 already in uniprot_topts\n",
      "u_id: A0A0P0UPC6 already in uniprot_topts\n",
      "u_id: X4YXV3 already in uniprot_topts\n",
      "u_id: A0A0K2GVV7 already in uniprot_topts\n",
      "u_id: P28999 already in uniprot_topts\n",
      "u_id: A8CGP1 already in uniprot_topts\n",
      "u_id: A0A088BCU2 already in uniprot_topts\n",
      "u_id: D2DRB6 already in uniprot_topts\n",
      "u_id: A0A1W6AKT2 already in uniprot_topts\n",
      "u_id: G0L5X4 already in uniprot_topts\n",
      "u_id: A0A346PYQ5 already in uniprot_topts\n",
      "u_id: G3BJX4 already in uniprot_topts\n",
      "u_id: G3BJX3 already in uniprot_topts\n",
      "u_id: B8DZM3 already in uniprot_topts\n",
      "u_id: A0LR95 already in uniprot_topts\n",
      "u_id: G0S9G3 already in uniprot_topts\n",
      "u_id: A0A0A0R2Z1 already in uniprot_topts\n",
      "u_id: A1XM14 already in uniprot_topts\n",
      "u_id: E2DQY5 already in uniprot_topts\n",
      "u_id: Q97YG7 already in uniprot_topts\n",
      "u_id: A3DJS9 already in uniprot_topts\n",
      "u_id: P10478 already in uniprot_topts\n",
      "u_id: A0A0P0ISK8 already in uniprot_topts\n",
      "u_id: O74641 already in uniprot_topts\n",
      "u_id: Q9AI65 already in uniprot_topts\n",
      "u_id: O06532 already in uniprot_topts\n",
      "u_id: P62694 already in uniprot_topts\n",
      "u_id: A7WNU2 already in uniprot_topts\n",
      "u_id: G0RVK1 already in uniprot_topts\n",
      "u_id: P10474 already in uniprot_topts\n",
      "u_id: A7UHH1 already in uniprot_topts\n",
      "u_id: A0A0U4VTN7 already in uniprot_topts\n",
      "u_id: O32106 already in uniprot_topts\n",
      "u_id: P28838 already in uniprot_topts\n",
      "u_id: Q8J2N2 already in uniprot_topts\n",
      "u_id: P25152 already in uniprot_topts\n",
      "u_id: P09960 already in uniprot_topts\n",
      "u_id: F0HXE4 already in uniprot_topts\n",
      "u_id: P40051 already in uniprot_topts\n",
      "u_id: A0A133P8W5 already in uniprot_topts\n",
      "u_id: Q9YBQ2 already in uniprot_topts\n",
      "u_id: P54422 already in uniprot_topts\n",
      "u_id: A0A1S5V3K5 already in uniprot_topts\n",
      "u_id: A7Z5E0 already in uniprot_topts\n",
      "u_id: Q65KZ6 already in uniprot_topts\n",
      "u_id: Q4KIM5 already in uniprot_topts\n",
      "u_id: A9YTT0 already in uniprot_topts\n",
      "u_id: A0A0K0M968 already in uniprot_topts\n",
      "u_id: P9WJN3 already in uniprot_topts\n",
      "u_id: O58235 already in uniprot_topts\n",
      "u_id: Q6F4N1 already in uniprot_topts\n",
      "u_id: Q6F4N1 already in uniprot_topts\n",
      "u_id: A0A077AY47 already in uniprot_topts\n",
      "u_id: A0A088BHP3 already in uniprot_topts\n",
      "u_id: Q6P7S1 already in uniprot_topts\n",
      "u_id: Q8KZT5 already in uniprot_topts\n",
      "u_id: A0A133Q8U5 already in uniprot_topts\n",
      "u_id: O41078 already in uniprot_topts\n",
      "u_id: P17618 already in uniprot_topts\n",
      "u_id: P25539 already in uniprot_topts\n",
      "u_id: Q9P6I7 already in uniprot_topts\n",
      "u_id: P53909 already in uniprot_topts\n",
      "u_id: Q46A53 already in uniprot_topts\n",
      "u_id: A9QXE0 already in uniprot_topts\n",
      "u_id: G5DDB2 already in uniprot_topts\n",
      "u_id: E1A0Z9 already in uniprot_topts\n",
      "u_id: A0A0E3D8K7 already in uniprot_topts\n",
      "u_id: Q08224 already in uniprot_topts\n",
      "u_id: Q583R7 already in uniprot_topts\n",
      "u_id: Q3LIE5 already in uniprot_topts\n",
      "u_id: O83774 already in uniprot_topts\n",
      "u_id: P0AB85 already in uniprot_topts\n",
      "u_id: Q99321 already in uniprot_topts\n",
      "u_id: Q3LIE5 already in uniprot_topts\n",
      "u_id: Q9UKK9 already in uniprot_topts\n",
      "u_id: P97687 already in uniprot_topts\n",
      "u_id: Q5DRK1 already in uniprot_topts\n",
      "u_id: Q9SPM5 already in uniprot_topts\n",
      "u_id: Q9SQG2 already in uniprot_topts\n",
      "u_id: Q99321 already in uniprot_topts\n",
      "u_id: Q96DE0 already in uniprot_topts\n",
      "u_id: Q7Z2E3 already in uniprot_topts\n",
      "u_id: Q6ZI95 already in uniprot_topts\n",
      "u_id: Q9X015 already in uniprot_topts\n",
      "u_id: V9NGI4 already in uniprot_topts\n",
      "u_id: V9NFX1 already in uniprot_topts\n",
      "u_id: V9NGQ2 already in uniprot_topts\n",
      "u_id: W8CQR1 already in uniprot_topts\n",
      "u_id: W8CQB2 already in uniprot_topts\n",
      "u_id: W8CQB1 already in uniprot_topts\n",
      "u_id: W8CR61 already in uniprot_topts\n",
      "u_id: Q56317 already in uniprot_topts\n",
      "u_id: Q56316 already in uniprot_topts\n",
      "u_id: Q8U1P6 already in uniprot_topts\n",
      "u_id: Q8U1P6 already in uniprot_topts\n",
      "u_id: Q684J7 already in uniprot_topts\n",
      "u_id: A3MSD2 already in uniprot_topts\n",
      "u_id: Q6PUG0 already in uniprot_topts\n",
      "u_id: Q6PUF9 already in uniprot_topts\n",
      "u_id: P31658 already in uniprot_topts\n",
      "u_id: Q51732 already in uniprot_topts\n",
      "u_id: Q99497 already in uniprot_topts\n",
      "u_id: P45470 already in uniprot_topts\n",
      "u_id: Q46948 already in uniprot_topts\n",
      "u_id: A4YDS4 already in uniprot_topts\n",
      "u_id: Q8KBL0 already in uniprot_topts\n",
      "u_id: Q5JDJ1 already in uniprot_topts\n",
      "u_id: Q9YGB0 already in uniprot_topts\n",
      "u_id: A0A2U8JPY3 already in uniprot_topts\n",
      "u_id: Q141Z6 already in uniprot_topts\n",
      "u_id: P0AC33 already in uniprot_topts\n",
      "u_id: P14407 already in uniprot_topts\n",
      "u_id: Q6L1T2 already in uniprot_topts\n",
      "u_id: Q8ZL58 already in uniprot_topts\n",
      "u_id: O30012 already in uniprot_topts\n",
      "u_id: D3RXI4 already in uniprot_topts\n",
      "u_id: A9QXE0 already in uniprot_topts\n",
      "u_id: Q485S0 already in uniprot_topts\n",
      "u_id: Q94AF2 already in uniprot_topts\n",
      "u_id: A0A3L6F337 already in uniprot_topts\n",
      "u_id: A0A1B1FUW4 already in uniprot_topts\n",
      "u_id: A0A1Z2QRE2 already in uniprot_topts\n",
      "u_id: G1EH67 already in uniprot_topts\n",
      "u_id: A0A346FH09 already in uniprot_topts\n",
      "u_id: Q9Z6D6 already in uniprot_topts\n",
      "u_id: A0A1B1Y7V0 already in uniprot_topts\n",
      "u_id: A0A1Z2QRD5 already in uniprot_topts\n",
      "u_id: I6P4V6 already in uniprot_topts\n",
      "u_id: Q21FJ0 already in uniprot_topts\n",
      "u_id: A0A0H3WC39 already in uniprot_topts\n",
      "u_id: G0L2Y1 already in uniprot_topts\n",
      "u_id: A0A0A1DYR7 already in uniprot_topts\n",
      "u_id: Q2XSC6 already in uniprot_topts\n",
      "u_id: A0A068L7N8 already in uniprot_topts\n",
      "u_id: Q2XSC6 already in uniprot_topts\n",
      "u_id: A0A0G3BIB7 already in uniprot_topts\n",
      "u_id: A8NU13 already in uniprot_topts\n",
      "u_id: A8NU13 already in uniprot_topts\n",
      "u_id: M1V9Q0 already in uniprot_topts\n",
      "u_id: M1VDX3 already in uniprot_topts\n",
      "u_id: M1VDX3 already in uniprot_topts\n",
      "u_id: Q2XSC6 already in uniprot_topts\n",
      "u_id: E3W202 already in uniprot_topts\n",
      "u_id: E3W203 already in uniprot_topts\n",
      "u_id: E3W204 already in uniprot_topts\n",
      "u_id: E3W202 already in uniprot_topts\n",
      "u_id: E3W203 already in uniprot_topts\n",
      "u_id: E3W204 already in uniprot_topts\n",
      "u_id: Q49SP3 already in uniprot_topts\n",
      "u_id: A0A385J6X8 already in uniprot_topts\n",
      "u_id: A0A1V0E492 already in uniprot_topts\n",
      "u_id: A0A0E3KJK7 already in uniprot_topts\n",
      "u_id: A0A385J6Y9 already in uniprot_topts\n",
      "u_id: P96273 already in uniprot_topts\n",
      "u_id: I4DST8 already in uniprot_topts\n",
      "u_id: P11544 already in uniprot_topts\n",
      "u_id: D5KS97 already in uniprot_topts\n",
      "u_id: P19021 already in uniprot_topts\n",
      "u_id: D4GYV5 already in uniprot_topts\n",
      "u_id: Q84IF9 already in uniprot_topts\n",
      "u_id: Q4FME3 already in uniprot_topts\n",
      "u_id: Q73GL9 already in uniprot_topts\n",
      "u_id: Q9X0Z7 already in uniprot_topts\n",
      "u_id: A0A223K4L6 already in uniprot_topts\n",
      "u_id: Q92MM4 already in uniprot_topts\n",
      "u_id: A6UKD1 already in uniprot_topts\n",
      "u_id: A6X7I5 already in uniprot_topts\n",
      "u_id: A1T974 already in uniprot_topts\n",
      "u_id: H0FT96 already in uniprot_topts\n",
      "u_id: Q7M181 already in uniprot_topts\n",
      "u_id: A3TM80 already in uniprot_topts\n",
      "u_id: F7Y223 already in uniprot_topts\n",
      "u_id: P9WPW9 already in uniprot_topts\n",
      "u_id: P94556 already in uniprot_topts\n",
      "u_id: Q73GL9 already in uniprot_topts\n",
      "u_id: Q9X0Z7 already in uniprot_topts\n",
      "u_id: S0APF4 already in uniprot_topts\n",
      "u_id: H3ZMH5 already in uniprot_topts\n",
      "u_id: B0M3E8 already in uniprot_topts\n",
      "u_id: A0A168GPR3 already in uniprot_topts\n",
      "u_id: Q9WYX9 already in uniprot_topts\n",
      "u_id: A0A0H2URG4 already in uniprot_topts\n",
      "u_id: D9RZ53 already in uniprot_topts\n",
      "u_id: P14060 already in uniprot_topts\n",
      "u_id: P26439 already in uniprot_topts\n",
      "u_id: Q980S1 already in uniprot_topts\n",
      "u_id: E1XUJ2 already in uniprot_topts\n",
      "u_id: Q6RET3 already in uniprot_topts\n",
      "u_id: A0A0S2SWE4 already in uniprot_topts\n",
      "u_id: P33247 already in uniprot_topts\n",
      "u_id: P0AA37 already in uniprot_topts\n",
      "u_id: P31115 already in uniprot_topts\n",
      "u_id: O30012 already in uniprot_topts\n",
      "u_id: Q5GJ60 already in uniprot_topts\n",
      "u_id: Q1EG72 already in uniprot_topts\n",
      "u_id: D7UQ40 already in uniprot_topts\n",
      "u_id: Q96FC9 already in uniprot_topts\n",
      "u_id: Q8CGC7 already in uniprot_topts\n",
      "u_id: P07814 already in uniprot_topts\n",
      "u_id: P00962 already in uniprot_topts\n",
      "u_id: O74725 already in uniprot_topts\n",
      "u_id: A4YGR1 already in uniprot_topts\n",
      "u_id: B3EY95 already in uniprot_topts\n",
      "u_id: B3TZD8 already in uniprot_topts\n",
      "u_id: B3TZD9 already in uniprot_topts\n",
      "u_id: Q8NQE1 already in uniprot_topts\n",
      "u_id: P0AES0 already in uniprot_topts\n",
      "u_id: Q5DM89 already in uniprot_topts\n",
      "u_id: P10378 already in uniprot_topts\n",
      "u_id: Q8DXM9 already in uniprot_topts\n",
      "u_id: D4N891 already in uniprot_topts\n",
      "u_id: D7P1H2 already in uniprot_topts\n",
      "u_id: D4N892 already in uniprot_topts\n",
      "u_id: A6VQP8 already in uniprot_topts\n",
      "u_id: M9UYB0 already in uniprot_topts\n",
      "u_id: Q8VTA5 already in uniprot_topts\n",
      "u_id: Q58178 already in uniprot_topts\n",
      "u_id: O28028 already in uniprot_topts\n",
      "u_id: P9WP79 already in uniprot_topts\n",
      "u_id: Q80WS1 already in uniprot_topts\n",
      "u_id: Q5JFW0 already in uniprot_topts\n",
      "u_id: Q877I4 already in uniprot_topts\n",
      "u_id: Q877I3 already in uniprot_topts\n",
      "u_id: Q877I5 already in uniprot_topts\n"
     ]
    }
   ],
   "source": [
    "uniprot_topts = defaultdict(list)\n",
    "uniprot_ec_num = defaultdict(list)\n",
    "for ec_num, ec_data in brenda_dict['data'].items():\n",
    "    if ec_num == \"spontaneous\" or 'proteins' not in ec_data:\n",
    "        continue\n",
    "    # print(ec_num)\n",
    "    if 'temperature_optimum' in ec_data:\n",
    "        topt_data = get_topt(ec_data, \n",
    "                             get_proteins(ec_data),\n",
    "                             get_organisms(ec_data),\n",
    "                             get_references(ec_data))\n",
    "        for u_id, data in topt_data.items():\n",
    "            data = [tuple(list(d) + [ec_num]) for d in data]\n",
    "            if u_id in uniprot_topts:\n",
    "                print(f\"u_id: {u_id} already in uniprot_topts\")\n",
    "            uniprot_topts[u_id] += data\n",
    "            uniprot_ec_num[u_id] += [ec_num]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "0f954d7c-d9dc-4026-9f30-4fee1e1636d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(40.0, ['22'], '#16# recombinant enzyme <22>'),\n",
       " (40.0, ['17', '25'], '#20# recombinant enzyme <25>')]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_topts['M9UYB0']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "850fb512-3129-447b-a7af-2599d37b906f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1.1.1.1']"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_ec_num['Q6L0S1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d728406a-1fa9-4aad-8d85-7b7a6362468d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fa7ef0f9-3436-4bc2-8f54-2c6b06889421",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_file = \"inputs/brenda/topt_data.tsv\"\n",
    "os.makedirs(\"inputs/brenda\", exist_ok=True)\n",
    "with open(out_file, 'w') as out:\n",
    "    out.write(f\"uniprot_id,topt,references,comments\\n\")\n",
    "    for u_id, rows in uniprot_topts.items():\n",
    "        for row in rows:\n",
    "            topt, refs, comments = row\n",
    "            out.write(f\"{u_id}\\t{topt}\\t{','.join(refs)}\\t{comments}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "71de7f64-f820-465d-9887-51662b8d0733",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove the \"assay at\" entries\n",
    "out_file = \"inputs/brenda/topt_data_no_assay_at.tsv\"\n",
    "os.makedirs(\"inputs/brenda\", exist_ok=True)\n",
    "with open(out_file, 'w') as out:\n",
    "    out.write(f\"uniprot_id\\ttopt\\torg\\tec_num\\treferences\\tcomments\\n\")\n",
    "    for u_id, rows in uniprot_topts.items():\n",
    "        for row in rows:\n",
    "            topt, org, refs, comments, ec_num = row\n",
    "            if comments:\n",
    "                # skip the assay at comments\n",
    "                comments = '; '.join(c for c in comments.split('; ') if 'assay at' not in c)\n",
    "                # if there are no comments left, then skip this entry\n",
    "                if len(comments) == 0:\n",
    "                    continue\n",
    "            else:\n",
    "                # TODO what if there are no comments at all\n",
    "                pass\n",
    "            out.write(f\"{u_id}\\t{topt}\\t{org}\\t{ec_num}\\t{','.join(refs)}\\t{comments}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "32400e13-6857-4007-bf53-e8facbfe5a67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>topt</th>\n",
       "      <th>org</th>\n",
       "      <th>ec_num</th>\n",
       "      <th>references</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q6L0S1</td>\n",
       "      <td>83.0</td>\n",
       "      <td>Picrophilus torridus</td>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>17989975</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P28625</td>\n",
       "      <td>25.0-30.0</td>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>1.1.1.265</td>\n",
       "      <td>32967812</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  uniprot_id       topt                       org     ec_num references  \\\n",
       "0     Q6L0S1       83.0      Picrophilus torridus    1.1.1.1   17989975   \n",
       "1     P28625  25.0-30.0  Saccharomyces cerevisiae  1.1.1.265   32967812   \n",
       "\n",
       "  comments  \n",
       "0     None  \n",
       "1     None  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(out_file, sep='\\t')\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "5abd96e3-4998-4217-b350-87364f1430f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3.2.1.8        118\n",
       "3.2.1.4        110\n",
       "3.2.1.1         74\n",
       "3.2.1.14        65\n",
       "3.1.3.8         64\n",
       "              ... \n",
       "2.4.1.177        1\n",
       "3.2.1.45         1\n",
       "3.2.1.48         1\n",
       "2.4.1.152        1\n",
       "1.14.13.242      1\n",
       "Name: ec_num, Length: 1481, dtype: int64"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.ec_num.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "ce4d7c4f-c83a-44fc-bce3-3beffb299f7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Homo sapiens                  257\n",
       "Saccharolobus solfataricus    104\n",
       "Arabidopsis thaliana          102\n",
       "Escherichia coli               90\n",
       "Bacillus subtilis              87\n",
       "                             ... \n",
       "Streptococcus equi              1\n",
       "Klebsiella sp.                  1\n",
       "Pyrobaculum sp.                 1\n",
       "Bacillus sp. SCSIO 15121        1\n",
       "Chlamydia pneumoniae            1\n",
       "Name: org, Length: 1719, dtype: int64"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.org.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "f456fdda-58ea-4f7a-9dcf-f7b8e67f07fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Q9YBQ2        9\n",
       "Q9ZEU2        7\n",
       "W6CX26        6\n",
       "Q2XSC6        6\n",
       "O58925        6\n",
       "             ..\n",
       "Q9X0D0        1\n",
       "Q6T1W6        1\n",
       "A0A172WCE4    1\n",
       "P35557        1\n",
       "O30538        1\n",
       "Name: uniprot_id, Length: 4910, dtype: int64"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.uniprot_id.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "1d1bb66f-5563-40a8-922a-76d6c7b3bf41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37.0         816\n",
       "30.0         583\n",
       "50.0         550\n",
       "40.0         469\n",
       "60.0         404\n",
       "            ... \n",
       "25.0-85.0      1\n",
       "40.0-65.0      1\n",
       "20.0-45.0      1\n",
       "96.0           1\n",
       "25.0-38.0      1\n",
       "Name: topt, Length: 230, dtype: int64"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.topt.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "02f58c2e-28f9-4b29-9f10-d1603176578e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>topt</th>\n",
       "      <th>org</th>\n",
       "      <th>ec_num</th>\n",
       "      <th>references</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2433</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>98.0</td>\n",
       "      <td>Aeropyrum pernix</td>\n",
       "      <td>3.1.1.1</td>\n",
       "      <td>21643808</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2434</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>90.0</td>\n",
       "      <td>Aeropyrum pernix K1</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>9651372,16143816,18930847,22655081,9651372</td>\n",
       "      <td>#13# at pH 5.4 &lt;11&gt;; #17# wild-type enzyme &lt;29&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2435</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>95.0</td>\n",
       "      <td>Aeropyrum pernix K1</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>9651372,18590836</td>\n",
       "      <td>#13# heat-activated enzyme &lt;11&gt;; #22# wild-typ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2436</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>77.0</td>\n",
       "      <td>Aeropyrum pernix K1</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18590836</td>\n",
       "      <td>#22# mutant DELTAN21 &lt;42&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2437</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>92.0</td>\n",
       "      <td>Aeropyrum pernix K1</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18590836</td>\n",
       "      <td>#22# mutant D15A above 92° &lt;42&gt;; #22# mutant R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2438</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Aeropyrum pernix K1</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18325786,19303951</td>\n",
       "      <td>#23# kinetic assay &lt;43&gt;; #27# recombinant wild...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2439</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Aeropyrum pernix</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18930847,20941418,23099882</td>\n",
       "      <td>#24# wild-type enzyme &lt;49&gt;; #22# activity assa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2440</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Aeropyrum pernix K1</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18930847,20941418,23099882</td>\n",
       "      <td>#24# wild-type enzyme &lt;49&gt;; #22# activity assa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2441</th>\n",
       "      <td>Q9YBQ2</td>\n",
       "      <td>74.0</td>\n",
       "      <td>Aeropyrum pernix</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18325786</td>\n",
       "      <td>#24# wild-type &lt;40&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     uniprot_id  topt                  org    ec_num  \\\n",
       "2433     Q9YBQ2  98.0     Aeropyrum pernix   3.1.1.1   \n",
       "2434     Q9YBQ2  90.0  Aeropyrum pernix K1  3.4.19.1   \n",
       "2435     Q9YBQ2  95.0  Aeropyrum pernix K1  3.4.19.1   \n",
       "2436     Q9YBQ2  77.0  Aeropyrum pernix K1  3.4.19.1   \n",
       "2437     Q9YBQ2  92.0  Aeropyrum pernix K1  3.4.19.1   \n",
       "2438     Q9YBQ2  70.0  Aeropyrum pernix K1  3.4.19.1   \n",
       "2439     Q9YBQ2  80.0     Aeropyrum pernix  3.4.19.1   \n",
       "2440     Q9YBQ2  80.0  Aeropyrum pernix K1  3.4.19.1   \n",
       "2441     Q9YBQ2  74.0     Aeropyrum pernix  3.4.19.1   \n",
       "\n",
       "                                      references  \\\n",
       "2433                                    21643808   \n",
       "2434  9651372,16143816,18930847,22655081,9651372   \n",
       "2435                            9651372,18590836   \n",
       "2436                                    18590836   \n",
       "2437                                    18590836   \n",
       "2438                           18325786,19303951   \n",
       "2439                  18930847,20941418,23099882   \n",
       "2440                  18930847,20941418,23099882   \n",
       "2441                                    18325786   \n",
       "\n",
       "                                               comments  \n",
       "2433                                               None  \n",
       "2434    #13# at pH 5.4 <11>; #17# wild-type enzyme <29>  \n",
       "2435  #13# heat-activated enzyme <11>; #22# wild-typ...  \n",
       "2436                          #22# mutant DELTAN21 <42>  \n",
       "2437  #22# mutant D15A above 92° <42>; #22# mutant R...  \n",
       "2438  #23# kinetic assay <43>; #27# recombinant wild...  \n",
       "2439  #24# wild-type enzyme <49>; #22# activity assa...  \n",
       "2440  #24# wild-type enzyme <49>; #22# activity assa...  \n",
       "2441                                #24# wild-type <40>  "
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.uniprot_id == \"Q9YBQ2\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5bf39365-1174-45c2-aaff-c27d08c30607",
   "metadata": {},
   "source": [
    "### TODO need to handle mutants\n",
    "There's lots of variety here. Not sure how to handle it.\n",
    "\n",
    "Some examples:\n",
    "- P12676\t60.0\tStreptomyces sp. SA-COO\t1.1.3.6\t9153088,9572974,19495743,30421362\t#18# wild-type and S103T, V121A, R135H and V145E mutants <27>; #5# strain ST-200 <37>; #9# reaction with cholesterol, enzyme CO1 <42>\n",
    "- Q93RE1\t60.0\tBurkholderia cepacia\t1.1.3.6\t9153088,9572974,19495743,30421362\t#18# wild-type and S103T, V121A, R135H and V145E mutants <27>; #5# strain ST-200 <37>; #9# reaction with cholesterol, enzyme CO1 <42>\n",
    "- Q6UG02\t50.0\tPhlebiopsis gigantea\t1.1.3.10\t15660220,14730138,29536215\t#26# mutant enzyme K312E with a C-terminal His6-tag <18>; #34# wild-type and mutants K312E, T166A/E539K <68>\n",
    "- A0A2P1MAF4\t50.0\tTrametes sp.\t1.1.3.10\t15660220,14730138,29536215\t#26# mutant enzyme K312E with a C-terminal His6-tag <18>; #34# wild-type and mutants K312E, T166A/E539K <68>\n",
    "- A0A2P1MAF4\t60.0\tTrametes sp.\t1.1.3.10\t29536215\t#34# mutant E359K, K312E/E539K <68>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "3d00c87f-a3b4-45c8-91e2-50042cccf1d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>topt</th>\n",
       "      <th>org</th>\n",
       "      <th>ec_num</th>\n",
       "      <th>references</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3780</th>\n",
       "      <td>Q97X08</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.2.1.4</td>\n",
       "      <td>11118302,17127051,15456402,25387612,25722023,1...</td>\n",
       "      <td>#103# hydrolysis of carboxymethylcellulose &lt;101&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3781</th>\n",
       "      <td>Q97X08</td>\n",
       "      <td>85.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.2.1.4</td>\n",
       "      <td>21557724,11772658,23627611,22577872,22170108,2...</td>\n",
       "      <td>#212,248# wild-type &lt;264,337&gt;; #212,222# chime...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4195</th>\n",
       "      <td>Q97ZS5</td>\n",
       "      <td>100.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.2.2.8</td>\n",
       "      <td>18355316</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4315</th>\n",
       "      <td>Q97VD6</td>\n",
       "      <td>90.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>9651372,16143816,18930847,22655081,9651372</td>\n",
       "      <td>#13# at pH 5.4 &lt;11&gt;; #17# wild-type enzyme &lt;29&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4319</th>\n",
       "      <td>Q7LX61</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.4.19.1</td>\n",
       "      <td>18930847,20941418,23099882</td>\n",
       "      <td>#24# wild-type enzyme &lt;49&gt;; #22# activity assa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4450</th>\n",
       "      <td>Q97YX7</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.4.23.42</td>\n",
       "      <td>24566144,19508193</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4451</th>\n",
       "      <td>Q97WS1</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.4.23.42</td>\n",
       "      <td>24566144,19508193</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4598</th>\n",
       "      <td>P95896</td>\n",
       "      <td>95.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.5.1.4</td>\n",
       "      <td>11453462,16243781</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4772</th>\n",
       "      <td>Q97YV9</td>\n",
       "      <td>50.0-60.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.1.11</td>\n",
       "      <td>12324325</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4816</th>\n",
       "      <td>Q980M3</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.5.3</td>\n",
       "      <td>16081328,21924318,21478358,20400571</td>\n",
       "      <td>#16# GTPase activity of domain 1 &lt;43&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4817</th>\n",
       "      <td>P35021</td>\n",
       "      <td>87.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.5.3</td>\n",
       "      <td>8051132,12586405</td>\n",
       "      <td>#46# SsMT3EF-1alpha &lt;63&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>P35021</td>\n",
       "      <td>91.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.5.3</td>\n",
       "      <td>12586405</td>\n",
       "      <td>#46# SsMT4EF-1alpha &lt;63&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>P35021</td>\n",
       "      <td>95.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.5.3</td>\n",
       "      <td>9893948</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4820</th>\n",
       "      <td>P30925</td>\n",
       "      <td>95.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.5.3</td>\n",
       "      <td>9893948</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4821</th>\n",
       "      <td>Q64214</td>\n",
       "      <td>95.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>3.6.5.3</td>\n",
       "      <td>9893948</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4925</th>\n",
       "      <td>Q9UWU1</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.1.1.19</td>\n",
       "      <td>20876533,18650422</td>\n",
       "      <td>#52# ADC is most active at temperatures above ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4947</th>\n",
       "      <td>Q97WG4</td>\n",
       "      <td>85.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.1.1.31</td>\n",
       "      <td>8772193,15516590</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4964</th>\n",
       "      <td>Q06121</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.1.1.48</td>\n",
       "      <td>23900843</td>\n",
       "      <td>#35# assay &lt;44&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4998</th>\n",
       "      <td>O54288</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.1.2.55</td>\n",
       "      <td>10527934,20736170,19478916,17176250,23315785</td>\n",
       "      <td>#3# mutant enzyme V193A &lt;9&gt;; #1# assay with 2-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4999</th>\n",
       "      <td>O54288</td>\n",
       "      <td>95.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.1.2.55</td>\n",
       "      <td>10527934</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>F9VNB1</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.1.99.3</td>\n",
       "      <td>27626127</td>\n",
       "      <td>#112# recombinant enzyme &lt;165&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5081</th>\n",
       "      <td>Q97U27</td>\n",
       "      <td>100.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.2.1.140</td>\n",
       "      <td>15474024</td>\n",
       "      <td>#3# above &lt;2&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5105</th>\n",
       "      <td>Q97U96</td>\n",
       "      <td>85.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.2.1.5</td>\n",
       "      <td>16849334</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5135</th>\n",
       "      <td>Q97UB2</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.2.1.9</td>\n",
       "      <td>16567425</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5136</th>\n",
       "      <td>Q97UB2</td>\n",
       "      <td>77.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>4.2.1.9</td>\n",
       "      <td>26102631</td>\n",
       "      <td>#22# dehydration of D-gluconate &lt;39&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5641</th>\n",
       "      <td>Q7LX98</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>5.4.99.15</td>\n",
       "      <td>9063973,11164309,15150700,9827330,29625930</td>\n",
       "      <td>#17# recombinant enzyme &lt;31&gt;; #18# recombinant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5722</th>\n",
       "      <td>Q9UXG1</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>5.6.2.4</td>\n",
       "      <td>18056710,17683280,21361871,20118258,11821426,2...</td>\n",
       "      <td>#43# the assay is carried out at either 60°C o...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5723</th>\n",
       "      <td>P95949</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>5.6.2.4</td>\n",
       "      <td>18056710,17683280,21361871,20118258,11821426,2...</td>\n",
       "      <td>#43# the assay is carried out at either 60°C o...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5724</th>\n",
       "      <td>D0KN27</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>5.6.2.4</td>\n",
       "      <td>18056710,17683280,21361871,20118258,11821426,2...</td>\n",
       "      <td>#43# the assay is carried out at either 60°C o...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5725</th>\n",
       "      <td>D0KN27</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Saccharolobus solfataricus</td>\n",
       "      <td>5.6.2.4</td>\n",
       "      <td>18056710</td>\n",
       "      <td>#43# the assay is carried out at either 60°C o...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     uniprot_id       topt                         org     ec_num  \\\n",
       "3780     Q97X08       80.0  Saccharolobus solfataricus    3.2.1.4   \n",
       "3781     Q97X08       85.0  Saccharolobus solfataricus    3.2.1.4   \n",
       "4195     Q97ZS5      100.0  Saccharolobus solfataricus    3.2.2.8   \n",
       "4315     Q97VD6       90.0  Saccharolobus solfataricus   3.4.19.1   \n",
       "4319     Q7LX61       80.0  Saccharolobus solfataricus   3.4.19.1   \n",
       "4450     Q97YX7       70.0  Saccharolobus solfataricus  3.4.23.42   \n",
       "4451     Q97WS1       70.0  Saccharolobus solfataricus  3.4.23.42   \n",
       "4598     P95896       95.0  Saccharolobus solfataricus    3.5.1.4   \n",
       "4772     Q97YV9  50.0-60.0  Saccharolobus solfataricus   3.6.1.11   \n",
       "4816     Q980M3       50.0  Saccharolobus solfataricus    3.6.5.3   \n",
       "4817     P35021       87.0  Saccharolobus solfataricus    3.6.5.3   \n",
       "4818     P35021       91.0  Saccharolobus solfataricus    3.6.5.3   \n",
       "4819     P35021       95.0  Saccharolobus solfataricus    3.6.5.3   \n",
       "4820     P30925       95.0  Saccharolobus solfataricus    3.6.5.3   \n",
       "4821     Q64214       95.0  Saccharolobus solfataricus    3.6.5.3   \n",
       "4925     Q9UWU1       80.0  Saccharolobus solfataricus   4.1.1.19   \n",
       "4947     Q97WG4       85.0  Saccharolobus solfataricus   4.1.1.31   \n",
       "4964     Q06121       75.0  Saccharolobus solfataricus   4.1.1.48   \n",
       "4998     O54288       70.0  Saccharolobus solfataricus   4.1.2.55   \n",
       "4999     O54288       95.0  Saccharolobus solfataricus   4.1.2.55   \n",
       "5036     F9VNB1       45.0  Saccharolobus solfataricus   4.1.99.3   \n",
       "5081     Q97U27      100.0  Saccharolobus solfataricus  4.2.1.140   \n",
       "5105     Q97U96       85.0  Saccharolobus solfataricus    4.2.1.5   \n",
       "5135     Q97UB2       80.0  Saccharolobus solfataricus    4.2.1.9   \n",
       "5136     Q97UB2       77.0  Saccharolobus solfataricus    4.2.1.9   \n",
       "5641     Q7LX98       75.0  Saccharolobus solfataricus  5.4.99.15   \n",
       "5722     Q9UXG1       60.0  Saccharolobus solfataricus    5.6.2.4   \n",
       "5723     P95949       60.0  Saccharolobus solfataricus    5.6.2.4   \n",
       "5724     D0KN27       60.0  Saccharolobus solfataricus    5.6.2.4   \n",
       "5725     D0KN27       45.0  Saccharolobus solfataricus    5.6.2.4   \n",
       "\n",
       "                                             references  \\\n",
       "3780  11118302,17127051,15456402,25387612,25722023,1...   \n",
       "3781  21557724,11772658,23627611,22577872,22170108,2...   \n",
       "4195                                           18355316   \n",
       "4315         9651372,16143816,18930847,22655081,9651372   \n",
       "4319                         18930847,20941418,23099882   \n",
       "4450                                  24566144,19508193   \n",
       "4451                                  24566144,19508193   \n",
       "4598                                  11453462,16243781   \n",
       "4772                                           12324325   \n",
       "4816                16081328,21924318,21478358,20400571   \n",
       "4817                                   8051132,12586405   \n",
       "4818                                           12586405   \n",
       "4819                                            9893948   \n",
       "4820                                            9893948   \n",
       "4821                                            9893948   \n",
       "4925                                  20876533,18650422   \n",
       "4947                                   8772193,15516590   \n",
       "4964                                           23900843   \n",
       "4998       10527934,20736170,19478916,17176250,23315785   \n",
       "4999                                           10527934   \n",
       "5036                                           27626127   \n",
       "5081                                           15474024   \n",
       "5105                                           16849334   \n",
       "5135                                           16567425   \n",
       "5136                                           26102631   \n",
       "5641         9063973,11164309,15150700,9827330,29625930   \n",
       "5722  18056710,17683280,21361871,20118258,11821426,2...   \n",
       "5723  18056710,17683280,21361871,20118258,11821426,2...   \n",
       "5724  18056710,17683280,21361871,20118258,11821426,2...   \n",
       "5725                                           18056710   \n",
       "\n",
       "                                               comments  \n",
       "3780   #103# hydrolysis of carboxymethylcellulose <101>  \n",
       "3781  #212,248# wild-type <264,337>; #212,222# chime...  \n",
       "4195                                               None  \n",
       "4315    #13# at pH 5.4 <11>; #17# wild-type enzyme <29>  \n",
       "4319  #24# wild-type enzyme <49>; #22# activity assa...  \n",
       "4450                                               None  \n",
       "4451                                               None  \n",
       "4598                                               None  \n",
       "4772                                               None  \n",
       "4816              #16# GTPase activity of domain 1 <43>  \n",
       "4817                           #46# SsMT3EF-1alpha <63>  \n",
       "4818                           #46# SsMT4EF-1alpha <63>  \n",
       "4819                                               None  \n",
       "4820                                               None  \n",
       "4821                                               None  \n",
       "4925  #52# ADC is most active at temperatures above ...  \n",
       "4947                                               None  \n",
       "4964                                    #35# assay <44>  \n",
       "4998  #3# mutant enzyme V193A <9>; #1# assay with 2-...  \n",
       "4999                                               None  \n",
       "5036                     #112# recombinant enzyme <165>  \n",
       "5081                                      #3# above <2>  \n",
       "5105                                               None  \n",
       "5135                                               None  \n",
       "5136               #22# dehydration of D-gluconate <39>  \n",
       "5641  #17# recombinant enzyme <31>; #18# recombinant...  \n",
       "5722  #43# the assay is carried out at either 60°C o...  \n",
       "5723  #43# the assay is carried out at either 60°C o...  \n",
       "5724  #43# the assay is carried out at either 60°C o...  \n",
       "5725  #43# the assay is carried out at either 60°C o...  "
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.org == \"Saccharolobus solfataricus\"].tail(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "39a70168-58b8-4ca1-894d-7cd432b805b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>topt</th>\n",
       "      <th>org</th>\n",
       "      <th>ec_num</th>\n",
       "      <th>references</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2916</th>\n",
       "      <td>O66037</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Bacillus sp. (in: Bacteria)</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>10679211,9821286,12693443,11594462,15935567,15...</td>\n",
       "      <td>#2,181# recombinant enzyme &lt;148,225&gt;; #1# var....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2917</th>\n",
       "      <td>O66037</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Bacillus sp. (in: Bacteria)</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2918</th>\n",
       "      <td>O66037</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Bacillus sp. (in: Bacteria)</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>22775008</td>\n",
       "      <td>#175# wild-type &lt;180&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2919</th>\n",
       "      <td>O66037</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Bacillus sp. (in: Bacteria)</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,14514072,12693443,12405788,15935567,16...</td>\n",
       "      <td>#3# immobilized enzyme &lt;93&gt;; #17# phytase LP11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2920</th>\n",
       "      <td>F1DH12</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Niallia nealsonii</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2921</th>\n",
       "      <td>G8GYH6</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Aspergillus niger</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2922</th>\n",
       "      <td>Q5XNQ9</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Aspergillus fumigatus</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2923</th>\n",
       "      <td>Q84B22</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2924</th>\n",
       "      <td>Q84B22</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2925</th>\n",
       "      <td>Q6T9Z6</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Aspergillus niger</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2926</th>\n",
       "      <td>D9D7K9</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Yersinia enterocolitica</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9449782,9603817,9821286,12111145,12693443,1127...</td>\n",
       "      <td>#104# strain 92 &lt;49&gt;; #180# recombinant His-ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2927</th>\n",
       "      <td>D9D7K9</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Yersinia enterocolitica</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,14727093,12693443,17897689,18815903,19...</td>\n",
       "      <td>#155# recombinant enzyme &lt;149&gt;; #153# wild-typ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2928</th>\n",
       "      <td>D9D7K9</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Yersinia enterocolitica</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2929</th>\n",
       "      <td>Q6U677</td>\n",
       "      <td>40.0-50.0</td>\n",
       "      <td>Obesumbacterium proteus</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>15251209</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2930</th>\n",
       "      <td>A0A2H4W727</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Morchella importuna</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>12693443,29017237</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2931</th>\n",
       "      <td>O31097</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2932</th>\n",
       "      <td>A2TEY6</td>\n",
       "      <td>37.0-55.0</td>\n",
       "      <td>Penicillium sp.</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>20826112</td>\n",
       "      <td>#154# broad optimum, mutant T11A/G56E/L65F/Q14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2933</th>\n",
       "      <td>A2TEY6</td>\n",
       "      <td>37.0-50.0</td>\n",
       "      <td>Penicillium sp.</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>20826112</td>\n",
       "      <td>#154# broad optimum, mutant T11A/H37Y/G56E/L65...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2934</th>\n",
       "      <td>H9C9P0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,14514072,12693443,12405788,15935567,16...</td>\n",
       "      <td>#3# immobilized enzyme &lt;93&gt;; #17# phytase LP11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2935</th>\n",
       "      <td>Q1KNE3</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Aspergillus oryzae</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,14514072,12693443,12405788,15935567,16...</td>\n",
       "      <td>#3# immobilized enzyme &lt;93&gt;; #17# phytase LP11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2936</th>\n",
       "      <td>A0A1S6WME7</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Pseudomonas sp.</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,12693443,14629009,11594462,18481215,18...</td>\n",
       "      <td>#3# free enzyme &lt;93&gt;; #83# recombinant enzyme ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2937</th>\n",
       "      <td>G0XZM1</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Janthinobacterium sp.</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,14727093,12693443,17897689,18815903,19...</td>\n",
       "      <td>#155# recombinant enzyme &lt;149&gt;; #153# wild-typ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2938</th>\n",
       "      <td>G4WZU4</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Serratia sp.</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>9821286,14727093,12693443,17897689,18815903,19...</td>\n",
       "      <td>#155# recombinant enzyme &lt;149&gt;; #153# wild-typ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2939</th>\n",
       "      <td>A7U7M0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>Kodamaea ohmeri</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>12111145,12653285,12693443,15163521,18040741,1...</td>\n",
       "      <td>#70# immobilized enzyme &lt;29&gt;; #193# recombinan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2940</th>\n",
       "      <td>A0A0F7VP60</td>\n",
       "      <td>65.0</td>\n",
       "      <td>Streptomyces sp.</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>12111145,12653285,12693443,15163521,18040741,1...</td>\n",
       "      <td>#70# immobilized enzyme &lt;29&gt;; #193# recombinan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2941</th>\n",
       "      <td>Q93GB6</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2942</th>\n",
       "      <td>Q70E78</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2943</th>\n",
       "      <td>Q9F657</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Bacillus subtilis</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2944</th>\n",
       "      <td>C4S833</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Yersinia mollaretii</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2945</th>\n",
       "      <td>A0A075FFV0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Pantoea sp. 3.5.1</td>\n",
       "      <td>3.1.3.8</td>\n",
       "      <td>2162321,18001837,20376636,20601512,20693453,21...</td>\n",
       "      <td>#7# mutants C215S, C264G &lt;124&gt;; #104# recombin...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      uniprot_id       topt                          org   ec_num  \\\n",
       "2916      O66037       60.0  Bacillus sp. (in: Bacteria)  3.1.3.8   \n",
       "2917      O66037       55.0  Bacillus sp. (in: Bacteria)  3.1.3.8   \n",
       "2918      O66037       80.0  Bacillus sp. (in: Bacteria)  3.1.3.8   \n",
       "2919      O66037       50.0  Bacillus sp. (in: Bacteria)  3.1.3.8   \n",
       "2920      F1DH12       55.0            Niallia nealsonii  3.1.3.8   \n",
       "2921      G8GYH6       55.0            Aspergillus niger  3.1.3.8   \n",
       "2922      Q5XNQ9       55.0        Aspergillus fumigatus  3.1.3.8   \n",
       "2923      Q84B22       55.0            Bacillus subtilis  3.1.3.8   \n",
       "2924      Q84B22       37.0            Bacillus subtilis  3.1.3.8   \n",
       "2925      Q6T9Z6       55.0            Aspergillus niger  3.1.3.8   \n",
       "2926      D9D7K9       55.0      Yersinia enterocolitica  3.1.3.8   \n",
       "2927      D9D7K9       45.0      Yersinia enterocolitica  3.1.3.8   \n",
       "2928      D9D7K9       37.0      Yersinia enterocolitica  3.1.3.8   \n",
       "2929      Q6U677  40.0-50.0      Obesumbacterium proteus  3.1.3.8   \n",
       "2930  A0A2H4W727       25.0          Morchella importuna  3.1.3.8   \n",
       "2931      O31097       37.0            Bacillus subtilis  3.1.3.8   \n",
       "2932      A2TEY6  37.0-55.0              Penicillium sp.  3.1.3.8   \n",
       "2933      A2TEY6  37.0-50.0              Penicillium sp.  3.1.3.8   \n",
       "2934      H9C9P0       50.0            Bacillus subtilis  3.1.3.8   \n",
       "2935      Q1KNE3       50.0           Aspergillus oryzae  3.1.3.8   \n",
       "2936  A0A1S6WME7       40.0              Pseudomonas sp.  3.1.3.8   \n",
       "2937      G0XZM1       45.0        Janthinobacterium sp.  3.1.3.8   \n",
       "2938      G4WZU4       45.0                 Serratia sp.  3.1.3.8   \n",
       "2939      A7U7M0       65.0              Kodamaea ohmeri  3.1.3.8   \n",
       "2940  A0A0F7VP60       65.0             Streptomyces sp.  3.1.3.8   \n",
       "2941      Q93GB6       37.0            Bacillus subtilis  3.1.3.8   \n",
       "2942      Q70E78       37.0            Bacillus subtilis  3.1.3.8   \n",
       "2943      Q9F657       37.0            Bacillus subtilis  3.1.3.8   \n",
       "2944      C4S833       37.0          Yersinia mollaretii  3.1.3.8   \n",
       "2945  A0A075FFV0       37.0            Pantoea sp. 3.5.1  3.1.3.8   \n",
       "\n",
       "                                             references  \\\n",
       "2916  10679211,9821286,12693443,11594462,15935567,15...   \n",
       "2917  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2918                                           22775008   \n",
       "2919  9821286,14514072,12693443,12405788,15935567,16...   \n",
       "2920  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2921  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2922  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2923  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2924  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2925  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2926  9449782,9603817,9821286,12111145,12693443,1127...   \n",
       "2927  9821286,14727093,12693443,17897689,18815903,19...   \n",
       "2928  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2929                                           15251209   \n",
       "2930                                  12693443,29017237   \n",
       "2931  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2932                                           20826112   \n",
       "2933                                           20826112   \n",
       "2934  9821286,14514072,12693443,12405788,15935567,16...   \n",
       "2935  9821286,14514072,12693443,12405788,15935567,16...   \n",
       "2936  9821286,12693443,14629009,11594462,18481215,18...   \n",
       "2937  9821286,14727093,12693443,17897689,18815903,19...   \n",
       "2938  9821286,14727093,12693443,17897689,18815903,19...   \n",
       "2939  12111145,12653285,12693443,15163521,18040741,1...   \n",
       "2940  12111145,12653285,12693443,15163521,18040741,1...   \n",
       "2941  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2942  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2943  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2944  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "2945  2162321,18001837,20376636,20601512,20693453,21...   \n",
       "\n",
       "                                               comments  \n",
       "2916  #2,181# recombinant enzyme <148,225>; #1# var....  \n",
       "2917  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2918                              #175# wild-type <180>  \n",
       "2919  #3# immobilized enzyme <93>; #17# phytase LP11...  \n",
       "2920  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2921  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2922  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2923  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2924  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2925  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2926  #104# strain 92 <49>; #180# recombinant His-ta...  \n",
       "2927  #155# recombinant enzyme <149>; #153# wild-typ...  \n",
       "2928  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2929                                               None  \n",
       "2930                                               None  \n",
       "2931  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2932  #154# broad optimum, mutant T11A/G56E/L65F/Q14...  \n",
       "2933  #154# broad optimum, mutant T11A/H37Y/G56E/L65...  \n",
       "2934  #3# immobilized enzyme <93>; #17# phytase LP11...  \n",
       "2935  #3# immobilized enzyme <93>; #17# phytase LP11...  \n",
       "2936  #3# free enzyme <93>; #83# recombinant enzyme ...  \n",
       "2937  #155# recombinant enzyme <149>; #153# wild-typ...  \n",
       "2938  #155# recombinant enzyme <149>; #153# wild-typ...  \n",
       "2939  #70# immobilized enzyme <29>; #193# recombinan...  \n",
       "2940  #70# immobilized enzyme <29>; #193# recombinan...  \n",
       "2941  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2942  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2943  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2944  #7# mutants C215S, C264G <124>; #104# recombin...  \n",
       "2945  #7# mutants C215S, C264G <124>; #104# recombin...  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.ec_num == \"3.1.3.8\"].tail(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "1d612602-7d63-435e-82dd-454c9b19dcfb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(list,\n",
       "            {'Q6L0S1': [83.0, ['173'], None],\n",
       "             'A3MVR8': [80.0, ['234', '236', '285'], '#109# assay at <234>'],\n",
       "             'D4GSN2': [80.0, ['234', '236', '285'], '#109# assay at <234>'],\n",
       "             'P39462': [80.0, ['234', '236', '285'], '#109# assay at <234>'],\n",
       "             'P28625': ['25.0-30.0', ['120', '307'], '#11# assay at <120>'],\n",
       "             'B2KJ46': [35.0,\n",
       "              ['122', '168', '196', '231', '284'],\n",
       "              '#11# immobilized enzyme <196>; #11# Zn-ADH and Co-ADH <122>; #46# reductive reaction, recombinant enzymes expressed from Saccharomyces cerevisiae and Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Saccharomyces cerevisiae <231>; #11# reduction of glycolaldehyde, furfural, butyraldehyde, and propionaldehyde <284>'],\n",
       "             'G2I689': [30.0,\n",
       "              ['84',\n",
       "               '144',\n",
       "               '182',\n",
       "               '185',\n",
       "               '205',\n",
       "               '224',\n",
       "               '284',\n",
       "               '287',\n",
       "               '296',\n",
       "               '303'],\n",
       "              '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>'],\n",
       "             'P00330': [30.0,\n",
       "              ['84',\n",
       "               '144',\n",
       "               '182',\n",
       "               '185',\n",
       "               '205',\n",
       "               '224',\n",
       "               '284',\n",
       "               '287',\n",
       "               '296',\n",
       "               '303'],\n",
       "              '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>',\n",
       "              20.0,\n",
       "              ['299'],\n",
       "              '#120# assay at <299>'],\n",
       "             'A0A380WRS7': [30.0,\n",
       "              ['84',\n",
       "               '144',\n",
       "               '182',\n",
       "               '185',\n",
       "               '205',\n",
       "               '224',\n",
       "               '284',\n",
       "               '287',\n",
       "               '296',\n",
       "               '303'],\n",
       "              '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>'],\n",
       "             'A0A0U5FNC1': [30.0,\n",
       "              ['84',\n",
       "               '144',\n",
       "               '182',\n",
       "               '185',\n",
       "               '205',\n",
       "               '224',\n",
       "               '284',\n",
       "               '287',\n",
       "               '296',\n",
       "               '303'],\n",
       "              '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>'],\n",
       "             'A0A3T0KHC6': [30.0,\n",
       "              ['84',\n",
       "               '144',\n",
       "               '182',\n",
       "               '185',\n",
       "               '205',\n",
       "               '224',\n",
       "               '284',\n",
       "               '287',\n",
       "               '296',\n",
       "               '303'],\n",
       "              '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>'],\n",
       "             'Q9HIM3': [75.0, ['213', '218'], None],\n",
       "             'Q4J702': [75.0,\n",
       "              ['213', '218'],\n",
       "              None,\n",
       "              65.0,\n",
       "              ['121', '217', '218', '243'],\n",
       "              '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>'],\n",
       "             'Q4J9F2': [75.0,\n",
       "              ['213', '218'],\n",
       "              None,\n",
       "              78.0,\n",
       "              ['217'],\n",
       "              '#127# the reaction rate increases up to 78°C and then decreases rapidly due to thermal inactivation <217>',\n",
       "              65.0,\n",
       "              ['121', '217', '218', '243'],\n",
       "              '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>'],\n",
       "             'Q9Y9P9': [95.0,\n",
       "              ['238'],\n",
       "              '#135# catalytic activity increased up to 95°C <238>'],\n",
       "             'Q8U259': [70.0,\n",
       "              ['82', '118', '126', '144', '181', '205', '225', '229'],\n",
       "              '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>'],\n",
       "             'P00327': [70.0,\n",
       "              ['82', '118', '126', '144', '181', '205', '225', '229'],\n",
       "              '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>',\n",
       "              25.0,\n",
       "              ['106',\n",
       "               '107',\n",
       "               '113',\n",
       "               '115',\n",
       "               '119',\n",
       "               '121',\n",
       "               '124',\n",
       "               '131',\n",
       "               '196',\n",
       "               '228',\n",
       "               '294',\n",
       "               '297'],\n",
       "              '#6,9,11,121# assay at <107,115,119,121,124,228,294,297>; #63# assay at, forward and reverse reaction <113>; #11# free enzyme, at 25°C <196>'],\n",
       "             'Q8L3C9': [70.0,\n",
       "              ['82', '118', '126', '144', '181', '205', '225', '229'],\n",
       "              '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>'],\n",
       "             'Q8TZM9': [100.0, ['138'], None],\n",
       "             'Q5JI09': [90.0, ['242'], None],\n",
       "             'V9SFA1': [45.0,\n",
       "              ['251', '288', '289', '292', '303'],\n",
       "              '#25# assay at <288,289,292>'],\n",
       "             'V9SDP6': [45.0,\n",
       "              ['251', '288', '289', '292', '303'],\n",
       "              '#25# assay at <288,289,292>'],\n",
       "             'H9ZGN0': [22.0,\n",
       "              ['128', '221'],\n",
       "              '#26,130# assay at room temperature <128,221>'],\n",
       "             'V9SCJ1': [55.0,\n",
       "              ['188', '243', '251'],\n",
       "              '#153# oxidation of crotyl aclohol <243>'],\n",
       "             'W6CX26': [55.0,\n",
       "              ['188', '243', '251'],\n",
       "              '#153# oxidation of crotyl aclohol <243>',\n",
       "              65.0,\n",
       "              ['121', '217', '218', '243'],\n",
       "              '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>'],\n",
       "             'E7FHL9': [37.0,\n",
       "              ['117', '290', '293'],\n",
       "              '#43,166,167# assay at <117,290,293>'],\n",
       "             'A4IP64': [60.0,\n",
       "              ['280', '304', '308'],\n",
       "              '#159# enzyme expressed in Pyrococcus furiosus <280>; #5# temperature optimum for oxidation reaction in glycine NaOH buffer using ethanol as the substrate <304>'],\n",
       "             'C7IV28': [60.0,\n",
       "              ['280', '304', '308'],\n",
       "              '#159# enzyme expressed in Pyrococcus furiosus <280>; #5# temperature optimum for oxidation reaction in glycine NaOH buffer using ethanol as the substrate <304>'],\n",
       "             'F9VMI9': [85.0,\n",
       "              ['205', '216'],\n",
       "              '#73# approximately 85°C <205>',\n",
       "              50.0,\n",
       "              ['137', '171', '216', '251'],\n",
       "              '#125# assay at <216>; #104# reaction with ethanol + NAD+ <171>'],\n",
       "             'G3FFC9': [50.0,\n",
       "              ['137', '171', '216', '251'],\n",
       "              '#125# assay at <216>; #104# reaction with ethanol + NAD+ <171>']})"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_topts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f8e43cd2-2926-45cf-9839-54ef9e1db976",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'proteins': ['103'],\n",
       "  'organisms': ['103'],\n",
       "  'references': ['173'],\n",
       "  'num_value': 83.0},\n",
       " {'comment': '#109# assay at <234>',\n",
       "  'proteins': ['163', '136', '109'],\n",
       "  'organisms': ['163', '136', '109'],\n",
       "  'references': ['234', '236', '285'],\n",
       "  'num_value': 80.0},\n",
       " {'organisms': ['11'],\n",
       "  'references': ['3'],\n",
       "  'min_value': 45.0,\n",
       "  'max_value': 50.0},\n",
       " {'comment': '#11# assay at <120>',\n",
       "  'proteins': ['178'],\n",
       "  'organisms': ['11', '178'],\n",
       "  'references': ['120', '307'],\n",
       "  'min_value': 25.0,\n",
       "  'max_value': 30.0},\n",
       " {'comment': '#29# assay at <129>; #11# Cu-ADH <122>; #11# enzyme covalently immobilized to magnetic Fe3O4 nanoparticles via glutaraldehyde <182>',\n",
       "  'organisms': ['11', '53', '57', '29'],\n",
       "  'references': ['99', '122', '129', '182', '278'],\n",
       "  'num_value': 40.0},\n",
       " {'comment': '#11# immobilized enzyme <196>; #11# Zn-ADH and Co-ADH <122>; #46# reductive reaction, recombinant enzymes expressed from Saccharomyces cerevisiae and Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Hansenula polymorpha <231>; #46# recombinant enzyme expressed in Saccharomyces cerevisiae <231>; #11# reduction of glycolaldehyde, furfural, butyraldehyde, and propionaldehyde <284>',\n",
       "  'proteins': ['101'],\n",
       "  'organisms': ['11', '46', '101'],\n",
       "  'references': ['122', '168', '196', '231', '284'],\n",
       "  'num_value': 35.0},\n",
       " {'comment': '#11# soluble enzyme <182>; #46,91,131,165,170,171,174,175# assay at <224,287,296,303>; #105# reduction of N-benzyl-3-pyrrolidinone <185>; #120# approximately 30°C <205>; #11# reduction of formaldehyde <284>',\n",
       "  'proteins': ['170', '120', '171', '174', '175'],\n",
       "  'organisms': ['131',\n",
       "   '170',\n",
       "   '120',\n",
       "   '46',\n",
       "   '105',\n",
       "   '91',\n",
       "   '171',\n",
       "   '174',\n",
       "   '11',\n",
       "   '165',\n",
       "   '175',\n",
       "   '72'],\n",
       "  'references': ['84',\n",
       "   '144',\n",
       "   '182',\n",
       "   '185',\n",
       "   '205',\n",
       "   '224',\n",
       "   '284',\n",
       "   '287',\n",
       "   '296',\n",
       "   '303'],\n",
       "  'num_value': 30.0},\n",
       " {'proteins': ['115', '126', '127'],\n",
       "  'organisms': ['115', '126', '127'],\n",
       "  'references': ['213', '218'],\n",
       "  'num_value': 75.0},\n",
       " {'comment': '#120# assay at <299>',\n",
       "  'proteins': ['120'],\n",
       "  'organisms': ['120'],\n",
       "  'references': ['299'],\n",
       "  'num_value': 20.0},\n",
       " {'comment': '#127# the reaction rate increases up to 78°C and then decreases rapidly due to thermal inactivation <217>',\n",
       "  'proteins': ['127'],\n",
       "  'organisms': ['127'],\n",
       "  'references': ['217'],\n",
       "  'num_value': 78.0},\n",
       " {'comment': '#135# catalytic activity increased up to 95°C <238>',\n",
       "  'proteins': ['135'],\n",
       "  'organisms': ['135'],\n",
       "  'references': ['238'],\n",
       "  'num_value': 95.0},\n",
       " {'comment': '#132# assay at <229>; #54# above <82>; #14# recombinant enzyme <126>; #121# approximately 70°C <205>',\n",
       "  'proteins': ['132', '121', '96'],\n",
       "  'organisms': ['121', '14', '88', '132', '54', '92', '96', '33'],\n",
       "  'references': ['82', '118', '126', '144', '181', '205', '225', '229'],\n",
       "  'num_value': 70.0},\n",
       " {'proteins': ['146'],\n",
       "  'organisms': ['146'],\n",
       "  'references': ['138'],\n",
       "  'num_value': 100.0},\n",
       " {'proteins': ['152'],\n",
       "  'organisms': ['152'],\n",
       "  'references': ['242'],\n",
       "  'num_value': 90.0},\n",
       " {'organisms': ['16', '31'], 'references': ['133', '134'], 'num_value': 33.0},\n",
       " {'comment': '#25# assay at <288,289,292>',\n",
       "  'proteins': ['142', '140'],\n",
       "  'organisms': ['17', '142', '25', '140'],\n",
       "  'references': ['251', '288', '289', '292', '303'],\n",
       "  'num_value': 45.0},\n",
       " {'comment': '#26,130# assay at room temperature <128,221>',\n",
       "  'proteins': ['130'],\n",
       "  'organisms': ['130', '26'],\n",
       "  'references': ['128', '221'],\n",
       "  'num_value': 22.0},\n",
       " {'comment': '#153# oxidation of crotyl aclohol <243>',\n",
       "  'proteins': ['141', '153'],\n",
       "  'organisms': ['28', '141', '153'],\n",
       "  'references': ['188', '243', '251'],\n",
       "  'num_value': 55.0},\n",
       " {'organisms': ['40'],\n",
       "  'references': ['85'],\n",
       "  'min_value': 50.0,\n",
       "  'max_value': 55.0},\n",
       " {'comment': '#43,166,167# assay at <117,290,293>',\n",
       "  'proteins': ['167'],\n",
       "  'organisms': ['43', '167', '166'],\n",
       "  'references': ['117', '290', '293'],\n",
       "  'num_value': 37.0},\n",
       " {'organisms': ['46'],\n",
       "  'references': ['114'],\n",
       "  'min_value': 30.0,\n",
       "  'max_value': 50.0},\n",
       " {'comment': '#46# reductive reaction, recombinant enzyme expressed from Arxula adeninivorans <231>; #46# recombinant enzyme expressed in Arxula adeninivorans <231>',\n",
       "  'organisms': ['46'],\n",
       "  'references': ['231'],\n",
       "  'num_value': 39.0},\n",
       " {'comment': '#159# enzyme expressed in Pyrococcus furiosus <280>; #5# temperature optimum for oxidation reaction in glycine NaOH buffer using ethanol as the substrate <304>',\n",
       "  'proteins': ['118', '159'],\n",
       "  'organisms': ['5', '118', '159'],\n",
       "  'references': ['280', '304', '308'],\n",
       "  'num_value': 60.0},\n",
       " {'organisms': ['50'],\n",
       "  'references': ['222'],\n",
       "  'min_value': 50.0,\n",
       "  'max_value': 60.0},\n",
       " {'comment': '#6,9,11,121# assay at <107,115,119,121,124,228,294,297>; #63# assay at, forward and reverse reaction <113>; #11# free enzyme, at 25°C <196>',\n",
       "  'proteins': ['121'],\n",
       "  'organisms': ['121', '6', '63', '11', '9', '27', '42'],\n",
       "  'references': ['106',\n",
       "   '107',\n",
       "   '113',\n",
       "   '115',\n",
       "   '119',\n",
       "   '121',\n",
       "   '124',\n",
       "   '131',\n",
       "   '196',\n",
       "   '228',\n",
       "   '294',\n",
       "   '297'],\n",
       "  'num_value': 25.0},\n",
       " {'organisms': ['7'], 'references': ['169'], 'num_value': 73.0},\n",
       " {'comment': '#73# approximately 85°C <205>',\n",
       "  'proteins': ['125'],\n",
       "  'organisms': ['73', '125'],\n",
       "  'references': ['205', '216'],\n",
       "  'num_value': 85.0},\n",
       " {'comment': '#73,126,127# assay at <121,217,218>; #153# reduction of crotonaldehyde <243>',\n",
       "  'proteins': ['126', '153', '127'],\n",
       "  'organisms': ['73', '126', '153', '127'],\n",
       "  'references': ['121', '217', '218', '243'],\n",
       "  'num_value': 65.0},\n",
       " {'comment': '#9,43# assay at <116>',\n",
       "  'organisms': ['9', '43'],\n",
       "  'references': ['116'],\n",
       "  'min_value': 30.0,\n",
       "  'max_value': 37.0},\n",
       " {'comment': '#125# assay at <216>; #104# reaction with ethanol + NAD+ <171>',\n",
       "  'proteins': ['125', '139'],\n",
       "  'organisms': ['125', '139', '97', '104'],\n",
       "  'references': ['137', '171', '216', '251'],\n",
       "  'num_value': 50.0}]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['1.1.1.1']['temperature_optimum']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "155ca3a8-b7e7-4bc3-ac58-22b7f8dabd13",
   "metadata": {},
   "source": [
    "### Check how much pH range data there is"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1abbf526-8740-42af-87c9-72fb6201f9a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'organisms': ['15'],\n",
       "  'references': ['42'],\n",
       "  'min_value': 4.0,\n",
       "  'max_value': 11.0},\n",
       " {'comment': '#27# in the biosynthetic direction <38>',\n",
       "  'proteins': ['27'],\n",
       "  'organisms': ['27'],\n",
       "  'references': ['38'],\n",
       "  'min_value': 5.0,\n",
       "  'max_value': 9.0},\n",
       " {'comment': '#27# in the degradative direction <38>',\n",
       "  'proteins': ['27'],\n",
       "  'organisms': ['27'],\n",
       "  'references': ['38'],\n",
       "  'min_value': 5.0,\n",
       "  'max_value': 7.8},\n",
       " {'comment': '#31# hydrolysis reaction <32>',\n",
       "  'proteins': ['31'],\n",
       "  'organisms': ['31'],\n",
       "  'references': ['32'],\n",
       "  'min_value': 7.8,\n",
       "  'max_value': 8.5},\n",
       " {'comment': '#31# synthesis reaction <32>',\n",
       "  'proteins': ['31'],\n",
       "  'organisms': ['31'],\n",
       "  'references': ['32'],\n",
       "  'min_value': 5.5,\n",
       "  'max_value': 6.9}]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['3.5.2.3']['ph_range']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "5e43ba4e-3d60-4ab8-87ca-4d136805f21a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2842\n"
     ]
    }
   ],
   "source": [
    "ph_range_data = defaultdict(list)\n",
    "ph_max_data = defaultdict(list)\n",
    "comments = defaultdict(list)\n",
    "for ec_number, ec_data in brenda_dict['data'].items():\n",
    "    # if '4.1.1.' not in ec_number:\n",
    "        # continue\n",
    "    if 'ph_range' not in ec_data:\n",
    "        continue\n",
    "    seen_prots = set()\n",
    "    for entry in ec_data['ph_range']:\n",
    "        if 'proteins' in entry:\n",
    "            prot = entry['proteins'][0]\n",
    "            if prot in seen_prots:\n",
    "                continue\n",
    "            seen_prots.add(prot)\n",
    "        # if 'num_value' in entry:\n",
    "            # val = entry['num_value']\n",
    "        if 'min_value' in entry:\n",
    "            # print(entry)\n",
    "            val = entry['min_value']\n",
    "            ph_range_data[ec_number].append(val)\n",
    "        if 'max_value' in entry:\n",
    "            ph_max_data[ec_number].append(entry['max_value'])\n",
    "        if 'comment' in entry:\n",
    "            comments[ec_number].append(entry['comment'])\n",
    "            \n",
    "print(len(ph_range_data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "640fc96d-1b6b-47a3-a0d1-c0b54d0f953b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'comment': '#11# oxidation of L-arabinitol with NAD+, recombinant enzyme, assay at <26>',\n",
       "  'proteins': ['11'],\n",
       "  'organisms': ['11'],\n",
       "  'references': ['26'],\n",
       "  'min_value': 8.0,\n",
       "  'max_value': 9.5},\n",
       " {'comment': '#15# recombinant enzyme <19>',\n",
       "  'proteins': ['13', '15'],\n",
       "  'organisms': ['13', '15'],\n",
       "  'references': ['14', '19'],\n",
       "  'num_value': 9.0},\n",
       " {'comment': '#16,17# assay at <22>',\n",
       "  'proteins': ['16'],\n",
       "  'organisms': ['17', '16'],\n",
       "  'references': ['22'],\n",
       "  'num_value': 9.6},\n",
       " {'comment': '#11# assay at <20>; #15# native enzyme <19>',\n",
       "  'proteins': ['11', '15'],\n",
       "  'organisms': ['11', '2', '15'],\n",
       "  'references': ['9', '19', '20', '21', '25'],\n",
       "  'num_value': 9.5}]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['1.1.1.12']['ph_optimum']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "36c1fdfd-7507-4fc1-b698-d54eefa0a9a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'comment': '#13# 92% of maximal activity at pH 6.0, 70% at pH 8.0, 60% at pH 9.0, and maximal activity at pH 8.0 <35>',\n",
       "  'organisms': ['13'],\n",
       "  'references': ['35'],\n",
       "  'min_value': 6.0,\n",
       "  'max_value': 8.0},\n",
       " {'comment': '#7# pH 4.5: about 55% of maximal activity, pH 8.5: about 70% of maximal activity, reductase activity <21>',\n",
       "  'organisms': ['7'],\n",
       "  'references': ['21'],\n",
       "  'min_value': 4.5,\n",
       "  'max_value': 8.5},\n",
       " {'comment': '#7# pH 7.0: about 40% of maximal activity, pH 10.5: about 60% of maximal activity, oxidase activity <21>',\n",
       "  'organisms': ['7'],\n",
       "  'references': ['21'],\n",
       "  'min_value': 7.0,\n",
       "  'max_value': 10.5},\n",
       " {'comment': '#8# reduction of L-xylulose <8>',\n",
       "  'organisms': ['8'],\n",
       "  'references': ['8'],\n",
       "  'min_value': 5.0,\n",
       "  'max_value': 8.5},\n",
       " {'comment': '#8# oxidation of xylitol <8>',\n",
       "  'organisms': ['8'],\n",
       "  'references': ['8'],\n",
       "  'min_value': 8.0,\n",
       "  'max_value': 11.5}]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_dict['data']['1.1.1.10']['ph_range']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "0e171585-ab9e-4171-9359-b697516ef051",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(list,\n",
       "            {'1.1.1.1': ['#101# pH 4.0: about 80% of maximal activity, pH 7.5: about 60% of maximal activity <168>',\n",
       "              '#103# pH 7: 41% of maximal activity, pH 8: 59% of maximal activity <173>',\n",
       "              '#11# pH 8.2: about 10% of maximal activity, pH 9.5: about 40% of maximal activity <87>',\n",
       "              '#131# reduction reaction activity range, profile overview <224>',\n",
       "              '#121# under the analyzed conditions, the activity of HLADH steadily increases with increasing pH up to pH 9.0 and then decreases at more alkaline conditions, but highest oxidative lactimization activity at pH 11.00 in the coupled assay with NADH oxidase <297>',\n",
       "              '#125# pH 5.5: about 35% of maximal activity, pH 7.5: about 55% of maximal activity, reduction of benzaldehyde <216>',\n",
       "              '#126# pH 6.7: about 40% of maximal activity, pH 8.5: about 85% of maximal activity, oxidation reaction <218>',\n",
       "              '#131# oxidation reaction activity range, profile overview <224>',\n",
       "              '#135# pH 7.0: about 70% of maximal activity, pH 9.6: about 50% of maximal activity, reduction of 2-pentanone <238>',\n",
       "              '#152# 50% of maximum activity for reduction of aldehydes <242>',\n",
       "              '#178# high catalytic activity within the range of pH 4.5-6.5, relative activity of Ymr152wp dropped quickly under alkaline conditions of pH 7.0-9.0, 40% of maximal activity at pH 4.5, 15% at pH 9.0 <307>',\n",
       "              '#22# more than 50% of maximum activity <283>',\n",
       "              '#45# pH 9.0: about 25% of maximal activity, pH 12.0: about 80% of maximal activity, pyrazole-sensitive enzyme <25>',\n",
       "              '#45# pH 10.0: about 50% of maximal activity, pH 11.0: about 75% of maximal activity, pyrazole-insensitive enzyme <25>',\n",
       "              '#49# pH 5.0: about 50% of maximal activity, pH 8.0: about 40% of maximal activity, reduction of phenylacetaldehyde <149>',\n",
       "              '#53# complete inactivation <278>',\n",
       "              '#59# pH 4.0: about 35% of maximal activity, pH 8.0: about 70% of maximal activity <147>',\n",
       "              '#71# pH 9.5: about 40% of maximal activity, pH 11.5: about 85% of maximal activity, ethanol oxidation enzyme form ADH-3 <60>',\n",
       "              '#72# pH 5.2: about 35% of maximal activity, pH 6.8: about 55% of maximal activity, oxidation of ethanol <84>',\n",
       "              '#72# pH 7.2: about 50% of maximal activity, pH 9.2: about 60% of maximal activity <84>',\n",
       "              '#9# pH 8.0: about 40% of maximal activity, pH 10.5: about 85% of maximal activity <96>',\n",
       "              '#9# about 30% of maximal activity at pH 8.0 and at pH 12.0 <14>; #49# pH 8.0: about 50% of maximal activity, pH 12.0: about 50% of maximal activity, oxidation of 2-phenylethanol <149>; #33# pH 8.0: about 70% of maximal activity, pH 12: about 60% of maximal activity <181>'],\n",
       "             '1.1.1.10': ['#13# 92% of maximal activity at pH 6.0, 70% at pH 8.0, 60% at pH 9.0, and maximal activity at pH 8.0 <35>',\n",
       "              '#7# pH 4.5: about 55% of maximal activity, pH 8.5: about 70% of maximal activity, reductase activity <21>',\n",
       "              '#7# pH 7.0: about 40% of maximal activity, pH 10.5: about 60% of maximal activity, oxidase activity <21>',\n",
       "              '#8# reduction of L-xylulose <8>',\n",
       "              '#8# oxidation of xylitol <8>'],\n",
       "             '1.1.1.100': ['#20# pH profile, detailed mechanistic pH-dependence analysis, overview <24>',\n",
       "              '#24# pH profile, pH-dependence analysis, overview <25>',\n",
       "              '#27# about 50% of activity maximum at pH 5.5 and 8.5 <11>',\n",
       "              '#48# pH 4: about 60% of maximal activity, pH 10: about 75% of maximal activity <40>',\n",
       "              '#9# both reaction directions <1>'],\n",
       "             '1.1.1.101': ['#1# about half-maximal activity at pH 5.5 and 8.2 <5>'],\n",
       "             '1.1.1.103': ['#16# 22% of maximal activity at pH 9.2, 50% at pH 9.5, and 52% at pH 12.0 <28>',\n",
       "              '#21# 31% of maximum activity <25>',\n",
       "              '#26# activity range, pH-dependent activity varies in different buffer systems, profile overview <39>'],\n",
       "             '1.1.1.105': ['#13# pH 7.5: 54% of maximal activity, pH 8: maximal activity <19>'],\n",
       "             '1.1.1.106': ['#2# linear activity increase <3>'],\n",
       "             '1.1.1.107': ['#3# not active below <1>',\n",
       "              '#4# pH 8.6: 66% of maximal activity, pH 10.0: 57% of maximal activity, 50 mM CHES-NaOH buffer <5>'],\n",
       "             '1.1.1.11': ['#3# pH 6.0: about 60% of maximum activity, pH 8.5: about 60% of activity maximum <6>',\n",
       "              '#3# at pH 7.5 and 10.0 about 50% relative activity, D-arabinitol <7>',\n",
       "              '#5# enzyme activity increases with increasing pH, measurements above pH 11.0 not feasible <2>',\n",
       "              '#9# oxidation of D-arabitol with NAD+, over 60% of maximal activity within this range, recombinant His-tagged enzyme <13>'],\n",
       "             '1.1.1.115': ['#1# about 55% of maximal activity at pH 9.0 and at pH 10.3 <1>'],\n",
       "             '1.1.1.116': ['#1# about 75% of maximal activity at pH 8.5 and 10.5 <4>',\n",
       "              '#1# about 60% of maximal activity at pH 9.4 and 11.8 <2>',\n",
       "              '#7# 20% of maximal activity at pH 6.5 and pH 9.0, pH profile <9>'],\n",
       "             '1.1.1.117': ['#2# about 80% of maximal activity at pH 7.3 and 9.2 <3>'],\n",
       "             '1.1.1.118': ['#7# pH 5.5: about 50% of maximal activity, pH 8.0: about 40% of maximal activity <9>'],\n",
       "             '1.1.1.119': ['#8# broad range of optimal activity, in Tris-buffer consistently higher than that in phosphate-buffers <1>'],\n",
       "             '1.1.1.12': ['#11# activity range, immobilized recombinant enzyme in nanoflowers, profile overview <26>',\n",
       "              '#15# activity range, inactive above or below <19>',\n",
       "              '#2# more than 60% of maximal activity within <9>'],\n",
       "             '1.1.1.121': ['#1# about 80% of maximal activity at pH 8.0 and 9.8 in Tris/acetate buffer <3>',\n",
       "              '#1# 66% of maximal activity at pH 7.2 and pH 9.3 in Tris/HCl buffer <4>',\n",
       "              '#1# about 80% of maximal activity at pH 8.5 and 10.4 in glycine/NaOH buffer <4>'],\n",
       "             '1.1.1.122': ['#5# active within <3>'],\n",
       "             '1.1.1.123': ['#4# pH 5.5 and 10.2: about 30% of activity maximum <3>',\n",
       "              '#8# pH 5: about 20% of activity maximum, pH 9: about 35% of activity maximum <1>'],\n",
       "             '1.1.1.129': ['#1# 50% of maximal activity at pH 8.9 and pH 11.5 <1>'],\n",
       "             '1.1.1.130': ['#1# 50% of maximal activity at pH 5.2 and pH 8.4 <1>'],\n",
       "             '1.1.1.132': ['#7# optimal pH range, enzyme inactive below pH 7 <20>'],\n",
       "             '1.1.1.136': ['#6# inactive above pH 8.4 <5>'],\n",
       "             '1.1.1.138': ['#11# reverse reaction, 81% of maximal activity at pH 5.0, 22% at pH 7.0, 72% at pH 5.6, and 60% at pH 7.4 <24>',\n",
       "              '#26# reduction of D-fructose, 35% of maximum activity <21>',\n",
       "              '#26# reduction of D-fructose, 50% of maximum activity <21>',\n",
       "              '#7# sharp decline below pH 8.5 and above pH 11.0, forward reaction, recombinant enzyme <27>',\n",
       "              '#9# 95% of its maximal activity for 5 min <10>'],\n",
       "             '1.1.1.14': ['#20# at pH 8 and 11 about 50% of activity maximum, sorbitol oxidation <3>',\n",
       "              '#20# at pH 4.7 and 8 about 50% of activity maximum, D-fructose reduction <3>',\n",
       "              '#20# less than 70% of maximal activity above and below, sorbitol oxidation <17>',\n",
       "              '#21# 40% of maximal activity at 7 and 70% of maximal activity at 10, sorbitol oxidation <13>',\n",
       "              '#23# less than 20% of maximal activity below pH 7.5, 100% at pH 9.0, sorbitol oxidation <28>',\n",
       "              '#23# less than 20% of maximal activity below pH 5, less than 80% above pH 7.5, fructose reduction <28>',\n",
       "              '#7# at pH 6 about 70% of maximal activity, at pH 9 about 40%, fructose reduction <8>',\n",
       "              '#7# at pH 7.5 about 70% of maximal activity, at pH 10 about 100%, sorbitol oxidation <8>',\n",
       "              '#7# at pH 7.5 and pH 9.5 about 70% of maximal activity, sorbitol oxidation <5>',\n",
       "              '#7# at pH 5.0 and pH 7.0 about 60% of maximal activity, fructose reduction <5>; #20# 30% of maximal activity at pH 5.0 and 60% of maximal activity at pH 7.0, fructose reduction <17>',\n",
       "              '#9# less than 50% of maximal activity above and below, sugar reduction <19>',\n",
       "              '#9# less than 50% of maximal activity above and below, sorbitol oxidation <19>'],\n",
       "             '1.1.1.140': ['#20# pH 2.5: about 40% of maximal activity, pH 5.5: about 45% of maximal activity <21>',\n",
       "              '#6# pH 5.7: about 50% of activity maximum, pH 7.5: about 35% of activity maximum <4>'],\n",
       "             '1.1.1.141': ['#4# strong increase in activity between pH 8.0 and 9.5, sharp decrease above <8>'],\n",
       "             '1.1.1.144': ['#1# pH 5: about 35% of activity maximum, pH 8: about 70% of activity maximum, NADH + perillyl aldehyde <1>',\n",
       "              '#1# at pH 6.3 and 10.5: about 10% of activity maximum, perillyl alcohol + NAD+ <1>'],\n",
       "             '1.1.1.145': ['#31# pH 6.5: about 55% of maximal activity, pH 10.0: about 45% of maximal activity <97>',\n",
       "              '#4# 52% of maximal activity <36>',\n",
       "              '#4# 37% of maximal activity at pH 7.4 <37>; #25,26# structure motif-dependent pH dependence, overview <36>',\n",
       "              '#5# approx. 50% of maximal activity at pH 4.0 and pH 10.0 <43>'],\n",
       "             '1.1.1.149': ['#12# steroid oxidation e.g. cortisol with Tris hydrochloride, NAD+ <3>',\n",
       "              '#12# steroid reduction e.g. 20alpha-dihydrocortisol with MES, MOPS or sodium phosphate, NADH <3>',\n",
       "              '#4# pH 6.8 favouring progesterone to 20alpha-hydroxyprogesterone reaction and pH 8.0 vice versa <8>',\n",
       "              '#8# 25°C, below and above range inactivation <22>'],\n",
       "             '1.1.1.153': ['#21# the enzymatic activity is high under the pH range of 3.0 to 6.0 <51>',\n",
       "              '#3# optimium for activity, highest activity in phosphate buffer <22>',\n",
       "              '#6# pH 5.5: optimum, pH 7.4: in 50 mM Tris-HCl buffer about 80% of maximum activity observed <6>'],\n",
       "             '1.1.1.157': ['#1# pH 5: about 35% of maximal activity, pH 7.5: about 55% of maximal activity, reduction of 3-acetoacetyl-CoA <3>'],\n",
       "             '1.1.1.159': ['#12# NAD reduction <15>'],\n",
       "             '1.1.1.16': ['#6# activity profile, overview. The enzyme shows over 60% of its maximum activity at pH values of 8.0-10.0 <5>'],\n",
       "             '1.1.1.162': ['#1# oxidation of acetoin to 2,3-butanediol <6>'],\n",
       "             '1.1.1.166': ['#1# active within, reductive direction <1>'],\n",
       "             '1.1.1.168': ['#1# at pH 5.2 and 8.0 about 50% of activity maximum <1>',\n",
       "              '#6# more than 40% activity between pH 6.0 and 8.0 <11>'],\n",
       "             '1.1.1.169': ['#3# pH 4.3: about 10% of activity maximum, pH 6.5: about 40% of activity maximum <3>'],\n",
       "             '1.1.1.17': ['#15# for the oxidative and reductive direction of the reactions under conditions where substrate is limiting (kcat/K) or saturating (kcat) <27>',\n",
       "              '#28# activity range, profile, overview <28>',\n",
       "              '#3# no sharp optimum <16>; #26# 65% and 50% of maximal activity, respectively <30>',\n",
       "              '#36# 12% of maximum activity <38>',\n",
       "              '#6# more than 95% of its original activity <13>'],\n",
       "             '1.1.1.179': ['#13# more than 50% of maximal activity <6>'],\n",
       "             '1.1.1.18': ['#6# half-maximal activity at pH 5.5 and pH 9.5 <4>'],\n",
       "             '1.1.1.183': ['#1# below 5 complete and irreversible inactivation after 5 min <1>',\n",
       "              '#4# activity range, profile overview <3>',\n",
       "              '#4# below 7 loss of activity <3>',\n",
       "              '#15# 68% of maximal activity is maintained at a pH range of pH 8.0-10.0 <14>; #4# 68% of the optimum activity is maintained at a pH range of 8.0-10.0 <14>',\n",
       "              '#6# over 72% of maximal activity at pH 8.5 and pH 10.0 <5>',\n",
       "              '#8# isoforms geraniol-DH I and geraniol-DH II show less than 40% activity below pH 6.5. Isoform geraniol-DH II shows no activity at pH 10.0, while isoform geraniol-DH I shows about 50% activity. Both isoforms are inactive at pH 6.0 and below <8>'],\n",
       "             '1.1.1.184': ['#15# the enzyme shows more than 70% activity between pH 6.5 and 8.0 <108>',\n",
       "              '#16# more than 80% of maximum activity <93>',\n",
       "              '#36# 15% of maximum activity <65>',\n",
       "              '#45# more than 90% of maximum activity <82>',\n",
       "              '#26# more than 80% activity at pH 6.0 and pH 8.0 <34>; #5# more than 50% activity between pH 6.0 and 8.0. The activity decreases rapidly when pH is below 6.0 or above 8.0 <110>; #14# more than 75% activity between pH 6.0 and 8.0 <113>'],\n",
       "             '1.1.1.195': ['#11# coniferyl aldehyde reduction, pH 6.0: about 70% of activity maximum, pH 8.0: about 30% of activity maximum <4>',\n",
       "              '#11# coniferyl alcohol oxidation, pH 7.2: about 40% of activity maximum, pH 9.7: about 35% of activity maximum <4>',\n",
       "              '#110# more than 70% of the maximum activity <58>',\n",
       "              '#129# profile overview <86>',\n",
       "              '#130# profile overview <86>',\n",
       "              '#135# activity range, profile overview <86>',\n",
       "              '#140# activity range, the enzyme is completely inactive at pH 5.0 and pH 9.0, profile overview <77>',\n",
       "              '#16# pH 4.5: about 55% of activity maximum, pH 8.5: about 30% of activity maximum <3>'],\n",
       "             '1.1.1.2': ['#10# oxidation reaction, increasing activity from pH 6.0 to 11.0 <62>',\n",
       "              '#14# oxidation reaction <65>',\n",
       "              '#62# more than 50% of maximum activity <105>',\n",
       "              '#78# assay pH <70>',\n",
       "              '#90# pH 6.7: about 80% of maximal activity, pH 9.0: about 80% of maximal activity <95>'],\n",
       "             '1.1.1.200': ['#6# about 50% of activity maximum at pH 8.5 and 10.1 <2>',\n",
       "              '#9# N- and the C-term tagged enzymes. Both recombinant enzymes similarly show optimal activity in a broad pH range in the direction of D-glucose 6-phosphate reduction, and a narrow range (pH 9.0-10.0) in the opposite oxidative direction <7>'],\n",
       "             '1.1.1.201': ['#12# activity range <10>'],\n",
       "             '1.1.1.202': ['#16# activity range, alcohol oxidation, profile overview <40>',\n",
       "              '#18# more than 80% of maximum activity for reduction of aldehyde <37>',\n",
       "              '#19# activity range, alcohol oxidation, profile overview <38>'],\n",
       "             '1.1.1.203': ['#11,15# activity range, profile overview <8>',\n",
       "              '#17# more than 40% activity between pH 6.0 and 8.5 <19>',\n",
       "              '#7# more than 50% activity between pH 6.0 and 9.0 <14>',\n",
       "              '#12,14# activity range, profile overview <8>; #8# activity is constantly increasing in this range, profile overview <8>'],\n",
       "             '1.1.1.205': ['#10# pH 7.5: about 50% of activity maximum, pH 9: about 50% of activity maximum <44>',\n",
       "              '#15# pH 7.5: about 55% of activity maximum, pH 9.5: about 70% of activity maximum <40>',\n",
       "              '#15# pH 7.5: about 60% of activity maximum, pH 8.7: about 40% of activity maximum <40>',\n",
       "              '#31# pH 8.5: about 60% of activity maximum, pH 10: about 75% of activity maximum <33>',\n",
       "              '#78# activity range, profile overview <153>',\n",
       "              '#82# activity range, profile overview <143>',\n",
       "              '#9# pH 7: about 45% of activity maximum, pH 9: about 85% of activity maximum <37>',\n",
       "              '#9# pH 6.7: about 40% of activity maximum, pH 9.0: about 55% of activity maximum <34>'],\n",
       "             '1.1.1.206': ['#28# narrow pH range <15>'],\n",
       "             '1.1.1.207': ['#1# 50% of maximal activity at pH 4.5 and pH 8.5 <3>'],\n",
       "             '1.1.1.208': ['#1# 50% of maximal activity at pH 6.5 and pH 10.5 <3>'],\n",
       "             '1.1.1.21': ['#1# below pH 4.4 and above pH 9.7 less than 10% activity <14>',\n",
       "              '#18# 59% activity at both threshold values <20>',\n",
       "              '#23# below pH 5.0 and above pH 8.0 inactivation at 25°C within 3-6 days <24>',\n",
       "              '#4# at pH 5.5 and pH 7.5 about 50% of activity maximum <7>',\n",
       "              '#41# 60% of maximum activity <197>'],\n",
       "             '1.1.1.213': ['#7# pH profile <17>'],\n",
       "             '1.1.1.214': ['#7# more than 40% of maximum activity <7>'],\n",
       "             '1.1.1.215': ['#19# no activity above <17>',\n",
       "              '#5# about half-maximal activity at pH 5.0 and 9.2, reduction of ketogluconate <2>',\n",
       "              '#5# about half-maximal activity at pH 10.8 and 12.5, oxidation of gluconate <2>',\n",
       "              '#7# about half-maximal activity at pH 4.5 and 8.5, reduction of ketogluconate <3>',\n",
       "              '#7# about half-maximal activity at pH 10 and 12, oxidation of gluconate <3>'],\n",
       "             '1.1.1.216': ['#3# reduction of trans,trans-farnesal: 60% of maximal activity at pH 6.5 and at pH 8 <1>',\n",
       "              '#3# oxidation of trans,trans-farnesol: 10% of maximal activity at pH 7.0, 30% of maximal activity at pH 10.2 <1>',\n",
       "              '#6# activity range, profile overview. At below pH 8.5 or above pH 10.0, the enzyme activity declines sharply <4>'],\n",
       "             '1.1.1.217': ['#1# less than 10% activity at pH 4 and 9 <1>'],\n",
       "             '1.1.1.219': ['#14# no significant variations in activity <5>',\n",
       "              '#18# about half-maximal activity at pH 5.1 and 7.2 <2>',\n",
       "              '#32# less than 10% residual activity <54>',\n",
       "              '#33# maximum DFR activities achieved towards dihydroflavonols and flavanones <19>'],\n",
       "             '1.1.1.22': ['#15# 80-83% of maximum activity <60>',\n",
       "              '#19# at pH 6.3 and 8.6 about 50% of activity maximum <8>',\n",
       "              '#25# 46% of maximum acitivity <35>; #43# 37% of maximum activity <57>',\n",
       "              '#40# pH 9.5: about 40% of maximal activity, pH 11.0: about 10% of maximal activity <53>',\n",
       "              '#41# no enzymic activity beyond <62>',\n",
       "              '#5# at pH 7.5 and 9.4 about 50% of activity maximum <4>',\n",
       "              '#50# half-maximal activity at about pH 7.0 and pH 9.5, maximal activity at pH 8.0 <65>',\n",
       "              '#8# at pH 7.8 and 9.4 about 50% of activity maximum <11>'],\n",
       "             '1.1.1.223': ['#1# half maximum activity at both pH <1>'],\n",
       "             '1.1.1.224': ['#1# inactivated at pH below 6 <2>'],\n",
       "             '1.1.1.229': ['#1# less than 10% of activity at both pH <1>'],\n",
       "             '1.1.1.234': ['#3# about half-maximal activity at pH 4.7 and 6.9 <1>'],\n",
       "             '1.1.1.236': ['#4# tolerant to pH changes over a wide range <7>',\n",
       "              '#7# reverse oxidative activities <7>'],\n",
       "             '1.1.1.238': ['#1# about half maximum activity <1>'],\n",
       "             '1.1.1.239': ['#1# sharp decrease above pH 10.0 <9>'],\n",
       "             '1.1.1.24': ['#6# at pH 8.5 and 9.8 about 50% of activity maximum <6>'],\n",
       "             '1.1.1.244': ['#10# activity range, forward reaction, inactive below pH 6.0 and above pH 10.5 <11>',\n",
       "              '#17# pH 8.5: about 50% of maximal activity, pH 10.0: about 90% of maximal activity <18>'],\n",
       "             '1.1.1.247': ['#2# 50% of maximal activity at pH 6.2 and pH 7.6, reduction of codeinone <1>',\n",
       "              '#2# 50% of maximal activity at pH 8.2 and pH 9.8, oxidation of codeine <1>'],\n",
       "             '1.1.1.248': ['#1# 50% of maximal activity at pH 5.8 and at pH 7.8, formationof salutaridine <1>',\n",
       "              '#1# 50% of maximal activity at pH 5.8 and at pH 7.8, formation of (7S)-salutaridinol <1>',\n",
       "              '#17# maximal activity at pH 9.5, sharp decline in activity at pH 10.0, oxidation reaction, recombinant enzyme <2>',\n",
       "              '#18# the enzyme shows a steep decrease by 90% of activity within 1.5 pH units on both sides of the optimum at pH 6.0 for the reduction reaction <3>'],\n",
       "             '1.1.1.25': ['#1# activity range, overview <49>',\n",
       "              '#13# no activity below and above <4>',\n",
       "              '#57# at pH 7.0, the forward reaction catalyzed by Corynebacterium glutamicum SDH proceeds at a rate about 10fold faster than the reverse reaction <43>',\n",
       "              '#6# pH 8.5: about 50% of activity maximum, pH 10.5: about 90% of activity maximum <8>'],\n",
       "             '1.1.1.250': ['#2# pH 8.0: about 40% of maximal activity, pH 8.5: about 60% of maximal activity, pH 10-11: maximal activity <2>',\n",
       "              '#3# 40% and 10% of maximal activity at pH 5.5 and pH 8.5, no activity below pH 4.5 and above pH 9.5 <3>; #4# oxidation of D-arabitol is not observed below pH 4.5 or above pH 9.5 <4>'],\n",
       "             '1.1.1.252': ['#1# pH 7.0: about 95% of maximal activity, pH 8.0: about 80% of maximal activity <7>',\n",
       "              '#1# pH 4.2: about 40% of maximal activity, pH 6.5: about 30% of maximal activity <7>'],\n",
       "             '1.1.1.255': ['#2# not greatly affected between <5>',\n",
       "              '#2# D-mannitol oxidation <6>',\n",
       "              '#2# D-mannose reduction <6>'],\n",
       "             '1.1.1.26': ['#6,7# the optimal pH for hydroxypyruvate reduction is more acidic than that for glyoxylate reduction <11>',\n",
       "              '#9# pH 5.1: about 60% of activity maximum, pH 7.8: about 50% of activity maximum <5>'],\n",
       "             '1.1.1.27': ['#109# pH 5.5: 40% of maximal activity of isoenzyme LDH-B4 and 60% of maximal activity of isoenzyme LDH-A2B2 and LDH-A4, pH 8.0: about 60% of maximal activity of isoenzyme LDH-A4, LDH-B4 and LDH-A4B4, reduction of pyruvate <48>',\n",
       "              '#110# pH 5.4: about 55% of maximal activity, pH 10.8: about 55% of maximal activity, reaction with lactate <45>',\n",
       "              '#110# pH 5.4: about 60% of maximal activity, pH 10.8: about 55% of maximal activity, reaction with pyruvate <45>',\n",
       "              '#111# 20% of maximal activity for oxidation of lactate at pH 8.0, 90% at pH 8.5, and maximal activity at pH 8.5-9.5, inactive at pH 7.5 <130>',\n",
       "              '#118# activity range, inactive at pH 8.0, profile overview <129>; #155# activity range, inactive at pH 8.0, profile overview. High level of activity between pH 4.0 and pH 7.5 <129>',\n",
       "              '#131# optimal pH range for pyruvate reduction and lactate oxidation is 6.5-7.0 and 9.0-10.0, respectively <105>',\n",
       "              '#132,133# LDHB, in Bis-Tris-propane buffer <98>',\n",
       "              '#157# activity range, inactive at pH 4.0 and pH 10.0 <133>',\n",
       "              '#23# pH 5.2: about 35% of maximal activity, pH 7.2: about 50% of maximal activity <6>; #145# LDH activity shows a broad plateau between pH 5.2 and pH 7.2, and no activity at pH 4.8 <87>',\n",
       "              '#30# activity range, the enzyme is inactive below pH 3.0 and above pH 9.0 <88>',\n",
       "              '#34# pH 5.3: about 50% of maximal activity, pH 8.0: about 60% of maximal activity, reduction of pyruvate <11>',\n",
       "              '#34# pH 8.5: about 60% of maximal activity, pH 10.7: about 70% of maximal activity, oxidation of lactate <11>',\n",
       "              '#47# pH 6.0: 85% of maximal activity, pH 8.0: 45% of maximal activity, reduction of pyruvate <30>',\n",
       "              '#47# pH 8.0: 60% of maximal activity, pH 9.6: 45% of maximal activity, oxidation of lactate <30>',\n",
       "              '#50# no activity below pH 4.3 and above pH 8.0 <2>',\n",
       "              '#68# about 55% of maximal activity at pH 5.0 and at pH 7.0 <39>',\n",
       "              '#94# pH 7.7: about 60% of maximal activity, pH 9.5: about 30% of maximal activity, reduction of pyruvate <17>'],\n",
       "             '1.1.1.271': ['#16# activity range <17>'],\n",
       "             '1.1.1.273': ['#5# 50% of maximal activity at pH 6.8 and pH 7.6 respectively <2>'],\n",
       "             '1.1.1.274': ['#2# reduction of 2,5-didehydro-D-gluconate <2>',\n",
       "              '#2# oxidation of 2-keto-L-gulonate <2>'],\n",
       "             '1.1.1.279': ['#1# rapid decrease of activity above <1>'],\n",
       "             '1.1.1.28': ['#20# more than 90% of maximum activity <73>',\n",
       "              '#20# 70% residual activity <73>',\n",
       "              '#21# activity range, profile overview <42>',\n",
       "              '#23# activity range, profile, overview <43>',\n",
       "              '#36# no activity observed <72>',\n",
       "              '#53# isozymes D-LDH1 and D-LDH2, activity range, profile, overview <46>',\n",
       "              '#53# isozyme D-LDH3, activity range, profile, overview <46>; #67# pyruvate reduction reaction, activity range, profile overview <44>',\n",
       "              '#67# lactate oxidation reaction, activity range, profile overview <44>',\n",
       "              '#70# activity range <54>',\n",
       "              '#77# activity range, profile overview <52>'],\n",
       "             '1.1.1.282': ['#8# the rate of activity decreases more rapidly, maximal 25% at pH 7.5, at pH values from 8.5 to 7 when shikimate rather than quinate is used as substrate <4>'],\n",
       "             '1.1.1.284': ['#10# pH 6.8: about 80% of maximal activity, pH 9.3: about 90% of maximal activity <10>',\n",
       "              '#11# active from pH 6.0 to pH 8.5 <9>',\n",
       "              '#21# pH 7.0: about 55% of maximal activity, pH 9.0: about 90% of maximal activity <11>',\n",
       "              '#21# 50% of maximal activity at pH 6.4 and at pH 9.3 <4>',\n",
       "              '#9# pH 6.5: about 70% of maximal activity, pH 9.5: about 80% of maximal activity <10>'],\n",
       "             '1.1.1.286': ['#1# pH-profile, overview <4>'],\n",
       "             '1.1.1.287': ['#3# ketone reduction <3>'],\n",
       "             '1.1.1.289': ['#2# oxidation of D-sorbitol <2>',\n",
       "              '#2# reduction of L-sorbose <2>'],\n",
       "             '1.1.1.292': ['#1# about 50% of maximal activity at pH 5.5 and at pH 8.8 <1>'],\n",
       "             '1.1.1.295': ['#1# pH 6.0: about 50% of maximal activity, pH 10.0: about 60% of maximal activity <1>'],\n",
       "             '1.1.1.298': ['#2# activity range <2>',\n",
       "              '#2# about 65% of maximum activity <2>',\n",
       "              '#2# about 30% of maximum acticity <2>',\n",
       "              '#6# half-maximal activity at pH 6.5 and pH 8.5 <6>; #3# more than 50% of maximla acitivty within <3>'],\n",
       "             '1.1.1.299': ['#2# oxidation of malate <1>'],\n",
       "             '1.1.1.3': ['#35# BsHSD is maximally active in L-HSE oxidation at pH 9.0 and is the least active at pH 7.0 with only 1.1% of the maximal activity <48>',\n",
       "              '#36# over 60% of maximal activity within this range, profile overview <46>'],\n",
       "             '1.1.1.30': ['#10# oxidation of beta-hydroxybutyrate <28>',\n",
       "              '#15# acetoacetate reduction, 50% activity at pH 5.0, 66% at pH 8.6 <4>',\n",
       "              '#16# acetoacetate reduction <23>',\n",
       "              '#23# optimal pH of BDH3 is 8.5 in the oxidation reaction and 6.5 in the reduction reaction <48>'],\n",
       "             '1.1.1.304': ['#4# reduction of diacetyl <10>',\n",
       "              '#6# 5 min, 30% loss of activity <2>',\n",
       "              '#6# 5 min, 20% loss of activity <2>',\n",
       "              '#6# stable within <2>',\n",
       "              '#6# 5 min, 60% loss of activity <2>'],\n",
       "             '1.1.1.305': ['#3# activity range, profile overview <7>'],\n",
       "             '1.1.1.307': ['#23# activity range <71>; #10# about 90% of maximal activity at pH 5.5, 60% at pH 9.0, and maximal activity at pH 6.0 <46>',\n",
       "              '#26# active from pH 5 to pH 8, wild-type and double-mutant (K271R/N273D) protein <18>; #15# the ratio of activities with NADH and NADPH is approximately constant between pH 5 and 8 <9>; #39# over 40% of maximal activity within this range, profile overview <69>',\n",
       "              '#17# pH 4.5: about 55% of maximal activity, pH 7.5: about 50% of maximal activity <1>',\n",
       "              '#2# pH 4.0: about 40% of maximal activity, pH 7.0: about 50% of maximal activity <7>',\n",
       "              '#27# pH 5.0: 81% of maximal activity, pH 7.0: about 65% of maximal activity <6>',\n",
       "              '#28# pH 4.0: about 55% of maximal activity, pH 6.5: about 60% of maximal activity <23>',\n",
       "              '#35# over 50% of maximal activity at pH 5.0-10.0, recombinant enzyme <66>',\n",
       "              '#38# 50% of maximal activity at pH 3.8 and pH 6.2, sugar reduction <65>',\n",
       "              '#46# over 80% of maximal activity, immobilized cells <60>',\n",
       "              '#47# recombinant enzyme, activity range <59>'],\n",
       "             '1.1.1.308': ['#1# pH 9-10: optimum. Activity decreases rapidly with decreasing pH. At pH 6.5 the enzyme is almost inactive <1>'],\n",
       "             '1.1.1.31': ['#10# oxidation of 3-hydroxypropionate by 3-HIBADH <13>',\n",
       "              '#13# approximately 40% of the activity is retained at pH 11.0. At pH 7.0, the activity is reduced to 30% of the activity observed at pH 9.0 <16>; #13# more than 60% activity between pH 8.0 and 10.0 <17>',\n",
       "              '#26# pH 7.5: about 50% of maximal activity, pH 9.5: about 50% of maximal activity <20>',\n",
       "              '#7# high activity between pH 8.0 and 9.0 <20>'],\n",
       "             '1.1.1.312': ['#2# at pH 5.5 and 9.3: about 50% of activity maximum <3>'],\n",
       "             '1.1.1.313': ['#1# IsfD shows a broad pH optimum from about pH 6.5 to pH 9.5. There is a steep increase in activity from pH 4.5 (negligible activity) and an equally steep decrease above pH 9.5 <1>'],\n",
       "             '1.1.1.317': ['#1# half-maximal activity <2>'],\n",
       "             '1.1.1.318': ['#19# activity range <13>'],\n",
       "             '1.1.1.319': ['#7# activity range <7>'],\n",
       "             '1.1.1.322': ['#1# 50% of maximal activity at pH 7.0 and pH 9.2 <1>'],\n",
       "             '1.1.1.323': ['#1# 50% of maximal activity at pH 6.7 and pH 9.3 <1>'],\n",
       "             '1.1.1.325': ['#2# activity range, profile overview, recombinant His-tagged enzyme, over 50% of maximal activity at pH 3.0-11.0 <3>'],\n",
       "             '1.1.1.34': ['#83# 19% of maximal activity at pH 5.5 and 29% at pH 10.0 <90>'],\n",
       "             '1.1.1.345': ['#3# activity is reduced by only about 10% at pH 5.0 and 8.0 <5>',\n",
       "              '#7# reduction of 4-methyl-2-oxopentanoate <5>',\n",
       "              '#7# oxidation of (R)-2-hydroxy-4-methylpentanoate <5>'],\n",
       "             '1.1.1.346': ['#2# for the reduction of 2,5-didehydro-D-gluconate <3>'],\n",
       "             '1.1.1.347': ['#1# as the pH increases from 7.0 to 7.5, there is a sharp increase in enzyme activity whereas the activity decreases sharply beyond pH 10.0 <1>'],\n",
       "             '1.1.1.35': ['#20# pH 9.0: about 45% of maximal activity, pH 11.0: about 55% of maximal activity <44>',\n",
       "              '#29# maximal activity at pH 6.0-7.0, 65% and 75% activity at pH 5.0 and pH 8.0, respectively, and 46% of the maximal activity at pH 10.0, and 26% of maximal activity at pH 4.0 <54>'],\n",
       "             '1.1.1.351': ['#3# about 70% of enzyme activities remain at pH 6.0 and 9.0 <2>'],\n",
       "             '1.1.1.353': ['#1# no activity is detected below pH 4.0 <1>'],\n",
       "             '1.1.1.354': ['#2# pH 8.0: about 75% of maximal activity, pH 10.5: about 35% of maximal activity <1>'],\n",
       "             '1.1.1.357': ['#11# activity range, profile overview <20>'],\n",
       "             '1.1.1.358': ['#4# over 40% of maximal activity within this range <6>',\n",
       "              '#5# over 40% of maximal activity within this range <7>'],\n",
       "             '1.1.1.36': ['#10# 70-35% of initial acetoacetyl-CoA reduced <2>',\n",
       "              '#23# no activity <10>'],\n",
       "             '1.1.1.365': ['#1# about 75, 65 and 18.5% of maximum activity at pH 6.5, 8.0 and 9.0, respectively <6>'],\n",
       "             '1.1.1.368': ['#1# recombinant enzyme, maximal activity at pH 6.0, 40% of maximal activity at pH 7.0, and 15% at pH 5.0, profile overview <2>'],\n",
       "             '1.1.1.37': ['#102# about 50% of maximal activity at pH 4.5 and pH 8.0 <78>',\n",
       "              '#105# activity range <120>',\n",
       "              '#136# activity range <87>',\n",
       "              '#142# pH 6.0: about 45% of maximal activity, pH 8.0: about 50% of maximal activity, reduction of oxaloacetate <67>',\n",
       "              '#142# pH 6.8: about 40% of maximal activity, pH 9.0: about 65% of maximal activity, oxidation of malate <67>',\n",
       "              '#178# over 50% of maximal activity within this range <143>',\n",
       "              '#187# profiles in different buffers, overview <108>',\n",
       "              '#199# maximal activity at pH 7.5, 70% of maximal activity at pH 10.0, 40% at pH 7.0, 20% at pH 5.0, very low activity below pH 5.0 and above pH 10.0 <137>',\n",
       "              '#28# malate oxidation, 10% of maximal activity at pH 6.0, maximal activity at pH 8.5 <121>',\n",
       "              '#28# oxaloacetate reduction, 50% of maximal activity at pH 5.5, maximal activity at pH 7.5 <121>',\n",
       "              '#88,195# activity range, recombinant enzyme <140>'],\n",
       "             '1.1.1.375': ['#2# enzymatic activity of the dimeric enzyme is controlled by a pH-dependent transition between an active and inactive dimeric state at pH 7 without dissociation of the subunits <2>'],\n",
       "             '1.1.1.376': ['#2# 50% of maximal activity is found at pH values of 6 and 10 <3>'],\n",
       "             '1.1.1.38': ['#21# over 70% of maximal activity within this range <27>'],\n",
       "             '1.1.1.380': ['#4# the optimum pH for the activity with L-galactonate is pH 8.0 while the remaining activity at pH 7.0, pH 7.5, and pH 9.0 is 73%, 97%, and 0%, respectively <3>'],\n",
       "             '1.1.1.39': ['#22# about 40% of maximal activity at pH 6.0 and at pH 8.5 <19>',\n",
       "              '#30# pH 7.0: about 30% of maximal activity, pH 8.7: about 45% of maximal activity <13>',\n",
       "              '#42# decarboxylation reaction, activity range <46>',\n",
       "              '#42# activity range, profile overview <46>',\n",
       "              '#42# carboxylation reaction, activity range <46>',\n",
       "              '#8# pH 5.8: about 60% of maximal activity, pH 7.6: about 40% of maximal activity <2>'],\n",
       "             '1.1.1.4': ['#34# for the reduction of acetoin, more than 85% of its maximal activity is in the pH range of 6.5 to 7.0 <32>',\n",
       "              '#11# temperature profile, overview <35>'],\n",
       "             '1.1.1.40': ['#101# pH: about 65% of maximal activity, pH: about 65% of maximal activity, recombinant NADP-ME3 <46>',\n",
       "              '#133# activity range, profile overview <86>',\n",
       "              '#23# broad range, 80% of maximal activity at pH 7.0 and pH 8.2 <29>',\n",
       "              '#23# pH 6.5: about 55% of maximal activity, pH 7.5: about 35% of maximal activity <26>',\n",
       "              '#33,45# the enzyme exhibits the highest catalytic performance at a relatively narrow pH range (7.0-8.0) <132>',\n",
       "              '#46# the enzyme is active in wide ranges of pH (pH 6.0-9.0) <119>',\n",
       "              '#54# pH 7.0: about 60% of maximal activity, pH 9.0: about 55% of maximal activity <11>',\n",
       "              '#59# pH 7.5: about 50% of maximal activity, pH 9.0: about 85% of maximal activity <28>',\n",
       "              '#67# activity range, profile overview <85>',\n",
       "              '#7# pH 6.8: about 40% of maximal activity, pH 8.0: about 70% of maximal activity, cytosolic enzyme, NADPH-dependent reductive carboxylation of pyruvate <19>',\n",
       "              '#7# pH 7.1: about 65% of maximal activity, pH 8.6: about 75% of maximal activity, NADP+-dependent decarboxylation of malate, at 10 mM malate, mitochondrial enzyme and cytosolic enzyme <19>',\n",
       "              '#91# pH 7.0: about 65% of maximal activity, pH 7.8: about 70% of maximal activity, recombinant NADP-ME2 <46>'],\n",
       "             '1.1.1.401': ['#1# little or no activity below pH 7.0, at pH 11.5 the enzme shows 75% of the activity at pH 9.5 <2>',\n",
       "              '#3# the enzyme has no or little activity at pH below 7.0, about 50% activity at pH 8.0, while the enzyme activity at pH 11.0 is about 75% of that obtained at pH 9.5 <2>'],\n",
       "             '1.1.1.404': ['#3# 70% of maximum activity <3>'],\n",
       "             '1.1.1.41': ['#26# activity range, profile overview <69>',\n",
       "              '#31# activity range, profile overview <70>',\n",
       "              '#47# more than 60% activity between pH 6.5 and 8.0 <78>'],\n",
       "             '1.1.1.414': ['#2# the optimum pH for the activity with L-galactonate is pH 8.0 while the remaining activity at pH 7.0, pH 7.5, and pH 9.0 is 73%, 97%, and 0 %, respectively <1>'],\n",
       "             '1.1.1.42': ['#10# pH-profile of wild-type and alanine-substituted mutant enzymes <59>',\n",
       "              '#11# pH 6.5: about 40% of maximal activity, pH 9.0: about 70% of maximal activity, isoenzyme IDP3 <77>; #68# pH 6.5: about 60% of maximal activity, pH 9.0: about 70% of maximal activity, isoenzyme IDP1 <77>',\n",
       "              '#117# measurements below pH 7.6 and above pH 8.8 lead to activities lower than 50% of its Vmax value <164>',\n",
       "              '#15# 80% of maximal activity in the pH range of pH 7.0-9.0 <98>',\n",
       "              '#33# isoform ICD-1, 30% of maximum activity, isoform ICD-2, 10% of maximum activity <63>',\n",
       "              '#33# isoform ICD-1, 90% of maximum activity, isoform ICD-2, less than 40% of maximum activity <63>',\n",
       "              '#59# activity range, profile overview <124>',\n",
       "              '#60# activity range <125>',\n",
       "              '#72# pH 7.0: about 60% of maximal activity, pH 9.0: about 95% of maximal activity, isoenzyme IDP1 <77>',\n",
       "              '#74# pH 7.0: about 60% of maximal activity, pH 8.0: about 40% of maximal activity, isoenzyme IDP2 <77>',\n",
       "              '#94# three-fold reduction in activity when the pH is shifted from the pH-optimum at 7.5 to pH 5.8 <119>',\n",
       "              '#96# high activity between pH 7.5 and 8.5 <148>'],\n",
       "             '1.1.1.424': ['#2# 50% of maximum activity <2>'],\n",
       "             '1.1.1.44': ['#52# about 70% of 6PGDH activities remain at pH 6.0 and 9.0 <119>; #85# about 70% of the activity remains at pH 6.0 and 9.0 <143>',\n",
       "              '#94# pH 7.0: about 60% of maximal activity, pH 8.0: about 55% of maximal activity <160>; #94# pH 7.0: about 60% of maximal activity, pH 8.0: about 60% of maximal activity <160>'],\n",
       "             '1.1.1.45': ['#1# oxidation of 3-hydroxybutanoate <4>',\n",
       "              '#3# reduction of acetoacetate <2>',\n",
       "              '#3# oxidation of L-3-hydroxybutanoate <2>'],\n",
       "             '1.1.1.46': ['#1# activity increases with increasing pH up to pH 12. At pH 13 the reaction is completely inhibited, cell free preparation <1>',\n",
       "              '#2# 50% activity at pH 6.0 and 10.0 <2>'],\n",
       "             '1.1.1.47': ['#1# pH 7.0: 70% of maximal activity with glucose and NAD+ and about 55% of the activity with glucose and NADP+, pH 8.5: about 55% of maximal activity with glucose and NAD+ and about 75% of the activity with glucose and NADP+ <30>',\n",
       "              '#1# the enzymes activity is more than 60% of its maximum at pH values ranging from 5.5 to 10.5 <71>',\n",
       "              '#13# pH 6.0: about 40% of maximal activity, pH 9.5: about 45% of maximal activity <28>',\n",
       "              '#16# when incubated for 6 h at 37°C, the enzyme is highly stable in acidic conditions (pH 5.0-6.0) but is completely inactivated at pH values above 8.0 <69>',\n",
       "              '#20# pH 8.5: about 35% of maximal activity, pH 12.0: about 65% of maximal activity <34>',\n",
       "              '#24# 50% of maximal activity at pH 4.5 and pH 11.0 <40>',\n",
       "              '#43# pH 7.0: about 40% of maximal activity, pH 10.0: about 45% of maximal activity <81>; #31# more than 60% activity between pH 7.0 and 10.0 <74>',\n",
       "              '#5# about 45% of maximal activity at pH 8.0 and pH 11.0 <35>',\n",
       "              '#6# at pH 7.5 and 10.0 respectively 75% and 50% activity compared to optimal pH <26>',\n",
       "              '#9,42# assay at <65>'],\n",
       "             '1.1.1.49': ['#10# about 50% of maximal activity at pH 7.5 and pH 9.5 <54>',\n",
       "              '#17# pH profile <61>',\n",
       "              '#53# more than 50% activity between pH 6.5 and 8.5 <135>',\n",
       "              '#54# recombinant enzyme <40>',\n",
       "              '#54# 50% of maximal activity at pH 5.9 and pH 9.0 <44>',\n",
       "              '#8# pH 6.2: about 50% of maximal activity, pH 9.8: about 50% of maximal activity <122>'],\n",
       "             '1.1.1.50': ['#10# pH profile, recombinant His-tagged wild-type and mutant enzymes <56>',\n",
       "              '#22# activity range, profile overview <85>'],\n",
       "             '1.1.1.53': ['#19# steroid reduction is observed within the pH range 4.0-10, retaining 19.9% of maximal activity at pH 4.0 and 29.5% of maximal activity at pH 10.0 <43>'],\n",
       "             '1.1.1.56': ['#2# 50% of maximal activity <11>',\n",
       "              '#2# 50% of maximal activity <11>',\n",
       "              '#6# sugar reduction <13>',\n",
       "              '#6# alcohol oxidation <13>',\n",
       "              '#7# pH 7.0: about 50% of maximal activity, pH 9.5: about 50% of maximal activity, in phosphate buffer <18>',\n",
       "              '#8# the enzyme shows about 25% activity at pH 9.0, while at pH 10.5 and 11.0 the activity is 85.6 and 68.11% respectively of that at pH 10.0 <16>'],\n",
       "             '1.1.1.58': ['#1# at pH 6 and 8: about 50% of activity maximum <4,5>',\n",
       "              '#1# pH 5.3: about 75% of activity maximum, pH 8.0: about 50% of activity maximum <2>'],\n",
       "             '1.1.1.59': ['#5# activity range <4>'],\n",
       "             '1.1.1.6': ['#26# pH 4.8: about 40% of maximal activity, pH 7.8: about 50% of maximal activity, reduction of dihydroxyacetone <8>',\n",
       "              '#3# pH 8.0: about 40% of maximal activity, pH 10.5: about 70% of maximal activity, oxidation of glycerol <14>',\n",
       "              '#3# dihydroxyaceton reduction reaction, activity range, profile overview <46>',\n",
       "              '#3# glycerol oxidation reaction, activity range, profile overview <46>',\n",
       "              '#30# oxidation activity range <50>',\n",
       "              '#6# pH 9.0: maximal activity, pH 11.0.: nearly inactive, oxidation of glycerol <12>',\n",
       "              '#6# the initial rate of reaction increases upon increasing the pH from pH 6.8 to 11.0, past pH 11.0 the initial rate falls rapidly, oxidation of glycerol <13>',\n",
       "              '#26# pH 7.0: about 60% of maximal activity, pH 10.0: about 85% of maximal activity, oxidation of glycerol <8>; #7# temperature profile, overview <54>',\n",
       "              '#8# glycerone reduction <55>',\n",
       "              '#8# glycerol oxidation <55>',\n",
       "              '#9# pH 5.5: about 70% of maximal activity, pH 9.0: about 45% of maximal activity, reduction of dihydroxyacetone <3>',\n",
       "              '#9# pH 9.0: about 45% of maximal activity, pH 11-12: maximal activity, oxidation of glycerol <3>'],\n",
       "             '1.1.1.61': ['#5# more than 80% activity between pH 7.0 and 9.0 <6>'],\n",
       "             '1.1.1.62': ['#10# about 60% of maximal activity at pH 8.5 and at pH 10.0 <1>',\n",
       "              '#3# estrone reduction <46>',\n",
       "              '#3# estradiol-17beta oxidation <46>',\n",
       "              '#40# pH 6.0: about 30% of maximal activity, pH 11.0: about 40% of maximal activity <154>',\n",
       "              '#8# pH 6.7: about 45% of maximal activity, pH 9.2: about 60% of maximal activity <15>'],\n",
       "             '1.1.1.65': ['#1# at pH 5.0 and 8.0 about 50% of activity maximum <1>'],\n",
       "             '1.1.1.66': ['#2# about half-maximal activity at pH 9.0 and 11.0 <1>'],\n",
       "             '1.1.1.67': ['#12# not active above, D-fructose reduction <23>',\n",
       "              '#15# more than 60% of the maximal activity in the pH range 6.0\\x969.0 <52>',\n",
       "              '#28# not active below, D-mannitol oxidation <17>',\n",
       "              '#21# pH 6.8: about 20% of maximal activity, pH 8.3: about 55% of maximal activity, oxidation of D-mannitol <35>',\n",
       "              '#21# pH 4.5: about 80% of maximal activity, pH 7.0: about 60% of maximal activity, reduction of D-fructose <35>',\n",
       "              '#38# about 80% activity at pH 5.5, 100% activity at pH 6.5, about 90% activity at pH 7.5, about 85% activity at pH 8.5, about 70% activity at pH 9.5 <54>',\n",
       "              '#9,12# not active below, D-mannitol oxidation <20,23>',\n",
       "              '#9# not active above, D-fructose reduction <20>'],\n",
       "             '1.1.1.69': ['#10# pH 8.0: about 30% of maximal activity, pH 12.0: 35% of maximal activity, oxidation of D-gluconate <6>',\n",
       "              '#10# pH 4.5: about 70% of maximal activity, pH 8.5: about 60% of maximal activity, reduction of 5-dehydro-D-gluconate <6>'],\n",
       "             '1.1.1.7': ['#1# hydroxyacetone phosphate reduction, values strongly dependent on buffer system <1>'],\n",
       "             '1.1.1.71': ['#16# strain B592 <5>',\n",
       "              '#16# strain B592, gradual increase up to pH 9.0 <5>',\n",
       "              '#19# aldehyde reduction, cofactor NADH, NADP dependent activity occurs only at pHs below 7 <7>',\n",
       "              '#19# ethanol oxidation, cofactor NAD+ <7>',\n",
       "              '#33# the enzyme shows around 50% of activity at pH 6.0, pH 6.5, and pH 8.0 <19>',\n",
       "              '#6,10# hexanal reduction, 95% of maximal rate <3>'],\n",
       "             '1.1.1.75': ['#2# cell free extracts, 50% activity at both pH <1>',\n",
       "              '#2# two peaks at pH 7 and 10 with very low activity at pH 8.5, at pH 7 50% of maximum activity <2>',\n",
       "              '#2# washed cell suspension, very low activity at 4.5, 50% activity at pH 8 <1>',\n",
       "              '#3# very low activity at pH 7.5, 60% of activity at pH 11 in liver homogenates <4>',\n",
       "              '#3# 40% of activity in soluble extract of disintegrated mitochondria <4>',\n",
       "              '#4# low activity at pH 6, 80% of activity at pH 10.5 <7>'],\n",
       "             '1.1.1.76': ['#14# activity range <11>',\n",
       "              '#8# the enzyme shows a relatively high activity over a wide pH range from 5.0 to 8.0 for racemic acetoin reduction and more than 60% of activity is retained between pH 4.5 and 10.0 <13>'],\n",
       "             '1.1.1.77': ['#1# reduction of lactaldehyde, pH 4.5: 50% of reductase activity maximum, pH 10: 10% of reductase activity maximum <5>',\n",
       "              '#2# reduction of DL-lactaldehyde, pH 6: 50% of reductase activity maximum, pH 7.8: 45% of reductase activity maximum <2>'],\n",
       "             '1.1.1.78': ['#1# pH 4.5 and pH 9.5: about 50% of activity maximum <5>'],\n",
       "             '1.1.1.79': ['#10# approximately 50% of the activity is retained at a pH of 3.5 and 8.0, respectively <32>',\n",
       "              '#15,30# enzyme isoform GLYR1 retains at least half of maximal activity over the pH range of 5.5-7.0 <34>',\n",
       "              '#17# 23% of maximal activity at pH 8.0 <9>',\n",
       "              '#21# approximately 10% of activity at pH 5.5 and 9 <19>',\n",
       "              '#22# recombinant protein <17>'],\n",
       "             '1.1.1.8': ['#1# pH-rate profile for the reduction of dihydroxyacetone phosphate shows rather sharp optimum between pH 7.5 and 8.0 <11>',\n",
       "              '#11# dihydroxyacetone reduction, activity decreases rapidly below pH 7.0 and above pH 8.0, more rapidly at acidic values <2>',\n",
       "              '#11# activity falls off rapidly as pH is increased <21>',\n",
       "              '#15# reduction of dihydroxyacetone phosphate occurs over a broad range between 5.8 to 7.2 <16>',\n",
       "              '#15# about half-maximal activity at pH 5.0 and 8.1, reduction of dihydroxyacetone phosphate <16>',\n",
       "              '#15# about half-maximal activity at pH 8.6 and 10.0, oxidation of glycerol 3-phosphate <16>',\n",
       "              '#18,20# highest activity, glycerol-3-phosphate formation <28>',\n",
       "              '#21# about half-maximal activity between pH 5.2 and 7.8, reduction of dihydroxyacetone phosphate <14>',\n",
       "              '#21# about half-maximal activity at pH 8.0 and 11.0, oxidation of L-glycerol-3-phosphate <14>',\n",
       "              '#23# about half-maximal activity at pH 8.0 and 10.0, oxidation of glycerol 3-phosphate, isozymes P-III, liver <5>',\n",
       "              '#24# maximum activity in pH interval 7.9-8.2 <25>',\n",
       "              '#27# maximal activities of the three allelic forms are obtained within a pH range of 6.0-6.5 for dihydroxyacetone phosphate reduction <19>',\n",
       "              '#27# maximal activities of the three allelic forms are obtained within a pH range of 10.0-10.5 for glycerol 3-phosphate oxidation <19>',\n",
       "              '#8# about 80% of maximal activity at pH 6.7 and 8.7, reduction of dihydroxyacetone phosphate, decrease of activity below and above these pH values <22>',\n",
       "              '#8# about 80% of maximal activity at pH 8.1 and 9.4, oxidation of glycerol-3-phosphate, decrease of activity below and rapidly above these pH values <22>',\n",
       "              '#9# about half-maximal activity at pH 6.5 and 8.2, reduction of dihydroxyacetone phosphate <8>'],\n",
       "             '1.1.1.81': ['#13# sharp drop of activity below and above <19>',\n",
       "              '#8# strong decrease above <12>'],\n",
       "             '1.1.1.82': ['#14# pH 5.5: about 90% of maximal activity, pH 8.0: about 40% of maximal activity <37>'],\n",
       "             '1.1.1.83': ['#3# at pH 7.0 and 10.6: about 50% of activity maximum <6>'],\n",
       "             '1.1.1.85': ['#7# about 75% of maximal activity at pH 7.5 and 11.5 <18>',\n",
       "              '#7# pH 8.0: 75% of maximal activity, pH 8.9-10.1: optimum <5>'],\n",
       "             '1.1.1.86': ['#1# pH 6.8: about 40% of maximal activity <2>',\n",
       "              '#2# pH 7.0: about 40% of maximal activity, pH 9.0: about 90% of maximal activity, with 2-hydroxy-2-ethyl-3-oxobutanoate as substrate, Tris-HCl buffer <3>',\n",
       "              '#47# pH 7.0: 87% of maximal activity, pH 10.0: 61% of maximal activity <57>',\n",
       "              '#6# pH 6.6: about 65% of maximal activity, pH 8.8: about 90% of maximal activity, reaction with 2-acetolactate, Tris buffer <4>',\n",
       "              '#8# pH 7.0: about 25% of maximal activity, pH 8.0-8.5: optimum, isoenzyme 1 <8>; #6# pH 7.0: about 60% of maximal activity, pH 8.5: about 65% of maximal activity <4>',\n",
       "              '#8# pH 6.5: about 35% of maximal activity, pH 8.5: about 90% of maximal activity, reaction with 2-acetolactate, isoenzyme 1 <8>'],\n",
       "             '1.1.1.87': ['#13# pH independence of the catalytic reaction over the range of pH 5.5-9.5, pH profiles <20>',\n",
       "              '#3# activity drops sharply below pH 6.5 and above pH 8.5 <6>',\n",
       "              '#3# pH 6.1: about 60% of activity maximum, pH 7.7: about 50% of activity maximum, reductive carboxylation <3>',\n",
       "              '#3# pH 7.4: about 50% of activity maximum, pH 9.3: about 90% of activity maximum, oxidative decarboxylation <3>'],\n",
       "             '1.1.1.9': ['#1# about 75% of maximal reduction activity at pH 4.0 and 8.0 <1>; #14# reduction reaction, activity range <50>; #36# over 60% of maximal activity within this range for oxidation of xylitol, profile overview <63>',\n",
       "              '#11# about 80% of maximal oxidation activity at pH 8.7 and 10.5 <8>',\n",
       "              '#14# oxidation reaction, activity range <50>',\n",
       "              '#14# over 60% of maximal activity within this range <67>',\n",
       "              '#16# fully active <21>',\n",
       "              '#32# over 50% of maximal activity within this range, profile overview <73>',\n",
       "              '#33# activity range, profile overview <60>',\n",
       "              '#33# the enzyme loses its activity at pH higher than 11.0 and lower than 4.0 <52>; #34# the enzyme lost its activity at pH lower than 4.0 and higher than 11.0, corresponding to NAD+ which is unstable at pH values greater than 10.5 <55>; #35# activity range, inactive above or below <55>',\n",
       "              '#40# over 50% of maximal activity within this range, profile overview <73>',\n",
       "              '#41# over 50% of maximal activity within this range, profile overview <73>'],\n",
       "             '1.1.1.90': ['#1# pH 3.5: about 60% of maximal activity, pH 6.0: about 45% of maximal activity <23>',\n",
       "              '#11# less than half-maximal activity above and below, oxidation of benzyl alcohol <16>',\n",
       "              '#11# less than half-maximal activity above and below, reduction of benzaldehyde <16>',\n",
       "              '#3# pH 7.5: about 15% of activity maximum, pH 10.5: about 20% of activity maximum <3>'],\n",
       "             '1.1.1.91': ['#12# pH 4.7: 45% of maximal activity, pH 9.2: 20% of maximal activity, coniferyl alcohol dehydrogenase activity <18>',\n",
       "              '#12# pH 7.1: about 45% of maximal activity, pH 8.5: about 20% of maximal activity, sinapyl alcohol dehydrogenase activity <18>',\n",
       "              '#23# at pH 8.0 and 10.0 the enzyme retains 62% activity, while the enzyme retains 13% activity at pH 4.0 <15>',\n",
       "              '#4# in phosphate buffer, activity of the enzyme is only one-half that shown in imidazole buffer <9>',\n",
       "              '#4# in imidazole buffer <9>',\n",
       "              '#4# in Tris buffer, activity is about 40% as in imidazole buffer <9>'],\n",
       "             '1.1.1.93': ['#2# at pH 7.0 and pH 10.6: 50% of activity maximum <4>',\n",
       "              '#3# at pH 7 and pH 9: about 50% of activity maximum, mesotartrate + NAD+ <5>'],\n",
       "             '1.1.1.95': ['#17# about 50% of activity maximum at pH 8.5 and 10.3, NAD+ reduction <16>',\n",
       "              '#17# about 50% of activity maximum at pH 5 and 7.8, NADH oxidation <16>',\n",
       "              '#19# enzymatic activity decreases below pH 6.5 <55>',\n",
       "              '#25# 3-phosphohydroxypyruvate reduction <45>',\n",
       "              '#25# 3-phospho-D-glycerate oxidation <45>',\n",
       "              '#36# activity is relatively constant between pH 7.5-9.5. Above pH 9.5 and below pH 7.5 activity falls off rapidly <44>',\n",
       "              '#46# pH 4.0: about 82% of maximal activity, pH 9.0: about 80% of maximal activity <71>',\n",
       "              '#6# activity drops sharply above pH 8.5 <22>',\n",
       "              '#6# Tris buffer and phosphate buffer <2>'],\n",
       "             '1.1.2.10': ['#10# the enzyme shows more than 50% activity between pH 7.5 and 9.5 <10>'],\n",
       "             '1.1.2.3': ['#14# pH 7.6: about 50% of maximal activity, pH 8.5: about 60% of maximal activity <55>',\n",
       "              '#5# almost no activity <2>'],\n",
       "             '1.1.2.4': ['#1# low activity with cytochrome c at pH 5.5, almost maximum activity at pH 8.5 <3>',\n",
       "              '#1# very low activity with ferricyanide <3>',\n",
       "              '#6# no activity at pH 5, almost maximum activity at pH 9 <8>',\n",
       "              '#9# pH 7.0: about 50% of maximal activity, pH 10.0: about 70% of maximal activity <24>'],\n",
       "             '1.1.2.5': ['#1# pH 7.0: about 50% of maximal activity, pH 9.0: about 80% of maximal activity, D-lactate + nitrotetrazolium blue <1>'],\n",
       "             '1.1.2.7': ['#10# the enzyme oxidation step becomes rate-limiting at pH 9.0 <1>',\n",
       "              '#16# pH profile, overview <10>'],\n",
       "             '1.1.3.10': ['#23# substrate D-glucose, measuring oxygen consumption (computer-interfaced Oxy-32 oxygen-monitoring system), 30°C, varying concentrations of both D-glucose and oxygen. Ping-pong kinetic mechanism at pH values below 7.0 and an ordered mechanism at pH above 7.0 <28>',\n",
       "              '#23# activity range <48>',\n",
       "              '#25# pH 5.5: about 40% of maximal activity, pH 11: about 40% of maximal activity <20>',\n",
       "              '#28# optimal buffer combination for the pyranose 2-oxidase systems determined <25>',\n",
       "              '#33# at pH 8.0, the enzyme still retains 45% of its maximal activity observed at pH 6.5 <65>',\n",
       "              '#8# at pH 5.0 in citrate buffer and pH 9.0 in glycine-NaOH buffer, the enzyme displays about 60% of its maximum activity. The activity begins to decline dramatically below pH 5.0 and above pH 9.0 <64>'],\n",
       "             '1.1.3.11': ['#1# pH 5.1: about 85% of maximal activity, pH 7.0: about 40% of maximal activity <1>'],\n",
       "             '1.1.3.12': ['#2# pH 7: 80% of maximal activity, pH 9: 50% of maximal acitvity <1>',\n",
       "              '#6# 48% of maximum activity <2>'],\n",
       "             '1.1.3.13': ['#11# pH 6.0: about 65% of maximal activity, pH 10.0: 85% of maximal activity <7>',\n",
       "              '#14# pH 5.5: 6% of maximal activity, pH 11.0: 12% of maximal activity <15>',\n",
       "              '#3# about half-maximal activity at pH 6.5 and 8.3 <6>',\n",
       "              '#31# more than 80% of the enzyme activity remains in the pH range from 5.5 to 6.5 <68>',\n",
       "              '#5# the free and immobilized enzyme show less than 45% of their maximum activities below pH 4.0 and approximately 85% at pH 10.0 <52>',\n",
       "              '#5# about half-maximal activity at pH 6.7 and 9.8 <6>',\n",
       "              '#61# 16% of maximal activity <38>',\n",
       "              '#64# more than 75% of maximum activity <74>; #64# more than 60% activity between pH 7.0 and 10.0 <62>',\n",
       "              '#70# more than 60% of maximum activity <87>'],\n",
       "             '1.1.3.15': ['#31# 80% of the maximum activity within <2>',\n",
       "              '#37# pH profiles, the enzyme shows a ping-pong bi-bi kinetic mechanism between pH 6.0 and 10.0, overview <68>',\n",
       "              '#5# L-LAC-OX activity is rather constant in 6.0-9.0 pH range <75>'],\n",
       "             '1.1.3.16': ['#4# pH 5.3: about 85% of activity maximum, pH 8.6: about 15% of activity maximum <7>'],\n",
       "             '1.1.3.17': ['#11# H99N mutant, lower KM compared to pH 8-11. Measuring the rate of oxygen consumption with a computer-interfaced Oxy-32 oxygen monitoring system <46>',\n",
       "              '#12# biosensor retains 60% of its maximum response <34>',\n",
       "              '#4# kcat/Kox with choline or 1,2-[2H4]choline as a substrate for choline oxidase is independent of pH between pH 6.0 and pH 10.0 <53>'],\n",
       "             '1.1.3.20': ['#1,2,3,4,5,6,7,8,9,10,11,13,14,15# high activity <16>',\n",
       "              '#11# about half-maximal activity at pH 6.6 and 9.3 <3>',\n",
       "              '#18# almost complete loss of activity <20>',\n",
       "              '#8# about half-maximal activity at pH 4.8 and 10.0 <2>'],\n",
       "             '1.1.3.21': ['#2# activity within, optimum at pH 8.5 <15>',\n",
       "              '#3# pH 5.0: about 50% of activity maximum, pH 6.7: about 30% of activity maximum <4>'],\n",
       "             '1.1.3.27': ['#2# pH 7.0: about 45% of maximal activity, pH 9.5: about 80% of maximal activity <2>'],\n",
       "             '1.1.3.29': ['#1# pH 5.0: about 70% of maximal activity, pH 11: about 50% of maximal activity <1>'],\n",
       "             '1.1.3.30': ['#9# almost constant activity in this range <4>'],\n",
       "             '1.1.3.37': ['#3# pH 5.0: about 20% of maximal activity, pH 8.0: about 40% of maximal activity <1>'],\n",
       "             '1.1.3.38': ['#1# pH 6.0: about 60% of maximal activity, pH 9.0: about 65% of maximal activity, reaction with creosol <15>',\n",
       "              '#1# about 30% of maximal activity at pH 7 and pH 12 <3>',\n",
       "              '#3# 5% residual activity <25>',\n",
       "              '#4# more than 90% of maximum activity <27>',\n",
       "              '#5# complete loss of activity <25>'],\n",
       "             '1.1.3.4': ['#10# 70% of maximal activity at pH 4.9 and pH 8.9 <75>',\n",
       "              '#18# pH 3.0: about 45% of maximal activity, pH 9.0: about 40% of maximal activity, 10 min oxidation-assay of beta-D-glucose <56>; #18# low activity at pH 3.0 and pH 9.0, high activity at pH 4.0-7.0 depending on the buffer system used <128>',\n",
       "              '#20# under O2 and Ar at 37°C, recombinant enzyme yGOXpenag <110>',\n",
       "              '#27# pH profile, overview <68>; #26# pH-profiles, 54% of maximal activity at pH 3.0, 20% of maximal activity at pH 8.0 for the recombinant enzyme, 50% for the native enzyme, overview <72>',\n",
       "              '#29# more than 88% of the maximum activity is observed between pH 4.0 and 8.0, outside this range the activity decreases dramatically <85>',\n",
       "              '#3# 80% of the maximal activity is observed in the pH 5.5-9.0 range <110>',\n",
       "              '#3# pH 4.5: about 60% of maximal activity, pH 7.0: about 50% of maximal activity <168>',\n",
       "              '#3# pH 4.5: about 85% of maximal activity, natural GOD and pmr1DELTA-mutant-derived GOD, about 90% of maximal activity, hyperglycosylated GOD expressed in wild-type cells of Saccharomyces cerevisiae, pH 10: about 75% of maximal activity, natural GOD and pmr1DELTA-mutant-derived GOD, about 90% of maximal activity, hyperglycosylated GOD expressed in wild-type cells of Saccharomyces cerevisiae <67>',\n",
       "              '#3# over 60% activity within this range at 50°C <142>',\n",
       "              '#3# the activities of modified GOD are about 40-55% of those of native GOD in the pH range tested (pH 2.0-7.0) <97>',\n",
       "              '#3# activity range, profile overview. 85% of maximal activity at pH 5.0, 51% activity at pH 7.0 <145>',\n",
       "              '#3# immobilized on activated carbon <16>',\n",
       "              '#3# soluble enzyme <16>',\n",
       "              '#43# pH 5.0: about 60% of maximal activity, pH 10.0: about 40% of maximal activity, recombinant wild-type enzyme <161>',\n",
       "              '#6# over 80% of maximal activity within this range <82>'],\n",
       "             '1.1.3.40': ['#1# at pH 6.5 and pH 10.0: about 50% of maximum activity <1>'],\n",
       "             '1.1.3.41': ['#1# over 80% of maximal activity within this range <4>'],\n",
       "             '1.1.3.47': ['#1# more than 65% of maximum activity <12>',\n",
       "              '#11# 50% of the activity at pH 7.0, substrate 5-hydroxymethylfurfural <7>',\n",
       "              '#12# complete loss of activity <12>',\n",
       "              '#3# 5% residual activity <12>'],\n",
       "             '1.1.3.49': ['#1# 80% of maximum activity <1>'],\n",
       "             '1.1.3.6': ['#1# activity range, profile overview <89>',\n",
       "              '#12# strong decrease in enzyme activity above <24>; #67# ChoL retains more than 80% activity in the pH range 6.5-9.0 <77>; #80# pH 6.5: about 50% of maximal activity, pH 9.0: about 50% of maximal activity <113>',\n",
       "              '#16# activity range, profile overview <99>',\n",
       "              '#18# wild-type, S103T, V121A, R135H and V145E mutants <27>',\n",
       "              '#33# activity range <86>; #18# activity range, profile overview <87>',\n",
       "              '#2# the activity increases with increasing pH value from 4.0 to 7.5 and then declines as pH increases further <75>',\n",
       "              '#21# about 50% of maximal activity at pH 4.5, about 90% of maximal activity at pH 8 <2>',\n",
       "              '#23# activity range <82>',\n",
       "              '#82# pH 6.0: about 20% of maximal activity, pH 9.0: about 55% of maximal activity <104>; #25# pH 6.0: about 50% of maximal activity, pH 9.0: about 40% of maximal activity <117>',\n",
       "              '#36# pH 5.5: about 50% of maximal activity, pH 8.5: about 40% of maximal activity, cholesterol oxidase I <45>; #77# pH 5.5: about 50% of maximal activity, pH 8.5: about 55% of maximal activity <111>; #81# pH 5.5: about 60% of maximal activity, pH 8.5: about 65% of maximal activity <111>',\n",
       "              '#36# about 50% of maximal activity at pH 5 and 9.5 <6>; #83# pH 5.0: about 50% of maximal activity, pH 9.5: about 40% of maximal activity <107>',\n",
       "              '#40# pH 5.0: about 70% of maximal activity, pH 7.5: about 30% of maximal activity <46>',\n",
       "              '#40# pH 5.0: about 90% of maximal activity, pH 6.5: about 60% of maximal activity, pH 7.0: about 20% of maximal activity <47>',\n",
       "              '#5# pH 4.0: native enzyme showes about 40% of maximal activity, enzyme immobilized on polyacrylonitrile hollow fibers shows about 65% of maximal activity, pH 8.0: native enzyme showes about 60% of maximal activity, enzyme immobilized on polyacrylonitrile hollow fibers shows about 85% of maximal activity <43>',\n",
       "              '#5# strain ST-200 <37>; #36# pH 5.0: about 75% of maximalm acttivity, pH 8.5: about 60% of maximal activity, cholesterol oxidase II <45>',\n",
       "              '#5# 33% of maximal activity at pH 5.0 and pH 9.0 <17>; #9# pH 5: 83% of maximal activity, pH 9.0: 77% of maximal activity, reaction with cholesterol, enzyme CO1 <42>; #6# more than 75% of the maximal activity is retained over the pH range of 5.0-9.0 <115>',\n",
       "              '#6# inducible enzyme, 80% of maximal activity at pH 6.0, 50% at pH 9.5, constitutive enzyme, 70% of maximal activity at pH 6.0, 50% at pH 9.5 <16>; #47# pH 6.0: about 65% of maximal activity, pH 9.5: about 60% of maximal activcity <57>',\n",
       "              '#66# relative high activities in a broad pH range of 6.0-8.0, however activity rapidly declines under pH 5.5 <81>',\n",
       "              '#74# the enzyme activity retains more than 95% of its maximal activity within the pH range of 6-7 at 25°C <100>',\n",
       "              '#9# pH 5: 78% of maximal activity, pH 8.0: 75% of maximal activity, reaction with cholestero, enzyme CO2 <42>; #78# pH 5.0: about 45% of maximal activity, pH 8.0: about 40% of maximal activity <110>'],\n",
       "             '1.1.3.7': ['#14# about 40% of maximal activity at pH 3.0 and pH 9.0 <2>; #20# pH-dependencies of wild-type and mutant enzymes, overview <34>',\n",
       "              '#16# pH 3.5: about 70% of maximal activity, pH 8.0: about 65% of maximal activity <5>',\n",
       "              '#28# 5% residual activity <50>',\n",
       "              '#31# the enzyme shows more than 90% of maximal activity, measured at pH 5.0, in the pH range between pH 4 and pH 6 and more than 80% between pH 3 and pH 7 <41>',\n",
       "              '#35# 50% of the activity at pH 7.0, substrate 5-hydroxymethylfurfural <46>',\n",
       "              '#39# complete loss of activity <50>',\n",
       "              '#4# pH 5.3: about 90% of maximal activity, pH 7.3: 35% of maximal activity <6>',\n",
       "              '#6# the enzyme displays over 63% of its activity in the pH range of 4.0-7.0. A further increase of these pH values decreases the relative activity to 17% <29>',\n",
       "              '#9# more than 65% of maximum activity <50>'],\n",
       "             '1.1.3.8': ['#3# activity stable <24>'],\n",
       "             '1.1.3.9': ['#10# pH 6: about 80% of maximal activity, pH 7: about 25% loss of activity, oxidation of glycolaldehyde <48>',\n",
       "              '#2# tested range, decrease of kcat/Km by 2 orders of magnitude compared to maximum kcat/Km <52>',\n",
       "              '#16# activity range, more than 95% activity in the range of pH 6.0-7.5, profile overview. Inactive at pH 5.0 and below <67>',\n",
       "              '#9# pH 6.0: about 40% of maximal activity, pH 9.0: about 65% of maximal activity <47>; #15# more than 80% of the maximum activity between pH 6.0 and 9.0, maximum activity occurs at pH 7.0 (phosphate buffer) and 8.0 (Tris buffer), respectively. At pH values below pH 5.0 the enzyme loses its activity <66>'],\n",
       "             '1.1.5.13': ['#3# more than 50% activity between pH 6.5 and 8.5 <6>'],\n",
       "             '1.1.5.14': ['#2# pH 4.5: about 85% of maximal activity, pH 6.4: about 50% of maximal activity <2>',\n",
       "              '#3# enzymatic activity is significantly inhibited at pH lower than 4.5 or higher than 6.5 <9>',\n",
       "              '#5# highest activity between pH 3.0 and 6.0 with 2,6-dichloroindophenol as electron acceptor <25>'],\n",
       "             '1.1.5.2': ['#13# the enzyme relative activity decreases from optimum at pH 5.0 to approximately 51% or 36% when the pH drops to 4.0 or increases to 6.0, respectively <81>',\n",
       "              '#18# pH 4.0: about 50% of maximal activity, pH 8.0: about 35% of maximal activity <16>',\n",
       "              '#29# pH 6.0: about 50% of maximal activity, pH 9.0: about 55% of maximal activity <83>',\n",
       "              '#9# pH 7.5: about 40% of maximal activity, pH 9.5: about 85% of maximal activity, membrane-bound enzyme, reaction with phenazine methosulfate, 2,6-dichlorophenol indophenol and pyrroloquinoline quinone <10,12>'],\n",
       "             '1.1.5.4': ['#16# no significant difference in activity from pH 6.8 to 8.2 <18>',\n",
       "              '#17# pH 6.0: about 60% of maximal activity, pH 12.0: about 50% of maximal activity <20>'],\n",
       "             '1.1.5.5': ['#13# pH 5.0: about 50% of maximal activity, pH 7.5: about 55% of maximal activity, substrate: ethanol <11>',\n",
       "              '#20# strong decrease of activity at pH levels below pH 4 and above pH 5.5, and no activity at pH 2.0 and pH 7.0, activity range, profile overview <18>'],\n",
       "             '1.1.5.9': ['#16# more than 70% activity between pH 6.0 and 8.5 <29>'],\n",
       "             '1.1.98.2': ['#1# pH 5: about 60% of maximal activity, pH 5.5: maximum, pH 7: about 50% of maximal activity, pH 8.0: maximal activity, pH 8.5: about 60% of maximal activity <2>'],\n",
       "             '1.1.98.5': ['#1# pH 3.0: about 30% of maximal activity, pH 7.0: about 30% of maximal activity <1>'],\n",
       "             '1.1.99.1': ['#6# about 50% of activity maximum at pH 7.5 and pH 10.0 <5>'],\n",
       "             '1.1.99.12': ['#2# pH 5.0: about 40% of activity maximum, pH 8.7: about 30% of activity maximum <1>'],\n",
       "             '1.1.99.13': ['#2# 85% of maximal activity at pH 5.2 and 7.2 <1>',\n",
       "              '#6# more than 60% activity between pH 6.0 and 9.0. When pH is decreased to 5.4, about 35% of the initial activity is observed. When pH is increased to 9.6, about 40% of the relative activity remains <16>'],\n",
       "             '1.1.99.14': ['#3# about half-maximal activity at pH 7.3 and 9.0 <1>'],\n",
       "             '1.1.99.18': ['#10# pH 4.5: about 60% of maximal activity, pH 8.0: about 40% of maximal activity, reaction with 2,6-dichlorophenol indophenol <26>',\n",
       "              '#11# pH 5.0: about 40% of maximal activity, pH 7.0: about 20% of maximal activity, cellobiose + cytochrome c, mutant enzyme N732D <30>',\n",
       "              '#11# pH 5.0: about 45% of maximal activity, pH 7.0: about 65% of maximal activity, cellobiose + 2,6-dichlorophenol-indophenol, mutant enzyme N732E <30>',\n",
       "              '#11# pH 5.5: about 50% of maximal activity, pH 8.0: about 60% of maximal activity, cellobiose + 2,6-dichlorophenol-indophenol, mutant enzyme N732D <30>',\n",
       "              '#11# the relative activity of purified enzyme is decreased when pH is over 4.5 and it is below 80% at 6.5 <90>',\n",
       "              '#11# highest activity at pH 3, decreases when the pH increases to pH 7.0, cellobiose + 2,6-dichlorophenol-indophenol, to about 10% with recombinant wild-type enzyme, to about 25% with mutant enzymes N732Q, to about 40% with mutant enzyme N732A and to about 50% with mutant enzyme N732H <30>',\n",
       "              '#11# highest activity at pH 3, decreases to 10-20% of the activity when the pH increases to pH 5.5, cellobiose + cytochrome c recombinant wild-type enzyme, mutant enzymes N732H, N732Q and N732A <30>',\n",
       "              '#11# pH 4.1: less than 50% of maximum activity, pH 5.7: less than 50% of maximum activity <20>',\n",
       "              '#30# pH 4.0: about 70% of maximal activity, pH 6.0: about 40% of maximal activity, lactose + 2,6-dichlorophenol indophenol, in citrate buffer <28>; #29# pH 4.0: about 75% of maximal activity, pH 6.0: about 65% of maximal activity, lactose + 2,6-dichlorophenol indophenol, in citrate buffer <28>; #11# pH 5.0: about 35% of maximal activity, pH 7.0: about 20% of maximal activity, cellobiose + cytochrome c, mutant enzyme N732E <30>; #27# below pH 4.0, and particularly above pH 6.0, the activity is observed to drop (65% activity at pH 3.0, 45% activity at pH 7.0 and no activity at pH 9.0) <70>',\n",
       "              '#18# pH 2.5: about 40% of maximal activity, pH 6.5: about 60% of maximal activity <33>; #57# lactose as substrate, 2,6-dichloroindophenol or 1,4-benzoquinone or ferricenium as electron acceptor <52>',\n",
       "              '#19# at 40°C <40>; #26# lactose as electron donor, 2,6-diclorindophenol as electron acceptor <51>',\n",
       "              '#26# the mediated electron transfer currents increase with pH and are 2.3 times lower at pH 5.5 and 6.0 for CDH adsorbed on electrosynthesised polyaniline modified electrode compared to CDH on plain screen printed electrodes <64>',\n",
       "              '#29# pH 3.0: about 60% of maximal activity, pH 5.0: about 70% of maximal activity, lactose + cytochrome c, in acetate buffer <28>; #30# pH 3.0: about 70% of maximal activity, pH 5.0: about 40% of maximal activity, lactose + cytochrome c, in acetate buffer <28>',\n",
       "              '#30# pH 3.5: about 50% of maximal activity, pH 5.5: about 20% of maximal activity, lactose + 2,6-dichlorophenol indophenol, in citrate buffer <28>; #29# pH 3.5: about 50% of maximal activity, pH 5.5: about 20% of maximal activity, lactose + cytochrome c, in citrate buffer <28>',\n",
       "              '#3# about 50% activity between pH 5.0 and 7.5 <75>',\n",
       "              '#3# more than 50% relative activity from pH 3.8 to 6.5 using 1,4-benzoquinone as electron acceptor <69>',\n",
       "              '#3# more than 50% relative activity from pH 3.8 to 6.5 using 2,6-dichlorophenolindol as electron acceptor <69>',\n",
       "              '#3# the CDH activity decreases very sharply below pH 4.5. The recombinant enzyme remains at 60.8% of the maximum CDH activity at pH 4.0, and at only 22.6% CDH activity at pH 3.5 <67>; #68# more than 50% activity between pH 3.4 and 6.0 <80>; #47# pH 3.5: about 35% of maximal activity, pH 6.0: about 40% of maximal activity, activity with 2,6-dichlorophenolindophenol as acceptor, enzyme from Phanerochaete chrysosporium <111>',\n",
       "              '#39# pH 3.2: about 50% of maximal activity, pH 5.3: about 45% of maximal activity, reaction with cellobiose and cytochrome c <34>',\n",
       "              '#68# pH 3.5: about 35% of maximal activity, pH 7.0: about 50% of maximal activity <110>'],\n",
       "             '1.1.99.2': ['#2# active over a wide pH range, without a clear optimum <2>; #1,2# the enzyme is active over a wide pH range (7.3-9.3) with no clear optimum <11>',\n",
       "              '#5# pH 5.0: about 45% of maximal activity, pH 8.5: 60% of maximal activity, reaction with alpha-oxoglutarate and NADH <1>'],\n",
       "             '1.1.99.20': ['#1,2# about half-maximal activity at pH 7.0 and 9.0 <3>',\n",
       "              '#2# about 75% of maximal activity at pH 6.5 and pH 8.5 <5>'],\n",
       "             '1.1.99.21': ['#2# activity range, profile overview <13>',\n",
       "              '#2# about 50% of activity at pH 5.0 <3>'],\n",
       "             '1.1.99.22': ['#2# about 50% of activity at pH 5.0 <2>'],\n",
       "             '1.1.99.28': ['#3# pH 5.0: about 35% of maximal activity, pH 7.1: about 30% of maximal activity <1>',\n",
       "              '#3# at pH 5.6 and 7.4 there are still 65% of enzyme activity remaining <27>'],\n",
       "             '1.1.99.29': ['#1# activity in the range is higher than 50% of the rate at the acidic optimum at pH 4 <1>',\n",
       "              '#4# in Tris, applied potential + 300 mV vs. Ag-AgCl, 1 mM glucose, flow rate 1.0 ml/min <8>',\n",
       "              '#4# in PBS, applied potential + 300 mV vs. Ag-AgCl, 1 mM glucose, flow rate 1.0 ml/min <8>',\n",
       "              '#4# depending on the buffer, highest biosensor response in 0.1 M ethanolamine buffer, applied potential + 300 mV vs. Ag-AgCl, 1 mM glucose, flow rate 1.0 ml/min <8>',\n",
       "              '#4# depending on the electron acceptor and with d-glucose as electron donor, has 2 pH optima for some substrates <10>; #5# depending on the electron acceptor and with D-glucose as the electron donor, has 2 pH optima for some substrates <9>',\n",
       "              '#5# in 0.1 M ethanolamine buffer, applied potential + 300 mV vs. Ag-AgCl, 1 mM glucose, flow rate 1.0 mL/min <8>',\n",
       "              '#7# more than 70% activity between pH 4.0 and 8.0 <29>'],\n",
       "             '1.1.99.3': ['#3# pH 4.0: about 40% of maximal activity, pH 6.0: about 60% of maximal activity <3>'],\n",
       "             '1.1.99.32': ['#4# no decrease in activity is observed in the pH range 5.5-8.0 <3>'],\n",
       "             '1.1.99.6': ['#4# pH 6.5: about 50% of maximal activity, pH 9.0: about 50% of maximal activity <6>',\n",
       "              '#6# pH 7.0: about 30% of maximal activit, pH 9.5: about 55% of maximal activity <9>'],\n",
       "             '1.1.99.7': ['#1# about 50% of maximal activity at pH 6.3 and 9.5 <1,2>'],\n",
       "             '1.1.99.9': ['#1# pH 5.0: about 30% of activity maximum, pH 7.0: 17% of activity maximum <1>',\n",
       "              '#2# activity at pH 9.5 in Tris is about 70% of that at pH 7.3 while activities at pH 6.4 and pH 6.0 in phosphate buffer are about 75% and 0% respectively of that at pH 7.3 <2>'],\n",
       "             '1.10.3.1': ['#10# pH-profile <44>; #39# activity range <81>; #23# PPO activities are determined in a pH range of 4.0-9.0 <65>',\n",
       "              '#13# 74% and 87% of maximal activity <46>',\n",
       "              '#26# highest activity and stability within a neutral and alkaline pH range <39>',\n",
       "              '#29# inactive at pH 3.0 and pH 7.5, half-maximal activity at pH 3.5 and pH 6.2 <56>',\n",
       "              '#41# 50% activity at pH 3.5 and pH 5.5, 25% activity at pH 3.0 <9>',\n",
       "              '#44# pH profiles with different substrates, overview <90>',\n",
       "              '#57# at pH 6.0, isozyme PPO 1 activity increases drastically, reaching maximum activity at pH 7.0 and then again radically decreases at pH 7.5 <80>',\n",
       "              '#70# pH optimum is determined over a pH 4.0-8.5 range <66>'],\n",
       "             '1.10.3.15': ['#1# more than 50% activity between pH 6.0 and 11.0 <1>'],\n",
       "             '1.10.3.2': ['#61# substrate guaiacol <86>; #1# pH 6.0: about 25% of maximal activity, pH 8.0: about 70% of maximal activity, substrate: 2,6-dimethoxyphenol <217>',\n",
       "              '#1# pH 3.0: about 40% of maximal activity, pH 4.0: about 30% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <217>',\n",
       "              '#103# sharp decline in activity above pH 6.0 <112>; #206# pH 2.0: about 30% of maximal activity, pH 6.0: about 30% of maximal activity <245>',\n",
       "              '#106# inactivation of isozyme Lacc I at pH 6.5, of isozyme Lacc II at pH 7.5 <99>',\n",
       "              '#106# pH 2.2: about 90% of maximal activity, pH 4.3: about 60% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <232>',\n",
       "              '#11# pH 7.5: about 45% of maximal activity, pH 8.5: about 35% of maximal activity, substrate: 2,6-dimethoxyphenol <252>',\n",
       "              '#11# pH 3.5: about 55% of maximal activity, pH 4.5: about 60% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <252>',\n",
       "              '#123# the enzyme retains more than 50% of its maximum activity in the pH range of 3.5-5.5 <123>; #178# pH 3.5: about 70% of maximal activity, pH 5.5: about 60% of maximal activity, substrate: guaiacol <240>',\n",
       "              '#124# pH 5.5: about 90% of maximal activity, pH 6.5: about 60% of maximal activity, oxidation of syringaldazine <126>',\n",
       "              '#124# pH 2.0: about 65% of maximal activity, pH 5.5: about 45% of maximal activity, oxidation of 2,2-azino-bis(3-ethylbenzthiazoline-6-sulfonate) <126>',\n",
       "              '#134# pH profiles of native and recombinant enzymes, overview <164>',\n",
       "              '#138# pH profile with substrate 2,6-dimethoxyphenol <161>',\n",
       "              '#141# activity range, 60-65% of maximal activity at pH 4.0 and pH 10.0, profile, overview <170>',\n",
       "              '#149# activity range <162>; #237# recombinant mutant enzyme MmPPOA-637, inactivation at pH 8.0 and above <259>',\n",
       "              '#15# pH 4.0: about 70% of maximal activity, pH 6.0: about 85% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <243>',\n",
       "              '#17# activity range, dependent on the substrate <147>; #227# pH 4.0: about 65% of maximal activity, pH 9.0: about 70% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <211>',\n",
       "              '#178# pH 2.2: about 40% of maximal activity, pH 4.2: about 60% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <240>; #178# pH 2.2: about 40% of maximal activity, pH 4.2: about 60% of maximal activity, substrate: 2,6-dimethoxyphenol <240>',\n",
       "              '#187# activity range, profile, overview <195>',\n",
       "              '#20# activity range <190>; #35# 60°C, more than 50% of maximal activity in the pH-range 2-8 <144>; #175# the activity rises abruptly when the pH is increased from slightly below 2.0. The activity declines sharply when the pH is raised from 4.0 to 5.0 and is indiscernible at pH 6.0-8.0 <182>',\n",
       "              '#201# for pH ranges between pH 2 and pH 11 enzyme activity ranges between 75 and 100% <225>',\n",
       "              '#217# pH 2.3: about 40% of maximal activity, pH 4.0: about 75% of maximal activity <257>',\n",
       "              '#230# pH 3.0: 93% of maximal activity, pH 11.0: 91% of maximal activity <231>',\n",
       "              '#237# recombinant wild-type enzyme, active range and broad optimum <259>',\n",
       "              '#239# no activity with substrate [Cu(I)(bicinchoninate)2]3- <276>',\n",
       "              '#38# the purified enzyme is highly active within this range <102>; #24# pH 2.0: about 75% of maximal activity, pH 5.0: about 60% of maximal activity, substrate: 2,2-azino-bis-(3-ethylbenzthiazoline-6-sulfonic acid) <246>',\n",
       "              '#202# pH 6.0: about 50% of maximal activity, pH 7.5: abput 80% of maximal activity <227>; #26# pH 6.0: about 55% of maximal activity, pH 7.5: about 55% of maximal activity, immobilized enzyme, substrate: 2,6-dimethoxyphenol <218>',\n",
       "              '#26# immobilized enzyme <4>; #38# a sharp drop in activity at above pH 6.0, pH profile <166>',\n",
       "              '#26# activity range <150>',\n",
       "              '#29# the purified laccase maintains stable at pH 3.0-11.0. The enzyme remains active at pH values from 3.0 to 12.0, and its activity at pH 2.0 and 12.0 are 56.6% and 85.5% of that at pH 7.0, respectively <177>',\n",
       "              '#32# pH 6.0: about 40% of maximal activity, pH 9.0: about 60% of maximal activity <250>',\n",
       "              '#191# activity range <198>; #120# pH 2.5: about 85% of maximal activity, pH 5.0: about 50% of maximal activity, oxidation of 2,2-azino-bis(3-ethylbenzothiazoline-6-sulfonate) <142>; #35# 40% of maximal activity at below pH 2.5 and above pH 5.0 <159>',\n",
       "              '#37# pH 7.5: about 65% of maximal activity, pH 9.0: 72% of maximal activity <230>',\n",
       "              '#41# retains more than 80% of activity after 17 h in this pH-range <84>',\n",
       "              '#46# decline of activity above <12>',\n",
       "              '#181# activity range, at pH 7.0, about 17%, 24%, 33%, 23% and 21% of the maximum oxidation rate is observed for substrate 2,2-azinobis(3-ethylbenzo-6-thiazolinesulfonic) acid, 2,6-dimethoxyphenol, guaiacol, syringaldazine, and anthraquinonic dye Acid Blue 62, respectively, while approximately 2%, 8%, 15%, 4% and 17% of the maximum activity in oxidation of these substrates is obtained at pH 8.0 <149>; #50# pH 5.0: about 20% of maximal activity, pH 8.0: about 65% of maximal activity. The enzyme is inactive at pH 4.0 and at pH 9.0 <210>',\n",
       "              '#55# activity range <158>',\n",
       "              '#56# activity range <155>',\n",
       "              '#58# 3% of maximal activity at pH 6.5, pH-activity profile at 30°C <94>',\n",
       "              '#61# substrate 2,2\\x92-azinobis(3-ethylbenzothiazoline-6-sulfonic acid) <86>',\n",
       "              '#61# substrate syringaldazine <86>',\n",
       "              '#62# depends on substrate <66>',\n",
       "              '#63# abrupt activity decline at pH 6.0, indiscernible at pH 8.0-9.0 <98>; #133# pH 3.0: about 45% of maximal activity, pH 6.0: about 35% of maximal activity, substrate: 2,6-dimethoxyphenol <123>',\n",
       "              '#77# activity range <189>',\n",
       "              '#95# no enzymatic activity above <73>',\n",
       "              '#97# no enzymic activity above <72>'],\n",
       "             '1.10.3.3': ['#10# at pH 4.5 and 8.3: approx. 50% activity <8>',\n",
       "              '#23# free enzyme: approx. 60% activity at pH 5.0, less than 10% activity at pH 9.0, immobilized enzyme: approx. 80% activity at pH 5.0, approx. 45% activity at pH 9.0 <25>'],\n",
       "             '1.10.3.4': ['#4# pH 4.6: about 50% of maximum activity, pH 7.6: about 65% of maximum activity <1>'],\n",
       "             '1.10.3.6': ['#1# 80% of maximum activity at pH 6.0, activity declines sharply at a pH above 7.0 <2>'],\n",
       "             '1.11.1.1': ['#20# four peroxidase isoenzymes <31>',\n",
       "              '#5# 90% of maximal activity at pH 5.0, 32% of maximal activity at pH 8.5 <1>'],\n",
       "             '1.11.1.10': ['#3# pH 2.0: about 60% of maximal acticity of soluble enzyme, about 65% of maximal activity of immobilized enzyme (covalent immobilization of chloroperoxidase on the magnetic p(GMA-MMA-EGDMA) beads). pH 4.0: about 55% of maximal activity of soluble enzyme, about 75% of maximal activity of immobilized enzyme (covalent immobilization of chloroperoxidase on the magnetic p(GMA-MMA-EGDMA) beads) <48>',\n",
       "              '#3# free and bio-conjugated-CPO poly(hydroxypropyl)methacrylateco-poly(ethylene glycol)-methacrylate-3 membrane <70>',\n",
       "              '#3# mainly <61>',\n",
       "              '#3# at pH values lower than 3.0 or higher than 4.0, the enzyme is completely inhibited <79>',\n",
       "              '#3# pH 3.7: about 90% of maximal activity, soluble and immobilized enzyme. pH 7.0: about 55% of maximal activity of immobilized enzyme, about 50% of maximal activity of soluble enzyme <32>'],\n",
       "             '1.11.1.11': ['#41# pH 5.0: 68% of maximal activity, pH 9.0: 23% of maximal activity <52>',\n",
       "              '#42# pH 5: about 50% of maximal activity, pH 7.5: about 50% of maximal activity <53>',\n",
       "              '#6# maximum activity maintained <4>',\n",
       "              '#67# activity 35 units mg-1 protein min-1 at pH 5.0 and pH 7.8 <66>',\n",
       "              '#9# both forms of the enzyme <2>; #32# the activity of APX decreases rapidly at pH above 8.0 <78>',\n",
       "              '#90# 80% of maximum activity <100>'],\n",
       "             '1.11.1.12': ['#14# pH 8.0: about 40% of maximal activity, pH 11: about 80% of maximal activity <52>'],\n",
       "             '1.11.1.13': ['#14# Mn2+ oxidation shows an optimum in the range 5.5-5.75 and negligible activity at pH values below 3.5 and above 6.5 <72>',\n",
       "              '#14# optima for Mn2+-independent 2,6-dimethoxyphenol oxidation <72>',\n",
       "              '#14# optima for Mn2+-dependent 2,6-dimethoxyphenol oxidation <72>',\n",
       "              '#14# the pH-activity profile for Mn2+ oxidation shows an optimum in the range 5.5 - 5.75 and negligible activity at pHs below 3.5 and above 6.5 <72>',\n",
       "              '#20# oxidation of veratryl alcohol decreases more than 95% when the pH is increased from 3 to 4.5 and nearly absent at pH 5 <35>',\n",
       "              '#20# pH 3.5: decrease of 95% of oxidation of Mn2+, pH 3: nearly no oxidation of Mn2+ <35>',\n",
       "              '#31# active in the range, below pH 2 and above pH 6: completely inactive <33>',\n",
       "              '#6# about half-maximal activity at pH 4.0 and 5.5, 2,2-azino-bis-3-ethyl-6-benzothiazolinesulfonate-oxidation <6>',\n",
       "              '#6# pH 3.5: about 30% of maximal activity, pH 4.0: about 70% of maximal activity, pH 6.0: about 45% of maximal activity <54>',\n",
       "              '#6# about 65% of maximal activity at pH 4.5 and about 50% at pH 5.7, Mn(III)-lactate formation <6>',\n",
       "              '#6# in presence of H2O2 the formation of enzyme compound I is independent of pH over the range <8>',\n",
       "              '#6# veratryl alcohol oxidation, activity increases with increasing pH <20>'],\n",
       "             '1.11.1.14': ['#6# pH range for tetracycline and oxytetracycline degradation, no removal is observed when pH was below 2.8 or above 5.4. The degradation percentage is comparatively stable and high when pH is between 3.6 and 4.2 for tetracycline and oxytetracycline <48>',\n",
       "              '#7,33,39,40,42# 25°C <34>'],\n",
       "             '1.11.1.16': ['#14# more than 60% activity between pH 3.0 and 6.5 <49>'],\n",
       "             '1.11.1.18': ['#20# pH 5.0: about 35% of maximal activity, pH 7.0: about 25% of maximal activity <21>',\n",
       "              '#8# pH 5.5: 70-90% of maximal activity depending on enzyme form, pH 10.0: about 40% of maximal activity <27>'],\n",
       "             '1.11.1.19': ['#32# the enzyme shows more than 35% activity at pH 4.0-9.0 <49>'],\n",
       "             '1.11.1.20': ['#4# pH 6.5: about 50% of maximal activity, pH 8.0: about 70% of maximal activity <1>'],\n",
       "             '1.11.1.21': ['#10# peroxidatic activity <29>',\n",
       "              '#26# pH range for both catalase and peroxidase activities, little or no activity is seen at pH 9.0 <17>',\n",
       "              '#35# for catalase activity <21>',\n",
       "              '#7# catalase activity <3>',\n",
       "              '#7# peroxidase activity <3>; #14# more than 75% of maximum activity within <58>'],\n",
       "             '1.11.1.24': ['#35# no activity at or below pH 5.4, optimal activity at pH 9.0 <19>'],\n",
       "             '1.11.1.25': ['#4# pH 7.5: about 50% of maximal activity, pH 9.3: about 55% of maximal activity, substrate: tert-butyl hydroperoxide <2>'],\n",
       "             '1.11.1.3': ['#1# highest activity <2>'],\n",
       "             '1.11.1.5': ['#29# by using the maximum current in the cathodic and anodic halfscans in the course of altering the pH a shift in potential is observed, at pH 7, a single redox couple (I) is dominant, but at more acidic pH values, a second couple (II) is clearly discernible at a more positive value of potential <86>',\n",
       "              '#4# the association rate constant for the binding of cyanide to H52L mutant varies almost 4 orders of magnitude in this pH range. Above pH 8 cyanide binds more rapidly to H52L mutant than to wild-type enzyme <34>'],\n",
       "             '1.11.1.6': ['#102# above 90% of maximum activity <101>',\n",
       "              '#111# 60 min, 30°C, no loss of activity <114>',\n",
       "              '#12# the activity of Eupergit C-immobilized catalase remains almost constant for pH values 6.0-7.5 whereas this is not case for the free catalase <159>',\n",
       "              '#139# activit range <166>',\n",
       "              '#142# activity range, 53.4% of maximal activity at pH 4.0, 18.4% at pH 3.0  <171>',\n",
       "              '#15# activity within <31>; #73# pH-profile, TvC-II <44>',\n",
       "              '#150# raspid decrease in activity below and above <183>',\n",
       "              '#65# activity within <19>; #69# more than 60% of catalase maximum activity within <90>; #18# activity range, free enzyme and immobilized enzyme on nanoparticles <162>',\n",
       "              '#21# isoforms Cat-1, Cat-3 <26>',\n",
       "              '#21# isoform Cat-2 <26>',\n",
       "              '#28# high activity within <13>',\n",
       "              '#47# more than 50% of maximum activity within <107>',\n",
       "              '#143# activity range <172>; #47# about 40% of maximal activity at pH 3 and at pH 10 <71>',\n",
       "              '#73# pH-profile, TvC-I <44>',\n",
       "              '#8# peroxidase activity <39>',\n",
       "              '#8# catalase activity <39>; #69# more than 80% of catalase maximum activity within <90>',\n",
       "              '#83# activity within <33>',\n",
       "              '#88# the purified CatP is stable and yields similar activity from pH 4.0 to 12.0 <157>',\n",
       "              '#96# rapid drop of activity below pH 6.0 and above pH 9.0 <43>'],\n",
       "             '1.11.1.7': ['#104# the activity of the enzyme for ABTS ranges between 100% and 55% at pH 3.0-4.0 and for 2,6-dimethoxyphenol at pH 2.0-4.0 <149>; #106# more than 93% of maximum activity within <166>',\n",
       "              '#13# approx. 50% of maximal activity at pH 4.5 and pH 6.5, respectively <48>',\n",
       "              '#15# approx. 60% of maximal activity at pH 6.0 and pH 8.0, approx. 30% of maximal activity at pH 9.5, almost no activity at pH 10.5 <45>',\n",
       "              '#17# pH 3: about 60% of maximal activity, pH 8: about 40% of maximal activity <76>',\n",
       "              '#30# almost no activity at pH 3.0 and pH 9.0, respectively <44>; #30# almost no activity at pH 3.0, approx. 20% of maximal activity at pH 9.0 <44>; #30# approx. 20% of maximal activity at pH 3.0 and pH 8.0, respectively <44>; #30# membrane-bound isoform <169>',\n",
       "              '#40# POD activities are determined in a pH range of 4.0-9.0 <91>',\n",
       "              '#49# 50% activity at pH 3.2 and 6.2 <5>',\n",
       "              '#56# although peroxidase activity is comparable in the pH range of 6.0-7.5, significant decline in enzyme activity is observed when the pH is increased to above 8.0 <139>',\n",
       "              '#56# assay at 4.5.and 7.0 <83>',\n",
       "              '#56# oxidation of 5-aminosalicylic acid, approx. 30% of maximal activity at pH 4.5 and pH 7.0, respectively <37>',\n",
       "              '#57# oxidative dehalogenation of 3,5-dibromo-4-hydroxybenzonitrile, approx. 5% of maximal activity at pH 2.5 and pH 6.0, respectively <41>',\n",
       "              '#57# oxidative dehalogenation of 2,2-methylene-bis[4-chlorophenol], approx. 2% of maximal activity at pH 2.5 and pH 6.0, respectively <41>',\n",
       "              '#57# oxidative dehalogenation of pentachlorophenol, approx. 50% of maximal activity at pH 2.0 and pH 6.5 respectively <41>',\n",
       "              '#79# pH 4: about 40% of maximal activity, pH 7: about 35% of maximal activity <56>',\n",
       "              '#9# approx. 30% of maximal activity at pH 2.0 <49>'],\n",
       "             '1.11.1.8': ['#4# activity independent of pH <1>',\n",
       "              '#6# 20% of activity at pH 2.5, 70% at pH 7.5 <2>'],\n",
       "             '1.11.1.9': ['#49# 26% of maximum activity, mutant C2S/C78S/C115S/C156S/C202S <143>',\n",
       "              '#7# less than 20% of maximal activity below <17>',\n",
       "              '#9# pH 7.5: about 45% of maximal activity, pH 9.0: about 50% of maximal activity <54>'],\n",
       "             '1.11.2.1': ['#1# more than 50% activity between pH 4.0 and 7.5 <35>',\n",
       "              '#2# pH 6.0: about 60% of maximal activity, pH 9.0: about 45% of maximal activity <22>',\n",
       "              '#3# for the reaction with methyl 3,4-dimethoxybenzyl ether <15>',\n",
       "              '#4# oxidation of veratryl alcohol and benzyl alcohol occurs in a broad pH range between 2.5 and 9.0 <4>',\n",
       "              '#5# more than 50% activity between pH 5.5 and 7.5 <35>',\n",
       "              '#6# more than 50% activity between pH 4.0 and 8.0 <35>',\n",
       "              '#7# activity range, profile overview. Analysis of pH dependencies of wild-type and mutant enzymes <38>',\n",
       "              '#8# there is substantial activity loss outside the pH range of 4.5-9.0 <30>'],\n",
       "             '1.11.2.2': ['#2# between pH 5.0 and 7.0, myeloperoxidase converts about 90% of available hydrogen peroxide to hypochlorous acid and the remainder to hypobromous acid. Above pH 7, there is an abrupt rise in the yield of hypobromous acid. At pH 7.8, it accounts for 40% of the hydrogen peroxide <26>'],\n",
       "             '1.12.1.2': ['#17# Tris-HCl buffer and potassium phosphate buffer pH 6.9: 20% activity, Tris-HCl buffer pH 9.1: 40% activity, potassium phosphate buffer pH 9.1: 65% activity <3>',\n",
       "              '#4# triethanolamine buffer, Tris buffer pH 7.0: 40-45% activity, triethanolamine buffer, Tris buffer pH 9.5: 60-75% activity <11>'],\n",
       "             '1.12.2.1': ['#3# pH 5.0: about 30% of maximal activity, pH 9.0: about 35% of maximal activity <2>'],\n",
       "             '1.12.7.2': ['#7# pH 6.0: about 60% of maximal activity, pH 8.0: about 70% of maximal activity <48>'],\n",
       "             '1.12.98.1': ['#14# the methylviologen-reducing activity increased with pH from pH 5.5-10 <20>',\n",
       "              '#2# F420 reduction <9>',\n",
       "              '#2# increasing methyl viologen activity <9>',\n",
       "              '#6# reverse reaction <15>'],\n",
       "             '1.12.98.2': ['#4# reaction rate increases hyperbolically with the proton concentration between pH 7 and pH 4.5 <8>'],\n",
       "             '1.12.98.4': ['#10# no activity is observed below pH 6.0, the activity gradually increases up to pH 8.0, above pH 8.0 the rate of non-enzymic sulfur disproportionation reaction is higher than the enzymic rate <16>'],\n",
       "             '1.12.99.6': ['#11# H2 uptake is constant over a pH range from pH 5 to pH 10 <5>',\n",
       "              '#20# the maximum H2 oxidation current increases as the pH is raised from 4.3 to 5.9, then it is constant up to pH 8.4 <46>',\n",
       "              '#51# almost no activity at pH 6.0, approx. 30% of maximal activity at pH 11.0 <33>',\n",
       "              '#97# optimal pH range for the hydrogen oxidation activity is 8.5 to 9.0, optimal pH range for the proton reduction activity is 6.0 to 6.5 <91>',\n",
       "              '#99# H2 oxidation, pH 5.5: 40% of maximal activity, pH 9: 60% of maximal activity <80>',\n",
       "              '#99# H2 evolution, pH 3.5 and 9.0: ca. 50% of maximal activity <80>'],\n",
       "             '1.13.11.1': ['#13# pH 6.0: about 60% of maximal activity, pH 9.5: about 60% of maximal activity <41>',\n",
       "              '#21# more than 50% activity between pH 4.5 and 8.0 <76>',\n",
       "              '#31# activity range, immobilized enzyme <72>',\n",
       "              '#33# the activity rapidly decreases outside this range <24>',\n",
       "              '#36# 45% of specific activity in pH range from 8-10.5 <59>',\n",
       "              '#41# increase of enzyme activity at pH 5.8-8.8, 29.3% decrease of enzyme activity at pH 9.4, 50% decrease of enzyme activity at pH 10.7 <63>',\n",
       "              '#5# high activity <46>',\n",
       "              '#34# for CD I and CD II respectively <11>; #6# pH 6.5: about 25% of maximal activity, pH 10: about 70% of maximal activity <49>; #33# below no enzyme activity, at pH 8.5 40% residual enzyme activity of the immobilized enzyme compared to inactive free form enzyme <61>',\n",
       "              '#62# differs between wild-type and mutant enzymes <60>',\n",
       "              '#66# activity range, profile overview <69>',\n",
       "              '#72# activity range, profile overview <74>'],\n",
       "             '1.13.11.12': ['#12# at pH 7.8, ca. 50% of maximal activity, at pH 8.4, ca. 30% of maximal activity, substrate alpha-linoleic acid <20>',\n",
       "              '#12# at pH 8.5, ca. 30% of maximal activity, at pH 8.9, ca. 80% of maximal activity, substrate linolenic acid <20>',\n",
       "              '#14# over 50% of maximal activity at pH 6.5-8.0, 40% of maximal activity at pH 8.5 <40>',\n",
       "              '#30# over 60% of maxmal activity within this range <44>',\n",
       "              '#6# activity range, overview <33>',\n",
       "              '#8# pH 5.6. about 50% of maximal activity, pH 8.5: about 70% of maximal activity <10>',\n",
       "              '#9# activity range, overview. Low activity below pH 7.0, maximum activity at pH 8.0, 75% of maximal activity at up to pH 10.5 <38>'],\n",
       "             '1.13.11.15': ['#3# 78% of maximal activity at pH 7.1 and 9.3 <18>',\n",
       "              '#4# activity range, profile overview. pH-rate profile of enzyme PaDHPAO reaction shows a bell-shaped plot that exhibits a maximum activity at pH 7.5 with two pKa values of pH 6.5 and pH 8.9 <43>'],\n",
       "             '1.13.11.17': ['#4# pH 3.8: about 80% of maximal activity, pH 5.8: about 65% of maximal activity, pH 6.2: about  30% of maximal activity <3>'],\n",
       "             '1.13.11.18': ['#15# activity range, recombinant enzyme <11>',\n",
       "              '#16# activity range, recombinant enzyme <11>'],\n",
       "             '1.13.11.2': ['#60# pH 4.0: 29% of maximal activity, pH 6.5: about 50% of maximal activity, pH 9.0: 61% of maximal activity <59>',\n",
       "              '#7# wild-type C23O only 50% residue activity at pH 9.5, mutated C23O still 75% residue activity at pH 9.5 <68>'],\n",
       "             '1.13.11.24': ['#3# Britton-Robinson buffer, composed of H3PO4, acetic acid, H3BO3 <7>',\n",
       "              '#3# 50 mM MES buffer <7>',\n",
       "              '#4# 50 mM MES-Tris buffer <23>'],\n",
       "             '1.13.11.25': ['#2# inactive below <2>'],\n",
       "             '1.13.11.27': ['#15# pH 6.0: about 55% of maximal activity, pH 9.5: about 50% of maximal activity <9>',\n",
       "              '#17# pH 5.5: about 50% of maximal activity, pH 7.7: about 55% of maximal activity <31>',\n",
       "              '#7# reaction rate is constant between pH 5 and pH 7 <1>'],\n",
       "             '1.13.11.28': ['#1# no activity above pH 7 <2>'],\n",
       "             '1.13.11.3': ['#26# pH 6.0: about 40% of maximal activity, pH 9.0: about 65% of maximal activity <45>',\n",
       "              '#28# 80% of maximal activity within this range <54>',\n",
       "              '#28# activity range, profile overview <54>',\n",
       "              '#35# the enzyme shows more than 50% activity between pH 7.0 and 11.0 and loses 98.3% of its original enzymatic activity at pH 4.0 and about 95.4% at pH 12.0 <60>',\n",
       "              '#36# the enzyme shows more than 50% activity between pH 5.5 and 8.5 <56>'],\n",
       "             '1.13.11.31': ['#12# pH 6.3: about 70% of maximal activity, pH 8.8: about 40% of maximal activity <15>',\n",
       "              '#3# pH 6.1: 37% of maximal activity, pH 8.6: 44% of maximal activity <19>'],\n",
       "             '1.13.11.33': ['#2# pH 6 about 50% of maximal activity, pH 9.5 about 60% of maximal activity <105>',\n",
       "              '#2# approximately level between pH 7 and 8 <104>'],\n",
       "             '1.13.11.34': ['#4# pH 7.0: 66% of maximal activity, pH 9.0: about 60% of maximal activity <13>',\n",
       "              '#9# almost inactive at pH 3.5 and at pH 7.0 <18>'],\n",
       "             '1.13.11.39': ['#33# pH 7.0: about 50% of maximal activity, pH 9.0: about 45% of maximal activity <33>',\n",
       "              '#30# activity range of immobilized enzyme <44>',\n",
       "              '#6,35,36# activity range, profile overview <41>; #26# over 60% at pH 6.0 and 9.0, and less than 50% catalytic efficiency at pH 5.0 and 10.0, strain BphCLA-4 <42>'],\n",
       "             '1.13.11.4': ['#22# activity range, isozymes Nagl2 and Nagl3 <37>'],\n",
       "             '1.13.11.45': ['#1# in the presence of 0.37 mM Ca2+ <1>',\n",
       "              '#1# in the absence of Ca2+ <1>',\n",
       "              '#5# at pH 5.0 64% of maximum activity, at pH 11 69% of maximum activity <2>',\n",
       "              '#5# 93% of maximum activity <5>',\n",
       "              '#5# 98% of maximum activity <5>'],\n",
       "             '1.13.11.48': ['#3# pH profile, C59S Hod, overview <5>'],\n",
       "             '1.13.11.49': ['#11# activity range, pH-rate profiles and pH-dependent ligand binding constants, active and inactive enzyme forms dependent on the protonation status, steady-state pH-rate profiles, overview <18>',\n",
       "              '#13# sodium phosphate buffer, 25°C <15>',\n",
       "              '#4# activity range <23>',\n",
       "              '#6# pH 5.5: about 85% of maximal activity, pH 7.0: about 40% of maximal activity <1>',\n",
       "              '#7# pH 4.8: 45% of maximal activity, pH 7.0: 45% of maximal activity, strain KJ <6>'],\n",
       "             '1.13.11.5': ['#4# approx. 15% of maximal activity at pH 5.0 and pH 8.0, respectively <16>'],\n",
       "             '1.13.11.52': ['#12# 5% activity loss at pH 5.5, 10% activity loss at pH 7, 20% activity loss at pH 8 <79>',\n",
       "              '#28# pH 7.0: about 40% of maximal activity, pH 9.5: about 50% of maximal activity <32>',\n",
       "              '#31# pH 6.0: about 45% of maximal activity, pH 9.0: about 40% of maximal activity <4>',\n",
       "              '#33# pH 6.5: about 50% of maximal activity, pH 9.0: about 70% of maximal activity <120>',\n",
       "              '#41# pH 5.0: about 50% of maximal activity, pH 7.5: about 65% of maximal activity <107>',\n",
       "              '#6# pH 5.0: about 50% of maximal activity, pH 7.0: about 65% of maximal activity <107>'],\n",
       "             '1.13.11.55': ['#1# activity range, recombinant enzyme, profile overview <16>',\n",
       "              '#15# activity range, proflie overview, at 50°C <22>',\n",
       "              '#2# the enzyme is active from pH 5.5 to 8 <10>; #2# activity range, profile overview, at 50°C <22>'],\n",
       "             '1.13.11.58': ['#10# pH 5.3: about 80% of maximal activity, pH 6.5: about 40% of maximal activity <13>',\n",
       "              '#13# pH 5.0: about 50% of maximal activity, pH 8.5: about 70% of maximal activity <1>',\n",
       "              '#17# pH 8: 30% of optimal activity, pH 9: about 15% of maximal activity <8>',\n",
       "              '#21# over 80% of maximal activity in a broad pH range with sharply decreasing relative activity below pH 5.0 and above pH 7.5 <26>',\n",
       "              '#6# pH 5.0: about 50% of maximal activity, pH 7: about 70% of maximal activity <2>'],\n",
       "             '1.13.11.6': ['#5# at pH 4.5: about 70% of activity maximum, at pH 7.0: about 50% of activity maximum <3>',\n",
       "              '#8# determination on heat-reactivated extracts after inactivation <6>'],\n",
       "             '1.13.11.61': ['#2# 50% of maximal activity at pH 4.5-6.5, maximal activity at pH 8.5 <4>'],\n",
       "             '1.13.11.63': ['#22# 70% of maximum activity <33>',\n",
       "              '#9# pH 7.0: about 50% of maximal activity, pH 9.0: about 35% of maximal activity, reaction with beta-apo-10-carotenol <1>'],\n",
       "             '1.13.11.66': ['#4# 20% of maximum activity <7>'],\n",
       "             '1.13.11.69': ['#4# activity range, profile overview <10>'],\n",
       "             '1.13.11.71': ['#4# 50 % of the maximal activity at pH 7.0 and pH 9.0, 20% at pH 6.0 and pH 9.5 <15>'],\n",
       "             '1.13.11.85': ['#3# pH values below pH 6 or above 8.5 result in only poor activities <7>'],\n",
       "             '1.13.11.86': ['#1# pH 6.5: about 50% of maximal activity, pH 9.0: about 40% of maximal activity <2>'],\n",
       "             '1.13.11.87': ['#4# pH 5.0: about 80% of maximal activity, pH 9.0: about 70% of maximal activity <8>'],\n",
       "             '1.13.12.16': ['#1# investigation of the pH effects on the kcat/Km-values <11>',\n",
       "              '#1# assays are carried out at pH 6, 8, 9.5, and 10 <14>',\n",
       "              '#5# when the enzyme is acidified to pH 3.0 and treated in the same way, the prosthetic groups do not dissociate from the protein and almost full activity remained <2>'],\n",
       "             '1.13.12.19': ['#4# over 50% of maximal activity within this range <7>'],\n",
       "             '1.13.12.2': ['#1# at pH 8.4 about 50% of activity maximum, at pH 9.4 about 40% of activity maximum <9>'],\n",
       "             '1.13.12.21': ['#1# pH 7.0: about 40% of maximal activity, pH 8.0: about 85% of maximal activity <1>'],\n",
       "             '1.13.12.3': ['#1# pH 5.5: about 50% of activity maximum, pH 10: about 55% of activity maximum <1>'],\n",
       "             '1.13.12.7': ['#1# pH 6.5: about 45% of activity maximum, pH 8.5: about 35% of activity maximum <21>',\n",
       "              '#1# pH up to 8 red light occurs instead of yellow-green light <8,9>; #1# pH below 7.5 emission shifts to longer wavelengths <9>',\n",
       "              '#1# peak emission at 560 nm <5>; #16# depending on mutation <106>',\n",
       "              '#18# bioluminescence spectra independent from pH, for mutants Y35N, Y35H and F16L/A40S <93>'],\n",
       "             '1.13.12.9': ['#1# 50% of maximal activity at pH 9.8 and 11.8, formation of beta-phenylpyruvate <3>'],\n",
       "             '1.13.99.1': ['#4# sharp decrease of activity below pH 6.5 and above pH 7.4 <1>',\n",
       "              '#8# almost no activity at pH 4.5, approx. 10% of maximal ativity at pH 10.0, approx. 70% of maximal activity at pH 8.0, i.e. a second optima <15>'],\n",
       "             '1.14.11.1': ['#4# half-maximal activity at pH 6.0 and 8.4, partially purified preparation <15>',\n",
       "              '#4# half-maximal activity at pH 6.0 and 7.5 <1>'],\n",
       "             '1.14.11.11': ['#10# pH 5.0: 20% of maximal activity, pH 7.4: optimum <10>',\n",
       "              '#3# about 50% of maximal activity at pH 6.0 and pH 9.0 <1>'],\n",
       "             '1.14.11.15': ['#1# about half-maximal activity at pH 7.0 and pH 9.0 <2>',\n",
       "              '#7# activity is reduced appreciably below pH 6 and above pH 9 <10>'],\n",
       "             '1.14.11.16': ['#4# about half-maximal activity at pH 6.5 and 7.9 <9>'],\n",
       "             '1.14.11.18': ['#2# at pH 6.5 over 80%, at pH 8.5 50% of control activity <4>'],\n",
       "             '1.14.11.43': ['#1# pH 6.0: about 30% of maximal activity, pH 7.5: about 50% of maximal activity <1>'],\n",
       "             '1.14.11.44': ['#1# pH 6.0: about 60% of maximal activity, pH 7.5: about 55% of maximal activity <1>'],\n",
       "             '1.14.11.45': ['#1# about 25% activity at pH 5.5 and 7.0, respectively, about 50% activity at pH 6.5 <5>'],\n",
       "             '1.14.11.52': ['#1# pH 6.5: 65% of maximal activity, pH 8.0: about 70% of maximal activity <1>'],\n",
       "             '1.14.12.10': ['#2# pH 6.0: about 55% activity, pH 8.0: about 60% activity <1>'],\n",
       "             '1.14.12.13': ['#4# assay at <7>'],\n",
       "             '1.14.12.24': ['#3# the enzyme is stable when incubated at pH 6 and 7, but its optimal activity is at pH 7.0. The activity is decreased significantly when incubated at pH 4.0, 5.0, 8.0, and 10.0 <3>'],\n",
       "             '1.14.13.1': ['#10# about 75% of activity maximum at pH 6.5 and 8.5 <21>',\n",
       "              '#4# about 50% of activity maximum at pH 6 and 10 <3>',\n",
       "              '#5# 65% of maximal activity at pH 6, 96% of maximal activity at pH 8 <20>'],\n",
       "             '1.14.13.128': ['#1# about 50% of maximal activity <2>'],\n",
       "             '1.14.13.130': ['#1# pH 7.2: 65% of maximal activity, pH 9.5: 68% of maximal activity <1>'],\n",
       "             '1.14.13.148': ['#2,7# the activity is approximately twice as high at pH 8.4 as at pH 7.4 <17>'],\n",
       "             '1.14.13.153': ['#1# 50% of maximal activity at pH 6.8 and pH 8.6 <1>'],\n",
       "             '1.14.13.161': ['#2# more than 60% of maximum activity within <4>'],\n",
       "             '1.14.13.163': ['#7# activity range, profile overview <6>'],\n",
       "             '1.14.13.189': ['#1# pH 7.0: about 45% of maximal activity, pH 10.0: about 80% of maximal activity <1>'],\n",
       "             '1.14.13.195': ['#2# pH profile, overview <5>',\n",
       "              '#4# activity range, pH profile <7>'],\n",
       "             '1.14.13.200': ['#2# pH 7.5: about 45% of maximal activity, pH 10: about 25% of maximal activity <1>'],\n",
       "             '1.14.13.22': ['#12# no activity below pH 5.5 and above pH 10.5 at 30°C <25>',\n",
       "              '#16# 70% of maximal activity at pH 7.0 within this range, His-tagged recombinant enzyme <40>',\n",
       "              '#6# pH 6.0: about 40% of activity maximum with isozymes type 1 and type 2, pH 10.5: about 10% of activity maximum with isozyme type 1, about 60% of activity maximum with isozyme type 2 <17>',\n",
       "              '#9# at pH 6.0 and 11.0: about 10% of activity maximum <4>'],\n",
       "             '1.14.13.24': ['#12# high activity range <15>',\n",
       "              '#13# at pH 5.5-8, the residual activity of soluble enzyme is less than that of the immobilized one <23>',\n",
       "              '#2# no activity below pH 6.5 and above pH 11 <6>'],\n",
       "             '1.14.13.25': ['#3# pH 6.4: about 20% of activity maximum, pH 7.4: about 25% of activity maximum <2>'],\n",
       "             '1.14.13.31': ['#2# pH 6.5: about 60% of activity maximum, pH 9.0: about 55% of activity maximum <2>'],\n",
       "             '1.14.13.33': ['#2# 50% of activity maximum at pH 6.5 and pH 8.8, NADH <1>',\n",
       "              '#2# pH 6.2: about 50% of activity maximum, pH 8.8: about 70% of activity maximum, NADPH <1>'],\n",
       "             '1.14.13.35': ['#1# pH 7.5: about 80% of activity maximum, pH 9.5: about 55% of activity maximum <12>; #1# at pH 7.5 or pH 9.0 about 60% of activity maximum <10>',\n",
       "              '#2# less than 1% of maximal activity at pH 5.5 and pH 9.8 <4>'],\n",
       "             '1.14.13.4': ['#2# at pH 5.0 and 9.0: about 30% of maximum activity <4>'],\n",
       "             '1.14.13.44': ['#2# more than 80% of maximal activity at pH 7.2 and pH 7.8, beyond pH 7.8 activity declines abruptly with increasing pH <4>'],\n",
       "             '1.14.13.50': ['#4# pH 6.0: about 28% of activity maximum, pH 7.5-8.5: activity maximum <3>'],\n",
       "             '1.14.13.54': ['#3# pH 6.0: about 35% of maximal activity, pH 9.0: about 65% of maximal activity, progesterone monooxygenase activity <2>',\n",
       "              '#3# pH 5.5: about 20% of maximal activity, pH 8.0: about 40% of maximal activity, androstenedione monooxygenase activity <1>'],\n",
       "             '1.14.13.58': ['#1# 50% of maximal activity at pH 6.5 and at pH 8.5 <2>'],\n",
       "             '1.14.13.59': ['#2# stable higher activity within this range <11>'],\n",
       "             '1.14.13.63': ['#2# 50% of maximal activity at pH 7.3 and 8.9 <1>'],\n",
       "             '1.14.13.7': ['#1# assay at <44>',\n",
       "              '#12# about 50% of activity maximum at pH 7.0 and 8.7 <1>',\n",
       "              '#20# pH 6.5: about 45% of maximal activity, pH 8.5: about 30% of maximal activity, oxygenase component PHO <29>',\n",
       "              '#7# more than 80% activity between pH 7.0 and 9.0 <60>'],\n",
       "             '1.14.13.8': ['#2# pH profile for the recombinant FMO1, FMO3, and FMO5, overview <71>',\n",
       "              '#2,8# about 50% of activity maximum at pH 7.6 and 9.6, liver, thiobenzamide S-oxidation <5>',\n",
       "              '#2,8# about 50% of activity maximum at pH 8.4 and 10.4, lung, thiobenzamide S-oxidation <5>',\n",
       "              '#3# isozyme FMO3, pH profile <48>',\n",
       "              '#3# pH 7.6: about 70% of maximal activity, pH 9.0: about 65% of maximal activity, p-tolyl sulfide S-oxidase activity <40>; #12# S-oxidase activity incrases from pH 7.6-9.0 <40>',\n",
       "              '#14# pH profile <48>; #3# isozyme FMO1, pH profile <48>',\n",
       "              '#5# pH 7.5: about 40% of activity maximum, pH 9.0: about 95% of activity maximum <6>',\n",
       "              '#9# pH profile <47>'],\n",
       "             '1.14.13.84': ['#1# effects of pH on enantioselectivity, overview <9>',\n",
       "              '#4# maximum activity pH at 8.0, below pH 6.0 the activity is very low, acetate buffer (pH 4.0-5.5), phosphate buffer (pH 7.5-8.0), Tris-HCl (pH 7.5-9.0), 25°, 30 min <10>'],\n",
       "             '1.14.13.9': ['#5# about 50% of maximal activity at pH 7 and 9 <18>',\n",
       "              '#5# about 50% of maximal activity at pH 7 and 9.3 <2>',\n",
       "              '#7# about 50% of maximal activity at pH 7.1 and 8.9 <7>'],\n",
       "             '1.14.13.92': ['#4# more than 80% of maximal activity <1>',\n",
       "              '#5# pH profile, and dependence of the enantioselectivity of the reaction, overview <9>'],\n",
       "             '1.14.14.1': ['#17# recombinant Escherichia coli cell expressing the enzyme, below 7.0 or over 8.0 the enzyme activity decreases drastically, pH profile, overview <63>'],\n",
       "             '1.14.14.10': ['#3# 35% activity of the purified enzyme remains at pH 7.0 and 9.6 <1>'],\n",
       "             '1.14.14.106': ['#2# half-maximal activity at pH 6.2 and pH 8.2 <1>'],\n",
       "             '1.14.14.108': ['#1# 80% of activity maximum at pH 6.5 and pH 8.5 <3>',\n",
       "              '#1# 70% of activity maximum at pH 7 and pH 8, NADH-dehydrogenase <5>'],\n",
       "             '1.14.14.135': ['#1# pH 7: 50% of maximum activity, no activity measurable above pH 9.5, probably because of the instability of the substrate <1>'],\n",
       "             '1.14.14.139': ['#5# pH 6.0: about 40% of maximal activity, pH 7.5: about 65% of maximal activity <11>'],\n",
       "             '1.14.14.144': ['#1# 60-70% of maximal activity at pH 7.0 and 8.0 <1>'],\n",
       "             '1.14.14.145': ['#2# 60-70% of maximal activity at pH 7.0 and 8.0 <3>; #4# about 80% of maximal activity at pH 7.0 and 8.0 <2>'],\n",
       "             '1.14.14.15': ['#1# pH 5.5: about 50% of maximal activity, pH 7.0: about 50% of maximal activity <2>'],\n",
       "             '1.14.14.16': ['#5# about 50% of activity maximum at pH 5.5 and 8.0 <1>'],\n",
       "             '1.14.14.174': ['#1# 50% of maximal activiyt at pH 6.7 and pH 8.3. Tris-Cl buffer <1>'],\n",
       "             '1.14.14.18': ['#1# pH 6.5: 63% activity, pH 7.9: 50% activity <4>',\n",
       "              '#8# binding capacity increases with the increasing pH from 7 to 9 <65>'],\n",
       "             '1.14.14.20': ['#7# activity range, profile overview <2>'],\n",
       "             '1.14.14.28': ['#4# both the wild-type and the mutants are active at pH values from 6.0 to 8.8 <2>'],\n",
       "             '1.14.14.33': ['#4# 75% of maximum activity <1>'],\n",
       "             '1.14.14.40': ['#2# 25% of maximum activity <1>'],\n",
       "             '1.14.14.51': ['#13# about 40% activity at pH 5.5, about 75% activity at pH 6.5, about 90% activity at pH 7.0, 100% activity at 8.0, about 75% activity at pH 9.0, about 40% activity at pH 10.0 <15>',\n",
       "              '#9# about half-maximal activity at pH 6.9 and 7.9 <2>'],\n",
       "             '1.14.14.52': ['#1# almost no activity at pH 5.5 and pH 8.0, respectively <2>',\n",
       "              '#2# about half-maximal activity at pH 7.3 and 8.3 <1>'],\n",
       "             '1.14.14.82': ['#3# half-maximal activity at pH 6.8 and 8.4 <7>'],\n",
       "             '1.14.14.87': ['#1# half-maximal activity <12>',\n",
       "              '#1# half-maximal activity <12>'],\n",
       "             '1.14.14.9': ['#12# substrate 4-aminophenylacetate can only bind to the wild-type enzyme at pH 6.0, substrate can bind to mutant S146A at both pH 6.0 and 9.0 <37>',\n",
       "              '#2# more than 80% of activity <14>',\n",
       "              '#8# assay range for determination of reaction kinetics of C2-FMNH with oxygen at various pH values investigated by stopped-flow and rapid quenched-flow techniques, detailed overview <29>'],\n",
       "             '1.14.14.91': ['#10# about 50% of maximal activity at pH 5.5 and at pH 8.8 <8>',\n",
       "              '#22# 50% of maximal activity at pH 6.3 and pH 8.2 <31>'],\n",
       "             '1.14.14.92': ['#1# approx. 35% of maximal activity at pH 4.5, approx. 15% of maximal activity at pH 8.0 <4>',\n",
       "              '#2# approx. 30% of maximal activity at pH 5.5, approx. 35% of maximal activity at pH 8.2 <3>'],\n",
       "             '1.14.14.94': ['#2# pH 6.8: about 50% of activity maximum, pH 9.5: about 70% of activity maximum <2>'],\n",
       "             '1.14.14.95': ['#1# 60% of maximal enzyme activity at pH 7.5 and 9.0 (no difference in activity between bis-Tris and Tris buffer) <2>'],\n",
       "             '1.14.14.97': ['#1# half-maximal activity at pH 6.5 and pH 9.7 <1>'],\n",
       "             '1.14.14.98': ['#1# pH 6.3: about 35% of maximal activity, pH 8.5: about 40% of maximal activity <1>'],\n",
       "             '1.14.14.99': ['#1# about half-maximal activity at pH 6.9 and 7.9 <1>',\n",
       "              '#2# half-maximal activity at pH 6.75 and 8.5, recombinant chimeric mutant enzyme <11>'],\n",
       "             '1.14.15.15': ['#2# pH 8.3: about 20% of activity maximum, pH 11.2: about 60% of activity maximum <3>',\n",
       "              '#3# 27-hydroxylation <9>'],\n",
       "             '1.14.15.18': ['#1# about 50% of activity maximum at pH 7.1 and pH 7.7 <2>'],\n",
       "             '1.14.15.3': ['#23# laurate hydroxylation in Tris buffer <3>'],\n",
       "             '1.14.15.6': ['#6# less than 50% of maximal activity below and above range <7>'],\n",
       "             '1.14.16.1': ['#3# recombinant enzyme, 25% loss of activity at pH 5.5, 15% loss of activity at pH 7.0-8.0 <33>'],\n",
       "             '1.14.16.2': ['#5# 4 isoforms <25>',\n",
       "              '#6# pH 5.0: less than 5% of maximal activity, pH 5.5-6.0: optimum, pH 6.8: about 30% of maximal activity, striatal homogenate <41>',\n",
       "              '#6# activity declines precipitously below pH 5.9 and above pH 7.4 <16>'],\n",
       "             '1.14.16.4': ['#4# about 50% activity at pH 6.0 and 8.7 <8>',\n",
       "              '#6# about 50% activity at pH 6 and 9.5 <9>'],\n",
       "             '1.14.16.5': ['#4# pH 7.5: about 40% of maximal activity, pH 9.5: about 75% of maximal activity <2>'],\n",
       "             '1.14.16.6': ['#1# half-maximal activity at pH 4.6 and pH 7.4 <3>'],\n",
       "             '1.14.17.1': ['#1# sharp decrease in activity between pH 4.5 and 5.0 and between pH 6.0 and 6.5 <8>'],\n",
       "             '1.14.17.3': ['#19# activity range, wild-type and mutant enzymes, pH profiles, overview <85>',\n",
       "              '#4# about 70% of activity maximum at pH 6.0 and 7.5 <11>',\n",
       "              '#4# recombinant enzyme <5>',\n",
       "              '#6# pH 7.5: about 50% of activity maximum, pH 10: about 60% of activity maximum, PAM-B <9>'],\n",
       "             '1.14.18.1': ['#101# less than 50% activity below pH 7.5 and above pH 10.5, L-dopa oxidation at 50°C <108>',\n",
       "              '#93# pH 5.0: about 30% of maximal activity, pH 8.0: about 50% of maximal activity, negligible activity below pH 4 and above pH 9, isoenzyme PPO II <24>; #11# high activity in the range pH 5-8 <15>; #36# pH 5.0: about 55% of maximal activity, pH 9.0: about 15% of maximal activity, catechol as substrate <58>; #54# active at this pH-range <187>; #165# determination of the pH optimum of TYR2 is carried out with 2 mM L-tyrosine as the substrate, at a pH range of 5-8, and measuring the enzyme activity spectrophotometrically <219>; #14# PPO is immobilized onto MAC400 and MAC200 at various enzyme activities 50000, 100000, 200000, 300000 U/l, at pH 5-8, and at temperature ranging from 10 to 40°C <164>',\n",
       "              '#110,111# pH 6.0: about 55% of maximal activity, pH 9.0: about 45% of maximal activity <119>',\n",
       "              '#123# measuring effect of pH on PPO activity and stability <179>',\n",
       "              '#156# enzyme activity decreases below pH 5.0 or above pH 8.0, however, the enzyme is still active at pH 3.0 and pH 9.0. PPO remains 77.96, 60.05 and 52.24% of its original activity between pH 5.0 and pH 7.0 using catechol, L-tyrosine and pyrogallol as substrates, respectively, and loses more than 57% of activity when using caffeic acid, gallic acid and chlorogenic acid as substrates at their optimum pH values <282>',\n",
       "              '#128# studies on effect of pH show retention of more than 90% activity over a pH range 5.0-6.5 <145>',\n",
       "              '#135# active at this pH-range <187>',\n",
       "              '#29# pH 4.5: about 45% of maximal activity, pH 7.5: about 60% of maximal activity, free enzyme <19>',\n",
       "              '#174# PPO activity increases with pH from pH 3.0 to pH 7.0 and decreases slightly as pH increases from pH 7.0 to pH 8.0, but this general pattern is affected by the substrate, pH optima for PPO activity depend on the phenolic substrates, detailed overview <350>',\n",
       "              '#178# activity range, profile overview <325>',\n",
       "              '#18# for diphenolase activity of tyrosinase <257>',\n",
       "              '#77# activity range <136>; #18# for monophenolase activity of tyrosinase <257>; #143# TyrA retains more than 90% of its activity after incubation at 4°C for 24 h in a pH range of 4.0-10.0 <223>; #25# activity range, profile overview <317>',\n",
       "              '#180# the proteolytically activated mushroom tyrosinase shows over 50% of its maximal activity in the range of pH 5-10 and accepts a wide range of substrates including mono- and diphenols, flavonols and chalcones <354>',\n",
       "              '#25# assay at pH 4.5.and 7.0 <161>',\n",
       "              '#25# the activity of mushrom tyrosinase is studied in this pH range <194>',\n",
       "              '#64# active at this pH-range <187>; #25# inhibition of the diphenolase activity of mushroom tyrosinase over the pH range of 5.5-8.0 is studied. The observed inhibitory activity at pH 6.8. and 5.8 is approximately 10times and 100times greater than at pH 7.8 <194>',\n",
       "              '#27# 74% and 87% of maximal activity <131>',\n",
       "              '#57# enzyme is active in the range pH 6-7.8 <6>; #27# pH 6: about 25% of maximal activity, pH 7.8: about 35% of maximal activity <66>',\n",
       "              '#36# pH 5.0: about 60% of maximal activity, pH 9.0: about 40% of maximal activity, dopa as substrate <58>; #29# pH 5.0: about 30% of maximal activity, pH 9.0: about 50% of maximal activity, enzyme in reversed micelles <19>; #76# some ca. 80% of catalase activity is retained between pH 6 and 8, while 80% of phenol oxidase activity is retained between pH 7 and 8 with only ca. 25% activity at pH 6 and ca. 50% at pH 8, retains ca. 80% of the initial activity across the pH range after 24 h incubation <146>',\n",
       "              '#34# the optimum pH for broccoli PPO is 5.7 with catechol and 4-methylcatechol as substrates <170>',\n",
       "              '#36,71# pH 4.0: about 65% of maximal activity, pH 8.0: about 75% of maximal activity, catecholase activity <26,107>',\n",
       "              '#45# no activity below <50>',\n",
       "              '#45# pH 4.3: 62% of maximal activity, pH 5.3: 79% of maximal activity <31>',\n",
       "              '#5# pH 6.5: about 60% of maximal activity, pH 10.0: about 10% of maximal activity <243>',\n",
       "              '#71# cresolase activity rises in this range without reaching a defined maximum <27>; #63# enzyme activity toward 5 mM 4-tert-butylcatechol is determined in a range of pH from 3.5 to 7.5 <176>',\n",
       "              '#51# highest activity and stability within a neutral and alkaline pH range <128>',\n",
       "              '#51# active at this pH-range <187>',\n",
       "              '#62# over 50% activity within this range, reduced activity above and below <351>',\n",
       "              '#7# less than 50% activity below pH 6.5 and above pH 8.5, L-dopa oxidation, recombinant enzyme <103>; #183# at pH below 6.0, PPO is inactivated. The PPO activity rapidly increases with pH in the range of 6.5-8.5, above pH 8.5, the activity decreases to about 80% of maximal activity at pH 9.0-10.0 <335>',\n",
       "              '#70# pH 3.7: 50% of maximal activity, pH 7.7: 50% of maximal activity <39>',\n",
       "              '#71# over 50% of maximal activity within this range <319>',\n",
       "              '#71# about 60% of maximal activity at pH 3.0 and pH 6.0. Marked decrease of activity above pH 6.0 <27>; #97# about 40% of maximal activity at pH 3.4 and pH 6.0, catechol oxidase in Na-acetate buffer <69>',\n",
       "              '#184# activity range, profile overview <336>',\n",
       "              '#80# pH 8.0: about 30% of maximal activity, pH 9.0: maximal activity, pH 9.5: no activity <121>',\n",
       "              '#83# activity range, drastic decrease in activity at pH 7.5 and above <330>',\n",
       "              '#93# pH 4.5: about 70% of maximal activity, pH 8.0: about 50% of maximal activity, negligible activity below pH 4 and above pH 9, isoenzyme PPO I <24>',\n",
       "              '#99# pH 5.7: 50% of maximal activity, pH 6.6-7.8: optimum <74>'],\n",
       "             '1.14.18.2': ['#10# about 60% of maximal activity at pH 5.6 and pH 6.8 <14>'],\n",
       "             '1.14.19.1': ['#6# pH 5.5: about 80% of maximum activity, pH 9.0: about 45% of maximum activity <18>'],\n",
       "             '1.14.19.17': ['#1,10,11,12,14# activity range <15>'],\n",
       "             '1.14.19.2': ['#3# pH 5.5: about 50% activity, pH 7.7: about 50% activity <10>'],\n",
       "             '1.14.19.23': ['#3# pH 6.0: about 65% of maximal activity, pH 9.0: about 65% of maximal activity <1>'],\n",
       "             '1.14.19.54': ['#1# 50% of maximum activity <3>',\n",
       "              '#1# 50% of maximum activity <3>'],\n",
       "             '1.14.19.59': ['#1# pH 5.0: about 75% of maximal activity, pH 7.0: about 70% of maximal activity <1>'],\n",
       "             '1.14.19.64': ['#1# about 40% of maximal activity at pH 7 and pH 9 <1>',\n",
       "              '#2# half optimal activity at pH 6.5 and pH 9 using S-cheilanthifoline as a substrate <3>'],\n",
       "             '1.14.19.65': ['#1# about 50% of maximal activity at pH 7 and pH 9 <1>',\n",
       "              '#2# half optimal activity at pH 6.5 and 8.5 <3>; #2# half optimal activity at pH 6.5 and pH 9 using S-cheilanthifoline as a substrate <3>'],\n",
       "             '1.14.19.67': ['#7# about 50% of maximal activity at pH 6.5 and pH 8.5 <1>'],\n",
       "             '1.14.19.68': ['#15# half optimal activity at pH 6.5 and pH 9 <5>'],\n",
       "             '1.14.20.13': ['#4# 20% of maximum activity <15>'],\n",
       "             '1.14.20.7': ['#7# over 50% of maximal activity within this range <7>'],\n",
       "             '1.14.99.1': ['#7# about 70% of maximum activity at pH 6.0 and 9.0 of prostaglandin E2 formation, about 50% of maximum activity at pH 6.0 and 9.0 of prostaglandin F2alpha formation <21>',\n",
       "              '#7# about 60% of maximum activity at pH 7.2 and 9.0 of prostaglandin D2 formation <21>'],\n",
       "             '1.14.99.12': ['#3# about 50% of activity maximum at pH 5.7 and 7.8 <1>'],\n",
       "             '1.14.99.15': ['#2# about half-maximal activity at pH 7.3 and 9.0, reductase <7>'],\n",
       "             '1.14.99.24': ['#3# entrapped cells <2>'],\n",
       "             '1.14.99.26': ['#1# half-maximal activity at pH 6.3 and pH 8.5 <1>'],\n",
       "             '1.14.99.53': ['#15# ammonium acetate buffer, 34% of maximum activity <10>'],\n",
       "             '1.14.99.54': ['#15,16# no activity below pH 4.5 <3>',\n",
       "              '#36# hardly any catalytic performance at a low pH <24>'],\n",
       "             '1.14.99.58': ['#3,4,5,6,7,9# high activity of PigA within this range <8>'],\n",
       "             '1.15.1.1': ['#104# Mn-SOD <51>; #250# 70% of maximal activity within this range <271>',\n",
       "              '#108# activity range, over 50% of the maximal activity at pH 6-10, profile overview <310>',\n",
       "              '#125# native and recombinant Cu,Zn-SOD, activity is pH-independent <84>',\n",
       "              '#130# activity range, the enzyme shows 30% of maximal activity at pH 10.0 and is inactive at pH 11.0 <253>',\n",
       "              '#19# activity decreases as pH increases <39>',\n",
       "              '#195# high enzyme activity <229>',\n",
       "              '#196# 50% of maximal activity at pH 8.0 and pH 10.5 <230>',\n",
       "              '#197# activity range <234>',\n",
       "              '#201# optimum pH-range <239>',\n",
       "              '#215# activity range, profile overview <299>',\n",
       "              '#232# 70% of maximal activity within this range <286>',\n",
       "              '#255# activity range, profile overview <281>; #235# activity does not change in the pH-range 4-10 <266>',\n",
       "              '#237# maximal activity at pH 7.6, 50% of miximal activity at pH 8.5, inactivation above pH 10.0 <274>',\n",
       "              '#24# 90% of maximal activity at pH 5.6-pH 7.5, 15% at pH 4.0, 60% at pH 12.0, 25°C, pH profile, overview <219>; #173# inactivation at pH 2.0, shrap drop of activity below pH 4.0, high activity at pH 4.0-pH 5.0, about 70% of maximal activity at pH 6.0-12.0 <209>',\n",
       "              '#253# over 70% of maximal activity within this range <282>',\n",
       "              '#26,102# Mn-SOD <25,81>',\n",
       "              '#265# high activity range <312>',\n",
       "              '#266# activity range, 75% of maximal activity at pH 1.0, 45% at pH 11.0, profile overview <315>',\n",
       "              '#270# active at <303>',\n",
       "              '#274# activity range <295>',\n",
       "              '#275# activity range, rapid loss of activity above pH 10.0, inactive at pH 5.0 and pH 11.0, profile overview <304>',\n",
       "              '#280,282# activity range <322,323>',\n",
       "              '#75# Fe-SOD <99>',\n",
       "              '#212# activity profile, overview <247>; #252# activity range, profile overview <276>; #263# about 70% of maximal activity within this range, 10% at pH 4.0, 40% at pH 11.0 <302>',\n",
       "              '#9# pH-profile, overview. Outside the range of 5.0-7.0, activity is rapidly lost <251>'],\n",
       "             '1.16.1.1': ['#1# pH 6.5: about 60% of maximal activity, pH 9.0: about 50% of maximal activity <15>',\n",
       "              '#97# activity range, profile overview <58>; #11# high activity within that range <48>',\n",
       "              '#63# activity range, profile overview, inactive at pH 5.0 <53>'],\n",
       "             '1.16.1.10': ['#1# activity range measured, inactivation at pH above 8.0, the enzyme is active at pH below pH 5.5 <5>',\n",
       "              '#3# pH 6.0: about 40% of maximal activity, pH 8.0: about 60% of maximal activity <2>'],\n",
       "             '1.16.1.2': ['#2# pH 7.0: about 40% of activity maximum, pH 10.0: about 35% of activity maximum <2>'],\n",
       "             '1.16.1.7': ['#6# pH 5.5: about 35% of maximal activity, pH 8.0: about 50% of maximal activity <1>'],\n",
       "             '1.16.3.1': ['#1# highest activity <84>'],\n",
       "             '1.16.3.4': ['#5# no activity with substrate [Cu(I)(bicinchoninate)2]3- <21>'],\n",
       "             '1.17.1.1': ['#2# pH 5.5: about 60% of activity maximum, pH 8.5: about 40% of activity maximum <2>'],\n",
       "             '1.17.1.3': ['#1,8# 80% activity at pH 6 and pH 7.5 <4>',\n",
       "              '#10# 95% activity at pH 6.2 and pH 7.8 <3>; #13# 95% of maximal activity at pH 76.2 and pH 7.8 <3>',\n",
       "              '#14,17,18,20# both isoforms LAR1 and LAR2 <10>',\n",
       "              '#7# no significant changes between pH 6.4 and pH 8.0 <1>'],\n",
       "             '1.17.1.4': ['#73# the enzyme is highly active at pH 5.0-8.0, shows low activity at pH 9.0, and is inactive above <127>',\n",
       "              '#77# activity range, bell-shaped pH-activity relationships for both the recombinant split variants <128>',\n",
       "              '#9# trends to increasing activities at higher values <5>'],\n",
       "             '1.17.1.5': ['#2# broad pH-range with maximum at pH 8.3 <3>'],\n",
       "             '1.17.1.8': ['#11# at pH 7.0 and 9.0, approyimately 75% and 90% of the enzymatic activity is retained. Below pH 7.0 and above pH 9.0 a significant drop in activity is observed. At pH 11.0 an almost complete loss of activity is observed <31>'],\n",
       "             '1.17.1.9': ['#10# pH 5.5: about 25% of maximal activity, pH 10.0: about 40% of maximal activity <29>',\n",
       "              '#10# 50% of maximal activity at pH 5.3 and at pH 10.1 <28>',\n",
       "              '#12# mutant enzyme D195Q/Y196H <70>',\n",
       "              '#27# pH 5.0: about 50% of maximal activity, pH 10.0: about 65% of maximal activity <35>; #16# completely inactive below pH 5.5 and above pH 10.0 <41>; #14# pH 5.0: about 80% of maximal activity, pH 10.0: about 65% of maximal activity <45>',\n",
       "              '#28# more than 50% activity between pH 5.0 and 11.0 <125>',\n",
       "              '#67# the specific activity at pH ranging from 5.5 to 9.5 is always more than 77% of the maximum activity <95>; #35# more than 50% activity between pH 5.0 and 11.0 <125>',\n",
       "              '#38# no activity at pH 5.0 or lower, pH 6.0: optimum <31>',\n",
       "              '#51# pH 5.0: about 60% of maximal activity, pH 5.3: about 80% of maximal activity, pH 9.6: about 80% of maximal activity, pH 10.5: about 65% of maximal activity <46>',\n",
       "              '#57# more than 50% activity between pH 6.0 and 10.0 <113>',\n",
       "              '#66# the enzyme displays a relatively wide pH spectrum and shows more than 98% of relative activity from pH 6.0 to 7.0 and still has 81 and 75% of relative activity under pH 7.5 and 8.0, respectively <93>',\n",
       "              '#71# the enzyme is highly active between pH 4.0 and 11.5. The relative activity decreases quite slowly above pH 5.0, and the enzyme shows considerable activity at pH 9.5\\x9611.5 <120>',\n",
       "              '#8# pH 7.2: about 45% of maximal activity, pH 8.4: about 65% of maximal activity <16>'],\n",
       "             '1.17.2.2': ['#1# enzyme is assayed over a range of pH from 7 to 10 in sodium/potassium phosphate, Tris/HCI and glycine/NaOH buffers <1>'],\n",
       "             '1.17.2.3': ['#2# pH 4.5: about 40% of activity maximum, pH 10: about 10% of activity maximum <2>'],\n",
       "             '1.17.3.1': ['#1# half-maximal activity at pH 5.2 and 8.1 <1>'],\n",
       "             '1.17.3.2': ['#12# pH 5.0: about 10% of activity maximum, pH 9.0: about 25% of activity maximum <14>',\n",
       "              '#33# potassium phosphate, Tris-HCl, Tris-acetate and glycine-NaOH buffer <148>',\n",
       "              '#7# under aerobic conditions, NO generation increases more than three times as the pH value decreases from pH 7.4 to 6.0 <50>',\n",
       "              '#7# pH 6.5: about 40% of maximal activity, pH 9.5: about 65% of maximal activity, conversion of dibromoacetonitrile to CN- <44>',\n",
       "              '#16# pH 6.0: about 25% of activity maximum, pH 9.0: about 20% of activity maximum <11>',\n",
       "              '#9# at pH 7.7 and 9.1: about 30% of activity maximum <24>'],\n",
       "             '1.17.4.1': ['#23# 60% activity at pH 7.2 <65>',\n",
       "              '#7# stable intermediate formation, change in the rate-limiting step at elevated pH, profiles, overview <141>'],\n",
       "             '1.17.4.4': ['#4# about half-maximal activity at pH 8.5 and 9.8 <13>'],\n",
       "             '1.17.5.1': ['#1# with 50 mM potassium phosphate buffer: 60% of optimum activity at pH 6.5, 87% of optimum activity at pH 7.5, with 50 mM Tris-HCl buffer: 60% of optimum activity at pH 7.5 <2>'],\n",
       "             '1.17.5.2': ['#1# 52% of maximum activity <1>'],\n",
       "             '1.17.7.3': ['#2# pH 8.0: about 65% of maximal activity, pH 9.5: about 75% of maximal activity <3>'],\n",
       "             '1.17.98.3': ['#1# pH 7.3: about 40% of maximal activity, pH 8.3: about 70% of maximal activity <4>'],\n",
       "             '1.17.98.4': ['#1# pH 6.5: about 40% of maximal activity, pH 9.7: about 40% of maximal activity <8,14>'],\n",
       "             '1.17.99.11': ['#1# pH 5.5: about 30% of maximal activity, pH 8-0: about 50% of maximal activity <1>'],\n",
       "             '1.17.99.2': ['#2# about maximum activity at both pH <1>'],\n",
       "             '1.17.99.3': ['#2# pH 6.5: about 30% of maximal activity, pH 8.5: about 60% of maximal activity <7>'],\n",
       "             '1.17.99.4': ['#3# about half-maximal activity at pH 8.0 and 10 with thymine as substrate <2>'],\n",
       "             '1.17.99.8': ['#1# potassium phosphate buffer <2>'],\n",
       "             '1.18.1.2': ['#12# assay range <146>',\n",
       "              '#31# ferredoxin-dependent activity at pH 6.5 is 70 micromol/min/mg, at pH 7 is 95 micromol/min/mg and at pH 10 is 30 micromol/min/mg <132>',\n",
       "              '#31# quinone-dependent activity at pH 7.5 is 60 micromol/min/mg, at pH 9.7 is 280 micromol/min/mg and at pH 10 is 260 micromol/min/mg <132>; #65# quinone-dependent activity at pH 7.5 is 60 micromol/min/mg, at pH 9.7 is 400 micromol/min/mg and at pH 10 is 325 micromol/min/mg <132>'],\n",
       "             '1.18.6.1': ['#37# pH 6.5: no activity below, pH 8.3: about 70% of activity maximum <30>',\n",
       "              '#48# narrow range <61>',\n",
       "              '#6# below and above no remaining activity <36>'],\n",
       "             '1.2.1.10': ['#1# 67% of maximal activity at pH 6.2, 79% of maximal activity at pH 7.4 <10>',\n",
       "              '#26# narrow activity range, inactivation at pH 5.5, profile overview <33>',\n",
       "              '#3# activity increases gradually over pH range from 6.5 to 8.5 <13>',\n",
       "              '#6# 70% of maximal activity at pH 8.5, 80% of maximal activity at pH 10 <1>',\n",
       "              '#6# less than 50% of maximal activity above and below <14>',\n",
       "              '#7# 30% of maximal activity at pH 7.0, 100% at pH 9.5 <6>',\n",
       "              '#8# activity of NADH oxidation increases between 6.5-9.5 <12>',\n",
       "              '#8# NAD+ reduction <12>',\n",
       "              '#9# complete inactivation <29>'],\n",
       "             '1.2.1.105': ['#10# pH 7.0: about 70% of maximal activity, pH 8.5: about 70% of maximal activity, 2-oxoglutarate dehydrogenase complex <12>',\n",
       "              '#24# about 50% of maximal activity at pH 6.6 and at pH 8.4, 2-oxoglutarate dehydrogenase complex <20>',\n",
       "              '#6# pH 6.5: about 40% of maximal activity, pH 8.0: about 50% of maximal activity, 2-oxoglutarate dehydrogenase complex <28>',\n",
       "              '#19# pH 6.5: about 60% of maximal activity, pH 8.5: about 30% of maximal activity, 2-oxoglutarate dehydrogenase complex <14>; #9# pH 6.5: about 25% of maximal activity, pH 8.5: about 30% of maximal activity, 2-oxoglutarate dehydrogenase complex <5>'],\n",
       "             '1.2.1.107': ['#11# high activity between pH 7.5 and 8.5 <1>',\n",
       "              '#4# half-maximum activity at pH 7.4 and 9.6 <10>'],\n",
       "             '1.2.1.11': ['#6# at pH 8.4 and pH 9.5: 50% of maximal activity, L-aspartate 4-semialdehyde + phosphate <10>',\n",
       "              '#7# pH 6.5: about 45% of activity maximum, mutant, about 8% of activity maximum, wild type, pH 9.0: about 55% of activity maximum, mutant, about 30% of activity maximum, wild type <8>'],\n",
       "             '1.2.1.12': ['#103# over 40% of maximal activity within this range, profile overview <266>',\n",
       "              '#12# pH 7.0: about 55% of maximal activity, pH 8.8: about 95% of maximal activity <69>',\n",
       "              '#13# 85% of maximal activity at pH 8.3 and pH 9.3, 50 mM sodium diphosphate buffer <78>',\n",
       "              '#13# about 85% of maximal activity at pH 7.5 and pH 8.5, 50 mM Tris-chloride buffer. About 75% of maximal activity at pH 7.5 and pH 8.5, 50 mM triethanolamine-chloride buffer <78>',\n",
       "              '#130# sharp decrease in activity above <217>',\n",
       "              '#134# lack of any pH dependence of the kinetic parameters in the pH range tested, recombinant His6-tagged wild-type enzyme <230>',\n",
       "              '#14# above, sharp decline in activity for enzyme of patients with chronic myeloid leukemia, no similar sharp decline for enzyme of healthy subjects <173>',\n",
       "              '#16# about 50% of maximal activity at pH 7.4 and at ph 9.6 <39,67>',\n",
       "              '#32# pH 5.5: about 50% of maximal activity, pH 9.0: about 55% of maximal activity, cytosolic enzyme <7>',\n",
       "              '#32# about 40% of maximal activity at pH 7.0 and at pH 9.0, glyoxysomal enzyme <7>',\n",
       "              '#37# pH 7.0: about 70% of maximal activity, pH 9.5: about 55% of maximal activity <75>',\n",
       "              '#57# pH 6.0: about 50% of maximal activity, pH 10.0: about 60% of maximal activity, oxidation of D-glyceraldehyde 3-phosphate, at 40°C <12>; #77# activity range, profile overview <258>'],\n",
       "             '1.2.1.13': ['#14# pH 7.0: about 40% of maximal activity, pH 9.1: about 90% of maximal activity <49>'],\n",
       "             '1.2.1.16': ['#10# KM value for NAD+ increases from 0.3 to 1 mM as the pH changes from pH 5 to 10. The pH profile data are obtained at a NAD+ concentration of 5 mM, essentially saturating across the entire pH range. Similarly, data for the NADP+ pH profile are obtained at 10 mM NADP+ <20>',\n",
       "              '#14# the enzyme cannot be assayed below pH 5.5, due to precipitation and loss of activity <27>',\n",
       "              '#17# specific activity increases steadily to a broad optimum <23>',\n",
       "              '#7# with NAD+, at pH 8.7: 45% of activity maximum, at pH 10.6: about 50% of activity maximum <9>',\n",
       "              '#7# with NADP+, at pH 8.7: about 30% of activity maximum, at pH 10.9: about 65% of activity maximum <9>',\n",
       "              '#8# at pH 7.0 and 9.3: about 30% of activity maximum <11>'],\n",
       "             '1.2.1.18': ['#2# about 65% of maximal activity at pH 7.0 and at pH 9.0 <6>',\n",
       "              '#2# pH 6.0: about 35% of maximal activity, pH 7.7: about 65% of maximal activity, activity with NADP+ <6>'],\n",
       "             '1.2.1.19': ['#1# at pH 4.0 and pH 7.0: about 60% of activity maximum <6>',\n",
       "              '#12# 4-aminobutyraldehyde, at pH 6.5: about 30% of activity maximum, at pH 9.0: about 70% of activity maximum <11>',\n",
       "              '#12# 4-guanidinobutyraldehyde, at pH 7.0: about 35% of activity maximum, at pH 10.0: about 50% of activity maximum <11>',\n",
       "              '#22,23# the enzyme retains approximately 20% of its maximal substrate-limited, 3-aminopropanal-dependent activity at pH 7.5 and is inactive at pH 6.5 <33>',\n",
       "              '#3# at pH 6.9: about 30% of activity maximum, at pH 8.5: about 65% of activity maximum <7>',\n",
       "              '#30# AMADH activity increases with a pH ranging from pH 3.0 to pH 8.0, but it decreases drastically when pH continuously rises from pH 8.0 to pH 10.0. At pH 9.0, the activity decreases by 75% compared to pH 8.0 <30>',\n",
       "              '#31# activity range, profile overview <31>',\n",
       "              '#32# activity range, profile overview <31>',\n",
       "              '#8# highly active between <5>'],\n",
       "             '1.2.1.20': ['#2# pH 7.2: about 50% of maximal activity, pH 8.7: about 40% of maximal activity <4>'],\n",
       "             '1.2.1.22': ['#2# pH 9.2: about 15% of maximum activity, pH 11.2: about 99% of maximum activity <7>',\n",
       "              '#9# 17% activity at pH 7.0, almost 100% activity at pH 10.0 <10>'],\n",
       "             '1.2.1.23': ['#4# not active below <10>',\n",
       "              '#4# not active below <7>',\n",
       "              '#4# activity increases up to, Tris buffer <10>',\n",
       "              '#4# activity increases up to, carbonate buffer <10>'],\n",
       "             '1.2.1.24': ['#20# pH 8.0: about 40% of maximal activity, pH 10.0: about 55% of maximal activity <17>',\n",
       "              '#20# pH 8.7: about 50% of maximal activity, pH 10.5: about 75% of maximal activity <11>',\n",
       "              '#21# pH 8.5: about 45% of maximal activity, pH 10.0: about 70% of maximal activity <24>',\n",
       "              '#3# at least 80% of its optimal activity <37>',\n",
       "              '#32# at pH 8.5 MalE-DmSSADH is 10fold more active than at pH 6.0 <40>',\n",
       "              '#4# about 50% of maximal activity at pH 7.0 and at pH 10.0 <19>; #4# pH 7.0: about 40% of maximal activity, pH 10.0: about 35% of maximal activity <12>'],\n",
       "             '1.2.1.25': ['#1# pH 6.0: about 25% of maximal activity, pH 7.5: about 45% of maximal activity <1>'],\n",
       "             '1.2.1.26': ['#14# more than 50% of maximal activity <7>',\n",
       "              '#5# not active below, 2,5-dioxopentanoate + NADP+ <3>'],\n",
       "             '1.2.1.27': ['#3# methylmalonate semialdehyde oxidation, less than 90% of maximal activity above and below <1>',\n",
       "              '#3# propanal oxidation, less than 50% of maximal activity above and below <1>',\n",
       "              '#4# methylmalonate semialdehyde oxidation, more than 90% of maximal activity <4>',\n",
       "              '#4# propanal oxidation, less than 50% of maximal activity below, less than 60% of maximal activity above <4>'],\n",
       "             '1.2.1.29': ['#2# enzymatic reactions carried out at pH 4.0 show only 5\\x9610% of the maximum activities observed at pH 10.0 <2>'],\n",
       "             '1.2.1.3': ['#33# pH 7.5: about 35% of maximal activity, pH 10.0: about 60% of maximal activity <9>; #29# pH 7.5: about 40% of maximal activity, pH 10.0: about 60% of maximal activity, reaction with acetaldehyde <52>',\n",
       "              '#36# about 60% activity at pH 6.0, about 70% activity at pH 7.0, 100% activity pH 8.0, about 85% activity at pH 9.0 <136>',\n",
       "              '#5# pH 8.0: about 35% of maximal activity, above pH 9.0: optimum, isoenzyme 2 <42>',\n",
       "              '#5# pH 8.5: about 40% of the activity of isoenzyme ALDH1 and 2, pH 10.5: about 50% of the activity of isoenzyme ALDH2, about 65% of the activity of isoenzyme ALDH1 <32>',\n",
       "              '#5# pH 7.0: about 66% of maximal activity, pH 9.0: about 85% of maximal activity, isoenzyme 1 <42>',\n",
       "              '#6# pH 9: about 35% of maximal activity, pH 10: optimum <2>',\n",
       "              '#6# 50% of maximal activity at pH 7.3 and at pH 9.5 <29>'],\n",
       "             '1.2.1.30': ['#2# not active below, benzoate + ATP <2>'],\n",
       "             '1.2.1.31': ['#4# pH 5: about 30% of activity maximum, pH 9: about 40% of activity maximum <9>',\n",
       "              '#8# pH 6: about 10% of activity maximum, pH 9: about 70% of activity maximum <11>'],\n",
       "             '1.2.1.32': ['#4# benzaldehyde oxidation, more than 50% of maximal activity between pH 8 and 11 <6>',\n",
       "              '#5# more than 50% of maximal activity between pH 7 and 9 <1>'],\n",
       "             '1.2.1.33': ['#2# activity increases from pH 7.0 to 9.5 <1>'],\n",
       "             '1.2.1.36': ['#4# exhibits a broad pH optimum <8>',\n",
       "              '#4# on chromatofocusing column, major activity peak is eluted at pH range 8.4-8.6 <9>'],\n",
       "             '1.2.1.38': ['#3# activity drops sharply below pH 9.0 and above 9.5 <4>'],\n",
       "             '1.2.1.4': ['#10# pH 8.5: about 60% of maximal activity, pH 11.5: about 80% of maximal activity <4>',\n",
       "              '#13# pH 5.0: 10% of maximal activity, pH 9.2: 15% of maximal activity <3>',\n",
       "              '#6# about 50% of maximal activity at pH 6.8 and 8.0 <11>'],\n",
       "             '1.2.1.41': ['#4# pH 6.0 and pH 8.5: about 20% of activity maximum <7>',\n",
       "              '#4# about 30% of activity maximum at pH 7.3, about 20% of activity maximum at pH 7.8 <11>'],\n",
       "             '1.2.1.42': ['#2# pH 6.5 and pH 9.0: about 20% of maximum activity <1>; #14# optimal activity is at pH 7.0, with activity falling dramatically below pH 6.5 and above 9.0 <10>'],\n",
       "             '1.2.1.44': ['#25# when the pH is above 6.5, the activity becomes negligible, feruloyl-CoA as substrate <20>',\n",
       "              '#3# pH 5.4: about 10% of activity maximum, pH 8.0: about 15% of activity maximum <2>',\n",
       "              '#47# activity range, profile overview <52>',\n",
       "              '#48# activity range, profile overview <52>',\n",
       "              '#49# activity range, inactive above pH 7.0, profile overview <43>'],\n",
       "             '1.2.1.46': ['#8# at pH 6.6 and 8.3 about 50% of activity maximum <4>',\n",
       "              '#9# optimal pH-value for the developed FdDH-based biosensor <18>'],\n",
       "             '1.2.1.47': ['#4# pH 6.2: about 30% of activity maximum, pH 10.5: about 15% of activity maximum <1>'],\n",
       "             '1.2.1.48': ['#11# pH 6.5: about 25% of activity maximum, pH 10: about 75% of activity maximum <3>'],\n",
       "             '1.2.1.49': ['#1# very low activity at physiological pH <10>',\n",
       "              '#2# activity falls abruptly above <9>',\n",
       "              '#3# not active below <6>'],\n",
       "             '1.2.1.5': ['#3# pH 6.2: about 80% of maximal activity, pH 6.7: about 80% of maximal activity <7>',\n",
       "              '#34# pH 6.0: about 60% of maximal activity, pH 9.0: about 90% of maximal activity <35>'],\n",
       "             '1.2.1.50': ['#8# activity falling dramatically below pH 6.5 and above 9.0 <16>'],\n",
       "             '1.2.1.54': ['#1# pH 7.5: about 20% of activity maximum, pH 10: about 70% of activity maximum <4>',\n",
       "              '#3# pH 7.0: about 20% of activity maximum, pH 10.0: about 80% of activity maximum <2>'],\n",
       "             '1.2.1.59': ['#10# considerable activity in the range of pH 7 to pH 10 <2>',\n",
       "              '#27# 30% of maximal activity at pH 5.0, 40% at pH 9.0 <17>'],\n",
       "             '1.2.1.65': ['#6# high activity within this range <14>'],\n",
       "             '1.2.1.67': ['#17# activity range, profile overview <17>'],\n",
       "             '1.2.1.7': ['#7# the activities measured at pH 6.85 and 10.7 are approximately 30% of the activity measured at pH 9.3 <12>'],\n",
       "             '1.2.1.73': ['#2# 50% of maximal activity at pH 7.2, 70% at pH 8.0, and 80% at pH 9.5-10.5 <3>'],\n",
       "             '1.2.1.75': ['#2# half-maximal activity <3>',\n",
       "              '#2# half-maximal activity <3>',\n",
       "              '#3# half-maximal activity <3>'],\n",
       "             '1.2.1.77': ['#1# 50% of maximal activity at pH 6.2 and pH 8.8 <2>'],\n",
       "             '1.2.1.78': ['#1# pH 7.0: about 60% of maximal activity, pH 8.6: about 70% of maximal activity <2>'],\n",
       "             '1.2.1.79': ['#10# GabD1 could not be assayed below pH 5.5, due to precipitation and loss of activity <11>',\n",
       "              '#18# 50% of maximal activity within this range, pH-profile overview <23>',\n",
       "              '#7# 75% of maximum activity <4>'],\n",
       "             '1.2.1.8': ['#10# pH 7.0: about 40% of maximal activity, pH 9.0: about 50% of maximal activity <13>',\n",
       "              '#14# pH 7.0: about 50% of maximal activity, pH 9.5: about 75% of maximal activity <32>; #14# pH 7.0: about 25% of maximal activity, pH 9.5: about 50% of maximal activity <15>',\n",
       "              '#19# pH 7.0: about 35% of maximal activity, pH 8.0: about 70% of maximal activity <36>',\n",
       "              '#65# assay at <55>'],\n",
       "             '1.2.1.80': ['#2# truncated mAtFAR6 protein <2>'],\n",
       "             '1.2.1.84': ['#15# inactive at pH 4.0, gradual increase up to maximal activity at pH 6.5, 42% of maximal activity at pH 7.5, reduction of activity above <14>'],\n",
       "             '1.2.1.86': ['#3# below pH 8.0, both isoenzymes show less than 50% of relative activity while complete inactivation is observed at pH 5.0-6.0 and pH 10.5-11.0 <4>'],\n",
       "             '1.2.1.88': ['#13# cofactor NAD+ <3>',\n",
       "              '#13# cofactor NADP+ <3>',\n",
       "              '#30# both isoforms <22>'],\n",
       "             '1.2.1.89': ['#2# pH 6.0: about 65% of maximal activity, pH 10.0: about 70% of maximal activity <1>',\n",
       "              '#3# pH 4.3: about 50% of maximal activity, pH 8.1: about 50% of maximal activity <2>'],\n",
       "             '1.2.1.9': ['#11# pH 7.5: about 35% of maximal activity, pH 12.0: about 60% of maximal activity <17>',\n",
       "              '#22# at pH 6.8: 19% of maximal activity, at pH 8.5: 92% of maximal activity <7>'],\n",
       "             '1.2.1.92': ['#3# the enzyme shows 44% and 42% activity at pH 5.0 and 8.0, respectively <3>',\n",
       "              '#4# the enzyme shows over 90% of its maximum within a small pH range from 8.0 to 11.0 and more than 60% activity between pH 6.0 and 12.0. At pH values lower than 5.0, activity of the enzyme is drastically reduced <2>',\n",
       "              '#5# more than 60% activity at pH 6.5, more than 90% activity from pH 7.0-8.0, 30% activity at pH 9.0 <4>'],\n",
       "             '1.2.1.99': ['#2# pH 7.0: about 25% of maximal activity, pH 9.0: about 20% of maximal activity <2>; #1# pH 7.0: about 50% of maximal activity, pH 9.0: about 80% of maximal activity <4>'],\n",
       "             '1.2.3.1': ['#84# more than 70% of maximum activity <131>',\n",
       "              '#13# broad <110>; #49# aldehyde oxidase active in either the organic solvent-free buffer or all nine water-miscible organic solvents, lowest enzyme activity is obtained at pH 6.0 with acetonitrile <119>',\n",
       "              '#14# approx. 10% of maximal activity at pH 3.0, approx. 20% of maximal activity at pH 10 <65>; #28# approx. 40% of maximal activity at pH 10 <65>',\n",
       "              '#15# more than 90% activity <60>',\n",
       "              '#38# approx. 5% of maximal activity at pH 4.0, approx. 10% of maximal activity at pH 10 <65>',\n",
       "              '#6# maximum aldehyde oxidase-catalyzed nitric oxide generation occurs at pH 6.0. When the pH is decreased to 5.0 or increased above 8.0, a decrease in the rate of nitric oxide generation is observed <117>'],\n",
       "             '1.2.3.15': ['#2# more than 30% activity at pH 4.0 and 6.0, no activity at pH 3.0 and above pH 7.0 <4>'],\n",
       "             '1.2.3.3': ['#3# 50% of the activity at pH 6.0 <10>'],\n",
       "             '1.2.3.4': ['#10# approx. 50% of maximal activity at pH 5.0 and pH 7.0, respectively, polyvinyl alcohol membrane immobilized oxalate oxidase <20>',\n",
       "              '#14# 50% activity at pH 4.0 and 7.0 <13,14>',\n",
       "              '#18# 30% activity at pH 2.5 and 4.5 <3>; #19# 60% activity at pH 2.5 and 4.5 <5>',\n",
       "              '#21# 100% activity at pH 2.0, 50% activity at pH 3.0, 10% activity at pH 3.5, no activity at pH 4.0 <55>',\n",
       "              '#26# about 57% activity at pH 3.5, 100% activity at pH 4.0, about 82% activity at pH 4.5, about 60% activity at pH 5.0, about 27% activity at pH 5.5 <48>',\n",
       "              '#6# the catalytic efficiency (Vmax/Km) increases continuously to lower pH <28>',\n",
       "              '#7# 20% activity at pH 2.0 and 4.5 <2>'],\n",
       "             '1.2.3.5': ['#3# at pH 6.5 and 8.5 50% activity <2>'],\n",
       "             '1.2.3.7': ['#4# about 50% of maximal activity at pH 2.9 and at pH 5.5 <3>',\n",
       "              '#5# pH 5.8: about 40% of maximal activity, pH 6.8: about 25% of maximal activity, crude extract <1>'],\n",
       "             '1.2.4.2': ['#31# 34% residual activity <32>'],\n",
       "             '1.2.5.2': ['#10# pH 5.0: about 50% of maximal activity, pH 7.5: about 55% of maximal activity, substrate: acetaldehyde <12>',\n",
       "              '#6# pH 5-6: optimum, pH 9: about 60% of maximal activity <11>'],\n",
       "             '1.2.5.3': ['#2# only modest loss of activity at these extreme pHs indicates that ionization of functional groups in the active site is not as critical to catalysis for CODH <8>'],\n",
       "             '1.2.7.1': ['#21# pH 6.0: about 45% of maximal activity, pH 10.0: about 60% of maximal activity, enzyme Ape2126/2128 <9>',\n",
       "              '#3# pH 5.5: about 60% of maximal activity, pH 8.5: about 35% of maximal activity <14>',\n",
       "              '#37# 20% residual activity <42>',\n",
       "              '#7# the enzyme retains more than 70% of the initial activity at pH 6.0-8.0, even after 30 min incubation <33>'],\n",
       "             '1.2.7.11': ['#1# pH 8.0: 65% of maximal activity, pH 9.0: optimum, inactive below pH 7.0 <1>',\n",
       "              '#2# pH 6.0: about 50% of maximal activity, pH 9.0: about 50% of maximal activity <5>',\n",
       "              '#7# pH 7.5: about 70% of maximal activity, pH 10.0: about 70% of maximal activity, enzyme Ape1473/1472 <4>'],\n",
       "             '1.2.7.12': ['#4# no activity below pH 6.5, maximal activity above pH 7.5 <4>'],\n",
       "             '1.2.7.3': ['#3# 50% activity at pH 7.0 and pH 9.5, inactive at pH 5.0 and pH 10.0 <3>',\n",
       "              '#9# pH 7.0: about 45% of maximal activity, pH 9.7: about 65% of maximal activity, recombinant For <8>'],\n",
       "             '1.2.7.4': ['#1# very little activity outside this range <2>',\n",
       "              '#17# pH 6.5: about 25% of maximal activity, pH 8-9: maximal activity <16>',\n",
       "              '#18# pH 8: about 55% of maximal activity, pH 11.5: about 50% of maximal activity <3>',\n",
       "              '#6# about 60% of activity at pH 9.0 <40>'],\n",
       "             '1.2.7.5': ['#1# activity increases with increasing pH at increasing temperature, nearly no activity below pH 7.0 <2>',\n",
       "              '#1# formaldehyde ferredoxin oxidoreductase, activity increases linearly from pH 5.5 to pH 10.0 at 80°C <7>',\n",
       "              '#2# oxidation of acetaldehyde with benzyl viologen as electron acceptor <1>'],\n",
       "             '1.2.7.6': ['#8# 60% of maximum activity <5>'],\n",
       "             '1.2.98.1': ['#4# pH 5: about 10% of activity maximum, pH 9.5: about 50% of activity maximum, no activity at pH 10 <5>'],\n",
       "             '1.2.99.6': ['#2# pH 3.9: 80% of maximal activity, pH 4.1: optimum, 4-fluorobenzoate and reduced carbamoyl methyl viologen as substrates <5>',\n",
       "              '#8# about 90% of maximal activity, butanal and NH2-CO-methyl viologen as substrates <9>'],\n",
       "             '1.20.1.1': ['#7# low activity <1>',\n",
       "              '#8# assay at, activity at pH 3.0 is about 2.2fold higher than at pH 7.0 <13>'],\n",
       "             '1.20.4.1': ['#11# 90% of maximal activity at pH 5.5, 60% at pH 7.5 <7>',\n",
       "              '#11# 60% of maximum activity <7>',\n",
       "              '#11# 90% of maximum activity <7>'],\n",
       "             '1.20.4.3': ['#5# more than 60% activity between pH 7.0 and 8.0 <6>'],\n",
       "             '1.20.4.4': ['#4# ArsC arsenate reductase activity is decreased >80% at pH 6.3 but is stable in the pH range 7.0-9.5 <8>'],\n",
       "             '1.21.1.1': ['#9# pH dependence of IYD binding and turnover, tightest binding was measured near a neutral pH of 7.6\\x967.7 <21>'],\n",
       "             '1.21.3.1': ['#5# 30% activity remaining above pH 9.0, highly reduced activity below pH 6.0 <3>'],\n",
       "             '1.21.3.3': ['#10# high conversion rates with this range <21>',\n",
       "              '#4,5# completely inactive at acidic or neutral pH <9>'],\n",
       "             '1.21.3.8': ['#1# half-maximal activities at pH values around 4.0 and 6.0 <2>'],\n",
       "             '1.21.4.5': ['#1# pH 8.0: about 50% of maximal activity, pH 9.5: about 40% of maximal activity <1>'],\n",
       "             '1.3.1.1': ['#5# pH 6.5: 33.6% of activity maximum, pH 8.2: 59% of activity maximum <1>'],\n",
       "             '1.3.1.10': ['#4# inactive above pH 8, no measurement below pH 6 <5>'],\n",
       "             '1.3.1.108': ['#1# pH 6.0: about 30% of maximal activity, pH 9.0: about 30% of maximal activity <1>'],\n",
       "             '1.3.1.110': ['#1# activity is rather insensitive to a pH ranging from 5.5 to 9.0 <1>'],\n",
       "             '1.3.1.12': ['#33# more than 75% of maximum activity within <58>',\n",
       "              '#34# 50% of maximum activity <60>'],\n",
       "             '1.3.1.14': ['#5# pH 5.5: 13% of the activity at pH 6.5 <2>; #5# pH 7.8: 65% of the activity at pH 6.5 <2>',\n",
       "              '#5# below and above the turnover number decreases <8>',\n",
       "              '#7# 70% of maximal activity, dihydrorotate oxidase reaction <15>'],\n",
       "             '1.3.1.15': ['#1# at pH 6.7 and 8.8: about 25% of activity maximum <2>'],\n",
       "             '1.3.1.16': ['#4# at pH 3 about 35% of activity maximum, at pH 7 about 60% of activity maximum <1>'],\n",
       "             '1.3.1.17': ['#3# 50% of maximum activity at pH 4.0 and pH 7.0 <10>'],\n",
       "             '1.3.1.19': ['#1# activity range <7>'],\n",
       "             '1.3.1.2': ['#3# about 80% of activity maximum at pH 6.7 and 7.9, uracil reduction <5>'],\n",
       "             '1.3.1.22': ['#10# recombinant monkey 5alpha-reductase isozyme 1 <49>',\n",
       "              '#10# recombinant monkey 5alpha-reductase isozyme 2 <49>',\n",
       "              '#3# recombinant human 5alpha-reductase isozyme 2 <49>',\n",
       "              '#4# pH .5: about 30% of activity maximum, pH 9.0: about 20% of activity maximum <35>; #3# recombinant human 5alpha-reductase isozyme 1 <49>',\n",
       "              '#4# at pH 5 and 8.5: about 50% of activity maximum <40>',\n",
       "              '#4# pH 5.0: about 20% of activity maximum, pH 8.0: about 20% of activity maximum <19>',\n",
       "              '#8# half maximal activity at pH 6.5 and 8.6 <43>'],\n",
       "             '1.3.1.24': ['#8# biliverdin + NADPH, pH 6: about 50% of activity maximum, pH 9: 27% of activity maximum <7>'],\n",
       "             '1.3.1.3': ['#4# about 50% of activity maximum at pH 5.2 and 8 <11>',\n",
       "              '#5# pH 5.7: about 50% of activity maximum, pH 8.0: about 70% of activity maximum <7>'],\n",
       "             '1.3.1.31': ['#27# about 65% of maximal activity at pH 4.5 and at pH 6.2 <5>',\n",
       "              '#44# active between pH 5.0 and pH 9.0 <18>',\n",
       "              '#45# only 30% of highest activity at pH 9 <25>',\n",
       "              '#46# conversion rate drops to 20% at pH 6.0 <24>',\n",
       "              '#48# pH 6.0: about 85% of maximal activity, pH 8.0: about 80% of maximal activity <36>',\n",
       "              '#5# optimal pH range <27>',\n",
       "              '#50# pH 6.8: about 75% of maximal activity, pH 8.0: about 80% of maximal activity <40>'],\n",
       "             '1.3.1.44': ['#3# less than 50% of maximal activity above pH 7.0 and below pH 5.2 <4>'],\n",
       "             '1.3.1.47': ['#1# no change in activity <1>'],\n",
       "             '1.3.1.48': ['#3# spontaneous oxidation of NADH below pH 6.0 <3>'],\n",
       "             '1.3.1.5': ['#1# pH 5.0: about 25% of activity at maximum, pH 8.0: about 80% of activity at maximum of crude enzyme preparation in presence of Zn2+, about 10% of activity at maximum of purified enzyme in presence of Mn2+ <1>'],\n",
       "             '1.3.1.51': ['#1# pH 5.5: about 40% of maximal activity, pH 9.0: about 45% of maximal activity <1>'],\n",
       "             '1.3.1.6': ['#14# in presence of NaCl <6>'],\n",
       "             '1.3.1.70': ['#2# 14-reductase has relatively broad pH profile in acidic region, activity drastically decreases with only moderate elevation of pH above 7.8 <2>',\n",
       "              '#4# about 50% of maximum activity <1>',\n",
       "              '#8# DELTA8,14-streol DELTA14-reductase has relatively broad pH profile in the basic region with optimal pH at pH 7.8, activity decreases rapidly at pH below 7 <5>'],\n",
       "             '1.3.1.73': ['#4# 50% of maximal activity at pH 5.2 and at pH 7.7 <1>'],\n",
       "             '1.3.1.74': ['#3# 50% activity at pH 5.3 and pH 6.6 <3>'],\n",
       "             '1.3.1.75': ['#6# about half-maximal activity at pH 5.5 and pH 8.5, no activity below pH 5.2 and above pH 11.0 <3>'],\n",
       "             '1.3.1.77': ['#3# pH 7.0: about 40% of maximal activity, pH 8.9: about 90% of maximal activity <2>',\n",
       "              '#38# activity range, profile overview <31>'],\n",
       "             '1.3.1.78': ['#18# enzyme activity dropps below pH 6.5 and above pH 8.5 <13>'],\n",
       "             '1.3.1.8': ['#3# with CoA changed from sigmoidal curve to flat one <12>'],\n",
       "             '1.3.1.81': ['#1# 70% of maximal activity within this range <3>'],\n",
       "             '1.3.1.82': ['#1# 50% of maximal activity within this range, wild-type enzyme <2>',\n",
       "              '#1# 50% of maximal activity within this range, deletion mutant enzyme <2>'],\n",
       "             '1.3.1.84': ['#3# half-maximal activity <1>'],\n",
       "             '1.3.1.92': ['#2# about 25% of maximum activity <2>'],\n",
       "             '1.3.1.95': ['#1# the activity of the acryloyl-CoA reductase decreases to about 30% at pH 3.9 or pH 8.3, respectively <1>'],\n",
       "             '1.3.1.98': ['#3# at pH 7.1 and pH 9.9 about 60% of maximal activity <5>'],\n",
       "             '1.3.2.3': ['#7# for activity with cytochrome c <19>'],\n",
       "             '1.3.3.14': ['#1# pH 3.8: about 40% of maximal activity, pH 7.0: about 65% of maximal activity <2>'],\n",
       "             '1.3.3.3': ['#16# activity range, recombinant enzyme <74>',\n",
       "              '#8,9# 80% of the maximum activity within <18,22>'],\n",
       "             '1.3.3.4': ['#13# at pH 2 and pH 10 less than 10% of activity <11>'],\n",
       "             '1.3.3.5': ['#13# activity range, inactive above and below. In reaction with electron donor substrates, the enzyme exhibits the maximal activity at acidic pH values: pH 4.0 for 2,2-azino-bis-[3-ethylbenzthiazoline-6-sulfonic acid] (ABTS) and pH 3.0 for potassium ferricyanide. Catalytic activity decreases on pH increase, and the enzyme becomes completely inactive at pH above 9.5. At neutral pH values, bilirubin oxidase retains about 50% maximal activity in oxidation of both substrates. In reaction with a hydrogen atom donor (catechol), the pH profile of the enzyme activity is shifted to alkaline values: enzymatic activity is not exhibited at pH below 6.0. This is probably related with the higher reactivity of the substrate as a phenolate anion <57>',\n",
       "              '#2# at pH 8.0 and 10.0: about 30% of maximum activity <3,6>',\n",
       "              '#8# using p-phenylenediamine or o-aminophenol as a substrate <32>',\n",
       "              '#8# activity range with substrate remazol brilliant blue R, profile overview <51>',\n",
       "              '#8# using 2,2-azino-bis(3-ethylbenzothiazoline-6-sulfonic acid) as a substrate <32>',\n",
       "              '#8# under acidic condition enzyme oxidizes only conjugated bilirubin <17>'],\n",
       "             '1.3.3.6': ['#10# both isoforms <32>',\n",
       "              '#5# pH 7: about 30% of activity maximum, pH 10: about 5% of activity maximum, inactive below pH 6.5 <6>'],\n",
       "             '1.3.5.1': ['#30# decrease of activity below <3>',\n",
       "              '#30# 80% of maximal activity <3>',\n",
       "              '#4# slow decline of activity at acidic pH, fast decline in alkaline range <7>',\n",
       "              '#80# pH 5.5: about 40% of maximal activity, pH 7.5: about 55% of maximal activity <119>'],\n",
       "             '1.3.5.2': ['#5# the rate constant for flavin reduction increases with pH <32>',\n",
       "              '#8# no activity below pH 6.0 and above pH 10.0 <15>'],\n",
       "             '1.3.7.1': ['#2# pH 5.5: about 60% of activity maximum, pH 8.2: about 20% of activity maximum, rapid decline in activity below pH 5.6 may be partly due to destruction of ferredoxin in acid <1>'],\n",
       "             '1.3.7.8': ['#24# 68% of maximal activity at pH 6.3 and 88% at pH 8.8 <22>',\n",
       "              '#25# inactive below pH 6.0 and above pH 9.0 <22>',\n",
       "              '#3# 50% of maximal activity at pH 6.6 and 8.2, less than 10% of maximal activity at pH 6.2 and 8.7 <4>'],\n",
       "             '1.3.8.1': ['#10# 30% resiudal activity <76>',\n",
       "              '#10# 30% residual activity <76>',\n",
       "              '#17# level of operon-mRNA not affected within this range <61>',\n",
       "              '#2# less than 50% of maximal activity above and below <6>',\n",
       "              '#5# 70% activity at pH 6, 48% activity at pH 10 <24>',\n",
       "              '#5# 42% activity at pH 6.5, 78% at pH 7.5 and 65% at pH 8.0 <44>',\n",
       "              '#9# 57% activity at pH 7.6, 72% activity at pH 9 <22>'],\n",
       "             '1.3.8.7': ['#4# approx. 20% of maximum activity at pH 7.5, sigmoidal increase between pH 7.0 and pH 10.0, approx. 80% of maximal activity at pH 9.0 <63>',\n",
       "              '#4# T168A mutant enzyme, approx. 20% of maximum activity at pH 6.0, sigmoidal increase between pH 5.5 and pH 9.0, approx. 80% of maximal activity at pH 7.8 <63>',\n",
       "              '#6# increasing activity with increasing pH <31>'],\n",
       "             '1.3.98.1': ['#10# best in sodium phosphate and sodium diphosphate buffer, not active below pH 6.0, 60% of activity maximum at pH 11.0 <3>',\n",
       "              '#7# assay in this range <8>'],\n",
       "             '1.3.99.4': ['#10# pH 8: about 35% of activity maximum, pH 11: about 25% of activity maximum, oxidase reaction <1>',\n",
       "              '#10# pH 7: about 60% of activity maximum, pH 10.5: about 85% of activity maximum, phenazine methosulfate dehydrogenation <1>',\n",
       "              '#32# over 50% of maximal activity within this range <39>; #5# over 40% of maximal activity within this range <41>'],\n",
       "             '1.3.99.5': ['#6# pH 7.0: about 40% of maximal activity, pH 8.0-9.0: maximal activity, reaction with Delta1-androstene-3,17-dione and N,N,N\\x91,N\\x91-tetramethyl-1,4-phenylenediamine <3>'],\n",
       "             '1.4.1.1': ['#15# pH 6.0: about 30% of maximal activity, pH 8.5: about 25% of maximal activity, reductive amination <53>',\n",
       "              '#15# pH 8.5: about 35% of maximal activity, pH 11.5: about 20% of maximal activity, oxidative deamination <53>',\n",
       "              '#2# pH 7.2: about 10% of activity maximum, pH 10.0: about 30% of activity maximum, reductive amination <21>',\n",
       "              '#2# pH 8: about 35% of activity maximum, pH 11: about 75% of activity maximum, oxidative deamination <21>; #9# pH 8.0: about 50% of maximal activity, pH 11.0: about 95% of maximal activity, reductive amination of pyruvate <4>',\n",
       "              '#30# pH 7.0: 13% of maximal activity, pH 10.8: 52% of maximal activity <1>',\n",
       "              '#36# pH 6.5: about 30% of activity maximum, pH 9.0: about 55% of maximal activity, reductive amination <8>',\n",
       "              '#36# pH 8.0: about 60% of activity maximum, pH 10.5: about 80% of activity maximum, oxidative deamination <8>',\n",
       "              '#40# pH 6.8: about 50% of maximal activity, pH 8.8: about 60% of maximal activity, reductive amination <40>',\n",
       "              '#40# pH 8.5: about 60% of maximal activity, pH 10.6: about 40% of maximal activity, oxidative deamination <40>',\n",
       "              '#41# pH 6.2: about 50% of maximal activity, pH 8.0: about 70% of maximal activity, oxidative deamination <67>',\n",
       "              '#41# pH 6.1: about 60% of maximal activity, pH 8.3: about 60% of maximal activity, reductive amination <67>',\n",
       "              '#5# pH 7.8: about 50% of activity maximum, pH 9.4: about 90% of activity maximum, reductive amination <30>',\n",
       "              '#6# pH 7.8: 54% of activity maximum, pH 10.2: 72% of activity maximum, reductive amination of pyruvate <3>',\n",
       "              '#6# pH 9.3: about 55% of activity maximum, pH 11.8: about 63% of activity maximum, oxidative deamination <3>',\n",
       "              '#61# pH 7.2: about 40% of maximal activity, pH 9.2: 40% of maximal activity, reductive amination of pyruvate <29>',\n",
       "              '#61# pH 7.8: about 50% of maximal activity, pH 10.0: about 65% of maximal activity, oxidative deamination of L-Ala <29>',\n",
       "              '#80# over 50% of maximal activity within this range wild-type enzyme <105>',\n",
       "              '#82# purified recombinant AsAlaDH lost activity rapidly at pH 11.0 or higher, while retaining more than 50% activity at pH 9.5. It shows less than 10% activity at neutral pH 7.0 and almost no activity at pH 3.0-6.5 <112>',\n",
       "              '#83# activity range <111>',\n",
       "              '#9# pH 7.0: about 50% of maximal activity, pH 10.0: about 50% of maximal activity, oxidative deamination of L-Ala <4>'],\n",
       "             '1.4.1.11': ['#2# pH 8.3: about 55% of maximal activity, pH 10.2: about 60% of maximal activity <1>'],\n",
       "             '1.4.1.12': ['#1# half maximal activity at pH 8.0 and 10.0, 2,4-diaminopentanoate + NAD+ + H2O <3>',\n",
       "              '#3# the enzyme shows more than 70% of maximum activity at pH 9.0-10.0 <5>'],\n",
       "             '1.4.1.13': ['#16# at pH 7.4 and 9.1 about 50% of activity maximum <21>',\n",
       "              '#3# at pH 6.5 and 8.7 about 50% of activity maximum <1>',\n",
       "              '#3# at pH 6.4 and 9.0 about 50% of activity maximum <1>'],\n",
       "             '1.4.1.14': ['#15# below and above activity declines rapidly <4>',\n",
       "              '#29# optimal pH-range with maximum at pH 7.5 <9>',\n",
       "              '#6# at pH 6 and 9 about 50% of activity maximum <16>'],\n",
       "             '1.4.1.16': ['#3# 2-amino-6-oxopimelate + NH3 + NADPH <16>',\n",
       "              '#3# meso-2,6-diaminopimelate + H2O + NADP+ <16>',\n",
       "              '#57# more than 85% of maximum activity <43>'],\n",
       "             '1.4.1.18': ['#15# approx. 35% of maximal activity at pH 7.0, approx. 50% of maximal activity at pH 10.5 <10>'],\n",
       "             '1.4.1.2': ['#14# about 90% of activity maximum at pH 7.9 and 8.5, reductive amination <9>',\n",
       "              '#14# about 90% of activity maximum at pH 8.4 and 9.2, oxidative deamination <9>',\n",
       "              '#20# about 50% of activity maximum at pH 7.0 and 8.8, reductive amination <8>',\n",
       "              '#20# 50% of activity maximum at pH 8.4 and 10.1, oxidative deamination <8>',\n",
       "              '#68# about 65% of activity maximum at pH 7.5 and 9.0, reductive amination <24>',\n",
       "              '#76# pH profile, overvew <117>'],\n",
       "             '1.4.1.20': ['#13# the enzyme shows more than 50% of the maximal activity between pH 9.0 and 10.5 <49>',\n",
       "              '#16# more than 70% of maximum activity <50>',\n",
       "              '#3# pH 9.3: about 40% of maximal activity, pH 10.2: about 35% of maximal activity <4>'],\n",
       "             '1.4.1.21': ['#9# L-Asp oxidation <3>; #9# L-aspartate oxidation <3>'],\n",
       "             '1.4.1.3': ['#12# NAD(P)H + 2-oxoglutarate + NH3, more than 50% of activity maximum at pH 7.5 and 8.8 <17>',\n",
       "              '#12# deamination, more than 50% of activity maximum at pH 8.0 and 9.2 <17>',\n",
       "              '#29# activity range, profile overview <92>',\n",
       "              '#45# pH 6.5: about 50% of maximal activity, pH 8.0: about 70% of maximal activity <108>',\n",
       "              '#46# pH 7.0: about 50% of maximal activity, pH 9.0: about 35% of maximal activity <108>',\n",
       "              '#6# maximal specific activity in TRA buffer <79>',\n",
       "              '#23# 90-100% activity <30>; #39# 25-30% less active at pH 7.0 than at pH 7.5 and 8.0 <60>; #39# equally active at pH 7.0 and 8.0 <60>; #6# lowering the pH of the buffer from pH 8.0 to pH 7.0 increases the Km for ammonia substantially, i.e. for hGDH1 from 12.8 mM to 57.5 mM, and for hGDH2: from 14.7 mM to 62.2 mM, thus essentially precluding reductive amination <95>',\n",
       "              '#7# half activity of glutamate deamination compared to pH 9.5 <42>'],\n",
       "             '1.4.1.4': ['#64# fully active after 60 min <20>',\n",
       "              '#108# more than 70% of maximum activity <107>',\n",
       "              '#62# fully active <11>',\n",
       "              '#62# above pH 9 or below 4.0 no activity <9>',\n",
       "              '#70# pH 8.5: about 60% of maximal activity, pH 10.0: about 40% of maximal activity <61>'],\n",
       "             '1.4.1.8': ['#10# above rates of reductive amination and oxidative deamination decrease dramatically <5>',\n",
       "              '#7# pH 8.5: 40% loss of activity, pH 9.5: 10% loss of activity <4>'],\n",
       "             '1.4.1.9': ['#17# activity range, profile, overview <50>',\n",
       "              '#19# pH 9: about 40% of maximal activity, pH 10.5: about 55% of maximal activity, oxidative deamination of L-Leu <43>',\n",
       "              '#19# pH 8: about 30% of maximal activity, pH 10: about 70% of maximal activity, reductive amination of 2-ketoisocaproate <43>; #32# more than 70% of maximum activity <64>',\n",
       "              '#3# pH 8.7: about 65% of maximal activity, pH 10.7: about 45% of maximal activity, reductive amination of 2-oxo-4-methylselenobutanoate <21>',\n",
       "              '#9# pH 10.0: about 40% of maximal activity, pH 11.5: about 80% of maximal activity, oxidative deamination of L-Leu <22>'],\n",
       "             '1.4.3.1': ['#25# activity assay range <39>',\n",
       "              '#47# the activity increases gradually as the pH is increased from 6.0 to 8.3, remains approximately constant at pH 8.3 to 12.5, and decreases rapidly when the pH is greater than 12.5. No enzymatic activity is detected at very low and high pH levels (5.0 and 13.0), respectively <46>',\n",
       "              '#6# largely destroyed below or over this range <23>'],\n",
       "             '1.4.3.10': ['#14# the enzyme exhibits more than 90% of the maximum activity at pH 7.0 to 10.0 <21>',\n",
       "              '#8# pH 6.8: about 50% of maximal activity, pH 9.0: about 60% of maximal activity <9>'],\n",
       "             '1.4.3.11': ['#1# pH 4.0: about 50% of maximal activity pH 10.0: about 40% of maximal activity <4>',\n",
       "              '#10# more than 50% activity between pH 5.0 and 7.5 <25>',\n",
       "              '#3# pH 4.0: about 75% of maximal activity, pH 7.0: about 80% of maximal activity <2>',\n",
       "              '#4# pH 5.5: about 30% of maximal activity, pH 8.5: about 80% of maximal activity <1>'],\n",
       "             '1.4.3.14': ['#16# about 50% activity at pH 6.0, about 80% activity at pH 7.0, 100% activity at pH 8.0, about 85% activity at pH 9.0, about 78% activity at pH 9.5 <16>',\n",
       "              '#2# enzyme activity does not fall below 60%, at pH 3.0 the activity is still 30% <8>',\n",
       "              '#3# the enzyme shows more than 70% of the maximal activity between pH 4.5 and 10.0 <22>'],\n",
       "             '1.4.3.16': ['#3# about 40% activity at pH 6.0, about 50% activity at pH 8.0 <17>',\n",
       "              '#9# pH 8.8: about 50% of maximal activity, pH 12.0: about 70% of maximal activity <16>'],\n",
       "             '1.4.3.19': ['#1# pH 8.0: about 55% of maximal activity, pH 11.0: about 55% of maximal activity, reaction with sarcosine <4>',\n",
       "              '#6# about 75% activity at pH 7.0, more than 50% activity is retained between pH 7.5 and 8.5, about 60% activity at pH 9.0. The enzyme activity decreases sharply below pH 7.0 and above pH 9.0. There is a negligible amount of activity at pH below 5.0 and above 10.0 <16>'],\n",
       "             '1.4.3.2': ['#11# half maximal activity at pH 3.5 and 12 <22>',\n",
       "              '#154# activity range, profile overview <164>',\n",
       "              '#156# complete loss of activity below <178>',\n",
       "              '#19# activity assay range <120>',\n",
       "              '#25# the enzyme shows a slight decrease in its activity at pH 5.0 and 9.6. Actually, the enzyme activity decreases by about 50% of its initial activity at the optimum at acidic pH (3.0) and by less than 3% at alkaline one (pH 10.4) <157>',\n",
       "              '#54# about 55% activity at pH 7.0, about 85% activity at pH 9.0 <135>',\n",
       "              '#82# L-alanine as substrate, pH 4.5, ca. 3% of the activity, pH 10.5, ca. 30% of the activity <62>',\n",
       "              '#97# pH-profile, overview <107>'],\n",
       "             '1.4.3.21': ['#39# piperideine-forming activity <76>',\n",
       "              '#5# assay range <64>'],\n",
       "             '1.4.3.22': ['#15# pH 5 and pH 10: 50% activity <25>',\n",
       "              '#5# pH 6.2: about 75% of maximal activity, pH 8.2: about 60% of maximal activity <39>',\n",
       "              '#5# pH 6.2 and pH 7.8: about 75% activity, putrescine <11>'],\n",
       "             '1.4.3.3': ['#36# pH 6.5: about 55% of maximal activity, pH 10.0: about 50% of maximal activity <47>; #14# pH 6.5: about 55% of maximal activity, pH 10.0: about 80% of maximal activity <47>',\n",
       "              '#5# pH 7.5 and pH 11: about 40% of maximal activity <62>',\n",
       "              '#5# less than 50% of maximal activity above and below, D-alanine, immobilized enzyme <6>',\n",
       "              '#5# less than 50% of maximal activity above and below, D-alanine, free enzyme <6>',\n",
       "              '#5# pH 8.4: about 60% of maximal activity, pH 9.0-10.0: maximum <47>',\n",
       "              '#59# pH 6.5: about 75% of maximal activity, pH 11.0: about 75% of maximal activity <48>'],\n",
       "             '1.4.3.4': ['#3# about 50% of activity maximum at pH 7.5 and 10 <18>',\n",
       "              '#3# no activity below and above <18>',\n",
       "              '#5# about 50% of activity maximum at pH 7 and 11 <11>'],\n",
       "             '1.4.3.5': ['#5# pyridoxamine phosphate oxidase activity is absent at pH 6.0, rises sharply from pH 6.0 to 8.5 and remains constantly high between pH 8.5 and 10.0 <37>',\n",
       "              '#5# pyridoxine phosphate oxidase activity <37>'],\n",
       "             '1.4.3.8': ['#1# half-maximal activities at pH 7 and pH 9.8 <2>',\n",
       "              '#2# pH 7.0: about 50% of activity maximum, pH 8.2: 37% of activity maximum <3>'],\n",
       "             '1.4.5.1': ['#5# pH 7.5: about 50% of maximal activity, pH 8.5: about 45% of maximal activity <3>'],\n",
       "             '1.4.7.1': ['#36# pH 5.0: about 45% of maximal activity, pH 10.0: about 55% of maximal activity <51>'],\n",
       "             '1.4.9.2': ['#3# 5.5: about 25% of activity maximum, 9.5: about 10% of activity maximum <3>'],\n",
       "             '1.5.1.10': ['#2# pH 5.5: about 45% of activity maximum, pH 7.8: about 35% of activity maximum <6>',\n",
       "              '#2# about 50% of activity maximum at pH 9 and 10 <7>',\n",
       "              '#2# about 50% of activity maximum at pH 8.3 and 10.3 <6>'],\n",
       "             '1.5.1.11': ['#65# about 70% of maximal activity at pH 6 and pH 8 <8>'],\n",
       "             '1.5.1.18': ['#1# pH 9.5: 60% of maximum activity, pH 12.2: 60% of maximum activity <1>'],\n",
       "             '1.5.1.2': ['#15# no proline dehydrogenase activity below pH 8.5 <14>',\n",
       "              '#35# pyrroline-5-carboxylate reductase activity <26>',\n",
       "              '#35# proline dehydrogenase activity <26>',\n",
       "              '#37,38# less than 50% of maximal activity above and below <10>'],\n",
       "             '1.5.1.20': ['#6# pH 6: about 70% of activity maximum, pH 8: about 35% of activity maximum with 5-methyl-5,6,7,8-tetrahydropteroylmonoglutamate, about 65% of activity maximum with 5-methyl-5,6,7,8-tetrahydropteroylpentaglutamate <26>',\n",
       "              '#7# half-maximal activity at pH 6.2 and pH 7.3, methylenetetrahydrofolate reductase reaction <1>',\n",
       "              '#7# about 60% of activity maximum at pH 5.8, about 50% of activity maximum at pH 8.6, menadione reductase activity <1>'],\n",
       "             '1.5.1.22': ['#14# pH optimum <9>'],\n",
       "             '1.5.1.23': ['#60# pH 8.0: 50% of maximal activity, pH 9.0: optimum, tauropine oxidation <7>',\n",
       "              '#60# 50% of maximal activity at pH 5.5 and at pH 7.4, tauropine formation <7>',\n",
       "              '#63# 50% of maximal activity at pH 5.1 and at pH 7.2, tauropine formation <9>',\n",
       "              '#63# 50% of maximal activity at pH 8.7 and at pH 9.2, tauropine oxidation <9>',\n",
       "              '#78# 50% of maximal activity at pH 5.5 and at pH 7.9, tauropine formation, enzyme form TaDH-1 <10>',\n",
       "              '#78# 50% of maximal activity at pH 6.8, optimum at pH 8.5, tauropine oxidation, enzyme form TaDH-1 <10>'],\n",
       "             '1.5.1.26': ['#47# half-maximal forward reaction at pH 5.8, half-maximal reverse reaction at pH 8.5 <5>'],\n",
       "             '1.5.1.3': ['#11# half-maximal activities at pH 4.2 and pH 8.8, acetate buffer, Tris buffer <22>',\n",
       "              '#12# half-maximal activities at pH 4.2 and pH 5.6 <65>',\n",
       "              '#12,36# activity at pH 9 is 8% of the activity at pH 5 <95>',\n",
       "              '#27# half-maximal activities at pH 5.5 and pH 8.8 <23>',\n",
       "              '#28# half-maximal activities at pH 5.3 and pH 8.0 <31>',\n",
       "              '#33# half-maximal activities at pH 5.9 and pH 8.5 <36>',\n",
       "              '#43# mutant W48Y, more than 60% of maximum activity <155>',\n",
       "              '#43# wild-type, more than 60% of maximum activity <155>',\n",
       "              '#59# rapid decrease in activity above, wild-type <114>',\n",
       "              '#61# half-maximal activities at pH 5.5 and pH 7.8 <23>',\n",
       "              '#48# recombinant, refolded enzyme <75>; #9# activity negligible below and above <49>',\n",
       "              '#91# activity range <264>'],\n",
       "             '1.5.1.30': ['#2# maximum activity at pH 6.0, more than 80% activity at pH 5.5, 6.5, and 7.0, about 75% activity at pH 7.5, about 30% activity at pH 8.5, less than 20% activity at pH 9-9.5, less than 10% activity at pH 4, 4.5, and 10, pH 4.0-6.0 citrate-phosphate buffer, pH 6.0-7.5 potassium-phosphate buffer, pH 7.5-8.5 Tris-HCl buffer, pH 8.5-10.0 glycine-NaOH buffer <27>',\n",
       "              '#4# pH 4.5: about 85% of maximal activity, pH 8.0: about 45% of maximal activity <1>'],\n",
       "             '1.5.1.33': ['#3# significant decline of activity at pH 7 and 4.7 <2>',\n",
       "              '#4# pH 4.5: about 50% of maximal activity, pH 7.0: about 30% of maximal activity with biopterin <7>'],\n",
       "             '1.5.1.34': ['#9# pH 6.8: about 45% activity, pH 7.8: about 15% activity <1>'],\n",
       "             '1.5.1.36': ['#10# activity irreversibly drops to under 5% at pH 5.3 indicating denaturation <10>',\n",
       "              '#4# optimal range <5>',\n",
       "              '#6# the enzyme maintains more than 80% of activity between pH values of 6.5 to 8 <2>'],\n",
       "             '1.5.1.37': ['#4# pH 6.0: 73% of maximal activity, pH 8.0: 81% of maximal activity <1>',\n",
       "              '#5# pH 5.0 (sodium succinate buffer): 52% of maximal activity, pH 7.6: 75% of maximal activity <4>'],\n",
       "             '1.5.1.38': ['#2# the enzyme exhibits a maximum activity at pH 5.5 which drops to a broad shoulder from pH 6.5 to pH 8.5 with an activity 75% that of maximum at pH 7.0. About 60% of maximal activity at pH 5.0, about 50% of maximal activity at pH 10.0 <2>'],\n",
       "             '1.5.1.40': ['#1# pH 4.3: about 50% of maximal activity, pH 6.7: about 50% of maximal activity <8>',\n",
       "              '#10# pH 7.0: about 60% of maximal activity, pH 9.0: about 50% of maximal activity, reduction of NADP+ with reduced coenzyme F420 <3>',\n",
       "              '#10# pH 4.5: about 45% of maximal activity, pH 6.5: about 35% of maximal activity, reduction of oxidized coenzyme F420 with NADPH <3>',\n",
       "              '#11# pH 7.5: about 50% of maximal activity, pH 10.0: about 90% of maximal activity, reduction of NADP+ with reduced coenzyme F420 <4>'],\n",
       "             '1.5.1.42': ['#1# over 50% of maximal activity within this range <7>',\n",
       "              '#2# bimodal, pH 5: 90% of maximal activity, pH 6.5: about 50% of maximal activity, pH 10: about 70% of maximal activity <5>',\n",
       "              '#2# the pH curve is bimodal, with maximal activity exhibited at pH 8.6, a minimum at pH 6.0. and a second maximum at pH 5.0 <2>'],\n",
       "             '1.5.1.43': ['#3# 85% and 18% of the maximum activity are observed at pH 8.0 and 6.3, respectively <3>'],\n",
       "             '1.5.1.47': ['#1# pH 4.8: about 55% of maximal activity, pH 6.3: about 70% of maximal activity <1>'],\n",
       "             '1.5.1.48': ['#1# pH 6.2: about 35% of maximal activity, pH 8.7: about 55% of maximal activity, reduction of 2-methyl-1-pyrroline <1>',\n",
       "              '#1# pH 6.2: about 60% of maximal activity, pH 8.7: about 60% of maximal activity, oxidation of (R)-2-methylpyrrolidine <1>'],\n",
       "             '1.5.1.5': ['#15# half-maximal activities at pH 5 and 9.8 <13>'],\n",
       "             '1.5.1.6': ['#5# more than 60% of maximal activity at pH 6.5 and 8.5 <1>'],\n",
       "             '1.5.1.7': ['#1# half-maximal activities at pH 5.2 and 9.0, L-lysine + 2-oxoglutarate + NADH <5>',\n",
       "              '#2# half-maximal activities at pH 5.6 and 7.8 <13>'],\n",
       "             '1.5.1.8': ['#18# pH 6.5: about 60% of maximal activity, pH 9.5: about 60% of maximal activity <23>',\n",
       "              '#3# pH 6.2: about 50% of activity maximum, pH 8.6: about 60% of activity maximum <7>',\n",
       "              '#3# about 50% of activity maximum at pH 7.2 and 8.8 <6>'],\n",
       "             '1.5.1.9': ['#2# half maximal activities at pH 7.9 and 9.3 <8>',\n",
       "              '#2# half maximal activities at pH 7.8 and 9.7 <9>',\n",
       "              '#8# no activity below pH 6 and over pH 10 <11>'],\n",
       "             '1.5.3.1': ['#5# activity range <54>',\n",
       "              '#5,9# pH 6: about 35% of maximum activity <2,7>; #8# pH 10: about 80% of maximum activity <1>; #5# pH 10: about 70% of maximum activity <7>; #9# pH 10: about 55% of maximum activity <2>; #8# pH 6: about 65% of maximum activity <1>'],\n",
       "             '1.5.3.12': ['#1# about 50% of maximal activity at pH 6.0 and 8.5 <3>',\n",
       "              '#2# about 50% of maximal activity at pH 5.5 and 8.0 <2>'],\n",
       "             '1.5.3.13': ['#2# pH profile, overview <6>'],\n",
       "             '1.5.3.16': ['#1# recombinant wild-type MmSMO enzymatic activity increases in the pH range, reaching a maximum at pH 8.5 and decreasing for higher pH values <23>',\n",
       "              '#11# pH 7: about 60% of maximal activity, pH 9.0: about 60% of maximal activity, substrate: spermine <5>; #11# pH 7: about 80% of maximal activity, pH 9.0: about 60% of maximal activity, substrate: norspermine <5>; #2# the activity markedly decreases from pH 9.0 to pH 7.0 <34>; #13# pH 7.0: about 60% of maximal activity, pH 9.0: about 45% of maximal activity, substrate: spermine <52>',\n",
       "              '#8# 7.5: about 30% of maximal activity, pH 9.0: about 40% of maximal activity, substrate: spermine <51>'],\n",
       "             '1.5.3.17': ['#11# pH 7.0: about 60% of maximal activity, pH 9.0: about 45% of maximal activity, substrate: spermine <25>',\n",
       "              '#8# pH 7: about 95% of maximal activity, pH 8.5: about 55% of maximal activity <4>'],\n",
       "             '1.5.3.18': ['#1# pH 5.0: about 45% of maximal activity at pH 5.0 and pH 7.5 <2>'],\n",
       "             '1.5.3.22': ['#2# 30% of the maximal activity at pH 6.5 and 8.5 <1>'],\n",
       "             '1.5.3.4': ['#4# pH 6: about 35% of activity maximum, pH 8: about 50% of activity maximum <1>'],\n",
       "             '1.5.3.5': ['#5# pH 7.0: about 65% of maximal activity, pH 9.0: about 25% of maximal activity <18>'],\n",
       "             '1.5.5.2': ['#22# enzyme ProDH shows a good activity in the range of pH 7.0-9.0, and optimum pH at pH 8.5 <27>'],\n",
       "             '1.5.8.3': ['#3# not active below, sarcosine + phenazine methosulfate <16>',\n",
       "              '#7# not active below, sarcosine + phenazine methosulfate <12>'],\n",
       "             '1.5.99.12': ['#4# pH 6.0: about 45% of maximal activity, pH 10.0: aout 55% of maximal activity <26>'],\n",
       "             '1.5.99.13': ['#3# 50% of maximal activity at pH 6.5 and at pH 8.5, proline dehydrogenation <1>'],\n",
       "             '1.6.1.1': ['#1# more than 50% activity between pH 7.0 and 9.0 <25>',\n",
       "              '#1# pH 7.0: about 55% of maximal activity, pH 10.5: about 45% of maximal activity <22>',\n",
       "              '#7# half maximal activity in this pH range in the absence of Ca2+ or Mg2+, no activity above <4>',\n",
       "              '#7# half maximal activity in this pH range in the presence of 5 mM Ca2+ or Mg2+, no activity above <4>'],\n",
       "             '1.6.1.2': ['#9# less than 50% of maximal activity above and below <42>'],\n",
       "             '1.6.1.3': ['#4# pH 7.6: about 40% of maximal activity, pH 9.4: about 70% of maximal activity <3>'],\n",
       "             '1.6.2.2': ['#4# depending on acceptor <1>',\n",
       "              '#8# enzyme activity is higher at pH 7.5 compared to pH 6.6 <113>'],\n",
       "             '1.6.2.4': ['#39# less than 50% of maximal activity above and below, grown on alkane or glycerol <27>'],\n",
       "             '1.6.2.6': ['#3# no activity below and above <9>'],\n",
       "             '1.6.3.2': ['#2# pH 6.5: about 70% of maximal activity, pH 8.5: about 45% of maximal activity <2>',\n",
       "              '#3# active over the range pH 7-9 <1>',\n",
       "              '#4# pH 7.5: about 70% of maximal activity, pH 8.5: about 70% of maximal activity <5>'],\n",
       "             '1.6.3.3': ['#1# pH 5.5: 73% of maximal activity (50 mM Mes buffer), pH 8.5: 81% of maximal activity (50 mM Mops buffer) <6>',\n",
       "              '#10# pH 6.0: about 60% of maximal activity, pH 9.5: about 60% of maximal activity <5>',\n",
       "              '#12# pH 6.5: about 55% of maximal activity, pH 9.0: about 55% of maximal activity <7>',\n",
       "              '#13# pH 5.0: about 50% of maximal activity, pH 8.0: about 50% of maximal activity <7>',\n",
       "              '#4# pH 7.8: about 60% of maximal activity, pH 11.0: about 90% of maximal activity <2>',\n",
       "              '#6# the activity decreases with pH in the range pH 4.5 to 9.8 <12>',\n",
       "              '#7# very active in the pH-range 6.5-7.5 <3>',\n",
       "              '#8# pH 4.4: about 70% of maximal activity, pH 8.4: about 90% of maximal activity <9>',\n",
       "              '#9# pH 4.5: about 90% of maximal activity, pH 8.0: about 50% of maximal activity <11>'],\n",
       "             '1.6.3.4': ['#10# pH 6.0: about 60% of maximal activity, pH 9.3: about 50% of maximal activity <8>',\n",
       "              '#15# 70% activity at pH 6.0 and pH 8.0 <20>',\n",
       "              '#3# higher than 70% of the maximum activity at pH values of 6.5\\x968.5 <3>',\n",
       "              '#5# pH 4.6: 68% of maximal activity, pH 6.0: 54% of maximal activity <5>',\n",
       "              '#7# the enzyme is most active between pH values 6.5-8.0 <14>'],\n",
       "             '1.6.5.2': ['#2# the enzyme maintains greater than 90% of its maximal activity between pH 5.5 and 7.5 <118>',\n",
       "              '#20# pH 3.7: about 45% of maximal activity, pH 4.8: about 55% of maximal activity <3>',\n",
       "              '#24# pH 5.0: about 45% of maximal activity, pH 9.0: about 55% of maximal activity <46>',\n",
       "              '#31# reduction of 2,6-dimethoxybenzoquinone <30>',\n",
       "              '#35# more than 95% of maximum activity within this range <60>',\n",
       "              '#36# more than 95% of maximum activity within this range <60>',\n",
       "              '#6# inactive above or below <23>'],\n",
       "             '1.6.5.4': ['#21# pH 5.5: about 60% of maximal activity, pH 9.5: about 75% of maximal activity <8>',\n",
       "              '#39,40,41# 85% or greater activity between pH 7.6 and pH 8.7 <44>',\n",
       "              '#9# about 50% of maximal activity at pH 6.7 and at pH 7.9 <1>'],\n",
       "             '1.6.5.5': ['#11# pH 4.5: about 60% of maximal activity, pH 10: about 50% of maximal activity <3>',\n",
       "              '#7# half-maximal activity at pH 7.5 and pH 9.2 <18>'],\n",
       "             '1.6.5.6': ['#4# with different buffers <2>'],\n",
       "             '1.6.5.9': ['#3# with ubiquinone-6 as acceptor the activity is almost constant between pH 4.5 and 9.5, at pH values lower than 4.5 and higher than 9.5, enzymic activity is irreversibly destroyed <1>'],\n",
       "             '1.7.1.1': ['#10# pH 6.0: about 35% of maximal activity, pH 8.5: about 45% of maximal activity <85>',\n",
       "              '#3# in 0.05 mM MOPS buffer <100>',\n",
       "              '#3# in MES buffer <100>'],\n",
       "             '1.7.1.10': ['#6# pH 5.5: about 70% of activity maximum, pH 7.2: 50% activity <6>',\n",
       "              '#6# pH 5.4: considerable activity in Tris-maleate buffer, pH 8.4: about 50% activity <6>'],\n",
       "             '1.7.1.13': ['#8# more than 60% of maximum activity <13>'],\n",
       "             '1.7.1.16': ['#8# considerable loss of activity above <10>'],\n",
       "             '1.7.1.2': ['#2# at pH 7 and 9: about 60% of activity maximum <13>',\n",
       "              '#23# at pH 7.1 and 8.2: about 50% of activity maximum <7>'],\n",
       "             '1.7.1.4': ['#3# pH 6.5: about 50% activity, pH 9.5: about 15% activity <7>',\n",
       "              '#4# pH 4.0: 40% activity, pH 11.0: 70% activity <13>'],\n",
       "             '1.7.1.6': ['#81# activity range, profile overview <71>',\n",
       "              '#20# pH 4.0: about 10% activity, pH 10.0: about 20% activity <1>',\n",
       "              '#55# maximal activity <34>',\n",
       "              '#6# more than 80% of maximum activity <58>',\n",
       "              '#7# pH 4.0: about 25% activity, pH 7.0: about 20% activity <2>',\n",
       "              '#75# more than 50% of maximum activity <54>',\n",
       "              '#79# recombinant chimeric fusion protein AZR-GDH, activity range, profile overview <69>',\n",
       "              '#83# activity range, profile overview <65>',\n",
       "              '#85# activity range, profile overview <61>'],\n",
       "             '1.7.1.7': ['#10# pH 7: about 25% of activity at maximum, pH 9.5: about 30% of activity at maximum <11>',\n",
       "              '#12# strong decrease above <16>',\n",
       "              '#4# pH 6: about 50% of activity at maximum, pH 9: about 70% of activity at maximum <12>'],\n",
       "             '1.7.2.1': ['#3# pH 5.2: about 60% of maximal activity, pH 6.5: about 55% of maximal activity, reaction with 5 mM nitrite and pseudoazurin as electron donor <60>',\n",
       "              '#3# pH 4.9: about 55% of maximal ativity, pH 6.2: about 65% of maximal activity, reaction with 0.5 mM nitrite and pseudoazurin as electron donor <60>',\n",
       "              '#48,50# 3fold decrease in the rate of NO formation compared to pH 6.5 <99>'],\n",
       "             '1.7.2.2': ['#16# the sulfite reductase activity of NiR has a maximum value at neutral pH, the sulfite reductase activity of NiR decreases with increasing pH and is absent at pH 9.0, the activity measured at pH 7.8 is 20% of the activity measured at pH 7.0 <32>'],\n",
       "             '1.7.2.4': ['#1# half maximum activity values at pH 6 and pH 9.2 <30>',\n",
       "              '#16# activity is maximal after incubation at high pH values <8>',\n",
       "              '#24# approx. 50% of maximal activity at pH 7.5, approx. 75% of maximal activity at pH 9.5 <34>',\n",
       "              '#32# perturbations of the protein conformation induced by pH variations, although the principal secondary structure elements are largely unaltered <48>'],\n",
       "             '1.7.2.5': ['#15# at pH 6.0 half of the activity of pH 4.8 <21>'],\n",
       "             '1.7.3.1': ['#1# half-maximal activity at pH 6.2 and 7.6 <1>'],\n",
       "             '1.7.3.2': ['#1# increase of activity from pH 7.0 to 9.0 <1>'],\n",
       "             '1.7.3.3': ['#10# 50% of activity maximum at pH 7.4 and pH 9.6, free enzyme <28>',\n",
       "              '#10# 50% of activity maximum at pH 8 and pH 11, immobilized enzyme <28>',\n",
       "              '#18# pH 8.0: about 35% of activity maximum, pH 10.2: about 55% of activity maximum <13>',\n",
       "              '#19# pH 7: about 50% of activity maximum, pH 11: about 40% of activity maximum <18>',\n",
       "              '#31# 50% of activity maximum at pH 8.5 and pH 10.5 <22>',\n",
       "              '#44# pH 6.5: about 40% of maximal activity, pH 10.0: about 50% of maximal activity <99>',\n",
       "              '#68# more than 80% of maximum activity <123>',\n",
       "              '#68# more than 80% of maximum activity <123>',\n",
       "              '#7# free uricase shows at least 50% relative activity between pH 6.5 and 9.5, around pH 7.5, free uricase remains 81.16% of its maximum activity, while the uricase loaded in the lipid vesicles remains almost the same high activity (178.26%) as its optimum activity (179.72%) <103>'],\n",
       "             '1.7.3.6': ['#1# activity with cytochrome c550 continues to increase to pH 11, activity at pH 8 is about 50% of the activity at pH 11 <2>',\n",
       "              '#1# pH 7.6: about 50% of maximal activity, pH 10.0: about 35% of maximal activity with horse heart cytochrome c <2>',\n",
       "              '#1# pH 7.3: about 35% of maximal activity, pH 11.5: about 50% of maximal activity activity with pseudoazurin <2>'],\n",
       "             '1.7.5.2': ['#1# approx. 50% of maximal activity at pH 5.0, almost no activity at pH 10.0 <2>'],\n",
       "             '1.7.7.1': ['#4# appreciable amounts of activity at pH 7.8 and pH 8.3 in Tris buffer <44>',\n",
       "              '#6# pH 6.8: about 75% of activity maximum, pH 8.3: about 70% of activity maximum <28>'],\n",
       "             '1.7.7.2': ['#14# at pH 7.3 and 9.2: 50% of activity maximum, aerobic assay, methyl viologen-linked and ferredoxin-linked activity <3>',\n",
       "              '#17# the activities at pH 6.0 and pH 8.0 are 83% and 81% of the activity at pH 7.2 <22>',\n",
       "              '#28# pH 5.0: about 50% of maximal activity, pH 11.0: about 80% of maximal activity <24>',\n",
       "              '#5# at pH 8.3 and 9.8: about 50% of activity maximum <9>'],\n",
       "             '1.8.1.12': ['#10# less than 15% activity at pH 6.0, about 50% activity at pH 6.5, about 80% activity at pH 7.0, 100% activity at pH 7.5, about 35% activity at pH 8.0, around 30% activity between pH 8.5 and 11.0, no activity at pH 12.0 and less than pH 5.0 <76>'],\n",
       "             '1.8.1.13': ['#1# about half-maximal activity at pH 6.3 and 8.2 <2>'],\n",
       "             '1.8.1.2': ['#3# 75% activity at pH 6.5 and pH 8.0 <38>'],\n",
       "             '1.8.1.4': ['#34# pH 5.5: about 75% of maximal activity, pH 7.5: about 40% of maximal activity, NADH-lipoamide oxidoreductase activity <37>',\n",
       "              '#34# pH 7.5: about 50% of maximal activity, pH 8.8: about 60% of maximal activity, dihydrolipoamide-NAD+ oxidoreductase activity <37>',\n",
       "              '#51# pH 5.0: about 10% of maximal activity, pH 7.0-7.5: 60-70% of maximal activity, pH 8.0-8.5: 90% of maximal activity, pH 10.0: 10% of maximal activity in carbonate buffer <30>',\n",
       "              '#6# pH 6.5: about 40% of maximal activity, pH 8.5: about 50% of maximal activity <28>',\n",
       "              '#79# high activity in a pH range of 6.0 to 7.0 <108>',\n",
       "              '#9# nitric oxide reduction <61>'],\n",
       "             '1.8.1.7': ['#14# at pH 5.2 and 9.0 about 20% of activity maximum <29>',\n",
       "              '#19# pH 7: about 40% of activity maximum, pH 9.5: about 55% of activity maximum <40>',\n",
       "              '#19# pH 7.3: about 65% of activity maximum, pH 8.6: about 90% of activity maximum <39>',\n",
       "              '#20# 50% of maximal activity at pH 6.1 and 9.1, root enzyme <27>',\n",
       "              '#20# pH 6.9: about 50% of activity maximum, pH 9.2: about 83% of activity maximum <27>',\n",
       "              '#30# pH 7: about 13% of activity maximum, pH 11: 36% of activity maximum <35>',\n",
       "              '#35# at pH 6.3-9.6 about 50% of activity maximum <49>',\n",
       "              '#48# in 100 mM Tris-HCl buffer <147>',\n",
       "              '#57# 50% activity at pH 6.5 and pH 9.0, 10% activity at pH 5.0 and pH10.0 <67>',\n",
       "              '#8,11,19# at saturating NADPH levels <13>'],\n",
       "             '1.8.1.8': ['#6# pH 6.2: about 50% of activity maximum, pH 7.8: about 20% of activity maximum <1>'],\n",
       "             '1.8.1.9': ['#11# wild-type and mutant <39>',\n",
       "              '#14# pH 6.5: about 50% of maximal activity, pH 8.5: about 60% of maximal activity <58>',\n",
       "              '#29# wild-type fusion protein <50>',\n",
       "              '#29# mutant enzyme, 50% activity at pH 6.5 <50>',\n",
       "              '#63# pH 6.5: about 40% of maximal activity, pH 9.0: about 50% of maximal activity, wild-type enzyme <104>',\n",
       "              '#7# around 60% activity at pH 6.0 and 6.5 <212>'],\n",
       "             '1.8.2.1': ['#11# 90% of maximal activity at pH 7.3 and 9.5 <1>',\n",
       "              '#18# less than 50% of maximal activity above and below <38>',\n",
       "              '#3# the greatest loss in activity occurs while the enzyme is at pH 10, while in the range above pH 6 and below pH 9 there is no significant variation, inactivation at pH 11.0 <41>',\n",
       "              '#6# wild type enzyme <24>',\n",
       "              '#6# mutant enzyme Y236F <24>'],\n",
       "             '1.8.2.2': ['#19# no residual activity <23>',\n",
       "              '#6# little change in activity between <12>'],\n",
       "             '1.8.2.5': ['#2# 8% of maximum activity <1>'],\n",
       "             '1.8.2.7': ['#1# more than 50% activity between pH 8.0 and 11.0 with potassium ferricyanide as electron acceptor, more than 50% activity between pH 8.5 and 10.5 with potassium ferricyanide as electron acceptor <3>'],\n",
       "             '1.8.3.1': ['#32# below: less than 50% of maximal activity <11>',\n",
       "              '#47# pH dependence of the maximum catalytic oxidation current at the GC/HSO electrode with 4 mM sulfite and in the presence of 10 mM [Fe(tacn)]23+ in 100 mM mixed buffer solution, overview <112>'],\n",
       "             '1.8.3.2': ['#1# pH 5.0: about 40% of maximal activity, pH 11: about 70% of maximal activity <23>',\n",
       "              '#34# more than 80% of maximum activity <60>'],\n",
       "             '1.8.3.3': ['#2,5# minimal activity below pH 5.0, active at least up to pH 10.0 <2,3>',\n",
       "              '#5# sharp decline in activity below pH 6.2 and above pH 8.8 <3>',\n",
       "              '#5# enzyme immobilized on microsensor surface out of cellulose acetate <4>'],\n",
       "             '1.8.4.11': ['#20# the enzyme shows more than 50% activity between pH 7.5 and 9.5 <100>',\n",
       "              '#59# sharp decrease below pH 6.5 and above pH 8.5 <13>'],\n",
       "             '1.8.4.12': ['#17# 3% of maximum activity <91>'],\n",
       "             '1.8.4.2': ['#10# pH 6.2: about 10% of activity maximum, pH 10.2: about 20% of activity maximum <41>',\n",
       "              '#12# reaction is negligible <76>'],\n",
       "             '1.8.4.3': ['#2# pH 7: about 30% of activity maximum, pH 9: about 75% of activity maximum <1>'],\n",
       "             '1.8.4.4': ['#2# pH 6.0: 10% of activity maximum, pH 8.5: activity maximum <1>'],\n",
       "             '1.8.5.1': ['#10,12,15,26,34,43,51# the enzyme almost completely loses its activity below pH 5.5 or above pH 10.0 <69>',\n",
       "              '#19# the activity of the enzyme at pH 6 was only 20% of that at pH 8.0 <26>',\n",
       "              '#21# pH 6.5: about 25% of maximal activity, pH 8.5: about 60% of maximal activity <16>; #46,47# more than 66% of maximum activity <53>',\n",
       "              '#25# isoforms DHAR1 and DHAR2 retain more than 80% of their maximum enzymatic activities at pH values between 6.0 and 10.0, respectively <68>',\n",
       "              '#36# pH 5.0: about 25% of maximal activity, pH 7.0: about 65% of maximal activity <40>',\n",
       "              '#45# 41% of maximum activity <53>',\n",
       "              '#49# more than 50% activity between pH 6.5 and 8.0 <67>'],\n",
       "             '1.8.5.2': ['#1# the native enzyme shows about 35% activity at pH 2.0 and 20% activity at pH 3.0, the recombinant enzyme shows about 80% activity at pH 2.0 and about 35% activity at pH 3.0 <3>',\n",
       "              '#2# pH 4.5: about 70% of maximal activity, pH 5.5: about 35% of maximal activity <1>'],\n",
       "             '1.8.5.8': ['#5# more than 70% activity between pH 8.0 and 9.0 <8>'],\n",
       "             '1.8.98.1': ['#1# activity with reduced benzyl viologen and N-(7-[(2-sulfoethyl)dithio]heptanoyl)-3-O-phospho-L-threonine increases continously with decreasing pH in the range between pH 7.5 and pH 5.0 <9>'],\n",
       "             '1.9.6.1': ['#12# pH 7.0: about 70% of maximal activity, pH 9.5: about 60% of maximal activity <11>',\n",
       "              '#2# sharp drop of activity below pH 5.5 and above pH 10 <1>'],\n",
       "             '1.9.98.1': ['#1# pH 5: about 50% of maximum activity, pH 6.7: about 25% of maximum activity <1>'],\n",
       "             '1.97.1.1': ['#19,20,21# 50% of maximum activity at pH 5.5 and 10 <8>',\n",
       "              '#36# pH 5.0: about 85% of maximal activity, pH 9.0: about 75% of maximal activity <18>'],\n",
       "             '1.97.1.9': ['#6# activity range, profile overview <22>'],\n",
       "             '2.1.1.1': ['#4# about 50% of maximum activity at pH 5.7 and 9.8 <2>'],\n",
       "             '2.1.1.104': ['#2# the enzyme has a very narrow pH range, it is most active at pH 7.5, and inactive below pH 6.5 and above pH 9.5 <57>',\n",
       "              '#2# 40% of maximum activity <44>',\n",
       "              '#2# 40% of maximum activity <44>',\n",
       "              '#6# about half-maximal activity at pH 6.9 and 8.6 <17>'],\n",
       "             '2.1.1.106': ['#1# half-maximal activity at pH 6.8 and pH 9.0 <1>'],\n",
       "             '2.1.1.11': ['#13# within activity range <19>',\n",
       "              '#4# pH 6.7: about 60% of maximal activity, pH 8.8: about 85% of maximal activity <1>'],\n",
       "             '2.1.1.116': ['#3# about 50% of maximal activity at pH 7 and at pH 9 <1>'],\n",
       "             '2.1.1.117': ['#4# 50% of maximal activity at pH 7.5 and at pH 10 <2>'],\n",
       "             '2.1.1.12': ['#15# about 50-55% of optimal activity at pH 5.8 and 9.0, only 1% of optimal activity at pH 5.0 <3>',\n",
       "              '#2# about 60% of activity maximum at pH 5.7 and pH 7.3 <1>'],\n",
       "             '2.1.1.122': ['#1# pH 6.5: about 55% of maximal activity, pH 8.0: about 65% of maximal activity <1>',\n",
       "              '#15# recombinant protein, half-maximal activity estimated at pH 6.8 and 8.2 <4>',\n",
       "              '#2# about 50% of maximal activity at pH 6.6 and 9.1 <2>'],\n",
       "             '2.1.1.128': ['#3# about 50% of maximal activity at pH 7.0 and 9.5 <2>'],\n",
       "             '2.1.1.129': ['#3# about 35% of maximal activity at pH 7.0 and pH 8.8 <2>'],\n",
       "             '2.1.1.13': ['#19# pH 6.0: about 75% of maximum activity, pH 7.5: about 35% of activity maximum <22>',\n",
       "              '#20# pH 6.5: about 30% of maximum activity, pH 8.3: about 85% of maximum activity <22>',\n",
       "              '#8# pH about 50% of maximum activity at pH 6.2 and 8.2 <5>'],\n",
       "             '2.1.1.136': ['#1# pH 7.0: about 50% of maximal activity, pH 7.0-8.5: optimum <2>'],\n",
       "             '2.1.1.137': ['#24# pH 7.5: about 40% of maximal activity, pH 11: about 35% of maximal activity <9>'],\n",
       "             '2.1.1.14': ['#29# at pH 4.6: 0.0096 micromol/min/mg, at pH 6.5: 0.155 micromol/min/mg, at pH 9.0: 0.119 micromol/min/mg <28>',\n",
       "              '#4# pH 6.0: about 70% of maximum activity, pH 8.7: about 90% of maximum activity <1>'],\n",
       "             '2.1.1.140': ['#8# about 50% of maximal activity at pH 6.0 and 9.0 <3>',\n",
       "              '#8# about 50% of maximal activity at pH 6.5 and at pH 8.0 <4>'],\n",
       "             '2.1.1.142': ['#10# half-maximal activities at pH 6.5 and pH 9.0 <18>',\n",
       "              '#5# slightly active at pH 5.5, no activity at pH 10 <3>'],\n",
       "             '2.1.1.147': ['#1# pH 7.0: about 45% of maximal activity, pH 9.5: about 30% of maximal activity <1>'],\n",
       "             '2.1.1.153': ['#1# 50% of the maximum activity at pH 6.4 and 8.6 <1>'],\n",
       "             '2.1.1.155': ['#2# 50% of maximum activity at pH 5.5 and 8.5, 4% of maximum activity at pH 5 <1>'],\n",
       "             '2.1.1.156': ['#2# glycine or N-methylglycine, very low activity below pH 8 <3>'],\n",
       "             '2.1.1.161': ['#1# very low activity below pH 7 <1>'],\n",
       "             '2.1.1.165': ['#15# about 60% of maximal activity at pH 6.0 and 8.0 <11>; #11# pH 6.0: about 35% of maximal activity, pH 8.0: about 75% of maximal activity <1>',\n",
       "              '#14# pH 5.4: about 50% of maximal activity, pH 10.0: about 80% of maximal activity <7>',\n",
       "              '#18# pH 6.0: about 40% of maximal activity, pH 9.0: about 35% of maximal activity, at pH greater than 9.2 no methyl bromide synthesis detected <14>'],\n",
       "             '2.1.1.182': ['#3# change of pH from 7.8 to 6.8 reduces methylation by 50% <10>'],\n",
       "             '2.1.1.184': ['#2# assay at <2>'],\n",
       "             '2.1.1.197': ['#5# less than 50% of maximal activity above and below <6>'],\n",
       "             '2.1.1.20': ['#5# pH 7.5: about 40% of maximal activity, pH 9.5: about 45% of maximal activity <24>'],\n",
       "             '2.1.1.218': ['#3# the bifunctional enzyme is active in a pH range 5.5-9.75. The intensity of m1A and m1G spots varies greatly as a function of the pH. At pH 5.5, m1A MTase activity of TK0422p is predominant over m1G. At pH 7 or higher, both m1A and m1G are detected, m1G intensity growing with increasing pH <3>'],\n",
       "             '2.1.1.221': ['#1# the bifunctional enzyme is active in a pH range 5.5-9.75. The intensity of m1A and m1G spots varies greatly as a function of the pH. At pH 5.5, m1A MTase activity of TK0422p is predominant over m1G. At pH 7 or higher, both m1A and m1G are detected, m1G intensity growing with increasing pH <2>'],\n",
       "             '2.1.1.228': ['#12# preincubation of Trm5 at pH 6.0, followed by adjustment of the solution to pH 8.0, does not cause irreversible inactivation of the enzyme. Enzyme treated in this manner retained equivalent activity in single-turnover assay <15>',\n",
       "              '#16,18# fairly active between pH 6.4 and pH 8.4 <1>',\n",
       "              '#7# pH 6.0: about 50% of maximal activity, pH 8.5: about 75% of maximal activity <3>'],\n",
       "             '2.1.1.230': ['#1# pH 7.0: about 60% of maximal activity, pH 7.8: about 60% of maximal activity <1>'],\n",
       "             '2.1.1.239': ['#1# pH 6.5: about 60% of maximal activity, pH 7.8: about 60% of maximal activity <1>'],\n",
       "             '2.1.1.258': ['#3# the forward and reverse reaction rates decrease as the pH is lowered <6>'],\n",
       "             '2.1.1.267': ['#1# loss of activity at pH 5.0-6.0, inactive below <7>',\n",
       "              '#15# recombinant enzyme, over 50% activity within this range <9>',\n",
       "              '#7# activity range, recombinant enzyme <10>'],\n",
       "             '2.1.1.269': ['#7# more than 50% of maximum activity within <1>; #4# maximum activity is observed at pH 7.0 to 8.0, about 50% activity at pH 6.5 and 8.3 <1>',\n",
       "              '#8# more than 50% of maximum activity within <1>; #8# maximum activity is observed at pH 7.0 to 8.0, about 50% activity at pH 6.0 and 8.8 <1>'],\n",
       "             '2.1.1.273': ['#3# with 65% of maximum activity at both pH 6.5 and 8.5 and at pH 5.5 and 9.5, the enzyme activity falls to about 50% of the optimal value <4>',\n",
       "              '#7# with 80\\x9690% of maximum activity <1>'],\n",
       "             '2.1.1.274': ['#1# about 90% activity at pH 7.0, about 40% activity at pH 8.0 <7>',\n",
       "              '#12# at pH 6.0 activity is 80% of optimal activity, at pH 8.0 it is 75%, and at pH 9 it is less than 50% <1>',\n",
       "              '#9# with 80\\x9690% of maximum activity <2>'],\n",
       "             '2.1.1.278': ['#2# at pH 6.5, the enzyme has about 34% of its maximal activity, whereas at pH 9.0, the enzyme displays about 40% of the maximal activity <4>',\n",
       "              '#4# pH 6.5: 20% of maximal activity, pH 9.0: 30% of maximal activity <3>'],\n",
       "             '2.1.1.279': ['#2# active in the range <1>'],\n",
       "             '2.1.1.283': ['#1# half maximal activity at pH 5.7 and pH 9.0 <1>',\n",
       "              '#2# activity range, recombinant enzyme <2>'],\n",
       "             '2.1.1.292': ['#2# pH 7.5: 40% of maximal activity, pH 8.0: optimum. No pH values above pH 8.0 are evaluated due to the inherent instability of S-adenosyl-L-methionine at alkaline pH <8>'],\n",
       "             '2.1.1.293': ['#1# the wild type enzyme shows about 42% activity at pH 6.0, about 78% activity at pH 6.5, and about 90% activity at pH 7.0 <1>'],\n",
       "             '2.1.1.302': ['#1# pH 7.5: about 50% of maximal activity, pH 10.0: about 55% of maximal activity <1>'],\n",
       "             '2.1.1.303': ['#1# pH 6.0: optimum, pH 8.0: about 50% of maximal activity, no activity below pH 5.5 <2>'],\n",
       "             '2.1.1.315': ['#1# pH 6.0: about 20% of maximal activity, pH 8.5: about 70% of maximal activity <1>'],\n",
       "             '2.1.1.336': ['#4# pH 6.5: about 35% of maximal activity, pH 9.3: about 90% of maximal activity <5>'],\n",
       "             '2.1.1.35': ['#3# 25% inhibition of enzyme activity at pH 7.8 and 8.8 <7>'],\n",
       "             '2.1.1.365': ['#1# the enzyme is active between pH 6.5 and 9.5 <1>'],\n",
       "             '2.1.1.37': ['#17# 50% of maximal activity at pH 7.6 and at pH 8.7 <12>',\n",
       "              '#9# about 50% of maximal activity at pH 6 and pH 9 <17>'],\n",
       "             '2.1.1.39': ['#2# pH 6.5: about 20% of activity maximum, pH 8.0: about 15% of activity maximum <2>'],\n",
       "             '2.1.1.4': ['#18# both recombinant N- and C-terminal fusions of the enzyme, no activity at pH 5.4 <35>'],\n",
       "             '2.1.1.40': ['#1# about 50% of activity maximum at pH 6.7 and 7.6 <1>'],\n",
       "             '2.1.1.41': ['#1# less than half-maximal activity above and below <1>',\n",
       "              '#14# half maximum velocities at 6.5 and 9.0 <32>',\n",
       "              '#14# half maximum velocities at 6.5 and 9.0 <32>',\n",
       "              '#14# optimum pH <32>',\n",
       "              '#7# recombinant enzyme <12>'],\n",
       "             '2.1.1.45': ['#11# phosphate buffer <2>',\n",
       "              '#11# Tris buffer <2>',\n",
       "              '#9# activity is negligible below pH 6.0 and above pH 9.0 <1>'],\n",
       "             '2.1.1.46': ['#1# pH 7.5: about 40% of maximal activity (Tris-HCl buffer), pH 10.0: about 25% of maximal activity (glycine-NaOH buffer) <1>'],\n",
       "             '2.1.1.49': ['#3# tryptamine, about half-maximal activity at pH 7.5 and 90% at pH 8.8, transferase B <3>'],\n",
       "             '2.1.1.56': ['#23# methyl transfer by Ecm1 declines sharply below pH 6.0 <27>',\n",
       "              '#3# half-maximal activity at pH 5.0, unstable above pH 8.0 <13>'],\n",
       "             '2.1.1.57': ['#2# about half-maximal activity at pH 6.6 and 8.1 <5>',\n",
       "              '#2# m7G- and A-capped RNA substrates <10>'],\n",
       "             '2.1.1.6': ['#6# about 50% of maximum activity at pH 6.8 and pH 8 <7>'],\n",
       "             '2.1.1.63': ['#18# pH 6.0: about 80% of maximal activity, pH 7.6: about 40% of maximal activity <34>'],\n",
       "             '2.1.1.68': ['#11# more than half-maximal activity at pH 6.5 and 8.0 <13>',\n",
       "              '#11# 37% of maximal activity at pH 7.0, 75% of maximal activity at pH 7.4 <12>',\n",
       "              '#2# about half-maximal activity at pH 5.4 and 7.6 <17>',\n",
       "              '#8# more than half-maximal activity at pH 5.0 and 8.5 <16>'],\n",
       "             '2.1.1.69': ['#1# pH 6.5: about 30% of maximal activity, pH 9.3: about 45% of maximal activity <4>'],\n",
       "             '2.1.1.7': ['#2# about 55% of maximum activity at pH 5.2 and 8.5 <1>',\n",
       "              '#3# pH 5.0: about 50% of maximum activity, pH 9.0: nearly inactive <3>',\n",
       "              '#3# half-maximal activity at pH 5.4 and pH 7.5 <4>'],\n",
       "             '2.1.1.72': ['#10# pH 2.8: about 65% of maximal activity, pH 9.0: about 45% of maximal activity <2>'],\n",
       "             '2.1.1.75': ['#1# about 50% of activity maximum at pH 7.5 and 10, glycine-NaOH buffer <1>',\n",
       "              '#2# the enzyme shows high catalytic activity over the pH range 7.5-9.0 <3>'],\n",
       "             '2.1.1.76': ['#5# about 50% of activity maximum at pH 7.2 and 9.3 <2>',\n",
       "              '#9# recombinant enzyme, over 50% activity within this range <10>'],\n",
       "             '2.1.1.77': ['#25# isoforms PIMT1 and PIMT2 show high activity between pH 5.0 and 9.0 <88>'],\n",
       "             '2.1.1.79': ['#6# about 50% of activity maximum at pH 5.5 and 8.5 <1>'],\n",
       "             '2.1.1.8': ['#4# about 50% of maximum activity at pH 7.3 and pH 10.3 <6>',\n",
       "              '#5# about 50% of maximum activity at pH 6.5 and pH 10.5, N-methylhistamine <3>'],\n",
       "             '2.1.1.80': ['#3# approx. 30% of maximal activity at pH 6.0, approx. 65% of maximal activity at pH 8.0 <3>'],\n",
       "             '2.1.1.82': ['#3# approx. 50% of maximal activity at pH 7.2 and 9.5 <1>'],\n",
       "             '2.1.1.83': ['#1# approx. 50% of maximal activity at pH 7.2 and 9.5 <1>'],\n",
       "             '2.1.1.85': ['#3# about 50% of activity maximum at pH 6.8 and 9.0 <2>'],\n",
       "             '2.1.1.86': ['#2# sharp decrease in activity below pH 6.0 and above pH 7.0 <1>'],\n",
       "             '2.1.1.88': ['#3# activity range, recombinant enzyme <3>'],\n",
       "             '2.1.1.9': ['#12# form III <12>',\n",
       "              '#12# form II <12>',\n",
       "              '#15# with furfuryl thiol as substrate <16>',\n",
       "              '#5# about 50% of maximum activity at pH 6.6 and pH 8.6 <3,5>'],\n",
       "             '2.1.1.91': ['#1# less than 50% of maximal activity below pH 9.0, almost no activity above pH 10.4 <1>'],\n",
       "             '2.1.1.95': ['#19# about 60% activity at pH 6.0, about 50% activity at pH 9.0 <17>'],\n",
       "             '2.1.2.1': ['#10# about 60% activity at pH 2.0, 100% activity at pH 3.0, about 20% activity at pH 4.0 <118>',\n",
       "              '#11# cytosol: about 60% of maximal activity at pH 6, about 70% at 8.5, mitochondria: about 40% of maximal activity at pH 6.0 and 8.5 <9>',\n",
       "              '#22# no activity above <14>',\n",
       "              '#22# about half-maximal activity at pH 7.3 and about 60% of maximal activity at 9.5 <14>',\n",
       "              '#27# about half-maximal activity at pH 6.5 and 9.5 <15>',\n",
       "              '#34# serine synthesis: about 80% of maximal activity at pH 7.0 and 9.0 <17>',\n",
       "              '#34# 3-phenylserine degradation: about half-maximal activity at pH 7.2 and 8.6 <17>',\n",
       "              '#5# activity range <91>',\n",
       "              '#52# activity range, profile overview <107>; #77# maximal activity at pH 7.0, 88% of the maximal activity at pH 7.0-8.0, less than 20% of maximal activity at pH 9.5, nearly no activity below pH 5.5 <110>',\n",
       "              '#6# the catalytic efficiency of the enzyme for tetrahydrofolate is about 29fold greater at pH 6.6 than at pH 8.3 <117>',\n",
       "              '#60# about 20% activity at pH 6.5, about 85% activity at pH 7.0, about 95% activity at pH 8.0, about 75% activity at pH 9.0, about 45% activity at pH 10.0 <130>',\n",
       "              '#78# almost inactive at pH 5.8, less than 10% of maximal activity at pH 6.0, maximal activity at pH 8.0, 80% of maximal activity at pH 7.5 and pH 8.5, inactive at pH 9.5 <115>',\n",
       "              '#85# the enzyme shows more than 40% activity between pH 7.0 and 8.0. The enzyme displays less than 20% of its maximal activity at pH 5.5 and nearly no activity is detected below pH 2.5 <123>'],\n",
       "             '2.1.2.11': ['#3# about 50% of activity maximum at pH 6 and 9, inactive below pH 5 <2>',\n",
       "              '#3# no ketopantoate formation above pH 10 <1>'],\n",
       "             '2.1.2.3': ['#22# more than 60% activity between pH 7.0 and 9.0 <42>',\n",
       "              '#26# pH 6.5: about 50% of maximal activity, pH 9.5: about 50% of maximal activity, AICAR transformylase activity of the bifunctional enzyme <42>'],\n",
       "             '2.1.2.5': ['#6# about 50% of activity maximum at pH 6.2 and 9.0 <4>',\n",
       "              '#7# about 50% of activity maximum at pH 6 and 9.8 <5>'],\n",
       "             '2.1.2.7': ['#2# about half-maximal activity at pH 6.0 and 10.5 <2>'],\n",
       "             '2.1.2.9': ['#4# below: rapid decrease in reaction rate <1>',\n",
       "              '#4# about 60% of maximal activity at pH 6.8 and 8.4 <1>',\n",
       "              '#8# good activity between <11>'],\n",
       "             '2.1.3.1': ['#6# assay at <18>'],\n",
       "             '2.1.3.10': ['#1# sharp decrease in activity from pH 5.5 to pH 4.5 <1>'],\n",
       "             '2.1.3.2': ['#18# steep increase in activity until pH 8.5, decrase above <52>'],\n",
       "             '2.1.3.3': ['#1# 50% of maximal activity at pH 9.0, 70% at pH 7.0 <24>',\n",
       "              '#10# approx. 50% of maximal activity at pH 6.7 and 8.5, triethanolamine/acetate buffer <36>',\n",
       "              '#10# 71% of maximal activity at pH 7.2, 94% of maximal activity at pH 8.3, HEPES/NaOH buffer <36>',\n",
       "              '#23# at 10 mM carbamoyl phosphate, sharp decrease in activity above <42>',\n",
       "              '#30# approx. 50% of maximal activity at pH 9.0 <29>',\n",
       "              '#32# sharp decrease in activity below pH 9.0 <20>',\n",
       "              '#45# activity range, anabolic and catabolic reactions <100>',\n",
       "              '#5# nearly constant activity in this range, half-maximal activity at pH 7.0, negliable activity at pH 6.0 <34>',\n",
       "              '#5# approx. 50% of maximal activity at pH 6.0 <34>',\n",
       "              '#9# approx. 85% of maximal activity at pH 6.8, approx. 25% of maximal activity at pH 8.5 <40>'],\n",
       "             '2.1.3.6': ['#5# activity range, anabolic reaction <8>',\n",
       "              '#5# activity range, catabolic reaction <8>'],\n",
       "             '2.1.3.7': ['#4# about 75% of maximal activity at pH 5.9 and 7.4 <1>'],\n",
       "             '2.1.4.1': ['#22# above pH 9, the activity dropped rapidly for all enzymes <32>'],\n",
       "             '2.2.1.1': ['#3# about half-maximal activity at pH 5.8 and 9.5 <6>',\n",
       "              '#40# more than 50% activity between pH 5.5 and 8.5 <111>',\n",
       "              '#5# more than 80% of maximum activity <102>',\n",
       "              '#5# more than 50% of maximum activity <102>',\n",
       "              '#6# about half-maximal activity at pH 6.8 and 9.0 <17>',\n",
       "              '#6# about half-maximal activity at pH 7.1 and 9.5 <15>',\n",
       "              '#7# about half-maximal activity at pH 6.5 and about 60% of maximal activity at pH 10.5 <19>',\n",
       "              '#7# about half-maximal activity at pH 6.5 and 10.2 <18>'],\n",
       "             '2.2.1.2': ['#32# native enzyme, 80% of maximal activity within this range <23>; #27# native enzyme <23>',\n",
       "              '#3# about 50% of activity maximum at pH 6.3 and 9.5, isoenzyme I, II, III <1,6>'],\n",
       "             '2.2.1.3': ['#1# about 80% of maximal activity at pH 6.5 and 7.5 <2>'],\n",
       "             '2.2.1.6': ['#107# activity range, overview <141>',\n",
       "              '#12# about 60% of maximal activity at pH 6.5 and pH 7.5, AHAS II <2>',\n",
       "              '#12# pH 5.5: about 80% of maximal activity, pH 8.0: about 55% of maximal activity, AHAS I <2>; #46# more than 50% between pH 5.5 and 8.0 <137>',\n",
       "              '#19# phosphate buffer <54>',\n",
       "              '#26# pH 6.0: about 50% of maximal activity, pH 6.9: about 30% of maximal activity <25>',\n",
       "              '#3# pH 6.0: about 55% of maximal activity, pH 8.0: about 50% of maximal activity <20>',\n",
       "              '#46# activity range, overview <137>',\n",
       "              '#5# pH 6.0: about 25% of maximal activity, pH 7.5: about 65% of maximal activity <45>; #32# the enzyme exhibits more than 60% activity in the range of pH 5.5-7.5 <164>',\n",
       "              '#58# pH 6.5: about 40% of maximal activity, pH 10.0: about 65% of maximal activity, pH profile <99>'],\n",
       "             '2.2.1.7': ['#26# the enzyme does not show substantial activity below pH 6.5 and above pH 8.5 <49>'],\n",
       "             '2.3.1.1': ['#16# pH 8.5: about 65% of maximal activity, pH 10.0: about 80% of maximal activity <27>',\n",
       "              '#9# 38% of maximal activity at pH 8.0., 25% of maximal activity at pH 9.5 <8>'],\n",
       "             '2.3.1.100': ['#1# 25% of maximal activity at pH 6, 90% of maximal activity at pH 8 <1>'],\n",
       "             '2.3.1.103': ['#1# 50% of maximal activity at pH 7.0 and pH 9.0 <1>'],\n",
       "             '2.3.1.105': ['#3# subcellular fractions R and P3 <3>'],\n",
       "             '2.3.1.107': ['#1# about 50% of maximal activity, pH 9.5: about 55% of maximal activity <5>',\n",
       "              '#1# activity increases from pH 5.5 to 7 and decreases rapidly between pH 9 and 11 <2>'],\n",
       "             '2.3.1.110': ['#3# formation of feruloyltyramine, half-maximal activity at pH 6.5 and pH 9.0 <1>',\n",
       "              '#5# half-maximal activity at pH 6.5 and pH 9.5 <9>'],\n",
       "             '2.3.1.112': ['#1# most active <1>'],\n",
       "             '2.3.1.113': ['#1# most active <1>'],\n",
       "             '2.3.1.115': ['#2# about 50% of activity maximum at pH 6.5 and 9.5 <3>'],\n",
       "             '2.3.1.117': ['#1# 50% of maximal activity at pH 7, 75% of maximal activity at pH 9.0 <2>'],\n",
       "             '2.3.1.122': ['#1# 45% of maximal activity at pH 5.0, 30% of maximal activity at pH 9.0 <1>'],\n",
       "             '2.3.1.123': ['#1# 20% of maximal activity at pH 5.0, 40% of maximal activity at pH 7.0 <1>'],\n",
       "             '2.3.1.13': ['#5# approx. 35% of maximal activity at pH 8.0, approx. 40% of maximal activity at pH 10.0, kidney enzyme <3>'],\n",
       "             '2.3.1.130': ['#1# about 80% of maximal activity at pH 7.1 and 7.6 <1>'],\n",
       "             '2.3.1.131': ['#1# about 80% of maximal activity at pH 6.2 and 8.0 <1>'],\n",
       "             '2.3.1.132': ['#1# about 80% of maximal activity at pH 7.4 and pH 8.2 <1>'],\n",
       "             '2.3.1.133': ['#14# about half-maximal activity at pH 5.5 and 7.5 <4>',\n",
       "              '#20,22# optimal pH range with shikimic acid as the acceptor, highest reaction rates are observed for both hydroxycinnamoyl-CoA derivatives between pH 7.5 and 7.9 using sodium phosphate buffer <15>',\n",
       "              '#21# optimal pH range with malate as the acceptor, highest reaction rates are observed for both hydroxycinnamoyl-CoA derivatives between pH 6.1 and 6.7 using sodium phosphate buffer <15>',\n",
       "              '#7# about 80% of maximal activity at pH 6.3 and 7.7 <1>',\n",
       "              '#8# about half-maximal activity at pH 6.2 in potassium phosphate buffer and 8.5 in Tris-HCl puffer <7>'],\n",
       "             '2.3.1.135': ['#7# 80% of maximum activity <49>'],\n",
       "             '2.3.1.138': ['#3# about 50% of activity maximum at pH 7.5 and 11 <3>',\n",
       "              '#3# about 50% of activity maximum at pH 7.2 and 10 <5>',\n",
       "              '#3# at 7.8 about 50% of activity maximum, at 10.0 activity maximum, caffeoyl-CoA + putrescine <4>'],\n",
       "             '2.3.1.139': ['#1# 75% of maximal activity at pH 6.0, 70% of maximal activity at pH 8.0 <1>'],\n",
       "             '2.3.1.140': ['#2# half-maximal activity at pH 6.5 and pH 8.5 <3>'],\n",
       "             '2.3.1.143': ['#1# 50% of maximal activity at pH 4.0 and pH 8.0 <1>'],\n",
       "             '2.3.1.15': ['#5# approx. half-maximal activity at pH 8.0 and 9.5, membrane-associated enzyme <26>',\n",
       "              '#5# approx. half-maximal activity at pH 7.5 and 9.2, reconstituted enzyme <26>',\n",
       "              '#5# approx. half-maximal activity at pH 7.0 and 8.6 <32>',\n",
       "              '#7# approx. half-maximal activity at pH 6.6 and 8.6 <22>',\n",
       "              '#8# maximal activity at pH 6.6, approx. 30% of maximal activity at pH 8.9 <24>'],\n",
       "             '2.3.1.151': ['#1# pH 6.0: about 20% of maximal activity, pH 8.0: about 45% of maximal activity <1>'],\n",
       "             '2.3.1.153': ['#2# active in this range <1>'],\n",
       "             '2.3.1.157': ['#8# activity range <14>'],\n",
       "             '2.3.1.159': ['#1# isozyme I, assay range <3>',\n",
       "              '#1# isozyme II, assay range <3>'],\n",
       "             '2.3.1.16': ['#14# pH 7.0: no significant activity below, pH 8.0: maximum activity <25>',\n",
       "              '#19# pH 7.5 and 9.5: 50% maximum activity, pH 6.5: no activity <30>',\n",
       "              '#7# pH 7.6: maximum activity, pH 8.3: about 75% of maximum activity <16>'],\n",
       "             '2.3.1.162': ['#1,3# 50% of maximal activity at pH 7.50 and pH 10.0, enzyme inactive at pH 11.0 <2>'],\n",
       "             '2.3.1.167': ['#1# half-maximal aactivity of recombinant enzyme at pH 6.0 and 8.0 <4>'],\n",
       "             '2.3.1.17': ['#3# about 13% of maximal activity at pH 5.2, about 50% of maximal activity at pH 8.0 <1>',\n",
       "              '#6# the enzymatic activity decreases at pH values below pH 7.5. A fit of the Vmax/Km data to a model which assumes that the protonation of a single group leads to loss of activity results in a pK value of 6.8 for a group that must be ionized for the enzyme to remain catalytically active. By contrast, the Vmax profile does not show substantial changes across the pH range <10>'],\n",
       "             '2.3.1.171': ['#1# 50% of maximal activity at pH 6.2 and pH 8.7 <2>',\n",
       "              '#5# 50% of maximal activity at pH 6.4 and at pH 9.7 <3>'],\n",
       "             '2.3.1.172': ['#1# active in the range <2>'],\n",
       "             '2.3.1.175': ['#1# pH 6.0: about 60% of maximal activity, pH 8.5: about 50% of maximal activity <6>'],\n",
       "             '2.3.1.178': ['#5# pH 8: about 50% of maximal activity, pH 10.5: about 70% of maximal activity <5>'],\n",
       "             '2.3.1.181': ['#2# pH 6.0: about 65% of maximal activity, pH 9.0: about 40% of maximal activity <6>'],\n",
       "             '2.3.1.183': ['#14# Catalytic activity reaches a maximum at pH 8 to pH 8.5. In Tris buffer at pH 7 and pH 9.5 20-30% catalytic activity remains. In phosphate buffer, the activity at pH 7 and pH 6.5 is 15% and 10%, respectively. At pH 6 the enzyme shows no activity. <17>',\n",
       "              '#8# no activity at pH 5.0 and pH 10.0 <3>'],\n",
       "             '2.3.1.184': ['#1# sharp drop of activity below pH 8.0 and above pH 10.0, inactive at pH 6.0 and pH 11.0 <5>',\n",
       "              '#1# 10% of maximal acivity at pH 7.0, 65% at pH 7.1, and 50% at pH 8.0 <11>'],\n",
       "             '2.3.1.185': ['#2# pH 7.0: less than 10% of maximal activity, pH 8.0: about 90% of maximal activity, pH 10.0: about 80% of maximal activity <3>'],\n",
       "             '2.3.1.186': ['#2# pH 7.0: about 35% of maximal activity, pH 10.0: about 20% of 20% of maximal activity <3>'],\n",
       "             '2.3.1.188': ['#1# half-maximal activity <1>'],\n",
       "             '2.3.1.19': ['#6# butanoylphosphate formation, 50% of maximal activity at pH 6.5, 80% of maximal activity at pH 8.0 <4>',\n",
       "              '#7# butanoyl-CoA formation, less than 50% of maximal activity above and below <5>',\n",
       "              '#7# butanoylphosphate formation, less than 50% of maximal activity above and below <5>'],\n",
       "             '2.3.1.195': ['#1# pH 6.3: 89% of maximal activity, pH 9.0: about 45% of maximal activity <1>'],\n",
       "             '2.3.1.196': ['#3# pH 6.5: 39% of maximal activity, pH 9.0: 69% of maximal activity <3>',\n",
       "              '#6,7# at least 50% of maximum activity <6>'],\n",
       "             '2.3.1.21': ['#6# L-CPT I <68>'],\n",
       "             '2.3.1.211': ['#4# activity range, profile overview <7>'],\n",
       "             '2.3.1.22': ['#21# maximum activity <36>'],\n",
       "             '2.3.1.23': ['#1# approx. half-maximal activity at pH 5.0 and 10.0 <19>',\n",
       "              '#13# pH 6.5: about 80% of maximal activity, pH 8.0: about 70% of maximal activity <51>',\n",
       "              '#18# activity range, 64% of maximal activity at pH 11.0, 34% at pH 12.0, and 10% at pH 6.0 <92>',\n",
       "              '#6# approx. half-maximal activity at pH 5.5 and 8.5 <23>',\n",
       "              '#6# pH 6.0: about 55% of maximal activity, pH 10.0: about 80% of maximal activity, Golgi-associated enzyme <46>',\n",
       "              '#9# almost no activity at pH 6.0 and pH 9.0 <38>; #6# approx. half-maximal activity at pH 6.0 and 9.6 <27>'],\n",
       "             '2.3.1.24': ['#1,2# about 50% of maximal activity within this range <12>',\n",
       "              '#14# optimal pH <4>',\n",
       "              '#21# over 50% ox maximal activity at pH 6.5-8.0, inactive at pH 4.5, maximally active at pH 7.5 <14>',\n",
       "              '#22# over 70% of maximal activity within this range <29>',\n",
       "              '#8# more than 70% of maximal activity between pH 7.0 and pH 8.0 <1>'],\n",
       "             '2.3.1.247': ['#2# 50% of maximal activity at pH 5.2 and pH 8.1 <1>'],\n",
       "             '2.3.1.255': ['#8# activity range, inactive below <36>'],\n",
       "             '2.3.1.26': ['#1# 2fold higher activity at pH 5.6 than at pH 7.4 <14>',\n",
       "              '#6# pH 6.5: 55% of activity maximum, pH 8.4: 75% of activity maximum <11>'],\n",
       "             '2.3.1.272': ['#1# the forward and the reverse reactions show optimum pH values at about 6.3-7.2 <1>'],\n",
       "             '2.3.1.275': ['#6# the enzyme is active between pH 6.5 and pH 9.5 <6>'],\n",
       "             '2.3.1.28': ['#1# pH 6.0: about 55% of maximal activity, pH 8.8: about 85% of maximal activity <2>',\n",
       "              '#17# about 50% of maximal activity at pH 5.0 and 9.0 <23>'],\n",
       "             '2.3.1.29': ['#4# about 30% of maximal activity at pH 6.0 and 43% at 9.0 <3>',\n",
       "              '#7# absorbtion spectrum does not change significantly <9>'],\n",
       "             '2.3.1.297': ['#1,2# about 50% of maximal activity within this range <7>'],\n",
       "             '2.3.1.30': ['#47# 50% of maximal activity at pH 5.0 and pH 10.0 <63>; #4# over 40% of maximal activity within this range <70>'],\n",
       "             '2.3.1.302': ['#1# more than 50% of maximum activity <6>'],\n",
       "             '2.3.1.304': ['#15# pH 7.0: about 40% of maximal activity, pH 8.8: about 65% of maximal activity, Tris/HCl buffer <19>',\n",
       "              '#36# pH 6.5: about 30% of maximal activity, pH 8.5: about 85% of maximal activity <5>'],\n",
       "             '2.3.1.31': ['#8# more than half-maximal activity over this range <4>'],\n",
       "             '2.3.1.32': ['#4# about half-maximal activity at pH 8.3, about 70% of maximal activity at pH 9.2 <1>'],\n",
       "             '2.3.1.33': ['#2# pH 7.0: 63% of activity maximum, pH 7.7: activity maximum <1>'],\n",
       "             '2.3.1.37': ['#10# about 50% of activity maximum at pH 5.8 and pH 8.6 <28>',\n",
       "              '#35# poorly active at <91>'],\n",
       "             '2.3.1.38': ['#1# about 60% of maximal activity at pH 5.0 and pH 8.5 <6>',\n",
       "              '#11# about half-maximal activity at pH 6.0 and pH 7.5 <11>',\n",
       "              '#4# about half-maximal activity at pH 7.5 and about 75% of maximal activity at pH 9.1 <7>'],\n",
       "             '2.3.1.39': ['#1# about half-maximal activity at pH 5.0, about 80% of maximal activity at pH 10.0 <12>',\n",
       "              '#17# about half-maximal activity at pH 6.2, about 65% of maximal activity at pH 9.8 <5>',\n",
       "              '#7# maximal activity <17>'],\n",
       "             '2.3.1.4': ['#10# approx. 90% of maximal activity at pH 6.0, approx. 95% of maximal activity at pH 8.0 <1>'],\n",
       "             '2.3.1.40': ['#1# about half-maximal activity at pH 5.0, about 60% of maximal activity at pH 10.0 <1>'],\n",
       "             '2.3.1.41': ['#14# synthase II, about half-maximal activity at pH 5.6 and pH 7.2 <15>',\n",
       "              '#14# synthase I, about half-maximal activity at pH 5.9 and pH 6.8 <15>',\n",
       "              '#31# synthase III <31>',\n",
       "              '#4# synthase I, about half-maximal activity at pH 6.3 and pH 8.0 <11>',\n",
       "              '#4# sharp drop in activity below pH 6.5 and above pH 8.0 <7>',\n",
       "              '#4# synthase II, about half-maximal activity at pH 5.2 and pH 7.6 <11>'],\n",
       "             '2.3.1.42': ['#2# detectable activity in this range <9>',\n",
       "              '#2# about half-maximal activity at pH 6.0 and pH 9.0, in the presence of asolectin <10>',\n",
       "              '#5# about half-maximal activity at pH 5.4 and pH 7.3 <18>',\n",
       "              '#6# about half-maximal activity at pH 6.5 and pH 8.5 <16>',\n",
       "              '#6# about half-maximal activity at pH 5.5 and pH 9.5 <4>',\n",
       "              '#6# about half-maximal activity at pH 5.1 and pH 8.5 <17>',\n",
       "              '#7# about half-maximal activity above <5>'],\n",
       "             '2.3.1.46': ['#2# crude extract, about 40% of maximal activity at pH 6.9, about 80% of maximal activity at pH 8.0 <1>',\n",
       "              '#2# pH profile of mutant K47R <8>',\n",
       "              '#3# 25°C, histidine-tagged enzyme, 0-50 mM acetic acid: 0 mM = pH 7.5, 10 mM = 7.0, 20 mM = 6.6, 30 mM = 6.2, 40 mM = 5.8, 50 mM = 5.4, the recombinant enzyme shows reduced activity at 25°C without great change at about 200 micromol/min/mg protein, about 3.5times higher activity at 40°C <10>; #2# 25°C, histidine-tagged enzymes, 0-50 mM acetic acid: 0 mM = pH 7.5, 10 mM = 7.0, 20 mM = 6.6, 30 mM = 6.2, 40 mM = 5.8, 50 mM = 5.4, sharp decrease of wild-type enzyme activity above 30 mM acetic acid, no activity at 50 mM, without acid about 2900 micromol/min/mg protein, the mutant strain 333 (S61T/E213V/I229T/N267D/N271K) enzyme retains almost 50% activity at 50 mM acetic acid, activity decreases above 30 mM from about 2000 micromol/min/mg protein, the mutants N267D and I229T do not greatly change their activity, about 2000-1800 micromol/min/mg protein, and about 2500-2000 micromol/min/mg protein, respectively <10>'],\n",
       "             '2.3.1.47': ['#6# about half-maximal activity at pH 6.5 and pH 7.5 <6>'],\n",
       "             '2.3.1.48': ['#5# PCAF protein <37>',\n",
       "              '#60# activity range, inactive below <203>',\n",
       "              '#7# about 50% of activity maximum at pH 6.6 and pH 8.4 <12>',\n",
       "              '#8# less than 10% of activity maximum below pH 6.5 and above pH 9.5 <10>'],\n",
       "             '2.3.1.49': ['#1# 25% of maximal activity at pH 6, 95% of maximal activity at pH 8 <1>'],\n",
       "             '2.3.1.5': ['#17# 80% activity <111>',\n",
       "              '#17# 80% activity <111>',\n",
       "              '#17# about half-maximal activity at pH 6.2 and 7.3 <14>',\n",
       "              '#19# about 25% of maximal activity pH 5.0 and pH 9.5 with 2-aminofluorene as substrate, about 65% of maximal activity at pH 5.0 and at pH 9.5 with p-aminobenzoate as substrate <66>',\n",
       "              '#46# about half-maximal activity at pH 9.3, maximal at 9.5 <17>',\n",
       "              '#55# pH profiles with cofactors acetyl-CoA and 4-nitrophenyl acetate, overview <73>',\n",
       "              '#88# more than 50% activity between pH 5.5 and 6.5 <155>'],\n",
       "             '2.3.1.50': ['#4# about 50% of activity maximum at pH 6.7 and pH 8.6 <9>'],\n",
       "             '2.3.1.51': ['#3# about half-maximal activity at pH 6.6 and 8.6 <6>',\n",
       "              '#54# over 50% of maximal activity within this range, profile overview <75>; #51# pH 7.5: about 70% of maximal activity, pH 11.0: about 50% of maximal activity <75>',\n",
       "              '#55# recombinant enzyme, over 90% of maximal activity at pH 6.5-7.5, 50% at pH 8.0, below 10% at pH 9.0 <78>'],\n",
       "             '2.3.1.54': ['#20# 40% of activity maximum at pH 6.5, between pH 7.0-8.3 80% of maximum activity <9>',\n",
       "              '#33# activity assay at, wild-type and mutant protein <39>'],\n",
       "             '2.3.1.56': ['#4# about 80% of maximal activity at pH 6.8 and pH 8.2 <4>',\n",
       "              '#4# about half-maximal activity at pH 5.6 and pH 9.6, nucleic acid adduct formation assay <6>'],\n",
       "             '2.3.1.57': ['#14# activity monitored every 0.3 pH units, bell shaped pH activity profile, depending on ionization state of two groups with apparent pKa values of 7.27 and 8.87 <32>',\n",
       "              '#18# about 50% of maximum activity <53>',\n",
       "              '#9# high conversion rates in the range of <4>'],\n",
       "             '2.3.1.59': ['#1# about half-maximal activity at pH 6.4 and 7.6, gentamicin C <2>'],\n",
       "             '2.3.1.64': ['#1# approx. half-maximal activity at pH 7.0 and 8.0 <2,3>',\n",
       "              '#3# half-maximal activity at pH 6.9 and pH 8.1 <4>'],\n",
       "             '2.3.1.65': ['#3# taurine: about half-maximal activity at pH 5.2 and pH 8.2, glycine: 80% of maximal activity at pH 8.2 <4>',\n",
       "              '#4# about 70% of maximal activity at pH 6.0, about half-maximal activity at pH 9.0, inactivation rate is similar for both taurine and glycine <5>'],\n",
       "             '2.3.1.66': ['#1# 10% of maximal activity at pH 7, maximal activity at pH 9 <1>'],\n",
       "             '2.3.1.67': ['#6# neuronal nuclear fraction and microsomal fraction <19>; #2# pH 6.0: about 80% of maximal activity, pH 9.0: about 70% of maximal activity <27>'],\n",
       "             '2.3.1.69': ['#1# pH 7.5: about 50% of activity maximum, increase of activity up to pH 9.5, pH 10: about 40% of activity maximum <2>'],\n",
       "             '2.3.1.7': ['#17# activity declines rapidly on either side of the optimum pH <26>'],\n",
       "             '2.3.1.75': ['#34# more than 50% activity between pH 6.0 and 8.0 <30>'],\n",
       "             '2.3.1.76': ['#4# maximum esterification between pH 7-8 <1>',\n",
       "              '#6# decreasing activity below pH 5 <10>'],\n",
       "             '2.3.1.78': ['#1# half-maximal activity at pH 5.0 and pH 7.0 <1>',\n",
       "              '#2# pH 5.0: about 30% of activity maximum, pH 5.5: maximum activity above <5>'],\n",
       "             '2.3.1.8': ['#12# about 50% of activity maximum at pH 6.6 pH and 8.2 <3>',\n",
       "              '#12# pH 6.8: about 25% of activity maximum, pH 8.6: about 90% of activity maximum <4>',\n",
       "              '#13# 90% loss of activity at pH 6.5, 25% loss of activity at pH 7.8 <34>',\n",
       "              '#13# pH 7.0: about 50% of maximal activity, pH 8.5: about 60% of maximal activity <40>',\n",
       "              '#18# pH 7.0: about 10% of activity maximum, pH 8.7: about 20% of activity maximum <17>'],\n",
       "             '2.3.1.82': ['#2# pH 5.3: about 50% of maximal activity, pH 7.6: about 20% of maximal activity, kanamycin A <2>'],\n",
       "             '2.3.1.83': ['#1# approx. half-maximal activity at pH 4.8 and 7.5 <1>'],\n",
       "             '2.3.1.84': ['#15# 75% of maximum activity <37>',\n",
       "              '#9# approx. 40% of maximal activity at pH 5.0, 80% of maximal activity at pH 8.5 <1>'],\n",
       "             '2.3.1.87': ['#37# pH 7.0: abut 40% of maximal activity, pH 8.0: about 50% of maximal activity <38>',\n",
       "              '#7# pH 6.5: about 30% of maximal activity, pH 8.0: about 50% of maximal activity <49>'],\n",
       "             '2.3.1.9': ['#12# approx. 80% of maximal activity at pH 6.0, approx. 40% at pH 9.0 <10>',\n",
       "              '#12# approx. 50% of maximal activity at pH 6.5 and pH 9.2, thiolysis <10>',\n",
       "              '#23# 69% of maximal activity at pH 9.0 <25>',\n",
       "              '#37# significant decrease in activiy below pH 7.0 or above pH 9.0 <53>'],\n",
       "             '2.3.1.90': ['#1# about half-maximal activity at pH 5.5 and pH 7.5 <3>'],\n",
       "             '2.3.1.91': ['#3# about half-maximal activity at pH 5.3 and pH 9.0 <2>'],\n",
       "             '2.3.1.92': ['#3# about half-maximal activity at pH 5.5 and 6.5 <3>',\n",
       "              '#3# about half-maximal activity at pH 5.7 and 6.8, various buffer systems <7>'],\n",
       "             '2.3.1.93': ['#1# half-maximal activity at pH 6.5, maximal activity at pH 8.0 <1>'],\n",
       "             '2.3.1.97': ['#4# about half-maximal activity above pH 9.0 and below pH 7.0, inactive at pH 6.0 <2>'],\n",
       "             '2.3.1.98': ['#1# 50% of maximal activity at pH 4.2 and pH 6.2 <1>'],\n",
       "             '2.3.1.99': ['#6# about 80% of maximal activity at pH 6.9 and 8.0 <1>',\n",
       "              '#9# about half-maximal activity at pH 5.6 and 7.9 <2>'],\n",
       "             '2.3.2.10': ['#1# L-seryl-tRNA, about half-maximal activity at pH 6.1 and about 90% of maximal activity at pH 8.6 <1>',\n",
       "              '#1# L-alanyl-tRNA, about 80% of maximal activity at pH 5.2 and pH 8.6 <1>'],\n",
       "             '2.3.2.13': ['#11# activity range <218>',\n",
       "              '#36# pH 5.0: about 80% of maximal activity, pH 8.0: about 80% of maximal activity <125>; #53# activity range of isozyme MTG-TX <250>',\n",
       "              '#4# transamidating activity of guinea pig TG2 was not significantly inhibited at pH 8.3 but minimal at pH 9.5 <174>',\n",
       "              '#41# 80-100% of maximal activity within this range <238>',\n",
       "              '#47# pH 6.5: about 65% of maximal activity, pH 10.0: about 60% of maximal activity <130>',\n",
       "              '#55,72# activity range, low activity at pH 2.0, high activity at pH 7.4-8.5, maximal activity at pH 8.5, 60% of maximal activity at pH 10.0, profile overview <237>',\n",
       "              '#71# activity range at 16°C, recombinant enzyme <236>'],\n",
       "             '2.3.2.15': ['#27# activity range <64>'],\n",
       "             '2.3.2.2': ['#10# about half-maximal activity at pH 7.3, maximal activity at pH 8.5 <27>',\n",
       "              '#12# no significant activities are detected below pH 7.0 or above pH 10.0 <214>',\n",
       "              '#21# active at, profile overview <223>; #62# over 60% of maximal activity within this range for the immobilized enzyme, profile overview <229>',\n",
       "              '#25# pH 6.0, about 55% of maximal activity, pH 9.0: about 80% of maximal activity, enzyme form GGT A <94>; #25# pH 6.0, about 55% of maximal activity, pH 9.0: about 80% of maximal activity, enzyme form GGT B <94>; #23# pH 6.0: about 40% of maximal activity, pH 9.0: about 70% of maximal activity, hydrolysis of 5-L-glutamyl-4-nitroanilide <99>',\n",
       "              '#29# about half-maximal activity at pH 6.5 and 9.5 <51>',\n",
       "              '#3# pH 7.5: about 40% of maximal activity, pH 10.0: about 35% of maximal activity, glutaminase activity of mutant enzyme D445A <83>',\n",
       "              '#3# 50% of maximal activity within this range, recombinant wild-type free enzyme, over 80% of maximal activity within this range, recombinant wild-type immobilized enzyme modified MTWs-GGT-Phar, the activity of recombinant immobilized enzyme MTWs-GGT is abolished above pH 10.0 <226>',\n",
       "              '#3# about half-maximal activity at pH 7.5, about 90% of maximal activity at pH 9.0 <47>',\n",
       "              '#3# pH 8.0: about 60% of maximal activity, pH 11: about 75% of maximal activity, transpeptidation with 5-L-glutamyl-4-nitroanilide and glycylglycine <82>; #3# pH 8.0: about 60% of maximal activity, pH 8.5-11: optimum, hydrolysis of 5-L-glutamyl-4-nitroanilide <82>',\n",
       "              '#5# about 35% activity at pH 7.0, about 55% activity at pH 8.0, about 85% activity at pH 9.0, 100% acitivity at pH 10.0, about 75% activity at pH 11.0 <197>; #3# the enzyme is active between pH 7.0 and 11.0 <210>',\n",
       "              '#25# pH 7.0, about 50% of maximal activity, pH 9.0: about 80% of maximal activity, enzyme form GGT I <94>; #25# pH 7.0, about 50% of maximal activity, pH 9.0: about 80% of maximal activity, enzyme form GGT II <94>; #3# pH 7.0: about 40% of maximal activity, pH 9.0: 65% of maximal activity <101>',\n",
       "              '#48# pH 4.0: about 50% of maximal activity, pH 13: about 40% of maximal activity <92>',\n",
       "              '#55# activity range, over 60% of maximal activity at pH 7.5-10.5, profile overview <218>',\n",
       "              '#6# pH 6.5: about 50% of maximal activity, pH 9.0: about 90% of maximal activity <84>',\n",
       "              '#60# more than 40% of maximal activity, hydrolysis of gamma-glutamyl-4-nitroanilide <203>',\n",
       "              '#69# activity range, recombinant enzyme <232>; #63# active at, inactive above, profile overview <224>',\n",
       "              '#64# active at, inactive below, profile overview <224>',\n",
       "              '#7# about 70% of maximal activity at pH 8.1, about 75% of maximal activity at pH 9.6 <17>',\n",
       "              '#70# BaGGT469 shows the highest activity towards L-gamma-glutamyl-4-nitroanilide with diglycine acceptor in 50 mM borate-NaOH buffer at pH 10.0. The enzymatic activity at the alkaline pH range of pH 8.0-9.0 is 92.6-95.9% relative to that at optimum pH. BaGGT469 is much less active in the acidic pH range from pH 4.0-6.0 <235>',\n",
       "              '#75# optimum activity at pH 7.0, 80% of maximal activity at pH 6.5-8.0, and residual activity sharply decreased to less than 50% at above pH 10.0 or below pH 4.0 <233>'],\n",
       "             '2.3.2.35': ['#5,6# the enzyme is active between pH 7.0 and 9.0 <6>'],\n",
       "             '2.3.2.5': ['#6# pH 7.5: about 55% of maximal activity, pH 9.3: about 45% of maximal activity, enzyme from spleen <4>',\n",
       "              '#6# pH 6.5: about 50% of maximal activity, pH 8.3: about 40% of maximal activity, pituitary enzyme <4>'],\n",
       "             '2.3.2.6': ['#1# about 80% of maximal activity at pH 7.0 and 9.0 <2>'],\n",
       "             '2.3.2.7': ['#1# about half-maximal activity at pH 8.5 and 9.4 <1>',\n",
       "              '#2# more than 40% activity between pH 6.0 and 11.0 <2>'],\n",
       "             '2.3.2.8': ['#6# about half-maximal activity at pH 8.0, about 80% of maximal activity at pH 9.8 <3>'],\n",
       "             '2.3.2.9': ['#2# approx. half-maximal activity at pH 6.6 and 7.8 <1>'],\n",
       "             '2.3.3.10': ['#28# pH 7.5: about 60% of maximal activity, pH 9.5: about 30% of maximal activity <35>'],\n",
       "             '2.3.3.12': ['#1# pH 7: about 60% of maximal activity, inactive at pH 9.5 <2>'],\n",
       "             '2.3.3.13': ['#16# pH 7.1: about 50% of maximal activity, pH 10.0: about 80% of maximal activity <29>',\n",
       "              '#9# pH 6.0: about 50% of maximal activity, pH 9.5: about 80% of maximal activity <25>'],\n",
       "             '2.3.3.15': ['#3# assay at <3>'],\n",
       "             '2.3.3.16': ['#74# about 30% of maximal activity at pH 6 and at pH 10 <53>',\n",
       "              '#88# nearly 50% of the activity is retained at pH 5.0 or 9.0 <95>'],\n",
       "             '2.3.3.17': ['#3# 50% activity at pH 6.3 and 9.0 <9>',\n",
       "              '#4# 50% activity at pH 7.4 and 9.7 <10>'],\n",
       "             '2.3.3.3': ['#2# approx. 50% activity at pH 7.0 <1>'],\n",
       "             '2.3.3.4': ['#1# 75% of maximal activity at pH 7.5 and 9 <1>'],\n",
       "             '2.3.3.5': ['#28,36# maximum activity <20>'],\n",
       "             '2.3.3.8': ['#18# about 50% of maximal activity at pH 7.3 and at pH 9.5 <6>',\n",
       "              '#19# pH 6.9: about 45% of maximal activity, pH 8.3: about 50% of maximal activity <17>',\n",
       "              '#26# about 50% of maximal activity at pH 7.0 and at pH 8.8 <21>',\n",
       "              '#28# more than 80% of maximum activity within this range <44>'],\n",
       "             '2.3.3.9': ['#11# 87% activity <54>',\n",
       "              '#11# 85% activity <54>',\n",
       "              '#14# pH 7.0: about 40% of maximal activity, pH 10.0: about 45% of maximal activity <65>',\n",
       "              '#30# pH 6.8 and 9.7: about 50% of maximal activity, phosphate-citrate-borate buffer <10>',\n",
       "              '#33# pH 7.0: about 80% of maximal activity, pH 8.7: about 60% of maximal activity, enzyme form MSL and MSH <4>',\n",
       "              '#37# pH 7.5: about 55% of maximal activity, pH 10.5: about 25% of maximal activity, enzyme form MSL and MSH <22>',\n",
       "              '#4# pH 5.0: about 35% of maximal activity, pH 9.5: about 70% of maximal activity <14>',\n",
       "              '#42,43# significant reduction of specific activity at pH 7.0 and below <37>',\n",
       "              '#47# malate synthase reaction <41>'],\n",
       "             '2.4.1.1': ['#14# approx. half-maximal activity at pH 5.7 and 8 <40>',\n",
       "              '#15# approx. half-maximal activity at pH 4.2 and 7.6, fast isozyme <49>',\n",
       "              '#15# approx. half-maximal activity at pH 4 and pH 8, slow isozyme <49>; #70# pH 4.0: about 60% of maximal activity, pH 8.0: about 50% of maximal activity <141>',\n",
       "              '#17# approx. half-maximal activity at pH 5.8 and 7.6, b-isozyme <23>',\n",
       "              '#17# approx. half-maximal activity at pH 5.8 and 7.2, a-isozyme <23>',\n",
       "              '#18# approx. half-maximal activity at pH 5.1 and 7.5 <14>',\n",
       "              '#21# approx. half-maximal activity at pH 5.4 and pH 8.2 <17>',\n",
       "              '#23# approx. half-maximal activity at pH 5.6 and pH 7.5, imidazole buffer, phosphorolysis <31>',\n",
       "              '#23# maximal activity at pH 7, approx. 60% of maximal activity at pH 8.1, glycylglycine buffer, phosphorolysis <31>',\n",
       "              '#26# approx. 70% of maximal activity at pH 5.2 and approx. half-maximal activity at pH 7.5 <24>',\n",
       "              '#26,51# approx. half-maximal activity at pH 5.2 and 7.2 <21,45>',\n",
       "              '#33# 50% of maximal activity at pH 6.7 and 7.9, direction of glucose-1-phosphate formation <121>',\n",
       "              '#33# 50% of maximal activity at pH 5.4 and 7.9, direction of polysaccharide synthesis <121>',\n",
       "              '#34# approx. half-maximal activity at pH 5.7 and 7.2 <15>',\n",
       "              '#4# approx. half-maximal activity at pH 6.0 and 7.5 <13>; #78# sharp decline in activity below pH 6.0 and above pH 7.5 <178>',\n",
       "              '#41# activity range <188>',\n",
       "              '#58# approx. half-maximal activity at pH 5.6 and approx. 65% of maximal activity at pH 8, citrate/phosphate-buffer <18>',\n",
       "              '#58# approx. 70% of maximal activity at pH 6 and about half-maximal activity at pH 8.6, Tris-buffer <18>'],\n",
       "             '2.4.1.10': ['#1# activity range, free and immobilized enzymes <67>',\n",
       "              '#1# 26% and 18% decrease in levansucrase activity is noticed at pH 4.2 and 9.2, respectively <106>',\n",
       "              '#10# pH activity profile, overview <69>',\n",
       "              '#12# activity range of free and immobilized enzymes, with higher activity of the immobilized enzyme, profiles, overview <62>',\n",
       "              '#15# activity range, no activity below pH 3.5 and above pH 6.5, profile, overview <83>; #71# the maximum enzymatic activity of the enzyme is obtained in the pH range of 5.0 to 6.0. The enzyme shows 60% activity at pH 4.0, while a significant loss in the enzymes activity is observed towards the basic pH <100>',\n",
       "              '#16# about half-maximal activity at pH 4.8 and pH 6.8 <23>',\n",
       "              '#20# the enzyme activity is greatly reduced at pH below 4.0 or above pH 7.0 <105>',\n",
       "              '#35# almost no activity under pH 3.0 and above pH 10.0 <6>',\n",
       "              '#37# about 85% activity at pH 5.0, 100% activity at pH 6.0, about 70% activity at pH 7.0 <108>; #73# more than 60% activity between pH 5.0 and 7.0 <94>',\n",
       "              '#39# recombinant enzyme <34>',\n",
       "              '#41# pH 3.5: about 55% of maximal activity, pH 8.3: about 60% of maximal activity, levan synthesis <44>',\n",
       "              '#43# pH 4.0: about 50% of the optimal activity, pH 6.2: about 50% of maximal activity. A pH above 7 leads to a complete inhibition of the levansucrase <41>',\n",
       "              '#45# 50% of the maximal activity are detected at pH 3 and pH 5.5, residual activity is measured at pH 7.5 <55>',\n",
       "              '#5# the enzyme activity significantly decreases at pH values below 5.6 and above 7.6 <53>; #5# wild-type, 50 mM Sorensen\\x92s phosphate buffer at 37°C, containing 500 mM sucrose using an enzyme concentration of 7.36 mg/l <53>',\n",
       "              '#56# activity range, profile overview <80>',\n",
       "              '#70# the best pH range for activity is between 5.5 and 6.5. Above pH 8.0 the enzyme shows an almost complete inhibition <101>',\n",
       "              '#72# more than 50% activity between pH 5.5 and 8.5 <107>',\n",
       "              '#76# the enzyme remains relative activity in the pH range of from 6.5 to 8.0. The enzyme retains more than 80% of initial activity when pH varying from 6.0 to 7.5 at 45°C, but loses almost 60% transfructosylation activity when the pH shifts to lower than 5.5 or higher than 8.0 <98>',\n",
       "              '#9# native and recombinant enzymes, 50% of maximal activity at pH 4.0 and pH 8.5 <89>'],\n",
       "             '2.4.1.100': ['#2# pH 5.0: about 70% of maximal activity, pH 8.0: about 65% of maximal activity <4>'],\n",
       "             '2.4.1.101': ['#24# activity decreases to less than 15% of the maximum at pH values lower than 5.0 and higher than 8.5 <46>',\n",
       "              '#5# recombinant fusion protein <21>',\n",
       "              '#9# about half-maximal activity at pH 5.1 and about 70% of maximal activity at pH 6.9 <10>'],\n",
       "             '2.4.1.102': ['#6# broad optimum <2>'],\n",
       "             '2.4.1.103': ['#1# about half-maximal activity at pH 6.7 and 7.6 <1>'],\n",
       "             '2.4.1.106': ['#1# pH 6.5: about 40% of maximal activity, pH 8.5: about 30% of maximal activity, reaction with isoorientin <2>'],\n",
       "             '2.4.1.111': ['#4# about half-maximal activity at pH 6.5 and 83% of maximal activity at pH 8.2, Tris buffer <3>',\n",
       "              '#7# about half-maximal activity at pH 6.5 and 8.4, Tris buffer <1>'],\n",
       "             '2.4.1.113': ['#2# half-maximal activity at pH 5.2 and pH 7.5 <2>'],\n",
       "             '2.4.1.114': ['#1# 25% of maximal activity at pH 6, 10% of maximal activity at pH 9 <1>'],\n",
       "             '2.4.1.115': ['#13# pH 6.6: about 40% of maximal activity, pH 9.4: about 50% of maximal activity <7>',\n",
       "              '#3# pH 6.0: about 44% of maximal activity, pH 8.8: about 60% of maximal activity, reaction with cyanidin and UDP-glucose <6>',\n",
       "              '#8# 50% of maximal activity at pH 6.6 and pH 8.0 <2>'],\n",
       "             '2.4.1.116': ['#1# about half-maximal activity at pH 7 and 8.5, reaction with cyanidin-3-rhamnosylglucoside <2>'],\n",
       "             '2.4.1.117': ['#9# pH 5.6: 25% of maximum activity, pH 8.8: about 40% of maximum activity <1>'],\n",
       "             '2.4.1.12': ['#13# pH 6.5: 13% of activity maximum, pH 7.5: 39% of activity maximum, pH 8.3: activity maximum, enzyme is not tested under more alkaline conditions because of instability of UDPglucose <4>'],\n",
       "             '2.4.1.120': ['#2# no activity detected below pH 4.5 and above pH 8.5 <4>',\n",
       "              '#3,11# over 70% of maximal activity at pH 6.0-7.5, 30% at pH 8.5, inactive at pH 9.0 <14>',\n",
       "              '#4# pH 5.0: 85% of maximal activity, pH 7.0: 63% of maximal activity <3>'],\n",
       "             '2.4.1.121': ['#2# pH 7.3-7.6: maximal activity, no activity below pH 6.7 <3>'],\n",
       "             '2.4.1.122': ['#8,11# assays at <10>'],\n",
       "             '2.4.1.123': ['#8# about half-maximal activity at pH 5 and 7, at 7 mM Mn2+ <1>',\n",
       "              '#8# about half-maximal activity at pH 5.1 and 7.8, at 2.0 mM Mn2+ <1>',\n",
       "              '#8# about half-maximal activity at pH 5.2 and 8.6, at 0.2 mM Mn2+ <1>'],\n",
       "             '2.4.1.127': ['#1# about 50% of maximal activity at pH 6.0 and pH 8.5, 70 mM Tris-sodium maleate buffer <1>'],\n",
       "             '2.4.1.13': ['#11# about half-maximal activity at pH 7.5 and about 90% of maximal activity at pH 10.0, sucrose synthesis <19>',\n",
       "              '#11# about 90% of maximal activity at pH 5.0 and about half-maximal activity at pH 8.2, sucrose cleavage <19>',\n",
       "              '#2# 96% of maximal activity at pH 6.2 and 6.6 <24>',\n",
       "              '#20# about half-maximal activity at pH 5.0 and 10.0, sucrose synthesis <22>',\n",
       "              '#20# about half-maximal activity at pH 5.0 and 8.0, sucrose cleavage <22>; #86# isoform SUS1 shows about 60% activity at pH 5.0, 100% activity at pH 5.5, about 90% activity at pH 6.5, about 40% activity at pH 7.5, and about 33% activity at pH 8.0. Isoform SUS2 shows about 50% activity at pH 5.0, 100% activity at pH 5.5, about 95% activity at pH 6.5, about 90% activity at pH 7.5, and about 70% activity at pH 8.0 <133>',\n",
       "              '#25# about 80% of maximal activity at pH 5.5 and about half-maximal activity at pH 9.0, sucrose cleavage <14>',\n",
       "              '#25# about half-maximal activity at pH 5.6 and about 90% of maximal activity at pH 10.0, sucrose synthesis <14>',\n",
       "              '#54# sucrose cleavage, activities with CDP, GDP, TDP, and UDP only occur at pH 7.0-8.0 and are very low at pH 9.0, while the activity with ADP is highest at pH 8.0 and higher than 60% of maximal activity in the pH range of 7.0-9.0 <114>',\n",
       "              '#59,61# about 10-15% of maximum activity <112>',\n",
       "              '#96,97,98,100# over 40% of maximal enzyme activity within this range <113>'],\n",
       "             '2.4.1.132': ['#10# activity range, donor substrate specificity, profile overview <21>'],\n",
       "             '2.4.1.135': ['#3# about 50% of maximal activity at pH 6.5 and pH 8.5 <2>'],\n",
       "             '2.4.1.136': ['#1# 50% of maximal activity at pH 5.5 and 8.5 <2>',\n",
       "              '#1# 50% of maximal activity at pH 4.7 and 9.0 <1>'],\n",
       "             '2.4.1.139': ['#1# 30% of maximal activity at pH 6.2, 45% of maximal activity at pH 7.5 <1>'],\n",
       "             '2.4.1.14': ['#5,26# very low activity at pH 4.5 <3>'],\n",
       "             '2.4.1.140': ['#1# pH 4.0: about 75% of maximal activity, pH 7.2: about 60% of maximal activity <1>',\n",
       "              '#1# about 50% of activity maximum at pH 4.6 and 6.4 <2>',\n",
       "              '#3# activity profile, overview <23>'],\n",
       "             '2.4.1.143': ['#16# the enzyme has an optimum pH of 6.7, with a broad shoulder of high activity from pH 7.1 to 8.1 <35>'],\n",
       "             '2.4.1.145': ['#1# depends on Triton X-100 concentration <2>'],\n",
       "             '2.4.1.149': ['#5# substantial reaction in the range, 65% of activity at pH 7 <6>'],\n",
       "             '2.4.1.15': ['#11# pH 5.5: about 75% of maximal activity, pH 8.0: about 70% of maximal activity <5>',\n",
       "              '#16# pH 5.0: about 85% of maximal activity, pH 6.0: about 50% of maximal activity <37>',\n",
       "              '#19# the enzyme is highly active at pH values from 5.0 to 8.0 <86>',\n",
       "              '#3# TPS activity is maximally at pH 8.5 with 2.11 U/mg, followed by 2.01 U/mg at pH 9.0, 1.93 U/mg at pH 9.5, and 1.21 U/mg at pH 10.5, but only 0.31 U/mg at pH 4.5 <58>',\n",
       "              '#45# pH 5.0: about 50% of maximal activity, pH 7.0: about 50% of maximal activity <66>',\n",
       "              '#47,48# activity range <75>',\n",
       "              '#55# drastic decrease in enzyme activity is observed at pH values below pH 6.0 and above pH 8.0 <70>',\n",
       "              '#66# more than 50% activity exists at pH 7.0-9.0 in Tris/HCl buffer <89>'],\n",
       "             '2.4.1.152': ['#22# pH range for FucTC is very broad <43>',\n",
       "              '#3# pH 6.0: about 40% of maximal activity, pH 8.8: about 50% of maximal activity, reaction with N-acetyllactosamine <25>'],\n",
       "             '2.4.1.16': ['#12# the enzyme is active between pH 5.0 and 8.0 <147>',\n",
       "              '#4# pH 6.5: about 85% of maximal activity, pH 8.5: about 70% of maximal activity <39>',\n",
       "              '#4# 50% of maximal activity at pH 7.1 and pH 8.3 <22>',\n",
       "              '#47# pH 5.5: 40% of maximal activity, pH 8.0: 95% of maximal activity <13>'],\n",
       "             '2.4.1.160': ['#1# about 50% of maximum activity at pH 7 and 9.5 <1>'],\n",
       "             '2.4.1.165': ['#2# about half-maximal activity at pH 5.8 and 9.5 <1>'],\n",
       "             '2.4.1.167': ['#1# 30% of maximal activity at pH 5.0, 70% of maximal activity at pH 7.0 <1>'],\n",
       "             '2.4.1.17': ['#6# substrate chenodeoxycholic acid <78>'],\n",
       "             '2.4.1.170': ['#1# active over this pH range <6>',\n",
       "              '#5# conversion rate about 10% <5>'],\n",
       "             '2.4.1.172': ['#1# 45% of maximal activity at pH 6.0, 10% of maximal activity at pH 10.5 <1>'],\n",
       "             '2.4.1.173': ['#24# the enzyme has an optimum pH for activity around 8.0 with half maximal activities at 7.0 and above 9.0 <28>'],\n",
       "             '2.4.1.178': ['#1# about 50% of activity maximum <2>',\n",
       "              '#2# about 50% of activity maximum at pH 6 and pH 9.5 <1>'],\n",
       "             '2.4.1.179': ['#1# about 80% of maximal activity at pH 5.3 and about 55% of maximal activity at pH 7.5 <2>',\n",
       "              '#3# about 75% of maximal activity at pH 5 and about half-maximal activity at pH 8.0 <3>'],\n",
       "             '2.4.1.18': ['#22# the enzyme shows at least 80% of maximal activity from pH 5.5 to pH 7.5. The enzyme activity decreases rapidly at pH values lower than 5.5 and higher than 9.0 <134>',\n",
       "              '#24# pH 5.0: about 40% of maximal activity, pH 9.0: about 50% of maximal activity <44>; #79# more than 60% of enzyme activity remains when assayed at pH 5.0-9.0 <133>',\n",
       "              '#35# active at <108>',\n",
       "              '#35# about 60% activity at pH 6.5, about 80% activity at pH 7.0, 100% activity at pH 7.5, more than 90% between pH 8.0, and 11.0 <129>',\n",
       "              '#51,55# iodine assay <75>',\n",
       "              '#67# isozyme BE2 shows nearly half of the maximum activity at pH 10.0 <105>',\n",
       "              '#80# the enzyme shows more than 70% activity between pH 5.5 and 7.0 <123>'],\n",
       "             '2.4.1.184': ['#1# approx. 50% of maximal activity at pH 5.3 and pH 8.0 <3>'],\n",
       "             '2.4.1.185': ['#1# 60% of maximal activity at pH 5.5 and pH 7.5 <1>',\n",
       "              '#3# activity range <6>'],\n",
       "             '2.4.1.187': ['#2# 40% of maximal activity at pH 6, 45% of maximal activity at pH 9 <1>'],\n",
       "             '2.4.1.189': ['#1# 50% of maximal activity at pH 5.6 and pH 9.0 <1>'],\n",
       "             '2.4.1.19': ['#15# strain 1011, wild-type and mutant enzymes retain 80% activity <28>',\n",
       "              '#15# No. 5 strain <2>',\n",
       "              '#15# ATCC 39612/Ha3-3-2 <9>; #17# C31 <12>',\n",
       "              '#17# ATCC 21783 <8>; #47# pH 5.0: about 70% of maximal activity, pH 10.0: about 50% of maximal activity <44>; #91# more than 60% activity between pH 6.0 and 10.0 <160>',\n",
       "              '#15# AL-6, quite stable at 40°C for 3 h <13,32>; #79# pH 5.0: about 40% of maximal activity, pH 8.0: about 50% of maximal activity <71>; #18# pH 5: soluble enzyme shows about 50% of maximal activity, soluble enzyme shows about 60% of maximal activity, pH 8: soluble enzyme shows 40% of maximal activity, immobilized enzyme shows about 50% of maximal activity <70>; #74# loss of more than 60% of activity above or below <118>; #77# more than 50% activity between pH 5.0 and 8.0 <128>',\n",
       "              '#73# more than 80% of maximum activity <114>',\n",
       "              '#17# enzyme immobilized on glyoxyl-agarose and free enzyme present similar pH/activities profiles, with two peaks at pH values of 5.5 and 7 and a minimum at pH 6.0-6.5 in both amylolytic and CGTase activities. Activities decrease after these two maximum values. The glyoxyl CGTase retains 30% of amylase activity at pH 4 and 50% at pH 9. The soluble enzyme retains 10% and 30%, respectively. In synthetic activities differences are not significant <66>',\n",
       "              '#17# ATCC 21783 <3>',\n",
       "              '#18# loss of cyclization activity above and below <117>',\n",
       "              '#18# IFO 3490 <32>; #28# pH 4.0: about 65% of maximal activity, pH 8.5: about 65% of maximal activity, wild-type enzyme <52>',\n",
       "              '#30# more than 80% activity in the range 5.0-6.7 <23>',\n",
       "              '#34# INMIA 3849 <32>',\n",
       "              '#45# pH 5.0: about 50% of maximal activity, pH 9.0: about 55% of maximal activity, native enzyme <72>; #41# more than 60% activity between pH 5.0 and 9.0 <141>; #86# the enzyme shows more than 50% activity between pH 5.0 and 9.0 <172>',\n",
       "              '#7# most active <153>; #15# pH 5.0: about 25% of maximal activity, pH 7.0: about 60% of maximal activity, the enzyme does not show any activity at pH 4.0 and at pH 10.0 <54>; #18# pH 5.0: about 70% of maximal activity, pH 9.0: about 70% of maximal activity, native enzyme <72>; #40# more than 70% activity between pH 5.0 and 7.0 <148>',\n",
       "              '#8# activity range, about 60% of maximal activity at pH 4.5-9.0 <78>',\n",
       "              '#89# most active <153>',\n",
       "              '#9# activity range, about 20% of maximal activity at pH 4.5, and 40% at pH 10.0 <99>',\n",
       "              '#90# there is no enzyme activity at pH lower than 6.0, but gradually increases with pH and then, after reaching its maxima pH 9.0 enzyme activity declines at pH 10 and lastly is lost as pH reaches to 11.0 <159>'],\n",
       "             '2.4.1.190': ['#1# 50% of maximal activity at pH 6.0 and pH 9.0 <1>'],\n",
       "             '2.4.1.191': ['#1# 50% of maximal activity at pH 6 and pH 9 <1>'],\n",
       "             '2.4.1.192': ['#1# at pH 5.8 about 50% of activity maximum, at pH 6.5-8.5 activity maximum <2>'],\n",
       "             '2.4.1.193': ['#1# pH 7.2: about 40% of maximal activity, pH 9.9: about 60% of maximal activity <1>'],\n",
       "             '2.4.1.194': ['#1# half-maximal activity at pH 6.6 and 8.6 <1>'],\n",
       "             '2.4.1.196': ['#3# at pH 5 nearly inactive, at pH 11 about 50% of activity maximum <3>'],\n",
       "             '2.4.1.197': ['#1# 20% of maximal activity at pH 5.5, 70% of maximal activity at pH 7.2 <1>'],\n",
       "             '2.4.1.2': ['#1# about half-maximal activity at pH 3.0 and 5.7 <2>'],\n",
       "             '2.4.1.20': ['#11# 50% activity <1>',\n",
       "              '#14# the enzyme shows more than 80% activity between pH 6.0 and 8.0 <41>',\n",
       "              '#19# the enzyme shows more than 50% activity between pH 4.0 and 9.5 <46>'],\n",
       "             '2.4.1.201': ['#1# about half-maximal activity at pH 5.0 and about 80% of maximal activity at pH 8.4 <2>'],\n",
       "             '2.4.1.202': ['#1# 10% of maximal activity at pH 6.5 and 30% of maximal activity at pH 10.0 <2>'],\n",
       "             '2.4.1.206': ['#3# cacodylate buffer <3>'],\n",
       "             '2.4.1.207': ['#10# isozyme XTH14 <78>',\n",
       "              '#10# pH 5: about 30% of maximal activity, pH 7.5: about 40% of maximal activity <9>',\n",
       "              '#10# isozyme XTH26 <78>',\n",
       "              '#10# both isoenzymes show a fast and symmetrical decrease in activity on both sides of their optima followed by an acidic shoulder between pH 4.0 and 5.0 for XTH14 and between pH 4.0 and 4.5 for XTH26. At pH 4.0 and 8.0, XTH14 retains about 30% and XTH26 40% of its maximal activity <78>; #185# AtXTH13 is more tolerant to basic environments, retaining 40% of its maximal activity at pH 8.0, but it displays only 20% activity at pH 4.0. The enzyme is inactive at pH 3.5. Bell-shaped curve of AtXTH13 <81>; #184# bell-shaped curve of AtXTH12 <81>',\n",
       "              '#68# XTR9 retains approximately 35% of its maximal activity at pH 4.5 and 30% at pH 8 <26>; #10# TCH4 and Meri-5 retain approximately 10-15% of its maximal activity at pH 4.5 and 65% at pH 8 <26>',\n",
       "              '#186# the activity-pH profile of AtXTH17 totally shifts towards the alkali side with its optimum at pH 7.2 and with an unusual acid intolerance. AtXTH18 shows the highest acid tolerance of these three enzymes. At pH 6.0 the enzyme still displays 60% of its activity at pH 7.0, whereas the activities of AtXTH17 and AtXTH19 drop below 30% of their maximal activity at pH 6.0. At or below pH 5.5, all three enzymes display less than 20% of their maximal activities. On the basic side, AtXTH17 shows the highest tolerance with activity of more than 80% of maximal activity at pH 8.0 <81>; #187# the activity-pH profile of AtXTH18 totally shifts towards the alkali side with optimum at pH 7.5 and with an unusual acid intolerance. At pH 6.0 the enzyme still displays 60% of its activity at pH 7.0. At or below pH 5.5, the enzyme displays less than 20% of its maximal activity. On the basic side, AtXTH18 shows the highest tolerance with activities of more than 80% of its maximal activity at pH 8.0 <81>; #188# the activity-pH profile of AtXTH19 totally shifts towards the alkali side with optimum at pH 7.0 and with an unusual acid intolerance. Activity of AtXTH19 drops below 30% of its maximal activity at pH 6.0. At or below pH 5.5, the enzymes displays less than 20% of its maximal activities. AtXTH19 retains 90% of its maximal activity at pH 7.5, but shows a 40% activity drop between pH 7.5 and pH 8.0 <81>',\n",
       "              '#191# isozyme DkXTH1 exhibits a bell-shaped pH profile, and a rapid loss of activity occurs when the pH ranges from pH 4 to pH 5, which is typical for XET isoenzymes <105>',\n",
       "              '#200# recombinant enzyme, activity range, profile overview <100>',\n",
       "              '#201# recombinant enzyme, activity range, profile overview <100>',\n",
       "              '#64# active between <8>',\n",
       "              '#69# pH-profile, rapid loss of activity below pH 4.5 and above pH 6.5 <31>',\n",
       "              '#7# pH 4: 25-35% of maximal activity, pH 7: 30-40% of maximal activity <1>; #192# isozyme DkXTH2 exhibits a bell-shaped pH profile, and a rapid loss of activity occurs below pH 4, which is typical for XET isoenzymes <105>; #204# over 30% of maximal activity within this range, recombinant enzyme, profile overview <112>'],\n",
       "             '2.4.1.21': ['#32# less than half-maximal activity above and below <29>'],\n",
       "             '2.4.1.210': ['#6# pH 6.5: about 70% of maximal activity, pH 9.5: about 55% of maximal activity <4>',\n",
       "              '#9# activity range <8>'],\n",
       "             '2.4.1.211': ['#22# in presence of Tris-HCl <4>',\n",
       "              '#22# in presence of MES-NaOH <4>',\n",
       "              '#22# in presence of Mops-NaOH <4>',\n",
       "              '#22# in presence of sodium acetate <4>'],\n",
       "             '2.4.1.212': ['#1# recombinant enzyme, pH-profile <16>'],\n",
       "             '2.4.1.214': ['#15# enzyme shows maximal activity in mildly alkaline conditions <22>',\n",
       "              '#18# more than 50% activity between pH 5.0 and 7.0 <31>',\n",
       "              '#19# more than 50% activity between pH 6.5 and 8.0 <31>',\n",
       "              '#4# enzyme is not affected by high pH but shows no activity at pH below pH 5.5 <12>'],\n",
       "             '2.4.1.217': ['#4# pH 6.0: about 60% of maximal activity, pH 8.0: about 75% of maximal activity <2>; #8# pH 6.0: about 40% of maximal activity, pH 8.0: about 40% of maximal activity <6>',\n",
       "              '#5# pH-profile <4>',\n",
       "              '#6# pH-profile, 30-40% of maximal activity at pH 5.0 and pH 9.0 <3>',\n",
       "              '#7# pH-profile of full length and truncated enzyme <5>'],\n",
       "             '2.4.1.220': ['#1# 6.8 and 10.2 half-maximal-activity <1>'],\n",
       "             '2.4.1.23': ['#1# 70% of maximal activity at pH 7, 85% of maximal activity at pH 9.0 <1>'],\n",
       "             '2.4.1.230': ['#5# over 90% of maximal activity at pH 4.0 and at pH 6.0 <15>'],\n",
       "             '2.4.1.231': ['#5# decrease in enzyme activity below pH 6.2 and above pH 6.6 <2>',\n",
       "              '#6# pH 5.5: about 75% of maximal activity, pH 7.8: about 80% of maxuimal activity, phosphorolysis <14>',\n",
       "              '#6# pH 5.5: about 80% of maximal activity, pH 75% of maximal activity, synthesis of alpha,alpha-trehalose <14>'],\n",
       "             '2.4.1.232': ['#1# the recombinant Och1p in Tris/malate buffer exhibits above 80% of maximum activity between pH 6.5 and pH 8.5 <16>',\n",
       "              '#5# relative activity at pH 2 28%, at pH 7 100%, strain J-1012 alpha-1,6-MT <20>'],\n",
       "             '2.4.1.234': ['#6# the activity of the forward reaction is 1.6fold higher at pH 7.5 than at 5.5, the activity of the reverse reaction is 8fold higher at pH 5.5 than at 7.5 <2>'],\n",
       "             '2.4.1.241': ['#2# pH 5.5: about 25% of maximal activity, pH 8.5: about 45% of maximal activity <2>'],\n",
       "             '2.4.1.242': ['#12# with substrate amylopectin, sharp drop of activity below pH 7.0 and above pH 9.0 <13>',\n",
       "              '#12# with substrate amylopectin, sharp drop of activity below pH 6.5 and more gradually above pH 8.0 <13>'],\n",
       "             '2.4.1.243': ['#3# more than 80% of maximum activity within this range <2>',\n",
       "              '#6# pH 5.0: about 80% of maximal activity, pH 6.0: about 90% of maximal activity, pH 4.3: about 20% of maximal activity, pH 6.5: about 20% of maximal activity <11>'],\n",
       "             '2.4.1.245': ['#2# pH 4.5: about 45% of maximal activity, pH 8.0: about 40% of maximal activity <10>',\n",
       "              '#4# 13% residual activity <2>'],\n",
       "             '2.4.1.247': ['#2# activity range, profile, overview <2>'],\n",
       "             '2.4.1.248': ['#6# pH 5.5: about 75% of maximal activity, pH 7.0: about 75% of maximal activity <17>',\n",
       "              '#6# pH 4.5: about 60% of maximal activity, pH 6.0: about 50% of maximal activity <18>'],\n",
       "             '2.4.1.25': ['#70# pH dependence of wild-type and mutant enzymes, D293N and E340Q are active below pH 6.5 and pH 5.5, respectively, but precipitate during the necessary prolonged incubation times, wild-type Tt AMase precipitates below pH 5.5 under similar extended incubation times as well, overview <49>',\n",
       "              '#78# more than 50% relative activity at pH values between 5.2 and 7.5 <84>'],\n",
       "             '2.4.1.254': ['#1# active over pH-range 6.5-8.5 <1>'],\n",
       "             '2.4.1.255': ['#3# activity remaining fairly high up to pH 7.5 while dropping rapidly below 6 <1>'],\n",
       "             '2.4.1.266': ['#10# pH 6.0: about 55% of maximal activity, pH 9.5: about 60% of maximal activity <4>',\n",
       "              '#2# pH 6.0: about 50% of maximal activity, pH 9.5: about 75% of maximal activity <4>',\n",
       "              '#3# pH 6.0: about 80% of maximal activity, pH 8.0: about 50% of maximal activity <7>',\n",
       "              '#5# pH 7.5: about 60% of maximal activity, pH 9.5: about 80% of maximal activity <1>',\n",
       "              '#9# pH 6.5: about 55% of maximal activity, pH 9.5: about 70% of maximal activity <2>'],\n",
       "             '2.4.1.268': ['#1# pH 4.5: about 50% of maximal activity, pH 9.0: about 35% of maximal activity <1>'],\n",
       "             '2.4.1.270': ['#1# pH 6.0: about 50% of maximal activity, pH 10.0: about 75% of maximal activity <1>'],\n",
       "             '2.4.1.294': ['#3# pH 7.0: about 40% of maximal activity, pH 8.5: about 20% of maximal activity, activity with cyanidin and UDP-galactose <3>'],\n",
       "             '2.4.1.295': ['#1# activity range <1>'],\n",
       "             '2.4.1.298': ['#1# 50% of maximal activity at pH 6.5 and pH 8.7 <1>'],\n",
       "             '2.4.1.305': ['#2# pH 5.0: about 60% of maximal activity, pH 9.0: about 40% of maximal activity <1>; #1# pH 5.0: about 75% of maximal activity, pH 9.0: about 75% of maximal activity <1>'],\n",
       "             '2.4.1.314': ['#2# more than 60% activity between pH 7.0 and 9.0 <3>'],\n",
       "             '2.4.1.32': ['#1# pH 6.5: 55% of activity maximum, pH 9.0: 10% of activity maximum <1>'],\n",
       "             '2.4.1.329': ['#4# more than 50% of maximum activity, synthesis of sucrose 6F-phosphate <2>'],\n",
       "             '2.4.1.33': ['#2# 65% of maximal activity at pH 5.6 and pH 8.5 <1>'],\n",
       "             '2.4.1.333': ['#3# more than 80% of maximum activity <4>'],\n",
       "             '2.4.1.336': ['#1# pH 5.0: about 45% of maximal activity, pH 8.7: about 50% of maximal activity <1>'],\n",
       "             '2.4.1.339': ['#2# pH 4.5: about 60% of maximal activity, pH 7.5: about 60% of maximal activity, synthetic reaction <2>'],\n",
       "             '2.4.1.34': ['#14# optimal conditions for callose synthese in vitro <27>',\n",
       "              '#15# 6.5 and 9: about 30% of activity maximum <10>',\n",
       "              '#34# pH 7.5: activity maximum, pH 8.5: about 50% of activity maximum <18>',\n",
       "              '#4# 6.8 and 8.8: about 50% of activity maximum <11>',\n",
       "              '#6# 7: about 60% of activity maximum, 7.8: about 70% of activity maximum <1>'],\n",
       "             '2.4.1.341': ['#1# about 60% of maximal activity <1>'],\n",
       "             ...})"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "dfa57cb9-9ceb-4b52-b56c-f2cd5aa82170",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9678 9680\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ec_num</th>\n",
       "      <th>ph_min</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>8.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.1.1.1</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9673</th>\n",
       "      <td>7.4.2.6</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9674</th>\n",
       "      <td>7.4.2.6</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9675</th>\n",
       "      <td>7.4.2.8</td>\n",
       "      <td>6.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9676</th>\n",
       "      <td>7.5.2.2</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9677</th>\n",
       "      <td>7.6.2.9</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9678 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ec_num  ph_min\n",
       "0     1.1.1.1     4.0\n",
       "1     1.1.1.1     7.0\n",
       "2     1.1.1.1     5.0\n",
       "3     1.1.1.1     8.2\n",
       "4     1.1.1.1     6.0\n",
       "...       ...     ...\n",
       "9673  7.4.2.6     6.0\n",
       "9674  7.4.2.6     5.5\n",
       "9675  7.4.2.8     6.5\n",
       "9676  7.5.2.2     7.0\n",
       "9677  7.6.2.9     5.5\n",
       "\n",
       "[9678 rows x 2 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ph_min_df = pd.DataFrame([(ec_num, ph_min) \n",
    "                          for ec_num, ph_vals in ph_range_data.items() \n",
    "                          for ph_min in ph_vals],\n",
    "                         columns=['ec_num', 'ph_min'],\n",
    "                        )\n",
    "ph_max_df = pd.DataFrame([(ec_num, ph_max) \n",
    "                          for ec_num, ph_vals in ph_max_data.items() \n",
    "                          for ph_max in ph_vals],\n",
    "                         columns=['ec_num', 'ph_max'],\n",
    "                        )\n",
    "print(len(ph_min_df), len(ph_max_df))\n",
    "ph_min_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "1d24e72b-8205-49fc-b226-be2939d0476c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9417 9418\n"
     ]
    }
   ],
   "source": [
    "ph_min_df = pd.DataFrame([(ec_num, ph_min) \n",
    "                          for ec_num, ph_vals in ph_range_data.items() \n",
    "                          for ph_min in ph_vals],\n",
    "                         columns=['ec_num', 'ph_min'],\n",
    "                        )\n",
    "ph_max_df = pd.DataFrame([(ec_num, ph_max) \n",
    "                          for ec_num, ph_vals in ph_max_data.items() \n",
    "                          for ph_max in ph_vals],\n",
    "                         columns=['ec_num', 'ph_max'],\n",
    "                        )\n",
    "print(len(ph_min_df), len(ph_max_df))\n",
    "# ph_min_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "5f1b6c8c-0e45-4073-a79a-50dd2d67f5d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "20d7ad76-15eb-4de1-a448-ce2284702f45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.legend.Legend at 0x7f4702ed3670>"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjEAAAHFCAYAAAADhKhmAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjYuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/P9b71AAAACXBIWXMAAA9hAAAPYQGoP6dpAABHVUlEQVR4nO3de1jUZf7/8dfIYRAFFIhTIZKZtp7yUB5qEytQSt2yzcoiNdPaTCO0g7UmtqVttVqr5aHtpxa62n6v7GgqtmZ2qWm6VB7WQ4umG0iZgojCCJ/fHy6D43AanGH4wPNxXVwx99xzz3ve4PDqcxqLYRiGAAAATKaZtwsAAACoC0IMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIM4EaLFy+WxWJx+LrkkkuUkJCgTz75xNvlVapt27YaNWpUvT/vwYMHZbFY9Oqrr1Z6/6uvviqLxaKDBw/Wb2ENXF1/XkVFRUpPT9cXX3zh9poAbyHEAB6waNEibd68WZs2bdLChQvl4+OjIUOG6OOPP/Z2aWiiioqKNH36dEIMGhVfbxcANEadO3dWr1697LcHDRqk1q1b6+9//7uGDBlS5eNKS0t19uxZWa3W+iiz0SoqKlJgYKC3ywDgYWyJAepBQECA/P395efnZx8r353y8ssv64UXXlB8fLysVqvWr18vSfrmm280dOhQhYaGKiAgQN27d9d7773nsG757qv169frD3/4g8LDwxUWFqZhw4bpp59+cphrs9n05JNPKioqSoGBgbr++uu1detWp1p//vlnPfLII/rNb36jli1bKiIiQjfeeKM2btxYq9fatm1bDR48WCtXrlTXrl0VEBCgyy+/XH/9619dbVutpKeny2KxaMeOHfr973+v1q1bq127dpLO9fDuu+9W27Zt1bx5c7Vt21b33HOPDh065LCGK30sLi7WpEmT7H284YYbtH379kp38+Tm5uqhhx7SZZddJn9/f8XHx2v69Ok6e/Zsja/LnT+vgwcP6pJLLpEkTZ8+3b6rs7zeAwcOaPTo0Wrfvr0CAwN16aWXasiQIfr+++9rrBPwJrbEAB5QvkXFMAwdPXpUr7zyik6dOqURI0Y4zf3rX/+qK6+8Uq+++qqCg4PVvn17rV+/XoMGDVLv3r01f/58hYSEaPny5brrrrtUVFTk9MfywQcf1K233qply5bp8OHDeuKJJ3Tffffpn//8p33O2LFj9c4772jy5MlKTEzUzp07NWzYMJ08edJhrV9//VWSNG3aNEVFRamwsFArV65UQkKCPv/8cyUkJNT4+rOyspSamqr09HRFRUVp6dKleuyxx1RSUqLJkyc7zC0rK6v0j3pZWVmNz3O+YcOG6e6779bDDz+sU6dOSTr3x7tDhw66++67FRoaqpycHM2bN0/XXHONdu/erfDwcIc1atPH0aNHa8WKFXryySd14403avfu3br99ttVUFDgsFZubq6uvfZaNWvWTM8995zatWunzZs364UXXtDBgwe1aNGial+PO39e0dHRWr16tQYNGqQxY8bowQcflCR7sPnpp58UFhaml156SZdccol+/fVXLVmyRL1799a//vUvdejQwaWfBVBvDABus2jRIkOS05fVajXefPNNh7nZ2dmGJKNdu3ZGSUmJw30dO3Y0unfvbthsNofxwYMHG9HR0UZpaanD8z3yyCMO815++WVDkpGTk2MYhmHs2bPHkGQ8/vjjDvOWLl1qSDJGjhxZ5Ws6e/asYbPZjJtuusm4/fbba+xBXFycYbFYjKysLIfxxMREIzg42Dh16pTD66/pKzs7u9rnmzZtmiHJeO6552qs7ezZs0ZhYaHRokUL4/XXX7eP17aPu3btMiQZTz31lMO8v//97059fOihh4yWLVsahw4dcpj76quvGpKMXbt2VVmnJ35eP//8syHJmDZtWpWPPX+NkpISo3379k41AA0Ju5MAD3jnnXe0bds2bdu2TZ999plGjhyp8ePHa+7cuU5zhw4d6rCb6cCBA/r3v/+te++9V5J09uxZ+9ctt9yinJwc7d2712mN83Xt2lWS7LtNyndRla9Zbvjw4fL1dd4gO3/+fPXo0UMBAQHy9fWVn5+fPv/8c+3Zs6dWr79Tp07q1q2bw9iIESNUUFCgHTt2OIw/9thj9l6d//XYY4/V6rnK3XHHHU5jhYWFeuqpp3TFFVfI19dXvr6+atmypU6dOlXpa6mpjxs2bJB0rm/n+/3vf+/Ux08++UQDBgxQTEyMw88wOTnZYa3K1PfP6+zZs5oxY4Z+85vfyN/fX76+vvL399f+/ftrvQbgDexOAjzgqquucjqw99ChQ3ryySd13333qVWrVvb7oqOjHR579OhRSdLkyZOddr2U++WXXxxuh4WFOdwuPzD49OnTkqRjx45JkqKiohzm+fr6Oj121qxZmjRpkh5++GH96U9/Unh4uHx8fDR16tRa/0G78HnOHyuvpdxll13m0Ktyrp5Fc2EfpXPB6fPPP9fUqVN1zTXXKDg4WBaLRbfccou9N+erbR8jIyMd5lXWx6NHj+rjjz92CKjnu/BneL76/nmlpaXpjTfe0FNPPaX+/furdevWatasmR588MFK+wQ0FIQYoJ507dpVa9as0b59+3Tttdfaxy0Wi8O88uM0pkyZomHDhlW6lqvHKJT/4cvNzdWll15qHz979qxTqMjIyFBCQoLmzZvnMH7hsRjVyc3NrXLswj/C7nJhH/Pz8/XJJ59o2rRpevrpp+3jxcXF9uNIXFVe+9GjR2vsY3h4uLp27aoXX3yx0rViYmJqfJ76+nllZGTo/vvv14wZMxzGf/nlF4fADTQ0hBignmRlZUmqOJiyKh06dFD79u317bffOv1Rqavyg3GXLl2qnj172sffe+89p4NqLRaL0yne3333nTZv3qzY2NhaPd+uXbv07bffOuxSWrZsmYKCgtSjR486vgrXWCwWGYbh9Fr+9re/qbS0tE5r3nDDDZKkFStWOLyO//u//3Pq4+DBg7Vq1Sq1a9dOrVu3dul5PPHzunCrUk1rfPrpp/rvf/+rK664wqXagfpEiAE8YOfOnfY/NseOHdP777+vzMxM3X777YqPj6/x8QsWLFBycrIGDhyoUaNG6dJLL9Wvv/6qPXv2aMeOHfrHP/7hUj1XXXWV7rvvPr322mvy8/PTzTffrJ07d9rPiDrf4MGD9ac//UnTpk1T//79tXfvXj3//POKj4+v1anB0rmtDEOHDlV6erqio6OVkZGhzMxM/fnPf66367cEBwfrhhtu0CuvvKLw8HC1bdtWGzZs0Ntvv13nrQudOnXSPffco7/85S/y8fHRjTfeqF27dukvf/mLQkJC1KxZxWGGzz//vDIzM9WvXz9NnDhRHTp00JkzZ3Tw4EGtWrVK8+fP12WXXVbp83ji5xUUFKS4uDh9+OGHuummmxQaGmrvy+DBg7V48WJ17NhRXbt21fbt2/XKK69UWR/QYHj7yGKgMans7KSQkBDj6quvNmbNmmWcOXPGPrf87JxXXnml0rW+/fZbY/jw4UZERITh5+dnREVFGTfeeKMxf/58p+fbtm2bw2PXr19vSDLWr19vHysuLjYmTZpkREREGAEBAUafPn2MzZs3G3FxcQ5nuxQXFxuTJ082Lr30UiMgIMDo0aOH8cEHHxgjR4404uLiauxBXFycceuttxr/93//Z3Tq1Mnw9/c32rZta8yaNcthXk2v/5VXXnHp7KSff/7Z6b4jR44Yd9xxh9G6dWsjKCjIGDRokLFz506n1+xKH8+cOWOkpaU59TEkJMTpTJ6ff/7ZmDhxohEfH2/4+fkZoaGhRs+ePY1nn33WKCwsrPZ1eeLntW7dOqN79+6G1Wp1OMvp+PHjxpgxY4yIiAgjMDDQuP76642NGzca/fv3N/r3719tnYA3WQzDMLySngA0Sm3btlXnzp0b7GdFecKmTZt03XXXaenSpZVeCwiAZ7A7CQBckJmZqc2bN6tnz55q3ry5vv32W7300ktq3759lQdiA/AMQgwAuCA4OFhr167Va6+9ppMnTyo8PFzJycmaOXOmAgICvF0e0KSwOwkAAJgSV+wFAACmRIgBAACmRIgBAACm1GgP7C0rK9NPP/2koKAgp8uRAwCAhskwDJ08eVIxMTEOF5CsTKMNMT/99FOtL5EOAAAalsOHD9d41ehGG2KCgoIknWvC+ZfpttlsWrt2rZKSkqr8dNmmgl5UoBfn0IcK9KICvahALyp4qhcFBQWKjY21/x2vTqMNMeW7kIKDg51CTGBgoIKDg/kFpBd29OIc+lCBXlSgFxXoRQVP96I2h4JwYC8AADAlQgwAADAlQgwAADAll4+J+fLLL/XKK69o+/btysnJ0cqVK3XbbbfZ769qH9bLL7+sJ554QpKUkJCgDRs2ONx/1113afny5fbbx48f18SJE/XRRx9JkoYOHao5c+aoVatWrpYMAGjCSktLZbPZ3LKWzWaTr6+vzpw5o9LSUresaVZ17YWfn598fHzcUoPLIebUqVPq1q2bRo8erTvuuMPp/pycHIfbn332mcaMGeM0d+zYsXr++eftt5s3b+5w/4gRI3TkyBGtXr1akjRu3DilpKTo448/drVkAEATZBiGcnNzdeLECbeuGRUVpcOHDzf5a5BdTC9atWqlqKioi+6hyyEmOTlZycnJVd4fFRXlcPvDDz/UgAEDdPnllzuMBwYGOs0tt2fPHq1evVpbtmxR7969JUlvvfWW+vbtq71796pDhw6ulg0AaGLKA0xERIQCAwPdEjrKyspUWFioli1b1nghtsauLr0wDENFRUXKy8uTJEVHR19UDR49xfro0aP69NNPtWTJEqf7li5dqoyMDEVGRio5OVnTpk2znxO+efNmhYSE2AOMJPXp00chISHatGkTIQYAUK3S0lJ7gAkLC3PbumVlZSopKVFAQAAhpo69KN/zkpeXp4iIiIvateTRELNkyRIFBQVp2LBhDuP33nuv4uPjFRUVpZ07d2rKlCn69ttvlZmZKelceo6IiHBaLyIiQrm5uZU+V3FxsYqLi+23CwoKJJ3bZ3f+vtDy7921f9TM6EUFenEOfahALyqYsRfFxcUyDEMBAQEqKytz27qGYdj/6851zehiehEQECDDMHT69GlZrVaH+1z5PfNoiPl//+//6d5771VAQIDD+NixY+3fd+7cWe3bt1evXr20Y8cO9ejRQ1LlBwgbhlHl5sCZM2dq+vTpTuNr165VYGCg03h5YAK9OB+9OIc+VKAXFczUC19fX0VFRenUqVMeCV8nT550+5pmVZdelJSU6PTp09qwYYPOnj3rcF9RUVGt1/FYiNm4caP27t2rFStW1Di3R48e8vPz0/79+9WjRw9FRUXp6NGjTvN+/vlnRUZGVrrGlClTlJaWZr9dftnipKQkpyv2ZmZmKjExkast0gs7enEOfahALyqYsRdnzpzR4cOH1bJlS6f/kb4Y5R9OyIcLX1wvzpw5o+bNm+uGG25w+vmU70mpDY+FmLfffls9e/ZUt27dapy7a9cu2Ww2+wE+ffv2VX5+vrZu3aprr71WkvT1118rPz9f/fr1q3QNq9XqtElKOncqV2X/6Koab4roRQV6cQ59qEAvKpipF6WlpbJYLGrWrJlbj10p321SvnZD8cUXX2jAgAE6fvx4vV2K5GJ60axZM1kslkp/p1z5HXM5xBQWFurAgQP229nZ2crKylJoaKjatGkj6VyK+sc//qG//OUvTo//4YcftHTpUt1yyy0KDw/X7t27NWnSJHXv3l3XXXedJOmqq67SoEGDNHbsWC1YsEDSuVOsBw8ezEG9AICLMjtzX50faxiGiouLZbVaa7314fHEK+v8fLXVr18/5eTkKCQkxKPPc35YOn8vhyS1bdtWqampSk1N9WgN53M5Rn7zzTfq3r27unfvLklKS0tT9+7d9dxzz9nnLF++XIZh6J577nF6vL+/vz7//HMNHDhQHTp00MSJE5WUlKR169Y5HKG8dOlSdenSRUlJSUpKSlLXrl317rvv1uU1AgDQqPn7+7vluitm43KISUhIkGEYTl+LFy+2zxk3bpyKiooqTYSxsbHasGGDjh07puLiYh04cECvv/66QkNDHeaFhoYqIyNDBQUFKigoUEZGBlfrBQA0egkJCZowYYJSU1PVunVrRUZGauHChTp16pRGjx6toKAgtWvXTp999pn9MV988YUsFov9wn6LFy9Wq1attGbNGl111VVq2bKlBg0a5HRB2vOVr/Hpp5+qW7duCggIUO/evfX99997+iXXWcPZoQcAACSdu0RJeHi4tm7dqgkTJugPf/iD7rzzTvXr1087duzQwIEDlZKSUu2ZPEVFRXr11Vf17rvv6ssvv9SPP/6oyZMn1/jcTzzxhF599VVt27ZNERERGjp0aIM9vd6jp1gDAOpg/UzXHzNgivvrgNd069ZNf/zjHyWdO/v2pZdeUnh4uP0SJc8995zmzZun7777Tn369Kl0DZvNpvnz56tdu3aSpEcffdTh436qMm3aNCUmJko6F6Yuu+wyrVy5UsOHD7fPueyyy5we58qp0e5CiAEAoIHp2rWr/XsfHx+FhYWpS5cu9rHyy42UX76/MoGBgfYAI527xH9188v17dvX/n1oaKg6dOigPXv2OMzZuHGjWrRo4fCxAwkJCTWu7W6EGAAAGpgLTzMuPx35/NuSqr1SbmVrlF9l11UXHjAcHx+v4OBgFRQUKDg4WM2aNZOvb/1HCo6JAQAAdlu2bLF/f/z4ce3bt08dO3b0YkVVY0sMAACwe/755xUWFqbIyEg9++yzCg8P12233ebtsirFlhgAAGD30ksv6bHHHlPPnj2Vk5Ojjz76SP7+/t4uq1JsiQEANCkXcwXdsrIyh+NAPOGLL75wGjt48KDT2PnHt5Rfw63cqFGjNGrUKIf5t912W62Oibn++uu1c+fOSu87/3kuPB6nsho9jS0xAADAlAgxAADAlNidBAAAnHZJmQFbYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgClxnRgAQNOyfmadH2oxDAUUF8titUoWS+0eNGBKnZ/PDHx8fLRy5UqnD4kcNWqUTpw4oQ8++MBjz82WGAAAYEqEGAAAGpCEhARNmDBBqampat26tSIjI7Vw4UKdOnVKo0ePVlBQkNq1a6fPPvvM/pjS0lKNGTNG8fHxat68uTp06KDXX3/dfv+ZM2fUqVMnjRs3zj6WnZ2tkJAQvfXWW1XWYrFYNG/ePCUnJ6t58+aKj4/XP/7xD8+88DogxAAA0MAsWbJE4eHh2rp1qyZMmKA//OEPuvPOO9WvXz/t2LFDAwcOVEpKioqKiiSd+0Tpyy67TO+99552796t5557Ts8884zee+89SVJAQICWLl2qJUuW6IMPPlBpaalSUlI0YMAAjR07ttpapk6dqjvuuEPffvut7rvvPt1zzz3as2ePx3tQG4QYAAAamG7duumPf/yj2rdvrylTpqh58+YKDw/X2LFj1b59ez333HM6duyYvvvuO0mSn5+fpk+frmuuuUbx8fG69957NWrUKHuIkaSrr75aL7zwgsaOHavHH39cP/zwg/72t7/VWMudd96pBx98UFdeeaX+9Kc/qVevXpozZ47DnHvuuUctW7Z0+Fq6dKl7m1IJDuwFAKCB6dq1q/17Hx8fhYWFqUuXLvaxyMhISVJeXp59bP78+frb3/6mQ4cO6fTp0yopKdHVV1/tsO6kSZP04Ycfas6cOfrss88UHh5eYy19+/Z1up2VleUwNnv2bN18880OY0899ZRKS0trXP9isCUGAIAGxs/Pz+G2xWJxGLP878yosrIySdJ7772nxx9/XA888IDWrl2rrKwsjR49WiUlJQ7r5OXlae/evfLx8dH+/fvrXJ/lgjOzoqKidMUVVzh8BQUF1Xn92iLEAABgchs3blS/fv30yCOPqHv37rriiiv0ww8/OM174IEH1LlzZ73zzjt68skntXv37hrX3rJli9Ptjh07uq32i8HuJAAATO6KK67QO++8ozVr1ig+Pl7vvvuutm3bpvj4ePucN954Q5s3b9Z3332n2NhYffbZZ7r33nv19ddfy9/fv8q1//GPf6hXr166/vrrtXTpUm3dulVvv/12fbysGrElBgAAk3v44Yc1bNgw3XXXXerdu7eOHTumRx55xH7/v//9bz3xxBN68803FRsbK+lcqDlx4oSmTp1a7drTp0/X8uXL1bVrVy1ZskRLly7Vb37zG4++ntpiSwwAoGm5iCvoGmVlOlNQIP/gYFmaeWY7wBdffOE0dvDgQedaDMP+vdVq1aJFi7Ro0SKHOTNnnrs6cceOHe2nY5cLDg5WdnZ2jfXExMRo7dq1Vd5fWlqqZpX0YvHixTWufbHYEgMAAEyJEAMAAEyJ3UkAAKBS5++yaojYEgMAAEyJEAMAaLQa+paEpspdPxdCDACg0Sm/uu2FZ+SgYSj/uVx4ZWJXcUwMAKDR8fHxUatWreyfLRQYGOh0qfy6KCsrU0lJic6cOVPpacVNSV16YRiGioqKlJeXp1atWsnHx+eiaiDEAAAapaioKEmOH5J4sQzD0OnTp9W8eXO3hCIzu5hetGrVyv7zuRiEGABAo2SxWBQdHa2IiAjZbDa3rGmz2fTll1/qhhtuuOhdIWZX1174+fld9BaYcoQYAECj5uPj47Y/mj4+Pjp79qwCAgKafIhpCL1o2jv0AACAaRFiAACAKbkcYr788ksNGTJEMTExslgs+uCDDxzuHzVqlCwWi8NXnz59HOYUFxdrwoQJCg8PV4sWLTR06FAdOXLEYc7x48eVkpKikJAQhYSEKCUlRSdOnHD5BQIAgMbJ5RBz6tQpdevWTXPnzq1yzqBBg5STk2P/WrVqlcP9qampWrlypZYvX66vvvpKhYWFGjx4sEpLS+1zRowYoaysLK1evVqrV69WVlaWUlJSXC0XAAA0Ui4f2JucnKzk5ORq51it1ipPncrPz9fbb7+td999VzfffLMkKSMjQ7GxsVq3bp0GDhyoPXv2aPXq1dqyZYt69+4tSXrrrbfUt29f7d27Vx06dHC1bAAA0Mh45OykL774QhEREWrVqpX69++vF198UREREZKk7du3y2azKSkpyT4/JiZGnTt31qZNmzRw4EBt3rxZISEh9gAjSX369FFISIg2bdpUaYgpLi5WcXGx/XZBQYGkc6eAnX9qXfn37jrdzszoRQV6cQ59qODVXhh1OFzRg3Xye1GBXlTwVC9cWc/tISY5OVl33nmn4uLilJ2dralTp+rGG2/U9u3bZbValZubK39/f7Vu3drhcZGRkcrNzZUk5ebm2kPP+SIiIuxzLjRz5kxNnz7daXzt2rUKDAx0Gs/MzKzLy2uU6EUFenEOfajgnV50dP0hF+y29wR+LyrQiwru7oUrHxXh9hBz11132b/v3LmzevXqpbi4OH366acaNmxYlY8zDMPhin+VXf3vwjnnmzJlitLS0uy3CwoKFBsbq6SkJAUHB9vHbTabMjMzlZiY2OTP8acXFZpSL95Yf6DK+yxGqdqe+UEHA9rJsJy7rsb4AVfUV2kNild/JzbOcv0xv02reU4dNaV/HzWhFxU81YvyPSm14fGL3UVHRysuLk779++XdO4y0CUlJTp+/LjD1pi8vDz169fPPufo0aNOa/3888+KjIys9HmsVqusVqvTuJ+fX6XNrWq8KaIXFZpCL8rDSU1zyuc19n7UxCu/E5Yy1x9TDzU2hX8ftUUvKri7F66s5fHrxBw7dkyHDx9WdHS0JKlnz57y8/Nz2PyUk5OjnTt32kNM3759lZ+fr61bt9rnfP3118rPz7fPAQAATZvLW2IKCwt14EDF5ujs7GxlZWUpNDRUoaGhSk9P1x133KHo6GgdPHhQzzzzjMLDw3X77bdLkkJCQjRmzBhNmjRJYWFhCg0N1eTJk9WlSxf72UpXXXWVBg0apLFjx2rBggWSpHHjxmnw4MGcmQQAACTVIcR88803GjBggP12+XEoI0eO1Lx58/T999/rnXfe0YkTJxQdHa0BAwZoxYoVCgoKsj9m9uzZ8vX11fDhw3X69GnddNNNWrx4scNnWyxdulQTJ060n8U0dOjQaq9NAwAAmhaXQ0xCQoIMw6jy/jVr1tS4RkBAgObMmaM5c+ZUOSc0NFQZGRmulgcAAJoIPjsJAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYkq+3CwCAerd+Zs1zjGaSOkobZ0mWsnNjA6Z4tCwArmFLDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCVCDAAAMCWXQ8yXX36pIUOGKCYmRhaLRR988IH9PpvNpqeeekpdunRRixYtFBMTo/vvv18//fSTwxoJCQmyWCwOX3fffbfDnOPHjyslJUUhISEKCQlRSkqKTpw4UacXCQAAGh+XQ8ypU6fUrVs3zZ071+m+oqIi7dixQ1OnTtWOHTv0/vvva9++fRo6dKjT3LFjxyonJ8f+tWDBAof7R4wYoaysLK1evVqrV69WVlaWUlJSXC0XAAA0Ur6uPiA5OVnJycmV3hcSEqLMzEyHsTlz5ujaa6/Vjz/+qDZt2tjHAwMDFRUVVek6e/bs0erVq7Vlyxb17t1bkvTWW2+pb9++2rt3rzp06OBq2QAAoJFxOcS4Kj8/XxaLRa1atXIYX7p0qTIyMhQZGank5GRNmzZNQUFBkqTNmzcrJCTEHmAkqU+fPgoJCdGmTZsqDTHFxcUqLi623y4oKJB0bheXzWazj5d/f/5YU0UvKjSlXliM0hrvO39Oo+yJUfNGaNv/5tjOn1tfvahFfU48WFtT+vdRE3pRwVO9cGU9j4aYM2fO6Omnn9aIESMUHBxsH7/33nsVHx+vqKgo7dy5U1OmTNG3335r34qTm5uriIgIp/UiIiKUm5tb6XPNnDlT06dPdxpfu3atAgMDncYv3GLUlNGLCk2hF/G1mNP2zA/271et2ue5YrymY61nZhZeWXFj1SoP1FKZ2tdnVw+1NYV/H7VFLyq4uxdFRUW1nuuxEGOz2XT33XerrKxMb775psN9Y8eOtX/fuXNntW/fXr169dKOHTvUo0cPSZLFYnFa0zCMSsclacqUKUpLS7PfLigoUGxsrJKSkhwClM1mU2ZmphITE+Xn53dRr9Hs6EWFptSLN9YfqPI+i1Gqtmd+0MGAdjIsPpKk8QOuqK/S6s/GWTVOsRnNlFl4pRJb7pOfpezc4G/Tqn+Qu9SiPicerK0p/fuoCb2o4KlelO9JqQ2PhBibzabhw4crOztb//znPx1CRGV69OghPz8/7d+/Xz169FBUVJSOHj3qNO/nn39WZGRkpWtYrVZZrVancT8/v0qbW9V4U0QvKjSFXpSHk5rmlM9rlP0oDyW14Gcpqwgx9dULF+qzq4famsK/j9qiFxXc3QtX1nL7dWLKA8z+/fu1bt06hYWF1fiYXbt2yWazKTo6WpLUt29f5efna+vWrfY5X3/9tfLz89WvXz93lwwAAEzI5S0xhYWFOnCgYnN0dna2srKyFBoaqpiYGP3+97/Xjh079Mknn6i0tNR+DEtoaKj8/f31ww8/aOnSpbrlllsUHh6u3bt3a9KkSerevbuuu+46SdJVV12lQYMGaezYsfZTr8eNG6fBgwdzZhIAAJBUhxDzzTffaMCAAfbb5cehjBw5Uunp6froo48kSVdffbXD49avX6+EhAT5+/vr888/1+uvv67CwkLFxsbq1ltv1bRp0+TjU7GZe+nSpZo4caKSkpIkSUOHDq302jQAAKBpcjnEJCQkyDCMKu+v7j5Jio2N1YYNG2p8ntDQUGVkZLhaHgAAaCL47CQAAGBKhBgAAGBKhBgAAGBKHv/YAQDS7EzXrnr7eOKVNU8CgCaOLTEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUCDEAAMCUfL1dAADAS9bPrN08o5mkjtLGWdKNT3m0JMAVbIkBAACmRIgBAACmRIgBAACmxDExAOBJtT3uBIDL2BIDAABMiRADAABMid1JALxmduY+l+Y/nnilhyoBYEZsiQEAAKZEiAEAAKZEiAEAAKZEiAEAAKZEiAEAAKZEiAEAAKZEiAEAAKZEiAEAAKbkcoj58ssvNWTIEMXExMhiseiDDz5wuN8wDKWnpysmJkbNmzdXQkKCdu3a5TCnuLhYEyZMUHh4uFq0aKGhQ4fqyJEjDnOOHz+ulJQUhYSEKCQkRCkpKTpx4oTLLxAAADROLoeYU6dOqVu3bpo7d26l97/88suaNWuW5s6dq23btikqKkqJiYk6efKkfU5qaqpWrlyp5cuX66uvvlJhYaEGDx6s0tJS+5wRI0YoKytLq1ev1urVq5WVlaWUlJQ6vEQAANAYufyxA8nJyUpOTq70PsMw9Nprr+nZZ5/VsGHDJElLlixRZGSkli1bpoceekj5+fl6++239e677+rmm2+WJGVkZCg2Nlbr1q3TwIEDtWfPHq1evVpbtmxR7969JUlvvfWW+vbtq71796pDhw51fb0AAKCRcOsxMdnZ2crNzVVSUpJ9zGq1qn///tq0aZMkafv27bLZbA5zYmJi1LlzZ/uczZs3KyQkxB5gJKlPnz4KCQmxzwEAAE2bWz8AMjc3V5IUGRnpMB4ZGalDhw7Z5/j7+6t169ZOc8ofn5ubq4iICKf1IyIi7HMuVFxcrOLiYvvtgoICSZLNZpPNZrOPl39//lhTRS8qeLoXFqO05knn8eTPpLpayu87f463aqmM22oxav7/N9v/5tjOn1uX56/Fc7mFB2tz6EUTf7/gfbOCp3rhynoe+RRri8XicNswDKexC104p7L51a0zc+ZMTZ8+3Wl87dq1CgwMdBrPzMystp6mhF5U8FQv4l2cv2qVa5/u7Ira1NL2zA8Nppbzua+WjrWemVl43idnr1rl0ee6KPVQW2bhlXV8nsaH980K7u5FUVFRree6NcRERUVJOrclJTo62j6el5dn3zoTFRWlkpISHT9+3GFrTF5envr162efc/ToUaf1f/75Z6etPOWmTJmitLQ0++2CggLFxsYqKSlJwcHB9nGbzabMzEwlJibKz8/vIl6t+dGLCp7uxRvrD7g0f/yAK9xeQ7nqarEYpWp75gcdDGgnw+Lj1Voq47ZaNs6qcYrNaKbMwiuV2HKf/Cxl5wZ/m1b9g+r4XG7hwdocenFDquvP04jwvlnBU70o35NSG24NMfHx8YqKilJmZqa6d+8uSSopKdGGDRv05z//WZLUs2dP+fn5KTMzU8OHD5ck5eTkaOfOnXr55ZclSX379lV+fr62bt2qa6+9VpL09ddfKz8/3x50LmS1WmW1Wp3G/fz8Km1uVeNNEb2o4KlelAcCV+rwlNrUYlh87PO8Xcv53FZLeSipzXNayipCTF2e34Xnuij1UJufpYz3iv/hfbOCu3vhylouh5jCwkIdOFDxf0/Z2dnKyspSaGio2rRpo9TUVM2YMUPt27dX+/btNWPGDAUGBmrEiBGSpJCQEI0ZM0aTJk1SWFiYQkNDNXnyZHXp0sV+ttJVV12lQYMGaezYsVqwYIEkady4cRo8eDBnJgEAAEl1CDHffPONBgwYYL9dvgtn5MiRWrx4sZ588kmdPn1ajzzyiI4fP67evXtr7dq1CgoKsj9m9uzZ8vX11fDhw3X69GnddNNNWrx4sXx8Kv6vbOnSpZo4caL9LKahQ4dWeW0aAADQ9LgcYhISEmQYRpX3WywWpaenKz09vco5AQEBmjNnjubMmVPlnNDQUGVkZLhaHgAAaCL47CQAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKvt4uAEDjMjtzn7dLANBEsCUGAACYEiEGAACYEruTAJNj9w2ApootMQAAwJQIMQAAwJQIMQAAwJQ4JgY4zxvrD8iw+NQ47/HEK+uhGgBAddgSAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIlPsQaAxmD9TG9XANQ7tsQAAABTIsQAAABTcnuIadu2rSwWi9PX+PHjJUmjRo1yuq9Pnz4OaxQXF2vChAkKDw9XixYtNHToUB05csTdpQIAABNze4jZtm2bcnJy7F+ZmZmSpDvvvNM+Z9CgQQ5zVq1a5bBGamqqVq5cqeXLl+urr75SYWGhBg8erNLSUneXCwAATMrtB/ZecsklDrdfeukltWvXTv3797ePWa1WRUVFVfr4/Px8vf3223r33Xd18803S5IyMjIUGxurdevWaeDAge4uGQAAmJBHz04qKSlRRkaG0tLSZLFY7ONffPGFIiIi1KpVK/Xv318vvviiIiIiJEnbt2+XzWZTUlKSfX5MTIw6d+6sTZs2VRliiouLVVxcbL9dUFAgSbLZbLLZbPbx8u/PH2uq6EWF8h5YjNpt7XO1Z7Vdty7ru7p2bdY6f01Pv1ZXuO131ah5I7Ttf3Ns58+ty/PX4rkaOodeNPH3C943K3iqF669/xmG4dZnP897772nESNG6Mcff1RMTIwkacWKFWrZsqXi4uKUnZ2tqVOn6uzZs9q+fbusVquWLVum0aNHOwQSSUpKSlJ8fLwWLFhQ6XOlp6dr+vTpTuPLli1TYGCg+18cAABwu6KiIo0YMUL5+fkKDg6udq5HQ8zAgQPl7++vjz/+uMo5OTk5iouL0/LlyzVs2LAqQ0xiYqLatWun+fPnV7pOZVtiYmNj9csvvzg0wWazKTMzU4mJifLz87vIV2hu9KJCeS8OBrSTYfGpcf74AVe4tP4b6w+4NN+V9V1duzoWo1Rtz/zg0AdPv1ZXuFpLlTbOqnGKzWimzMIrldhyn/wsZecGf5vmkedq6Bx6cUOqt8vxKt43K3iqFwUFBQoPD69ViPHY7qRDhw5p3bp1ev/996udFx0drbi4OO3fv1+SFBUVpZKSEh0/flytW7e2z8vLy1O/fv2qXMdqtcpqtTqN+/n5VdrcqsabInpRwbD41CrEuNqv2qxZ1/VdXbu2a5av6+nX6gq3/Z6Wh5LaPKelrCLE1OX5XXiuhs7PUsZ7xf/wvlnB3b1wZS2P7axdtGiRIiIidOutt1Y779ixYzp8+LCio6MlST179pSfn5/9rCbp3NaanTt3VhtiAABA0+KRLTFlZWVatGiRRo4cKV/fiqcoLCxUenq67rjjDkVHR+vgwYN65plnFB4erttvv12SFBISojFjxmjSpEkKCwtTaGioJk+erC5dutjPVgIAAPBIiFm3bp1+/PFHPfDAAw7jPj4++v777/XOO+/oxIkTio6O1oABA7RixQoFBQXZ582ePVu+vr4aPny4Tp8+rZtuukmLFy+Wj4/nNlMDAABz8UiISUpKUmXHCzdv3lxr1qyp8fEBAQGaM2eO5syZ44nyAABAI2D+CxgAAIAmiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMiRADAABMydfbBQBAuT4/Lqx+wvow57EBUzxTDIAGjy0xAADAlNgSA8Dc1s/0dgUAvIQtMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJQIMQAAwJTcHmLS09NlsVgcvqKiouz3G4ah9PR0xcTEqHnz5kpISNCuXbsc1iguLtaECRMUHh6uFi1aaOjQoTpy5Ii7SwUAACbmkS0xnTp1Uk5Ojv3r+++/t9/38ssva9asWZo7d662bdumqKgoJSYm6uTJk/Y5qampWrlypZYvX66vvvpKhYWFGjx4sEpLSz1RLgAAMCFfjyzq6+uw9aWcYRh67bXX9Oyzz2rYsGGSpCVLligyMlLLli3TQw89pPz8fL399tt69913dfPNN0uSMjIyFBsbq3Xr1mngwIGeKBkAAJiMR0LM/v37FRMTI6vVqt69e2vGjBm6/PLLlZ2drdzcXCUlJdnnWq1W9e/fX5s2bdJDDz2k7du3y2azOcyJiYlR586dtWnTpipDTHFxsYqLi+23CwoKJEk2m002m80+Xv79+WNNFb2oUN4Di1G7rX2u9qy269ZlfVfXrs1a56/p6dd6vjKLT7X324z6O4yv/LkcnrMu/1bqsWZPcehFE3+/4H2zgqd64dr7n2EY7nzyzz77TEVFRbryyit19OhRvfDCC/r3v/+tXbt2ae/evbruuuv03//+VzExMfbHjBs3TocOHdKaNWu0bNkyjR492iGQSFJSUpLi4+O1YMGCSp83PT1d06dPdxpftmyZAgMD3fkSAQCAhxQVFWnEiBHKz89XcHBwtXPdviUmOTnZ/n2XLl3Ut29ftWvXTkuWLFGfPn0kSRaLxeExhmE4jV2opjlTpkxRWlqa/XZBQYFiY2OVlJTk0ASbzabMzEwlJibKz8/PpdfW2NCLCuW9OBjQTkYNWwMkafyAK1xa/431B1ya78r6rq5dHYtRqrZnfnDog6df6/muObK42vuvbRta57VdZTOaKbPwSiW23Cc/S9m5wd+mVf+gymyc5d7CvMChFzekerscr+J9s4KnelG+J6U2PLI76XwtWrRQly5dtH//ft12222SpNzcXEVHR9vn5OXlKTIyUpIUFRWlkpISHT9+XK1bt3aY069fvyqfx2q1ymq1Oo37+flV2tyqxpsielHBsPjUKsS42q/arFnX9V1du7Zrlq/r6dd6vmY17Iqyh4l65Gcpq3jeuvw78ULNnuJnKeO94n9436zg7l64spbHd9YWFxdrz549io6OVnx8vKKiopSZmWm/v6SkRBs2bLAHlJ49e8rPz89hTk5Ojnbu3FltiAEAAE2L27fETJ48WUOGDFGbNm2Ul5enF154QQUFBRo5cqQsFotSU1M1Y8YMtW/fXu3bt9eMGTMUGBioESNGSJJCQkI0ZswYTZo0SWFhYQoNDdXkyZPVpUsX+9lKgLfNztzn7RIAoMlze4g5cuSI7rnnHv3yyy+65JJL1KdPH23ZskVxcXGSpCeffFKnT5/WI488ouPHj6t3795au3atgoKC7GvMnj1bvr6+Gj58uE6fPq2bbrpJixcvlo+P+zebAwAAc3J7iFm+fHm191ssFqWnpys9Pb3KOQEBAZozZ47mzJnj5uoAAEBj4fEDewHAWzb/51it5/a9PMyDlQDwBEIMAKBhWj/T9ccMmOL+OtBgmf9SkgAAoEkixAAAAFMixAAAAFPimBh4lSvXW3k88UoPVgIAMBu2xAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFMixAAAAFPy9XYBAOpXnx8XVnlfmcVHv4T10zVHFquZUXpucH2YNGBKPVUHALXHlhgAAGBKhBgAAGBK7E6CaczO3OfS/McTr/RQJQCAhoAQA6Bam/9zTPrP5FrP7/O//25pM84zBQHA/xBiAACetX6mtytAI0WIAQD9b4vTecosPlKYtPXgr/Yztbacrdilye5KwPvcfmDvzJkzdc011ygoKEgRERG67bbbtHfvXoc5o0aNksVicfjq06ePw5zi4mJNmDBB4eHhatGihYYOHaojR464u1wAAGBSbg8xGzZs0Pjx47VlyxZlZmbq7NmzSkpK0qlTpxzmDRo0SDk5OfavVatWOdyfmpqqlStXavny5frqq69UWFiowYMHq7S01N0lAwAAE3L77qTVq1c73F60aJEiIiK0fft23XDDDfZxq9WqqKioStfIz8/X22+/rXfffVc333yzJCkjI0OxsbFat26dBg4c6O6yAaBGDhcKXB/mvUIASKqHY2Ly8/MlSaGhoQ7jX3zxhSIiItSqVSv1799fL774oiIiIiRJ27dvl81mU1JSkn1+TEyMOnfurE2bNlUaYoqLi1VcXGy/XVBQIEmy2Wyy2Wz28fLvzx9rqhpCLyyG57asufK6yud6sh5XuFK7qzWXWXxqvK+6ObVVl17W9Lw2w7WNxxfzOmrqhau1mFn5a7UZzaS6vF/UZ688/H7WEN43GwpP9cK19z/DMNz67OcxDEO/+93vdPz4cW3cuNE+vmLFCrVs2VJxcXHKzs7W1KlTdfbsWW3fvl1Wq1XLli3T6NGjHUKJJCUlJSk+Pl4LFixweq709HRNnz7daXzZsmUKDAx0/4sDAABuV1RUpBEjRig/P1/BwcHVzvXolphHH31U3333nb766iuH8bvuusv+fefOndWrVy/FxcXp008/1bBhw6pczzAMWSyWSu+bMmWK0tLS7LcLCgoUGxurpKQkhybYbDZlZmYqMTFRfn5+dX1pjUJD6MUb6w94bO3xA66o9dzyXhwMaCfDDVshLpYrtbvaw2uOLK7yvjKLj34N7a3QX7+u+OykOtp22SiXH1NdbZJ0bdvQau+/0NaDv7pcQ7maeuFqLWZmM5ops/BKJbbcJ78bUl1fYOMst9dUpd+m1TznIjSE982GwlO9KN+TUhseCzETJkzQRx99pC+//FKXXXZZtXOjo6MVFxen/fv3S5KioqJUUlKi48ePq3Xr1vZ5eXl56tevX6VrWK1WWa1Wp3E/P79Km1vVeFPkzV54MjDU5TUZFp8GEWJcqd3VemsTTpoZpRcdYurSx5qe089S5tb1artGZeu4Wktj4Gcpq9t7RX32qp7ey/gbUsHdvXBlLbfvqDQMQ48++qjef/99/fOf/1R8fHyNjzl27JgOHz6s6OhoSVLPnj3l5+enzMxM+5ycnBzt3LmzyhADAACaFrdviRk/fryWLVumDz/8UEFBQcrNzZUkhYSEqHnz5iosLFR6erruuOMORUdH6+DBg3rmmWcUHh6u22+/3T53zJgxmjRpksLCwhQaGqrJkyerS5cu9rOVAABA0+b2EDNv3jxJUkJCgsP4okWLNGrUKPn4+Oj777/XO++8oxMnTig6OloDBgzQihUrFBQUZJ8/e/Zs+fr6avjw4Tp9+rRuuukmLV68WD4+3t/UDwAAvM/tIaamk52aN2+uNWvW1LhOQECA5syZozlz5rirNAAA0Ijw2UkATOPCzzcC0LQ1nas1AQCARoUQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATIkQAwAATImPHWiCZmfukyRZjFLFS3pj/QEZlso/WPPxxCvrsTIAAGqPLTEAAMCUCDEAAMCUCDEAAMCUOCYGaIDKj1sCUA/Wz6z9XKOZpI4eKwWuIcQADUSfHxd6uwQAMBV2JwEAAFNiSwwA1MHm/xxzaX7fy8M8VAnQdLElBgAAmBIhBgAAmBK7k1AtV8+S4Qq/AID6wpYYAABgSoQYAABgSuxOAuARXPcGgKexJQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSB/YCABoPVz6RGqbHlhgAAGBKbIkBANQeWzrQgBBiAKCB4ROygdohxADVqMsF27a0GeeBSgAAFyLEoNFy5cMrLUap4j1YCwDA/TiwFwAAmBIhBgAAmBK7k+qBK7s1JOnxxCs9VEnTVt3xLWUWH/0S1k/XHFmsZkZpPVYFAKgrtsQAAABTYktMA2TmLTeu1g4AQF01+BDz5ptv6pVXXlFOTo46deqk1157Tb/97W+9XRYANBhcVwZNVYMOMStWrFBqaqrefPNNXXfddVqwYIGSk5O1e/dutWnTxtvlwU0a27VY6vJ60Pi5GjQASXW7QvKAKe6vo4Fq0CFm1qxZGjNmjB588EFJ0muvvaY1a9Zo3rx5mjmTS183ZQQFoH5s/s8xlVl8pDBp68Ffazzwna081eAjG9yuwYaYkpISbd++XU8//bTDeFJSkjZt2uSlqio0pGM/GlItAADUlwYbYn755ReVlpYqMjLSYTwyMlK5ublO84uLi1VcXGy/nZ+fL0n69ddfZbPZ7OM2m01FRUU6duyY/Pz86lxfcWF+nR/bUFiMUhUVF6m4NF+Gxcfp/h4/LXV5zR0x97r8mJNnzrr8GHcrsxgqKirSyTNnm/Qp1vShQmPuxdrdR12a70ovXF3bFT3btPbY2rVlM5rV/W9IYYlnirrQsfrZdemuv6cXOnnypCTJMIwa5zbYEFPOYrE43DYMw2lMkmbOnKnp06c7jcfHczH5+vW6twsAgCYu3dsFuMXJkycVEhJS7ZwGG2LCw8Pl4+PjtNUlLy/PaeuMJE2ZMkVpaWn222VlZfr1118VFhbmEHoKCgoUGxurw4cPKzg42HMvwAToRQV6cQ59qEAvKtCLCvSigqd6YRiGTp48qZiYmBrnNtgQ4+/vr549eyozM1O33367fTwzM1O/+93vnOZbrVZZrVaHsVatWlW5fnBwcJP/BSxHLyrQi3PoQwV6UYFeVKAXFTzRi5q2wJRrsCFGktLS0pSSkqJevXqpb9++WrhwoX788Uc9/PDD3i4NAAB4WYMOMXfddZeOHTum559/Xjk5OercubNWrVqluLg4b5cGAAC8rEGHGEl65JFH9Mgjj7htPavVqmnTpjntemqK6EUFenEOfahALyrQiwr0okJD6IXFqM05TAAAAA0Mn2INAABMiRADAABMiRADAABMiRADAABMqcmFmDfffFPx8fEKCAhQz549tXHjRm+XVK9mzpypa665RkFBQYqIiNBtt92mvXv3erusBmHmzJmyWCxKTU31dile8d///lf33XefwsLCFBgYqKuvvlrbt2/3dln17uzZs/rjH/+o+Ph4NW/eXJdffrmef/55lZWVebs0j/vyyy81ZMgQxcTEyGKx6IMPPnC43zAMpaenKyYmRs2bN1dCQoJ27drlnWI9rLpe2Gw2PfXUU+rSpYtatGihmJgY3X///frpp5+8V7CH1PQ7cb6HHnpIFotFr732Wr3V16RCzIoVK5Samqpnn31W//rXv/Tb3/5WycnJ+vHHH71dWr3ZsGGDxo8fry1btigzM1Nnz55VUlKSTp065e3SvGrbtm1auHChunbt6u1SvOL48eO67rrr5Ofnp88++0y7d+/WX/7yl2qvet1Y/fnPf9b8+fM1d+5c7dmzRy+//LJeeeUVzZkzx9uledypU6fUrVs3zZ07t9L7X375Zc2aNUtz587Vtm3bFBUVpcTERPsH9jUm1fWiqKhIO3bs0NSpU7Vjxw69//772rdvn4YOHeqFSj2rpt+Jch988IG+/vrrWn1UgFsZTci1115rPPzwww5jHTt2NJ5++mkvVeR9eXl5hiRjw4YN3i7Fa06ePGm0b9/eyMzMNPr372889thj3i6p3j311FPG9ddf7+0yGoRbb73VeOCBBxzGhg0bZtx3331eqsg7JBkrV6603y4rKzOioqKMl156yT525swZIyQkxJg/f74XKqw/F/aiMlu3bjUkGYcOHaqforygqj4cOXLEuPTSS42dO3cacXFxxuzZs+utpiazJaakpETbt29XUlKSw3hSUpI2bdrkpaq8Lz8/X5IUGhrq5Uq8Z/z48br11lt18803e7sUr/noo4/Uq1cv3XnnnYqIiFD37t311ltvebssr7j++uv1+eefa9++fZKkb7/9Vl999ZVuueUWL1fmXdnZ2crNzXV4D7Varerfv3+Tfg8tl5+fL4vF0uS2XpaVlSklJUVPPPGEOnXqVO/P3+Cv2Osuv/zyi0pLS50+ATsyMtLpk7KbCsMwlJaWpuuvv16dO3f2djlesXz5cu3YsUPbtm3zdile9Z///Efz5s1TWlqannnmGW3dulUTJ06U1WrV/fff7+3y6tVTTz2l/Px8dezYUT4+PiotLdWLL76oe+65x9uleVX5+2Rl76GHDh3yRkkNxpkzZ/T0009rxIgRTe5DIf/85z/L19dXEydO9MrzN5kQU85isTjcNgzDaaypePTRR/Xdd9/pq6++8nYpXnH48GE99thjWrt2rQICArxdjleVlZWpV69emjFjhiSpe/fu2rVrl+bNm9fkQsyKFSuUkZGhZcuWqVOnTsrKylJqaqpiYmI0cuRIb5fndbyHOrLZbLr77rtVVlamN99809vl1Kvt27fr9ddf144dO7z2O9BkdieFh4fLx8fHaatLXl6e0/9ZNAUTJkzQRx99pPXr1+uyyy7zdjlesX37duXl5alnz57y9fWVr6+vNmzYoL/+9a/y9fVVaWmpt0usN9HR0frNb37jMHbVVVc1qYPeyz3xxBN6+umndffdd6tLly5KSUnR448/rpkzZ3q7NK+KioqSJN5Dz2Oz2TR8+HBlZ2crMzOzyW2F2bhxo/Ly8tSmTRv7e+ihQ4c0adIktW3btl5qaDIhxt/fXz179lRmZqbDeGZmpvr16+elquqfYRh69NFH9f777+uf//yn4uPjvV2S19x00036/vvvlZWVZf/q1auX7r33XmVlZcnHx8fbJdab6667zulU+3379jXJT4wvKipSs2aOb40+Pj5N4hTr6sTHxysqKsrhPbSkpEQbNmxoUu+h5coDzP79+7Vu3TqFhYV5u6R6l5KSou+++87hPTQmJkZPPPGE1qxZUy81NKndSWlpaUpJSVGvXr3Ut29fLVy4UD/++KMefvhhb5dWb8aPH69ly5bpww8/VFBQkP3/qkJCQtS8eXMvV1e/goKCnI4FatGihcLCwprcMUKPP/64+vXrpxkzZmj48OHaunWrFi5cqIULF3q7tHo3ZMgQvfjii2rTpo06deqkf/3rX5o1a5YeeOABb5fmcYWFhTpw4ID9dnZ2trKyshQaGqo2bdooNTVVM2bMUPv27dW+fXvNmDFDgYGBGjFihBer9ozqehETE6Pf//732rFjhz755BOVlpba30tDQ0Pl7+/vrbLdrqbfiQvDm5+fn6KiotShQ4f6KbDezoNqIN544w0jLi7O8Pf3N3r06NHkTi2WVOnXokWLvF1ag9BUT7E2DMP4+OOPjc6dOxtWq9Xo2LGjsXDhQm+X5BUFBQXGY489ZrRp08YICAgwLr/8cuPZZ581iouLvV2ax61fv77S94eRI0cahnHuNOtp06YZUVFRhtVqNW644Qbj+++/927RHlJdL7Kzs6t8L12/fr23S3ermn4nLlTfp1hbDMMw6icuAQAAuE+TOSYGAAA0LoQYAABgSoQYAABgSoQYAABgSoQYAABgSoQYAABgSoQYAABgSoQYAABgSoQYAKaUkJCg1NRUp/HFixerVatW9V4PgPpHiAEAAKbUpD4AEoB5JCQk2D+IMyMjQz4+PvrDH/6gP/3pT7JYLF6uDkBDwJYYAA3WkiVL5Ovrq6+//lp//etfNXv2bP3tb3/zdlkAGgg+ABJAg5SQkKC8vDzt2rXLvuXl6aef1kcffaTdu3crISFBmzZtkr+/v8Pjzp49q4CAAJ04ccILVQOoT2yJAdBg9enTx2HXUd++fbV//36VlpZKku69915lZWU5fD3//PPeKhdAPeOYGACmFRISoiuuuMJhLCIiwkvVAKhvbIkB0GBt2bLF6Xb79u3l4+PjpYoANCSEGAAN1uHDh5WWlqa9e/fq73//u+bMmaPHHnvM22UBaCDYnQSgwbr//vt1+vRpXXvttfLx8dGECRM0btw4b5cFoIHg7CQADVJCQoKuvvpqvfbaa94uBUADxe4kAABgSoQYAABgSuxOAgAApsSWGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEr/H7hP9Gac5J0GAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ax = ph_min_df.ph_min.hist(bins=30, alpha=0.5, label='min pH')\n",
    "ph_max_df.ph_max.hist(bins=30, ax=ax, alpha=0.5, label='max pH')\n",
    "ax.set_title('Brenda pH range data')\n",
    "ax.set_xlabel('pH')\n",
    "plt.legend()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1a451223-843d-41cd-81fc-db764785d136",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.legend.Legend at 0x7f47013c6410>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjEAAAHFCAYAAAADhKhmAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjYuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/P9b71AAAACXBIWXMAAA9hAAAPYQGoP6dpAABHE0lEQVR4nO3de1yUZf7/8fcIwwAKKBCnFZHMQ+spD3nqIFaglvrdbNfKIi3T2swidCtrTWxL2w7arpaHtq9a6mr7/WVHM7FM7aGm6VJ5WA8tmm0gZQoiCgPcvz9cJseBgUFg5obX8/HgAXPd133fn/saHN5e933PWAzDMAQAAGAyzbxdAAAAQG0QYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYgAAgCkRYoA6tGTJElksFqevSy65RElJSfrggw+8XV6l2rZtq7Fjxzb4fg8fPiyLxaIXX3yx0uUvvviiLBaLDh8+3LCF+bjaPl9FRUXKyMjQZ599Vuc1Ad5CiAHqweLFi7V161Zt2bJFixYtkp+fn4YPH67333/f26WhiSoqKtKMGTMIMWhU/L1dANAYdenSRb1793Y8HjJkiFq1aqW///3vGj58eJXrlZWVqbS0VDabrSHKbLSKiooUHBzs7TIA1DNmYoAGEBgYqICAAFmtVkdbxemU559/Xs8884wSExNls9m0YcMGSdKXX36pESNGKDw8XIGBgerRo4feeustp+1WnL7asGGDfv/73ysyMlIREREaOXKkfvjhB6e+drtdjz76qGJiYhQcHKyrr75a27dvd6n1xx9/1AMPPKBf//rXatGihaKionTddddp8+bNNTrWtm3batiwYVq9erW6deumwMBAXXrppfrrX//q6bDVSEZGhiwWi3bt2qXf/va3atWqldq1ayfp3Bjedtttatu2rYKCgtS2bVvdfvvtOnLkiNM2PBnH4uJiTZ482TGO1157rXbu3FnpaZ7c3Fzdd999at26tQICApSYmKgZM2aotLS02uOqy+fr8OHDuuSSSyRJM2bMcJzqrKj30KFDuvvuu9W+fXsFBwfrV7/6lYYPH65vvvmm2joBb2ImBqgHFTMqhmHo2LFjeuGFF3T69GmNHj3ape9f//pXdejQQS+++KJCQ0PVvn17bdiwQUOGDFHfvn21YMEChYWFaeXKlbr11ltVVFTk8sfy3nvv1U033aQVK1bo6NGj+sMf/qA777xTn376qaPP+PHj9cYbb2jKlClKTk7W7t27NXLkSJ06dcppWz///LMkafr06YqJiVFhYaFWr16tpKQkffLJJ0pKSqr2+LOyspSWlqaMjAzFxMRo+fLlevjhh1VSUqIpU6Y49S0vL6/0j3p5eXm1+znfyJEjddttt+n+++/X6dOnJZ37492xY0fddtttCg8PV05OjubPn68rr7xSe/fuVWRkpNM2ajKOd999t1atWqVHH31U1113nfbu3aubb75ZBQUFTtvKzc1Vnz591KxZMz311FNq166dtm7dqmeeeUaHDx/W4sWL3R5PXT5fsbGxWrt2rYYMGaJx48bp3nvvlSRHsPnhhx8UERGh5557Tpdccol+/vlnLV26VH379tU///lPdezY0aPnAmgwBoA6s3jxYkOSy5fNZjNeffVVp77Z2dmGJKNdu3ZGSUmJ07JOnToZPXr0MOx2u1P7sGHDjNjYWKOsrMxpfw888IBTv+eff96QZOTk5BiGYRj79u0zJBmPPPKIU7/ly5cbkowxY8ZUeUylpaWG3W43rr/+euPmm2+udgwSEhIMi8ViZGVlObUnJycboaGhxunTp52Ov7qv7Oxst/ubPn26Icl46qmnqq2ttLTUKCwsNJo3b2785S9/cbTXdBz37NljSDIee+wxp35///vfXcbxvvvuM1q0aGEcOXLEqe+LL75oSDL27NlTZZ318Xz9+OOPhiRj+vTpVa57/jZKSkqM9u3bu9QA+BJOJwH14I033tCOHTu0Y8cOffTRRxozZowmTpyoefPmufQdMWKE02mmQ4cO6V//+pfuuOMOSVJpaanj68Ybb1ROTo7279/vso3zdevWTZIcp00qTlFVbLPCqFGj5O/vOiG7YMEC9ezZU4GBgfL395fVatUnn3yiffv21ej4O3furO7duzu1jR49WgUFBdq1a5dT+8MPP+wYq/O/Hn744Rrtq8Itt9zi0lZYWKjHHntMl112mfz9/eXv768WLVro9OnTlR5LdeO4ceNGSefG7Xy//e1vXcbxgw8+0KBBgxQXF+f0HA4dOtRpW5Vp6OertLRUM2fO1K9//WsFBATI399fAQEBOnjwYI23AXgDp5OAenD55Ze7XNh75MgRPfroo7rzzjvVsmVLx7LY2FindY8dOyZJmjJlisuplwo//fST0+OIiAinxxUXBp85c0aSdPz4cUlSTEyMUz9/f3+XdWfPnq3Jkyfr/vvv15/+9CdFRkbKz89P06ZNq/EftAv3c35bRS0VWrdu7TRWFTy9i+bCcZTOBadPPvlE06ZN05VXXqnQ0FBZLBbdeOONjrE5X03HMTo62qlfZeN47Ngxvf/++04B9XwXPofna+jnKz09Xa+88ooee+wxDRw4UK1atVKzZs107733VjpOgK8gxAANpFu3bvr444914MAB9enTx9FusVic+lVcpzF16lSNHDmy0m15eo1CxR++3Nxc/epXv3K0l5aWuoSKZcuWKSkpSfPnz3dqv/BaDHdyc3OrbLvwj3BduXAc8/Pz9cEHH2j69Ol6/PHHHe3FxcWO60g8VVH7sWPHqh3HyMhIdevWTc8++2yl24qLi6t2Pw31fC1btkx33XWXZs6c6dT+008/OQVuwNcQYoAGkpWVJemXiymr0rFjR7Vv315fffWVyx+V2qq4GHf58uXq1auXo/2tt95yuajWYrG43OL99ddfa+vWrYqPj6/R/vbs2aOvvvrK6ZTSihUrFBISop49e9byKDxjsVhkGIbLsfztb39TWVlZrbZ57bXXSpJWrVrldBz/93//5zKOw4YN05o1a9SuXTu1atXKo/3Ux/N14axSddv48MMP9Z///EeXXXaZR7UDDYkQA9SD3bt3O/7YHD9+XG+//bYyMzN18803KzExsdr1Fy5cqKFDh2rw4MEaO3asfvWrX+nnn3/Wvn37tGvXLv3jH//wqJ7LL79cd955p15++WVZrVbdcMMN2r17t+OOqPMNGzZMf/rTnzR9+nQNHDhQ+/fv19NPP63ExMQa3RosnZtlGDFihDIyMhQbG6tly5YpMzNTf/7znxvs/VtCQ0N17bXX6oUXXlBkZKTatm2rjRs36vXXX6/17ELnzp11++2366WXXpKfn5+uu+467dmzRy+99JLCwsLUrNkvlxk+/fTTyszM1IABA/TQQw+pY8eOOnv2rA4fPqw1a9ZowYIFat26daX7qY/nKyQkRAkJCXr33Xd1/fXXKzw83DEuw4YN05IlS9SpUyd169ZNO3fu1AsvvFBlfYDP8PaVxUBjUtndSWFhYcYVV1xhzJ492zh79qyjb8XdOS+88EKl2/rqq6+MUaNGGVFRUYbVajViYmKM6667zliwYIHL/nbs2OG07oYNGwxJxoYNGxxtxcXFxuTJk42oqCgjMDDQ6Nevn7F161YjISHB6W6X4uJiY8qUKcavfvUrIzAw0OjZs6fxzjvvGGPGjDESEhKqHYOEhATjpptuMv7v//7P6Ny5sxEQEGC0bdvWmD17tlO/6o7/hRde8OjupB9//NFl2ffff2/ccsstRqtWrYyQkBBjyJAhxu7du12O2ZNxPHv2rJGenu4yjmFhYS538vz444/GQw89ZCQmJhpWq9UIDw83evXqZTz55JNGYWGh2+Oqj+dr/fr1Ro8ePQybzeZ0l9OJEyeMcePGGVFRUUZwcLBx9dVXG5s3bzYGDhxoDBw40G2dgDdZDMMwvJKeADRKbdu2VZcuXXz2s6Lqw5YtW3TVVVdp+fLllb4XEID6wekkAPBAZmamtm7dql69eikoKEhfffWVnnvuObVv377KC7EB1A9CDAB4IDQ0VOvWrdPLL7+sU6dOKTIyUkOHDtWsWbMUGBjo7fKAJoXTSQAAwJR4x14AAGBKhBgAAGBKhBgAAGBKjfbC3vLycv3www8KCQlxeTtyAADgmwzD0KlTpxQXF+f0BpKVabQh5ocffqjxW6QDAADfcvTo0WrfNbrRhpiQkBBJ5wah4m267Xa71q1bp5SUlCo/WbYpY3zcY3zcY3zcY3zcY3yq11TGqKCgQPHx8Y6/4+402hBTcQopNDTUKcQEBwcrNDS0Uf8C1Bbj4x7j4x7j4x7j4x7jU72mNkY1uRSEC3sBAIApEWIAAIApEWIAAIApNdprYgAAkKSysjLZ7XZvl3HR7Ha7/P39dfbsWZWVlXm7nFqzWq3y8/Ork20RYgAAjZJhGMrNzdXJkye9XUqdMAxDMTExOnr0qOnf/6xly5aKiYm56OMgxAAAGqWKABMVFaXg4GDT/+EvLy9XYWGhWrRoUe2bwPkqwzBUVFSkvLw8SVJsbOxFbY8QAwBodMrKyhwBJiIiwtvl1Iny8nKVlJQoMDDQtCFGkoKCgiRJeXl5ioqKuqhTS+YdBQAAqlBxDUxwcLCXK0FlKp6Xi71WiRADAGi0zH4KqbGqq+eFEAMAAEyJEAMAgMl99tlnslgsjeZOrJriwl4AQJMyJ/NAg+7vkeQO9b6PAQMGKCcnR2FhYfW6n88++0yDBg3SiRMn1LJlS6dlbdu2VVpamtLS0uq1hvMRYgAAMLmAgADFxMR4u4wGx+kkAAB8SFJSkiZNmqS0tDS1atVK0dHRWrRokU6fPq2JEycqLCxM7dq100cffeRY58LTSUuWLFHLli318ccf6/LLL1eLFi00ZMgQ5eTkVLnfim18+OGH6t69uwIDA9W3b19988039X3IteZxiNm0aZOGDx+uuLg4WSwWvfPOO07LLRZLpV8vvPCCo09SUpLL8ttuu81pOydOnFBqaqrCwsIUFham1NTUJneuDwDQNC1dulSRkZHavn27Jk2apN///vcaNWqU+vTpoy+//FKDBw9WamqqioqKqtxGUVGRXnzxRb355pvatGmTvvvuO02ZMqXaff/hD3/Qiy++qB07digqKkojRozw2Y9t8Ph00unTp9W9e3fdfffduuWWW1yWX5jyPvroI40bN86l7/jx4/X00087Hle8+U2F0aNH6/vvv9fatWslSRMmTFBqaqref/99T0sGgKZjwyzP1xk0te7rwEXp3r27/vjHP0qSpk6dqueee06RkZEaM2aMQkND9dRTT2n+/Pn6+uuv1a9fv0q3YbfbtWDBArVr106S9OCDDzr93a3K9OnTlZycLOlcmGrdurVWr16tUaNGOfq0bt3aZT13gaq+eBxihg4dqqFDh1a5/MJzcu+++64GDRqkSy+91Kk9ODi4yvN3+/bt09q1a7Vt2zb17dtXkvTaa6+pf//+2r9/vzp27Ohp2QAAmEa3bt0cP/v5+SkiIkJdu3Z1tEVHR0uS4+37KxMcHOwIMNK5t/h3179C//79HT+Hh4erY8eO2rdvn1OfzZs3KyQkxKktKSmp2m3XtXq9sPfYsWP68MMPtXTpUpdly5cv17JlyxQdHa2hQ4dq+vTpjgHZunWrwsLCHAFGkvr166ewsDBt2bKl0hBTXFys4uJix+OCggJJ55JoxTTYhd/hjPFxj/Fxj/Fxr8HGx6jFpY4+8JzV9fjY7XYZhqHy8nKVl5c7LTMMo072UVMX7r8m/P39ndazWCzy9z/3J/v8+ktLS52OseLn8vJyWa1Wp20YhuEYE3d1VjZmF7YnJCS43J3k7+/vdvsXbsswDNntdpePHfDkd6BeQ8zSpUsVEhKikSNHOrXfcccdSkxMVExMjHbv3q2pU6fqq6++UmZmpqRzH9oVFRXlsr2oqCjl5uZWuq9Zs2ZpxowZLu3r1q1zedvpiv2gcoyPe4yPe4yPe/U/Pp08X2XNmrovo5bqanz8/f0VExOjwsJClZSUOC07/z+8DaHiP9U1VVpaqpKSEqf1ysvLHXWfOnXK0X7mzBkVFBQ4TuWcOnVKzZo109mzZ2UYhtM2zpw547aeim1s2LBBN998syTp5MmTOnDggNq0aVPpfs5XXl6us2fP1uh4S0pKdObMGW3atEmlpaWV1lET9Rpi/vd//1d33HGHAgMDndrHjx/v+LlLly5q3769evfurV27dqlnz56SKn9LYsMwqnyr4qlTpyo9Pd3xuKCgQPHx8UpJSVFoaKikc+kuMzNTycnJslqtF318jQ3j4x7j4x7j416Djc/m2Z6vc0169X3qWV2Pz9mzZ3X06FG1aNHC5W+QzWa76O17ouJvUE35+/srICDAab1mzZo56g4JCXH8LQwKClJoaKjjP+shISEKDQ1VYGCgLBaL0zYqrj2tqp6Kbbz00ktq3bq1oqOj9cc//lGRkZG6/fbbFRAQ4LKf8zVr1kyBgYE1Ot6zZ88qKChI1157rcvz40noq7cQs3nzZu3fv1+rVq2qtm/Pnj1ltVp18OBB9ezZUzExMTp27JhLvx9//NFxHvBCNput0l9Mq9Xq8g+isjb8gvFxj/Fxj/Fxr97Hx+L5qQv50PNVV+NTVlYmi8WiZs2aucwYNPTnKdXmE6crar+w7cJlFcfn7vGFdVRVT0X7c889p0ceeUQHDx5U9+7d9d577zmCxoX7qUndVe3LYrFU+Te6puotxLz++uvq1auXunfvXm3fPXv2yG63KzY2VtK5i4ry8/O1fft29enTR5L0xRdfKD8/XwMGDKivkgEATUBDvIPuxfjss89c2g4fPqzy8nKnWYrzr41JSkpyejx27FiNHTvWaRu/+c1vanQ90NVXX63du3dXuuzC/VxYY0PzOMQUFhbq0KFDjsfZ2dnKyspSeHi42rRpI+ncVNA//vEPvfTSSy7rf/vtt1q+fLluvPFGRUZGau/evZo8ebJ69Oihq666SpJ0+eWXa8iQIRo/frwWLlwo6dwt1sOGDePOJAAAIKkWb3b35ZdfqkePHurRo4ckKT09XT169NBTTz3l6LNy5UoZhqHbb7/dZf2AgAB98sknGjx4sDp27KiHHnpIKSkpWr9+vdMVysuXL1fXrl2VkpKilJQUdevWTW+++WZtjhEAADRCHs/EuJtKqjBhwgRNmDCh0mXx8fHauHFjtfsJDw/XsmXLPC0PAADUQk3+vvsaPjsJAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYUr1+ACQAAD5nw6yG3d+gqQ27vwZmsVi0evVq/eY3v3FqHzt2rE6ePKl33nmn3vbNTAwAADAlQgwAAD4kKSlJkyZNUlpamlq1aqXo6GgtWrRIp0+f1sSJExUWFqZ27drpo48+cqxTVlamcePGKTExUUFBQerYsaP+8pe/OJafPXtWnTt3dno3/ezsbIWFhem1116rshaLxaL58+dr6NChCgoKUmJiov7xj3/Uz4HXAiEGAAAfs3TpUkVGRmr79u2aNGmSfv/732vUqFHq06ePvvzySw0ePFipqakqKiqSJJWXl6t169Z66623tHfvXj311FN64okn9NZbb0mSAgMDtXz5ci1dulTvvPOOysrKlJqaqkGDBmn8+PFua5k2bZpuueUWffXVV7rzzjt1++23a9++ffU+BjVBiAEAwMd0795df/zjH9W+fXtNnTpVQUFBioyM1JgxY9S+fXs99dRTOn78uL7++mtJktVq1YwZM3TllVcqMTFRd9xxh8aOHesIMZJ0xRVX6JlnntH48eP1yCOP6Ntvv9Xf/va3amv53e9+p3vvvVcdOnTQn/70J/Xu3Vtz58516nP77berRYsWTl/Lly+v20GpBBf2AgDgY7p16+b42c/PTxEREerataujLTo6WpKUl5fnaFuwYIH+9re/6ciRIzpz5oxKSkp0xRVXOG138uTJevfddzV37lx99NFHioyMrLaW/v37uzzOyspyapszZ45uuOEGp7bHHntMZWVl1W7/YjATAwCAj7FarU6PLRaLU5vFYpF07jSSJL311lt65JFHdM8992jdunXKysrS3XffrZKSEqft5OXlaf/+/fLz89PBgwdrXV/F/ivExMTosssuc/oKCQmp9fZrihADAIDJbd68WQMGDNADDzygHj166LLLLtO3337r0u+ee+5Rly5d9MYbb+jRRx/V3r17q932tm3bXB536tSpzmq/GJxOAgDA5C677DK98cYb+vjjj5WYmKg333xTO3bsUGJioqPPK6+8oq1bt+rrr79WfHy8PvroI91xxx364osvFBAQUOW2//GPf6h37966+uqrtXz5cm3fvl2vv/56QxxWtZiJAQDA5O6//36NHDlSt956q/r27avjx4/rgQcecCz/17/+pT/84Q969dVXFR8fL+lcqDl58qSmTZvmdtszZszQypUr1a1bNy1dulTLly/Xr3/963o9nppiJgYA0LT4+DvofvbZZy5thw8fVnl5uQoKChxthmE4frbZbFq8eLEWL17stN6sWefenbhTp06O27ErhIaGKjs7u9p64uLitG7duiqXn1/H+ZYsWVLtti8WMzEAAMCUCDEAAMCUOJ0EAAAqVdWpIl/BTAwAADAlQgwAoNHy9ZmEpqqunhdCDACg0al4d9sL78iBb6h4Xi58Z2JPcU0MAKDR8fPzU8uWLR2fLRQcHOzyVvlmU15erpKSEp09e1bNmplzDsIwDBUVFSkvL08tW7aUn5/fRW2PEAMAaJRiYmIkOX9IopkZhqEzZ84oKCjI9IGsZcuWjufnYhBiAACNksViUWxsrKKiomS3271dzkWz2+3atGmTrr322os+DeNNVqv1omdgKhBiAACNmp+fX5390fQmPz8/lZaWKjAw0NQhpi6Z86QaAABo8ggxAADAlAgxAADAlAgxAADAlAgxAADAlAgxAADAlAgxAADAlAgxAADAlAgxAADAlAgxAADAlDwOMZs2bdLw4cMVFxcni8Wid955x2n52LFjZbFYnL769evn1Ke4uFiTJk1SZGSkmjdvrhEjRuj777936nPixAmlpqYqLCxMYWFhSk1N1cmTJz0+QAAA0Dh5HGJOnz6t7t27a968eVX2GTJkiHJychxfa9ascVqelpam1atXa+XKlfr8889VWFioYcOGqayszNFn9OjRysrK0tq1a7V27VplZWUpNTXV03IBAEAj5fEHQA4dOlRDhw5128dms1X5Edv5+fl6/fXX9eabb+qGG26QJC1btkzx8fFav369Bg8erH379mnt2rXatm2b+vbtK0l67bXX1L9/f+3fv18dO3b0tGwAANDI1Ms1MZ999pmioqLUoUMHjR8/Xnl5eY5lO3fulN1uV0pKiqMtLi5OXbp00ZYtWyRJW7duVVhYmCPASFK/fv0UFhbm6AMAAJo2j2diqjN06FD97ne/U0JCgrKzszVt2jRdd9112rlzp2w2m3JzcxUQEKBWrVo5rRcdHa3c3FxJUm5urqKioly2HRUV5ehzoeLiYhUXFzseFxQUSJLsdrvsdrvj5/O/wxnj4x7j4x7j416DjY9Ri/+b+sBzxu9P9ZrKGHlyfHUeYm699VbHz126dFHv3r2VkJCgDz/8UCNHjqxyPcMwZLFYHI/P/7mqPuebNWuWZsyY4dK+bt06BQcHO7VlZmZWexxNGePjHuPjHuPjXv2PTyfPV7ngukVv4veneo19jIqKimrct85DzIViY2OVkJCggwcPSpJiYmJUUlKiEydOOM3G5OXlacCAAY4+x44dc9nWjz/+qOjo6Er3M3XqVKWnpzseFxQUKD4+XikpKQoNDZV0Lt1lZmYqOTlZVqu1zo6xsWB83GN83GN83Guw8dk82/N1rkmvvk894/enek1ljCrOpNREvYeY48eP6+jRo4qNjZUk9erVS1arVZmZmRo1apQkKScnR7t379bzzz8vSerfv7/y8/O1fft29enTR5L0xRdfKD8/3xF0LmSz2WSz2VzarVary5NdWRt+wfi4x/i4x/i4V+/jYyn3fB0fer74/aleYx8jT47N4xBTWFioQ4cOOR5nZ2crKytL4eHhCg8PV0ZGhm655RbFxsbq8OHDeuKJJxQZGambb75ZkhQWFqZx48Zp8uTJioiIUHh4uKZMmaKuXbs67la6/PLLNWTIEI0fP14LFy6UJE2YMEHDhg3jziQAACCpFiHmyy+/1KBBgxyPK07hjBkzRvPnz9c333yjN954QydPnlRsbKwGDRqkVatWKSQkxLHOnDlz5O/vr1GjRunMmTO6/vrrtWTJEvn5+Tn6LF++XA899JDjLqYRI0a4fW8aAADQtHgcYpKSkmQYRpXLP/7442q3ERgYqLlz52ru3LlV9gkPD9eyZcs8LQ8AADQR9X5NDIDGYU7mAbfLLUaZEiW9suGQDMu5WdVHkjs0QGUAmio+ABIAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJgSIQYAAJiSv7cLAIAmYfNsSZ3OfbeU12ydQVPrtSTA7JiJAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApkSIAQAApuRxiNm0aZOGDx+uuLg4WSwWvfPOO45ldrtdjz32mLp27armzZsrLi5Od911l3744QenbSQlJclisTh93XbbbU59Tpw4odTUVIWFhSksLEypqak6efJkrQ4SAAA0Ph6HmNOnT6t79+6aN2+ey7KioiLt2rVL06ZN065du/T222/rwIEDGjFihEvf8ePHKycnx/G1cOFCp+WjR49WVlaW1q5dq7Vr1yorK0upqamelgsAABopf09XGDp0qIYOHVrpsrCwMGVmZjq1zZ07V3369NF3332nNm3aONqDg4MVExNT6Xb27duntWvXatu2berbt68k6bXXXlP//v21f/9+dezY0dOyAQBAI1Pv18Tk5+fLYrGoZcuWTu3Lly9XZGSkOnfurClTpujUqVOOZVu3blVYWJgjwEhSv379FBYWpi1bttR3yQAAwAQ8nonxxNmzZ/X4449r9OjRCg0NdbTfcccdSkxMVExMjHbv3q2pU6fqq6++cszi5ObmKioqymV7UVFRys3NrXRfxcXFKi4udjwuKCiQdO46Hbvd7vj5/O9wxvi419THx2KU1Wj5+f2a6lhVxm40c/pes5VqMX6ebP9i9lPHmvq/r5poKmPkyfHVW4ix2+267bbbVF5erldffdVp2fjx4x0/d+nSRe3bt1fv3r21a9cu9ezZU5JksVhctmkYRqXtkjRr1izNmDHDpX3dunUKDg52arvwlBecMT7uNdXxSaxhv7Znv3X8vGbNgfopxpQ6SJIyCzvUfJU1a2qxn06er1Kr/dSPpvrvyxONfYyKiopq3LdeQozdbteoUaOUnZ2tTz/91GkWpjI9e/aU1WrVwYMH1bNnT8XExOjYsWMu/X788UdFR0dXuo2pU6cqPT3d8bigoEDx8fFKSUlx7N9utyszM1PJycmyWq0XcYSNE+PjXlMfn1c2HHK73GKUqe3Zb3U4sJ0Mi58kaeKgyxqiNFOwb3pZmYUdlNzigKyW8pqtdE169X0utHm25+vUZj91rKn/+6qJpjJGFWdSaqLOQ0xFgDl48KA2bNigiIiIatfZs2eP7Ha7YmNjJUn9+/dXfn6+tm/frj59+kiSvvjiC+Xn52vAgAGVbsNms8lms7m0W61Wlye7sjb8gvFxr6mOT0UwqUm/ir5NcZyq9N/gYrWU1zzE1Gb8arrti91PPWmq/7480djHyJNj8zjEFBYW6tChX/5Hlp2draysLIWHhysuLk6//e1vtWvXLn3wwQcqKytzXMMSHh6ugIAAffvtt1q+fLluvPFGRUZGau/evZo8ebJ69Oihq666SpJ0+eWXa8iQIRo/frzj1usJEyZo2LBh3JkEAAAk1SLEfPnllxo0aJDjccUpnDFjxigjI0PvvfeeJOmKK65wWm/Dhg1KSkpSQECAPvnkE/3lL39RYWGh4uPjddNNN2n69Ony8/vlf3rLly/XQw89pJSUFEnSiBEjKn1vGgAA0DR5HGKSkpJkGEaVy90tk6T4+Hht3Lix2v2Eh4dr2bJlnpYHAACaCD47CQAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmJLHHwAJoH7NyTzg8TqPJHeoh0oAwLcxEwMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEyJEAMAAEzJ39sFAMDFmJN5wON1HknuUA+VAGhozMQAAABTIsQAAABT4nQSAKBhbJhV9TKjmaRO0ubZkqX8l/ZBU+u9LJgXMzEAAMCUCDEAAMCUPA4xmzZt0vDhwxUXFyeLxaJ33nnHablhGMrIyFBcXJyCgoKUlJSkPXv2OPUpLi7WpEmTFBkZqebNm2vEiBH6/vvvnfqcOHFCqampCgsLU1hYmFJTU3Xy5EmPDxAAADROHoeY06dPq3v37po3b16ly59//nnNnj1b8+bN044dOxQTE6Pk5GSdOnXK0SctLU2rV6/WypUr9fnnn6uwsFDDhg1TWVmZo8/o0aOVlZWltWvXau3atcrKylJqamotDhEAADRGHl/YO3ToUA0dOrTSZYZh6OWXX9aTTz6pkSNHSpKWLl2q6OhorVixQvfdd5/y8/P1+uuv680339QNN9wgSVq2bJni4+O1fv16DR48WPv27dPatWu1bds29e3bV5L02muvqX///tq/f786duxY2+MFAACNRJ3enZSdna3c3FylpKQ42mw2mwYOHKgtW7bovvvu086dO2W32536xMXFqUuXLtqyZYsGDx6srVu3KiwszBFgJKlfv34KCwvTli1bKg0xxcXFKi4udjwuKCiQJNntdtntdsfP53+HM8bHvYYaH4tRVn2nCzTEc1ZdXRXLz+/nC3VVxhu/43ajmdP3mq1Uizo92f7F7Kc23NRW5fjweuTQVF6jPTm+Og0xubm5kqTo6Gin9ujoaB05csTRJyAgQK1atXLpU7F+bm6uoqKiXLYfFRXl6HOhWbNmacaMGS7t69atU3BwsFNbZmZmDY+oaWJ83Kvv8UmsxTpr1nj+rrWeqmldbc9+6/jZl+o6X0PU5ercuwRnFnrwbsFr1tRiP508X6VW+6mN6mtzGZ8Gq808GvtrdFFRUY371sv7xFgsFqfHhmG4tF3owj6V9Xe3nalTpyo9Pd3xuKCgQPHx8UpJSVFoaKikc+kuMzNTycnJslqtNT6epoLxca+hxueVDYc8XmfioMvqoRJn1dVlMcrU9uy3OhzYTobFz2fqqkxD1HUh+6aXlVnYQcktDsh6/vuguHNNevV9LrR5tufr1GY/teGmNrvRrPLxaajaTKCpvEZXnEmpiToNMTExMZLOzaTExsY62vPy8hyzMzExMSopKdGJEyecZmPy8vI0YMAAR59jx465bP/HH390meWpYLPZZLPZXNqtVqvLk11ZG37B+LhX3+NTEQA80RDPV03rMix+jr6+VNf5vPL7/d8/zFZLec1DTG3qrOm2L3Y/tVGD2lzGh9ciF439NdqTY6vT94lJTExUTEyM01RXSUmJNm7c6AgovXr1ktVqdeqTk5Oj3bt3O/r0799f+fn52r59u6PPF198ofz8fEcfAADQtHk8E1NYWKhDh36Zvs3OzlZWVpbCw8PVpk0bpaWlaebMmWrfvr3at2+vmTNnKjg4WKNHj5YkhYWFady4cZo8ebIiIiIUHh6uKVOmqGvXro67lS6//HINGTJE48eP18KFCyVJEyZM0LBhw7gzCQAASKpFiPnyyy81aNAgx+OK61DGjBmjJUuW6NFHH9WZM2f0wAMP6MSJE+rbt6/WrVunkJAQxzpz5syRv7+/Ro0apTNnzuj666/XkiVL5Of3y7Tw8uXL9dBDDznuYhoxYkSV700DAACaHo9DTFJSkgzDqHK5xWJRRkaGMjIyquwTGBiouXPnau7cuVX2CQ8P17JlyzwtDwAANBF8dhIAADAlQgwAADAlQgwAADAlQgwAADClennHXgBAHdgwy9sVAD6NmRgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBKhBgAAGBK/t4uAAAqzMk84O0SAJgIMzEAAMCUCDEAAMCUOJ0ENFGcugFgdszEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAU+IWa8BDr2w4JMPiV+P+jyR3qMdqAKDpYiYGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYEiEGAACYUp2HmLZt28pisbh8TZw4UZI0duxYl2X9+vVz2kZxcbEmTZqkyMhINW/eXCNGjND3339f16UCAAATq/MQs2PHDuXk5Di+MjMzJUm/+93vHH2GDBni1GfNmjVO20hLS9Pq1au1cuVKff755yosLNSwYcNUVlZW1+UCAACTqvN37L3kkkucHj/33HNq166dBg4c6Giz2WyKiYmpdP38/Hy9/vrrevPNN3XDDTdIkpYtW6b4+HitX79egwcPruuSAQCACdXrxw6UlJRo2bJlSk9Pl8VicbR/9tlnioqKUsuWLTVw4EA9++yzioqKkiTt3LlTdrtdKSkpjv5xcXHq0qWLtmzZUmWIKS4uVnFxseNxQUGBJMlut8tutzt+Pv87nDE+7lWMi8XwbEbQ0/H0dPu12Udt91OT7Z2/3YY49trwxu+43Wjm9N2nNNR4uDn2KseH1yOHpvIa7cnxWQzDMOqrkLfeekujR4/Wd999p7i4OEnSqlWr1KJFCyUkJCg7O1vTpk1TaWmpdu7cKZvNphUrVujuu+92CiSSlJKSosTERC1cuLDSfWVkZGjGjBku7StWrFBwcHDdHxwAAKhzRUVFGj16tPLz8xUaGuq2b72GmMGDBysgIEDvv/9+lX1ycnKUkJCglStXauTIkVWGmOTkZLVr104LFiyodDuVzcTEx8frp59+cgyC3W5XZmamkpOTZbVa6+AIGxfGx72K8Tkc2M6jD4CcOOgyj/bzyoZDnpbm8T5qux93LEaZ2p791ml8GuLYa6M243Wx7JteVmZhByW3OCCrpbzB9+/WNekNs5/Ns6tcZDeaVT4+DVWbCTSV1+iCggJFRkbWKMTU2+mkI0eOaP369Xr77bfd9ouNjVVCQoIOHjwoSYqJiVFJSYlOnDihVq1aOfrl5eVpwIABVW7HZrPJZrO5tFutVpcnu7I2/ILxcc+w+HkUYjwdS0+2Xdt91HY/Nd1uxbYb4thrwyu/3//9w2y1lPteiGmo8ajBcbuMD69FLhr7a7Qnx1ZvJ2cXL16sqKgo3XTTTW77HT9+XEePHlVsbKwkqVevXrJarY67mqRzszW7d+92G2IAAEDTUi8zMeXl5Vq8eLHGjBkjf/9fdlFYWKiMjAzdcsstio2N1eHDh/XEE08oMjJSN998syQpLCxM48aN0+TJkxUREaHw8HBNmTJFXbt2ddytBAAAUC8hZv369fruu+90zz33OLX7+fnpm2++0RtvvKGTJ08qNjZWgwYN0qpVqxQSEuLoN2fOHPn7+2vUqFE6c+aMrr/+ei1ZskR+fg0z1QwAAHxfvYSYlJQUVXa9cFBQkD7++ONq1w8MDNTcuXM1d+7c+igPAAA0Aj74hgUAAADVI8QAAABTIsQAAABTIsQAAABTIsQAAABTqtcPgAQANFIbZnm7AoCZGAAAYE6EGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEqEGAAAYEr+3i4AAOpKv+8W1azjhohffh40tX6KAVDvmIkBAACmxEwMgKZtwyzP12H2BvAJzMQAAABTIsQAAABTIsQAAABTIsQAAABTIsQAAABTIsQAAABTIsQAAABTIsQAAABTIsQAAABTqvMQk5GRIYvF4vQVExPjWG4YhjIyMhQXF6egoCAlJSVpz549TtsoLi7WpEmTFBkZqebNm2vEiBH6/vvv67pUAABgYvUyE9O5c2fl5OQ4vr755hvHsueff16zZ8/WvHnztGPHDsXExCg5OVmnTp1y9ElLS9Pq1au1cuVKff755yosLNSwYcNUVlZWH+UCAAATqpfPTvL393eafalgGIZefvllPfnkkxo5cqQkaenSpYqOjtaKFSt03333KT8/X6+//rrefPNN3XDDDZKkZcuWKT4+XuvXr9fgwYPro2QAAGAy9RJiDh48qLi4ONlsNvXt21czZ87UpZdequzsbOXm5iolJcXR12azaeDAgdqyZYvuu+8+7dy5U3a73alPXFycunTpoi1btlQZYoqLi1VcXOx4XFBQIEmy2+2y2+2On8//DmeMj3sV42IxPJsR9HQ8Pd1+bfZR2/3UZHvnb7chjv185Ra/GvWzGxc5CV2L8a7Y50Xvuz7U5t98HR9HlePD65FDU3mN9uT4LIZhGHW5848++khFRUXq0KGDjh07pmeeeUb/+te/tGfPHu3fv19XXXWV/vOf/yguLs6xzoQJE3TkyBF9/PHHWrFihe6++26nQCJJKSkpSkxM1MKFCyvdb0ZGhmbMmOHSvmLFCgUHB9flIQIAgHpSVFSk0aNHKz8/X6GhoW771vlMzNChQx0/d+3aVf3791e7du20dOlS9evXT5JksVic1jEMw6XtQtX1mTp1qtLT0x2PCwoKFB8fr5SUFMcg2O12ZWZmKjk5WVar1eNja+wYH/cqxudwYDsZNfwfvyRNHHSZR/t5ZcMhT0vzeB+13Y87FqNMbc9+6zQ+DXHs57vy+yU16tenbfhF7UfXpFff5wL2TS8rs7CDklsckNVSfnH7r2u1OB5tnl2nJdiNZpWPT21qa6Saymt0xZmUmqiX00nna968ubp27aqDBw/qN7/5jSQpNzdXsbGxjj55eXmKjo6WJMXExKikpEQnTpxQq1atnPoMGDCgyv3YbDbZbDaXdqvV6vJkV9aGXzA+7hkWP49CjKdj6cm2a7uP2u6nptut2HZDHPv5mtXwdNRFh4ja/Pv47z6tlnLfCzEXcTx1zWV8eC1y0dhfoz05tno/OVtcXKx9+/YpNjZWiYmJiomJUWZmpmN5SUmJNm7c6AgovXr1ktVqdeqTk5Oj3bt3uw0xAACgaanzmZgpU6Zo+PDhatOmjfLy8vTMM8+ooKBAY8aMkcViUVpammbOnKn27durffv2mjlzpoKDgzV69GhJUlhYmMaNG6fJkycrIiJC4eHhmjJlirp27eq4WwkAAKDOQ8z333+v22+/XT/99JMuueQS9evXT9u2bVNCQoIk6dFHH9WZM2f0wAMP6MSJE+rbt6/WrVunkJAQxzbmzJkjf39/jRo1SmfOnNH111+vJUuWyM+vfqa/AQCA+dR5iFm5cqXb5RaLRRkZGcrIyKiyT2BgoObOnau5c+fWcXVAw5uTecDbJQBAo+SDb1gAAABQPUIMAAAwpXq/xRoAzG7rv487Pd5WWv0pwkeSO9RXOQD+ixADAMCGWZ6vM2hq3dcBj3A6CQAAmBIhBgAAmBKnk+CTanNbMtcgAEDTwkwMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJX9vFwDUlTmZBzzq/0hyh3qqBBU8fU4AwBOEGAAe6ffdokrbyy1++iligK78fomaGWUXvZ9tbSZc9Da86cIA1+foz1KEtP3wz1WOT/9LIxqiNKDRqPMQM2vWLL399tv617/+paCgIA0YMEB//vOf1bFjR0efsWPHaunSpU7r9e3bV9u2bXM8Li4u1pQpU/T3v/9dZ86c0fXXX69XX31VrVu3ruuSAQCNyYZZ3q4ADaTOr4nZuHGjJk6cqG3btikzM1OlpaVKSUnR6dOnnfoNGTJEOTk5jq81a9Y4LU9LS9Pq1au1cuVKff755yosLNSwYcNUVnbx/8MDAADmV+czMWvXrnV6vHjxYkVFRWnnzp269tprHe02m00xMTGVbiM/P1+vv/663nzzTd1www2SpGXLlik+Pl7r16/X4MGD67psAABgMvV+TUx+fr4kKTw83Kn9s88+U1RUlFq2bKmBAwfq2WefVVRUlCRp586dstvtSklJcfSPi4tTly5dtGXLlkpDTHFxsYqLix2PCwoKJEl2u112u93x8/nf4cyXxsdSB9dUVMfT46zo3xC1eao2z1ltj6Pc4ue2varlnqpNfTXdt93wbBL6wu1eTG3uavS0rjpTm3/zdVxrxbG7jIEP1FalBn6t9KXX6PrkyfFZDMMw6qsQwzD0P//zPzpx4oQ2b97saF+1apVatGihhIQEZWdna9q0aSotLdXOnTtls9m0YsUK3X333U6hRJJSUlKUmJiohQsXuuwrIyNDM2bMcGlfsWKFgoOD6/7gAABAnSsqKtLo0aOVn5+v0NBQt33rdSbmwQcf1Ndff63PP//cqf3WW291/NylSxf17t1bCQkJ+vDDDzVy5Mgqt2cYhiwWS6XLpk6dqvT0dMfjgoICxcfHKyUlxTEIdrtdmZmZSk5OltVqvZhDa5R8aXxe2XCo3vcxcdBlHvWvGJ/Dge1k1NFsQ13x9Fik2o/xld8vqbS93OKnn8P7KvznL+rk7qQdrcd6vE5VtV2oT9vw6judZ/vhn50e16a2Xv95s9rx8bSuOnNNevV9LrR5dp2WYDeaKbOwg5JbHJDVUv7LAh+orUq1qe0i+NJrdH2qOJNSE/UWYiZNmqT33ntPmzZtqvaOotjYWCUkJOjgwYOSpJiYGJWUlOjEiRNq1aqVo19eXp4GDBhQ6TZsNptsNptLu9VqdXmyK2vDL3xhfBoiJNT2GA2Ln8+FmNocS22PobqA0swoq5MQU5v6arpfpz+StdjuxdTmbnw8ravO1ObfQj3VarWUO4+DD9Xmwkuvk77wGl2fPDm2Oj9xaBiGHnzwQb399tv69NNPlZiYWO06x48f19GjRxUbGytJ6tWrl6xWqzIzMx19cnJytHv37ipDDAAAaFrqfCZm4sSJWrFihd59912FhIQoNzdXkhQWFqagoCAVFhYqIyNDt9xyi2JjY3X48GE98cQTioyM1M033+zoO27cOE2ePFkREREKDw/XlClT1LVrV8fdSgAAoGmr8xAzf/58SVJSUpJT++LFizV27Fj5+fnpm2++0RtvvKGTJ08qNjZWgwYN0qpVqxQSEuLoP2fOHPn7+2vUqFGON7tbsmSJ/Px8axofAAB4R52HmOpudgoKCtLHH39c7XYCAwM1d+5czZ07t65KAwBUhne4hUnx2UkAmpyt/z7u7RIA1AEvvbMSAADAxSHEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAUyLEAAAAU+JjB5q4OZkHHD9bjDIlSnplwyEZlqo/aPOR5A4NUBngu/p9t8jjdcrd/JsCUDvMxAAAAFNiJgZowmozowAAvoKZGAAAYErMxACNwPnXNgFAU0GIAQDAl22Yde670UxSJ2nzbMlS7n6dQVPrvSxfQIgBfFBtrlXZ1mZCPVQCAL6LEAMAPmLrv4971L//pRH1VAlgDlzYCwAATImZGHisNheR8gZ5AIC6xkwMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJUIMAAAwJW6xBuCT+IRtANVhJgYAAJgSMzEAANRGxQczeqKJfDBjQ2EmBgAAmBIzMQDQhPAhk2hMCDEAAN9Vm1M2aDIIMWiyPP0gS4tRpsRa7Ie7bACgfnBNDAAAMCVmYgAPXfn9EjUzyrxdBgA0eYSYBuLpqQtJeiS5Qz1UAgBA40CI8WEEHwAAqubzIebVV1/VCy+8oJycHHXu3Fkvv/yyrrnmGm+XBQ/VJpDBM1xAjPrg6S3ZErdlo+H4dIhZtWqV0tLS9Oqrr+qqq67SwoULNXToUO3du1dt2rTxdnkA4FW1CRhAY+LTIWb27NkaN26c7r33XknSyy+/rI8//ljz58/XrFm8d0Bj1VAzCtvaTGiQ/QCo3vbDP0sR577X5MJ5Znuq0VDvr+Plj1Hw2RBTUlKinTt36vHHH3dqT0lJ0ZYtW7xUle/jtA0A+DDevK9O+WyI+emnn1RWVqbo6Gin9ujoaOXm5rr0Ly4uVnFxseNxfn6+JOnnn3+W3W6XJNntdhUVFen48eOyWq0XVd9rm/59Uev7IotRpqLiIhWX5cuw+Lks7/nDco+3uSvuDo/XOXW21ON1aqPzgVc96l9u8dOJVr116mwpt1hXotxiqKioiPGpQlMan3V7j3m8TrnFz6Pxqc0+aqNXm1YNsp+asBvNzv0Ns5TIain3djnnHK/7U5qnTp2SJBmGUW1fnw0xFSwWi9NjwzBc2iRp1qxZmjFjhkt7YmJt3mMVdecv3i4AAFBvMupty6dOnVJYWJjbPj4bYiIjI+Xn5+cy65KXl+cyOyNJU6dOVXp6uuNxeXm5fv75Z0VERDhCT0FBgeLj43X06FGFhobW7wGYEOPjHuPjHuPjHuPjHuNTvaYyRoZh6NSpU4qLi6u2r8+GmICAAPXq1UuZmZm6+eabHe2ZmZn6n//5H5f+NptNNpvNqa1ly5aVbjs0NLRR/wJcLMbHPcbHPcbHPcbHPcanek1hjKqbgangsyFGktLT05WamqrevXurf//+WrRokb777jvdf//93i4NAAB4mU+HmFtvvVXHjx/X008/rZycHHXp0kVr1qxRQkKCt0sDAABe5tMhRpIeeOABPfDAA3WyLZvNpunTp7ucdsI5jI97jI97jI97jI97jE/1GCNXFqMm9zABAAD4mGbeLgAAAKA2CDEAAMCUCDEAAMCUCDEAAMCUmkyIefXVV5WYmKjAwED16tVLmzdv9nZJPmHWrFm68sorFRISoqioKP3mN7/R/v37vV2Wz5o1a5YsFovS0tK8XYpP+c9//qM777xTERERCg4O1hVXXKGdO3d6uyyfUFpaqj/+8Y9KTExUUFCQLr30Uj399NMqL/eRz75pYJs2bdLw4cMVFxcni8Wid955x2m5YRjKyMhQXFycgoKClJSUpD179ninWC9wNz52u12PPfaYunbtqubNmysuLk533XWXfvjhB+8V7GVNIsSsWrVKaWlpevLJJ/XPf/5T11xzjYYOHarvvvvO26V53caNGzVx4kRt27ZNmZmZKi0tVUpKik6fPu3t0nzOjh07tGjRInXr1s3bpfiUEydO6KqrrpLVatVHH32kvXv36qWXXqryHbObmj//+c9asGCB5s2bp3379un555/XCy+8oLlz53q7NK84ffq0unfvrnnz5lW6/Pnnn9fs2bM1b9487dixQzExMUpOTnZ8KGBj5258ioqKtGvXLk2bNk27du3S22+/rQMHDmjEiBFeqNRHGE1Anz59jPvvv9+prVOnTsbjjz/upYp8V15eniHJ2Lhxo7dL8SmnTp0y2rdvb2RmZhoDBw40Hn74YW+X5DMee+wx4+qrr/Z2GT7rpptuMu655x6ntpEjRxp33nmnlyryHZKM1atXOx6Xl5cbMTExxnPPPedoO3v2rBEWFmYsWLDACxV614XjU5nt27cbkowjR440TFE+ptHPxJSUlGjnzp1KSUlxak9JSdGWLVu8VJXvys/PlySFh4d7uRLfMnHiRN1000264YYbvF2Kz3nvvffUu3dv/e53v1NUVJR69Oih1157zdtl+Yyrr75an3zyiQ4cOCBJ+uqrr/T555/rxhtv9HJlvic7O1u5ublOr9c2m00DBw7k9boK+fn5slgsTXbm0+ffsfdi/fTTTyorK3P55Ovo6GiXT8hu6gzDUHp6uq6++mp16dLF2+X4jJUrV2rXrl3asWOHt0vxSf/+9781f/58paen64knntD27dv10EMPyWaz6a677vJ2eV732GOPKT8/X506dZKfn5/Kysr07LPP6vbbb/d2aT6n4jW5stfrI0eOeKMkn3b27Fk9/vjjGj16dKP/QMiqNPoQU8FisTg9NgzDpa2pe/DBB/X111/r888/93YpPuPo0aN6+OGHtW7dOgUGBnq7HJ9UXl6u3r17a+bMmZKkHj16aM+ePZo/fz4hRueuyVu2bJlWrFihzp07KysrS2lpaYqLi9OYMWO8XZ5P4vW6ena7XbfddpvKy8v16quverscr2n0ISYyMlJ+fn4usy55eXkuab8pmzRpkt577z1t2rRJrVu39nY5PmPnzp3Ky8tTr169HG1lZWXatGmT5s2bp+LiYvn5+XmxQu+LjY3Vr3/9a6e2yy+/XP/v//0/L1XkW/7whz/o8ccf12233SZJ6tq1q44cOaJZs2YRYi4QExMj6dyMTGxsrKOd12tndrtdo0aNUnZ2tj799NMmOwsjNYG7kwICAtSrVy9lZmY6tWdmZmrAgAFeqsp3GIahBx98UG+//bY+/fRTJSYmerskn3L99dfrm2++UVZWluOrd+/euuOOO5SVldXkA4wkXXXVVS635R84cIBPm/+voqIiNWvm/FLr5+fXZG+xdicxMVExMTFOr9clJSXauHEjr9f/VRFgDh48qPXr1ysiIsLbJXlVo5+JkaT09HSlpqaqd+/e6t+/vxYtWqTvvvtO999/v7dL87qJEydqxYoVevfddxUSEuKYsQoLC1NQUJCXq/O+kJAQl+uDmjdvroiICK4b+q9HHnlEAwYM0MyZMzVq1Cht375dixYt0qJFi7xdmk8YPny4nn32WbVp00adO3fWP//5T82ePVv33HOPt0vzisLCQh06dMjxODs7W1lZWQoPD1ebNm2UlpammTNnqn379mrfvr1mzpyp4OBgjR492otVNxx34xMXF6ff/va32rVrlz744AOVlZU5XrPDw8MVEBDgrbK9x7s3RzWcV155xUhISDACAgKMnj17cgvxf0mq9Gvx4sXeLs1ncYu1q/fff9/o0qWLYbPZjE6dOhmLFi3ydkk+o6CgwHj44YeNNm3aGIGBgcall15qPPnkk0ZxcbG3S/OKDRs2VPqaM2bMGMMwzt1mPX36dCMmJsaw2WzGtddea3zzzTfeLboBuRuf7OzsKl+zN2zY4O3SvcJiGIbRkKEJAACgLjT6a2IAAEDjRIgBAACmRIgBAACmRIgBAACmRIgBAACmRIgBAACmRIgBAACmRIgBAACmRIgBYEpJSUlKS0tzaV+yZIlatmzZ4PUAaHiEGAAAYEpN4gMgAZhPUlKS40M2ly1bJj8/P/3+97/Xn/70J1ksFi9XB8AXMBMDwGctXbpU/v7++uKLL/TXv/5Vc+bM0d/+9jdvlwXAR/ABkAB8UlJSkvLy8rRnzx7HzMvjjz+u9957T3v37lVSUpK2bNmigIAAp/VKS0sVGBiokydPeqFqAA2JmRgAPqtfv35Op4769++vgwcPqqysTJJ0xx13KCsry+nr6aef9la5ABoY18QAMK2wsDBddtllTm1RUVFeqgZAQ2MmBoDP2rZtm8vj9u3by8/Pz0sVAfAlhBgAPuvo0aNKT0/X/v379fe//11z587Vww8/7O2yAPgITicB8Fl33XWXzpw5oz59+sjPz0+TJk3ShAkTvF0WAB/B3UkAfFJSUpKuuOIKvfzyy94uBYCP4nQSAAAwJUIMAAAwJU4nAQAAU2ImBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmBIhBgAAmNL/B8Areo42jjkxAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ax = ph_min_df.ph_min.hist(bins=30, alpha=0.5, label='min pH')\n",
    "ph_max_df.ph_max.hist(bins=30, ax=ax, alpha=0.5, label='max pH')\n",
    "ax.set_title('Brenda pH range data')\n",
    "ax.set_xlabel('pH')\n",
    "plt.legend()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59df74c4-a7ca-42b3-a8a1-599795fb9253",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44d4f835-87a8-42d1-991b-fe4dc2cdea5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "094d4434-0668-486d-8094-eca57c8b0c97",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
